{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import json\n",
    "import github_api\n",
    "import search_for_url\n",
    "import numpy as np\n",
    "from urllib import parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "papes = glob.glob('../git_papes/*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "papes = sorted(papes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "176"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(papes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(papes[5]) as infile:\n",
    "    dat = json.load(infile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "def standardize_url(u):\n",
    "    pp = parse.urlparse(u)\n",
    "    u = pp.netloc + pp.path\n",
    "    if u[-1]=='/':\n",
    "        u = u[:-1]\n",
    "    return u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "pairs = []\n",
    "for p in papes:\n",
    "    with open(p) as infile:\n",
    "        dat = json.load(infile)\n",
    "        for d in dat:\n",
    "            #assert(len(d['documents'])==1)\n",
    "            pmcid = d['documents'][0]['id']\n",
    "            urls = search_for_url.search_for_url(' '.join([i['text'] for i in d['documents'][0]['passages']]))\n",
    "            for u in urls:\n",
    "                u = standardize_url(u)\n",
    "                if 'github' in u:\n",
    "                    pairs.append((pmcid, u))\n",
    "pairs = np.array(pairs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'github.com/hello'"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "standardize_url('https://github.com/hello/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "urls_in_papes = pd.DataFrame({'pmcid': np.array(pairs)[:, 0], 'url': np.array(pairs)[:, 1]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmcid</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4910214</td>\n",
       "      <td>github.com/lh3/wgsim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5071354</td>\n",
       "      <td>github.com/JAvRZ/andi-dataprocessing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5013563</td>\n",
       "      <td>github.com/AharoniLab/MatchWeiz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4958917</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5107194</td>\n",
       "      <td>github.com/ijmarshall/picotron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5127779</td>\n",
       "      <td>github.com/awilfert/PSAP-pipeline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4928833</td>\n",
       "      <td>gist.github.com/darmitage/3179407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5140850</td>\n",
       "      <td>github.com/confunguido/PokemonDataAnalysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4966778</td>\n",
       "      <td>uhkniazi.github.io/dismiss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5045566</td>\n",
       "      <td>github.com/alexandrovteam/spatial-corals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5048062</td>\n",
       "      <td>uubram.github.io/RTCR/along</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5137812</td>\n",
       "      <td>github.com/bulik/ldsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5141584</td>\n",
       "      <td>github.com/tanghaibao/treecut/blob/master/scri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5142241</td>\n",
       "      <td>github.com/aboyle/F-seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5142633</td>\n",
       "      <td>www.github.com/rhenley/Pythion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5145274</td>\n",
       "      <td>github.com/Caleydo/caleydo_clue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5146994</td>\n",
       "      <td>github.com/Caleydo/pathfinder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>5147733</td>\n",
       "      <td>github.com/GauravPandeyLab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5148158</td>\n",
       "      <td>github.com/tdsmith/eleven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>5152673</td>\n",
       "      <td>github.com/nmcnulty/COPRO-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>5152685</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5154552</td>\n",
       "      <td>github.com/rhenley/Pyth-Ion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5154821</td>\n",
       "      <td>github.com/rgcgithub/clamms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>5154867</td>\n",
       "      <td>github.com/ZWang-Lab/LSKAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>5159287</td>\n",
       "      <td>github.com/markus-nilsson/md-dmri</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>5161348</td>\n",
       "      <td>github.com/divara01/PSB2017_ReproducibilityOfBNs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>5164934</td>\n",
       "      <td>github.com/bulik/ldsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/torognes/swarm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/oscar-franzen/oclust</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/fangly/AmpliCopyrighter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29223</th>\n",
       "      <td>6168466</td>\n",
       "      <td>github.com/bulik/ldsc/wiki/Genetic-Correlation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29224</th>\n",
       "      <td>6168517</td>\n",
       "      <td>github.com/baj12/clean_ngs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29225</th>\n",
       "      <td>6168540</td>\n",
       "      <td>github.com/SantanderMetGroup/notebooks</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29226</th>\n",
       "      <td>6168587</td>\n",
       "      <td>github.com/CamachoDejay/FRET-calculations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29227</th>\n",
       "      <td>6168625</td>\n",
       "      <td>github.com/BigNeuron/Data/releases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29228</th>\n",
       "      <td>6168662</td>\n",
       "      <td>github.com/CuppenResearch/IAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29229</th>\n",
       "      <td>6168678</td>\n",
       "      <td>k-basinski.github.io/mia_music_prefs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29230</th>\n",
       "      <td>6168856</td>\n",
       "      <td>github.com/lh3/seqtk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29231</th>\n",
       "      <td>6169016</td>\n",
       "      <td>github.com/svkucheryavski/mdatoolsm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29232</th>\n",
       "      <td>6169378</td>\n",
       "      <td>github.com/arunsethuraman/inrelate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29233</th>\n",
       "      <td>6169389</td>\n",
       "      <td>github.com/chaozhi/RABBIT.git</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29234</th>\n",
       "      <td>6169393</td>\n",
       "      <td>github.com/danrdanny/Drosophila15GenomesProject</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29235</th>\n",
       "      <td>6169397</td>\n",
       "      <td>github.com/danrdanny/Nanopore_ISO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29236</th>\n",
       "      <td>6169397</td>\n",
       "      <td>github.com/esolares/DMI1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29237</th>\n",
       "      <td>6169740</td>\n",
       "      <td>github.com/cancerit/CaVEMan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29238</th>\n",
       "      <td>6169915</td>\n",
       "      <td>github.com/ChusCarro/MatCardiacMLab/releases/t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29239</th>\n",
       "      <td>6169918</td>\n",
       "      <td>github.com/TheJacksonLaboratory/GDCSlicing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29240</th>\n",
       "      <td>6169937</td>\n",
       "      <td>github.com/VHIRHepatiques/QSutils</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29241</th>\n",
       "      <td>6170166</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29242</th>\n",
       "      <td>6170167</td>\n",
       "      <td>github.com/satijalab/seurat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29243</th>\n",
       "      <td>5526259</td>\n",
       "      <td>github.com/james-e-barrett</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29244</th>\n",
       "      <td>5526664</td>\n",
       "      <td>github.com/swiederm/predictive-gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29245</th>\n",
       "      <td>5526664</td>\n",
       "      <td>github.com/elifesciences-publications/predicti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29246</th>\n",
       "      <td>5526665</td>\n",
       "      <td>github.com/kescobo/kvasir</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29247</th>\n",
       "      <td>5526665</td>\n",
       "      <td>github.com/elifesciences-publications/kvasir</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29248</th>\n",
       "      <td>5526666</td>\n",
       "      <td>github.com/JaneliaSciComp/CohenBolstadLee_eLif...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29249</th>\n",
       "      <td>5526896</td>\n",
       "      <td>cole-trapnell-lab.github.io/cufflinks</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29250</th>\n",
       "      <td>5526946</td>\n",
       "      <td>drpowell.github.io/vennt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29251</th>\n",
       "      <td>5526970</td>\n",
       "      <td>github.com/vaibhhav/phage_competition_paper_fr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29252</th>\n",
       "      <td>5526973</td>\n",
       "      <td>github.com/alexdobin/STAR/releases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29253 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         pmcid                                                url\n",
       "0      4910214                               github.com/lh3/wgsim\n",
       "1      5071354               github.com/JAvRZ/andi-dataprocessing\n",
       "2      5013563                    github.com/AharoniLab/MatchWeiz\n",
       "3      4958917                    broadinstitute.github.io/picard\n",
       "4      5107194                     github.com/ijmarshall/picotron\n",
       "5      5127779                  github.com/awilfert/PSAP-pipeline\n",
       "6      4928833                  gist.github.com/darmitage/3179407\n",
       "7      5140850         github.com/confunguido/PokemonDataAnalysis\n",
       "8      4966778                         uhkniazi.github.io/dismiss\n",
       "9      5045566           github.com/alexandrovteam/spatial-corals\n",
       "10     5048062                        uubram.github.io/RTCR/along\n",
       "11     5137812                              github.com/bulik/ldsc\n",
       "12     5141584  github.com/tanghaibao/treecut/blob/master/scri...\n",
       "13     5142241                            github.com/aboyle/F-seq\n",
       "14     5142633                     www.github.com/rhenley/Pythion\n",
       "15     5145274                    github.com/Caleydo/caleydo_clue\n",
       "16     5146994                      github.com/Caleydo/pathfinder\n",
       "17     5147733                         github.com/GauravPandeyLab\n",
       "18     5148158                          github.com/tdsmith/eleven\n",
       "19     5152673                      github.com/nmcnulty/COPRO-Seq\n",
       "20     5152685                    broadinstitute.github.io/picard\n",
       "21     5154552                        github.com/rhenley/Pyth-Ion\n",
       "22     5154821                        github.com/rgcgithub/clamms\n",
       "23     5154867                         github.com/ZWang-Lab/LSKAT\n",
       "24     5159287                  github.com/markus-nilsson/md-dmri\n",
       "25     5161348   github.com/divara01/PSB2017_ReproducibilityOfBNs\n",
       "26     5164934                              github.com/bulik/ldsc\n",
       "27     5164958                          github.com/torognes/swarm\n",
       "28     5164958                    github.com/oscar-franzen/oclust\n",
       "29     5164958                 github.com/fangly/AmpliCopyrighter\n",
       "...        ...                                                ...\n",
       "29223  6168466     github.com/bulik/ldsc/wiki/Genetic-Correlation\n",
       "29224  6168517                         github.com/baj12/clean_ngs\n",
       "29225  6168540             github.com/SantanderMetGroup/notebooks\n",
       "29226  6168587          github.com/CamachoDejay/FRET-calculations\n",
       "29227  6168625                 github.com/BigNeuron/Data/releases\n",
       "29228  6168662                      github.com/CuppenResearch/IAP\n",
       "29229  6168678               k-basinski.github.io/mia_music_prefs\n",
       "29230  6168856                               github.com/lh3/seqtk\n",
       "29231  6169016                github.com/svkucheryavski/mdatoolsm\n",
       "29232  6169378                 github.com/arunsethuraman/inrelate\n",
       "29233  6169389                      github.com/chaozhi/RABBIT.git\n",
       "29234  6169393    github.com/danrdanny/Drosophila15GenomesProject\n",
       "29235  6169397                 github.com/danrdanny/Nanopore_ISO1\n",
       "29236  6169397                           github.com/esolares/DMI1\n",
       "29237  6169740                        github.com/cancerit/CaVEMan\n",
       "29238  6169915  github.com/ChusCarro/MatCardiacMLab/releases/t...\n",
       "29239  6169918         github.com/TheJacksonLaboratory/GDCSlicing\n",
       "29240  6169937                  github.com/VHIRHepatiques/QSutils\n",
       "29241  6170166                    broadinstitute.github.io/picard\n",
       "29242  6170167                        github.com/satijalab/seurat\n",
       "29243  5526259                         github.com/james-e-barrett\n",
       "29244  5526664                github.com/swiederm/predictive-gain\n",
       "29245  5526664  github.com/elifesciences-publications/predicti...\n",
       "29246  5526665                          github.com/kescobo/kvasir\n",
       "29247  5526665       github.com/elifesciences-publications/kvasir\n",
       "29248  5526666  github.com/JaneliaSciComp/CohenBolstadLee_eLif...\n",
       "29249  5526896              cole-trapnell-lab.github.io/cufflinks\n",
       "29250  5526946                           drpowell.github.io/vennt\n",
       "29251  5526970  github.com/vaibhhav/phage_competition_paper_fr...\n",
       "29252  5526973                 github.com/alexdobin/STAR/releases\n",
       "\n",
       "[29253 rows x 2 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "urls_in_papes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "urls_in_papes.to_csv('../data/papes_and_github_urls.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16510"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(urls_in_papes.pmcid))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Diagnose some cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "pmcid_to_doc = {}\n",
    "for p in papes:\n",
    "    with open(p) as infile:\n",
    "        dat = json.load(infile)\n",
    "        for d in dat:\n",
    "            #assert(len(d['documents'])==1)\n",
    "            pmcid = d['documents'][0]['id']\n",
    "            pmcid_to_doc[pmcid] = d['documents'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'passages': [{'text': '‘Genome skimming’ with the MinION hand-held sequencer identifies CITES-listed shark species in India’s exports market',\n",
       "   'offset': 0,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Cantu;given-names:Vito Adrian',\n",
       "    'name_3': 'surname:Fellows;given-names:Sam R.',\n",
       "    'name_0': 'surname:Johri;given-names:Shaili',\n",
       "    'name_1': 'surname:Solanki;given-names:Jitesh',\n",
       "    'name_6': 'surname:Vyas;given-names:Asit',\n",
       "    'article-id_publisher-id': '40940',\n",
       "    'license': 'Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons license, and indicate if changes were made. The images or other third party material in this article are included in the article’s Creative Commons license, unless indicated otherwise in a credit line to the material. If material is not included in the article’s Creative Commons license and your intended use is not permitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this license, visit http://creativecommons.org/licenses/by/4.0/.',\n",
       "    'name_5': 'surname:Moreno;given-names:Isabel',\n",
       "    'elocation-id': '4476',\n",
       "    'article-id_doi': '10.1038/s41598-019-40940-9',\n",
       "    'section_type': 'TITLE',\n",
       "    'article-id_pmid': '30872700',\n",
       "    'volume': '9',\n",
       "    'article-id_pmc': '6418218',\n",
       "    'year': '2019',\n",
       "    'name_4': 'surname:Edwards;given-names:Robert A.',\n",
       "    'type': 'front',\n",
       "    'name_7': 'surname:Dinsdale;given-names:Elizabeth A.'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes - sharks, rays, skates, and chimeras, are among the most threatened and data deficient vertebrate species. Global demand for shark and ray derived products, drives unregulated and exploitative fishing practices, which are in turn facilitated by the lack of ecological data required for effective conservation of these species. Here, we describe a Next Generation Sequencing method (using the MinION, a hand-held portable sequencing device from Oxford Nanopore Technologies), and analyses pipeline for molecular ecological studies in Chondrichthyes. Using this method, the complete mitochondrial genome and nuclear intergenic and protein-coding sequences were obtained by direct sequencing of genomic DNA obtained from shark fin tissue. Recovered loci include mitochondrial barcode sequences- Cytochrome oxidase I, NADH2, 16S rRNA and 12S rRNA- and nuclear genetic loci such as 5.8S rRNA, Internal Transcribed Spacer 2, and 28S rRNA regions, which are commonly used for taxonomic identification. Other loci recovered were the nuclear protein-coding genes for antithrombin or SerpinC, Immunoglobulin lambda light chain, Preprogehrelin, selenium binding protein 1(SBP1), Interleukin-1\\xa0beta (IL-1β) and Recombination-Activating Gene 1 (RAG1). The median coverage across all genetic loci was 20x and sequence accuracy was ≥99.8% compared to reference sequences. Analyses of the nuclear ITS2 region and the mitochondrial protein-encoding loci allowed accurate taxonomic identification of the shark specimen as Carcharhinus falciformis, a CITES Appendix II species. MinION sequencing provided 1,152,211\\u2009bp of new shark genome, increasing the number of sequenced shark genomes to five. Phylogenetic analyses using both mitochondrial and nuclear loci provided evidence that Prionace glauca is nested within Carcharhinus, suggesting the need for taxonomic reassignment of P. glauca. We increased genomic information about a shark species for ecological and population genetic studies, enabled accurate identification of the shark tissue for biodiversity indexing and resolved phylogenetic relationships among multiple taxa. The method was independent of amplification bias, and adaptable for field assessments of other Chondrichthyes and wildlife species in the future.',\n",
       "   'offset': 124,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Introduction',\n",
       "   'offset': 2405,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes (sharks, rays, skates, and chimaeras) are among the oldest extant vertebrates. They are vital to top-down regulation and maintenance of oceanic ecosystems, and to sustain healthy commercial fisheries and ecosystem services derived from marine environments. The high demand for shark and ray fins and an emerging demand for shark meat and oil has made world trade in Chondrichthyes a >$US 800 million industry. Demand for shark fins and meat has caused overfishing and exploitation of Chondrichthyes worldwide, driving an unprecedented number of chondrichthyan species towards extinction. Of the ~1,178 chondrichthyan species known today, 25% are threatened with extinction globally, making Chondrichthyes one of the most endangered vertebrate species groups. The threat is even more severe in areas such as the Arabian Seas Region (ASR) with up to 50% of chondrichthyan species in the region facing extinction threat.',\n",
       "   'offset': 2418,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyan fisheries are driven by 20 nations and a vast majority of these countries experience the highest levels of data deficiency. An estimated 46% of Chondrichthyes are considered data deficient, suggesting that information is insufficient to determine their conservation status. These data deficiencies restrict design and implementation of efficient conservation measures and allow continued exploitation of the most vulnerable Chondrichthyes. Not surprisingly, nations with high data deficiency also have some of the highest percentages of threatened Chondrichthyes. For effective conservation of remaining chondrichthyan populations continued monitoring and assessment is essential in order to determine taxonomic units, population structure, population size and identify priority conservation units with respect to evolutionary and management significance. Once these data are available, they can be used to design effective conservation measures including but not limited to, the designation of protected areas, reduction of targeted and incidental fishing of priority concern units, and CITES listing of priority concern units to protect vulnerable populations.',\n",
       "   'offset': 3351,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Data deficiency of Chondrichthyes can be addressed by increasing the rate of ecological investigations in areas with a high density of threatened and data deficient species. In the current scenario, molecular ecological studies in Chondrichthyes are typically restricted to DNA barcoding and microsatellite analyses for taxonomic identification and population studies respectively. Molecular taxonomy is conducted by PCR amplification and sequencing of DNA barcodes such as Cytochrome oxidase 1(COI)– NADH2 16S rRNA and 12S rRNA, of the mitochondrial genome and the Recombination-Activating Gene 1(RAG1) and Internal transcribed spacer 2 (ITS) regions of the nuclear genome. Sampling each one of these barcodes from a specimen requires a priori sequence information, PCR optimization, PCR amplification, post PCR clean up and finally sequencing. Single barcode PCR based identification is not the most efficient method for taxonomic identification and can be limiting due to the following reasons. First, single barcodes do not always guarantee accurate taxonomic identification, due to non-amplification for certain species using universal primers, or identical barcode sequences for multiple species such as Carcharhinid sharks or different barcode sequences for geographically isolated populations of the same nominal species. Second, the laboratory required for amplification and sequencing of barcodes from chondrichthyan specimens is often inaccessible to researchers and fisheries scientists in areas with heavy chondrichthyan fishing and exports, requiring them to outsource detection, adding significantly to the processing time. Last, single barcode PCRs provides up to 1000\\u2009bp of data and restrict the amount of genetic data derived for the respective species. Microsatellites used for population studies among Chondrichthyes are also PCR based, only include neutral genetic loci, require standardization for use across different laboratories and do not provide genome-wide assessments of neutral as well as adaptive loci. Accurate estimates of population size and structure as well as understanding of evolutionary units are only obtained from screening multiple genomic loci such as single nucleotide polymorphisms (SNPs) from genomic data sets. Thus, PCR based approaches for taxonomic and population studies are limiting to molecular ecological studies in Chondrichthyes. Adopting a genomics approach could allow assessment of multiple genomic loci without amplification bias and without the need for a priori sequence information. Such an approach would be ideal to increase ecological and evolutionary investigations in Chondrichthyes, similar to those of other commercial fisheries. Increasing capacity for use of genome-wide data addresses taxonomic inadequacies and enables complex ecological and evolutionary questions to be answered by providing information about population size, structure, biogeography and priority conservation units. These data together will address chondrichthyan data deficiencies and assist conservation efforts.',\n",
       "   'offset': 4529,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genomic analyses have limited adoption among chondrichthyan researchers due to complex instrumentation and expensive sequencing for whole genome shotgun sequencing or reduced representation methods such as Restriction site-Associated-DNA-Sequencing (RADseq) and Hybridization based gene capture. A crucial aim of the current study was to reduce the burden of sample processing and analyses, and to increase accuracy, speed and breadth of ecological investigations of Chondrichthyes through genome sequencing with sophisticated yet accessible technology. We explored the use of multiple mitochondrial and nuclear barcode regions which form the high copy number fraction of the genome to increase accuracy of taxonomic identification. We explored direct sequencing of genomic DNA (gDNA) to reduce sample processing times and to avoid amplification biases, thus increasing the potential for application of the assay to multi- species assessments. ‘Genome skimming’ which allows deep sequencing of high copy number fractions of the genome was used to access multiple mitochondrial and nuclear barcode sequences from shark genomic DNA. Sequencing was done on the MinION portable sequencer from Oxford Nanopore Technologies (ONT) which requires library preparation and loading times of as little as thirty minutes, allows sequence acquisition as early as 2\\u2009hours into sequencing, and allows local sequencing and analyses of genomic DNA with a user-friendly software interface installed on a portable computer.',\n",
       "   'offset': 7592,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We present characterization of ‘genome skimming’ on the MinION portable sequencer for taxonomic identification and phylogenetic analyses of Chondrichthyes. We report identification of a shark fin specimen in India’s export market to Carcharhinus falciformis, based on the complete mitochondrial sequence and nuclear genetic regions of the specimen. In addition, we present two sequence analyses methods which can be adapted to other wildlife species and which are suitable for standalone analyses in the field or laboratory, respectively. Last we resolved the placement of P. glauca within the Carcharhinus genus and described its relationships with other taxa in the genus, with high statistical power using the entire mitochondrial genome and nuclear genes from C. falciformis. Our sequencing and analyses methods increase the available genomic information about C. falciformis and provide an accessible tool for genome-wide studies across Chondrichthyes.',\n",
       "   'offset': 9102,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Results',\n",
       "   'offset': 10066,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genus level taxonomic resolution through photo identification and COI fragment analyses',\n",
       "   'offset': 10074,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sampling location and sample morphology of a dismembered shark specimen at time of sampling.',\n",
       "   'offset': 10162,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig1',\n",
       "    'file': '41598_2019_40940_Fig1_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The shark specimen encountered in Veraval, Gujarat, was processed for international fin and meat exports, and the head and fins of the specimen could be sampled for tissue collection and photo-identification purposes (Fig.\\xa01). Based on features of the head and mouth cavities the specimen was identified to the Carcharhinidae family. In order to obtain a species-level taxonomic resolution, we used DNA extracted from the specimen to PCR amplify three different fragments from COI and NADH2 mitochondrial loci. We were successful in amplifying one short 160\\u2009bp fragment of the cytochrome oxidase 1 (COI) gene, while PCR amplification of two longer fragments from COI and NADH2 did not yield a PCR product. The 160\\u2009bp PCR fragment was cross-referenced with the BOLD database which resulted in 100% sequence similarity to six species of the Carcharhinus genus including Carcharhinus brevipinna, Carcharhinus falciformis, Carcharhinus melanopterus, Carcharhinus sorrah, Carcharhinus altimus and Carcharhinus limbatus. These results obtained using a single short barcode supported identification based on morphology but were inconclusive for species-level identification.',\n",
       "   'offset': 10255,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Assessment of complete mitochondrial sequence using ‘genome skimming’ on the MinION',\n",
       "   'offset': 11428,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to obtain higher taxonomic resolution by assessment of longer and multiple barcode sequences ‘genome skimming’ using genomic DNA of the specimen was conducted on the MinION sequencer. We obtained 74,536 sequences with read lengths between 5–98,997\\u2009bp within\\xa036\\u2009hours of a single sequencing run.',\n",
       "   'offset': 11516,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Two workflow designs used in this study one, for field or forensic applications with no connectivity and average laptop computing power and two, for detailed laboratory analyses with connectivity and high computing power.',\n",
       "   'offset': 11831,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig2',\n",
       "    'file': '41598_2019_40940_Fig2_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(A) Sequencing coverage for complete mitochondrial genome of the specimen. The mean coverage for 16, 675 bases of the mitogenome was 19.4 with a Std Dev of 2.0, and minimum and maximum coverage of 16x and 25x respectively. Coverage in the forward direction was 9.5 and 9.9 in the reverse. Sequencing coverage for regularly used individual mitochondrial barcodes is shown in Fig.\\xa0S1. (B) Sequencing coverage for mitochondrial and nuclear genes of the specimen. Sequencing coverage for nine annotated nuclear genome contigs and one mitochondrial contig recovered with Workflow II ranged from 1x to >50x.',\n",
       "   'offset': 12053,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig3',\n",
       "    'file': '41598_2019_40940_Fig3_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The sequences were analyzed using two parallel workflows (Fig.\\xa02). In the first workflow, raw reads from the MinION were mapped to reference mitogenomes of Carcharhinus spp. identified to match the shark specimen by PCR fragment alignment. Mitogenomes were available for four of six Carcharhinus spp. (Carcharhinus melanopterus Carcharhinus sorrah, Carcharhinus brevipinna and Carcharhinus falciformis (Supplementary Table\\xa0S1) that matched the\\xa0specimen using PCR. These 4 mitogenomes were used to account for biases in the database and increase the mapping\\xa0accuracy of MinION sequences. All of the 74,536 reads from the MionION were mapped onto each of the 4 mitogenomes, which identified the same set of 47 reads (~0.06%) from each mapping. Assembly of the 47 reads generated a complete mitochondrial contig of 16,675\\u2009bp (Fig.\\xa03A, Genbank accession #MK092088), a genome size similar to mitochondrial genomes reported for Carcharhinus spp., (eg. Carcharhinus sorrah (16,707\\u2009bp), Carcharhinus brevipinna (16,706\\u2009bp), Carcharhinus melanopterus (16,706\\u2009bp), Carcharhinus plumbeus (16,706\\u2009bp), and Carcharhinus obscurus (16,706\\u2009bp), but significantly different than the published mitogenome for Carcharhinus falciformis at 17,639\\u2009bp. Mean coverage for the complete mitochondrial genome was 19.4\\u2009×\\u2009±\\u20092\\u2009×\\u2009(Fig.\\xa03A, Supplementary Fig.\\xa0S1).',\n",
       "   'offset': 12656,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In the second workflow (Fig.\\xa02), all 74,536 sample reads were first assembled into contigs, quality controlled by trimming read ends and aligned against a database of chondrichthyan mitochondrial and nuclear sequences to annotate the contigs. This method yielded a mitochondrial sequence of 16,477\\u2009bp which was 98% identical to the sequence in workflow 1 and nuclear genetic loci such as the Internal transcribed spacer 2 (ITS2) region, 5.8S, 28S and 18S\\u2009 rRNA coding regions, and protein-coding regions for antithrombin or SerpinC, Immunoglobulin lambda light chain, Preprogehrelin, selenium binding protein 1(SBP1), Interleukin-1\\xa0beta (IL-1β) and Recombination-Activating ene 1 (RAG1) (SRA accession #PRJNA503780). With the second workflow, we obtained an almost complete mitochondrial sequence and in addition, nuclear coding and non-coding regions. Mean coverage for nuclear coding and non-coding genes recovered ranged from 1x to >50x\\xa0(Fig.\\xa03B, Supplementary Fig.\\xa0S1).',\n",
       "   'offset': 14023,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Annotated mitochondrial sequence indicating gene position and G/C richness regions. Outer circle represents aligned genes with outermost representing the heavy strand (H) and adjacent inner ring the light strand (L). Green represents protein coding regions, pink tRNA, blue rRNA, and grey is the D-loop. Middle circle in black represents GC content: Protruding outwards indicates GC rich while inward protrusion shows AT rich regions. The inner most circle is GC skew (G\\u2009−\\u2009C/G\\u2009+\\u2009C) with positive skew (between 0 and 1) represented by green and negative as purple (between 0 and −1).',\n",
       "   'offset': 15007,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig4',\n",
       "    'file': '41598_2019_40940_Fig4_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genes identified in the Carcharhinus falciformis mitochondrial sequence with their start and end positions on the forward (+) or reverse (−) strands\\xa0and a score representing the heuristic measure of the annotation content for each entry.',\n",
       "   'offset': 15602,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab1',\n",
       "    'file': 'Tab1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Start\\tStop\\tType\\tName\\xa0(Direction)\\tScore\\n1\\t70\\ttRNA\\t\\ntRNA-Phe\\xa0(+)\\n\\t1.50E-14\\n72\\t1031\\trRNA\\t\\n12s rRNA\\xa0(+)\\n\\t0\\n1029\\t1100\\ttRNA\\t\\ntRNA-VAL\\xa0(+)\\n\\t1.00E-14\\n1124\\t2764\\trRNA\\t\\n16s rRNA\\xa0(+)\\n\\t0\\n2764\\t2838\\ttRNA\\t\\ntRNA-Leu2\\xa0(+)\\n\\t1.40E-12\\n2839\\t3813\\tgene\\t\\nND1\\xa0(+)\\n\\t453393778\\n3814\\t3883\\ttRNA\\t\\ntRNA-Ile\\xa0(+)\\n\\t2.00E-14\\n3885\\t3956\\ttRNA\\t\\ntRNA-Gln\\xa0(-)\\n\\t5.00E-15\\n3956\\t4024\\ttRNA\\t\\ntRNA-Met\\xa0(+)\\n\\t6.10E-15\\n4025\\t5071\\tgene\\t\\nND2\\xa0(+)\\n\\t324312000.6\\n5070\\t5140\\ttRNA\\t\\ntRNA-Trp\\xa0(+)\\n\\t1.80E-15\\n5142\\t5210\\ttRNA\\t\\ntRNA-Ala\\xa0(-)\\n\\t9.80E-14\\n5211\\t5281\\ttRNA\\t\\ntRNA-Asn\\xa0(-)\\n\\t2.00E-12\\n5317\\t5384\\ttRNA\\t\\ntRNA-Cys\\xa0(-)\\n\\t1.50E-11\\n5386\\t5454\\ttRNA\\t\\ntRNA-Tyr\\xa0(-)\\n\\t2.60E-16\\n5456\\t7012\\tgene\\t\\nCOI\\xa0(+)\\n\\t1166241108\\n7013\\t7083\\ttRNA\\t\\ntRNA-Ser2\\xa0(-)\\n\\t6.40E-15\\n7087\\t7156\\ttRNA\\t\\ntRNA-Asp\\xa0(+)\\n\\t6.30E-13\\n7164\\t7854\\tgene\\t\\nCOII\\xa0(+)\\n\\t280357893.9\\n7855\\t7925\\ttRNA\\t\\ntRNA-Lys\\xa0(+)\\n\\t4.90E-13\\n7927\\t8094\\tgene\\t\\nATPase 8\\xa0(+)\\n\\t3849114.9\\n8085\\t8719\\tgene\\t\\nATPase 6\\xa0(+)\\n\\t135520179.4\\n8763\\t9548\\tgene\\t\\nCOIII\\xa0(+)\\n\\t361705805.6\\n9551\\t9619\\ttRNA\\t\\ntRNA-Gly\\xa0(+)\\n\\t1.90E-13\\n9620\\t9970\\tgene\\t\\nND3\\xa0(+)\\n\\t43327351.7\\n9969\\t10038\\ttRNA\\t\\ntRNA-Arg\\xa0(+)\\n\\t2.90E-12\\n10039\\t10335\\tgene\\t\\nND4L\\xa0(+)\\n\\t25640182.3\\n10329\\t11709\\tgene\\t\\nND4\\xa0(+)\\n\\t770745173.3\\n11710\\t11778\\ttRNA\\t\\ntRNA-His\\xa0(+)\\n\\t1.40E-11\\n11779\\t11843\\ttRNA\\t\\ntRNA-Ser1\\xa0(+)\\n\\t5.60E-08\\n11844\\t11915\\ttRNA\\t\\ntRNA-Leu1\\xa0(+)\\n\\t8.30E-22\\n11916\\t13745\\tgene\\t\\nND5\\xa0(+)\\n\\t1036110292\\n13741\\t14262\\tgene\\t\\nND6\\xa0(-)\\n\\t47160329.7\\n14263\\t14331\\ttRNA\\t\\ntRNA-Glu\\xa0(-)\\n\\t1.10E-11\\n14334\\t15479\\tgene\\t\\nCyt b\\xa0(+)\\n\\t760558631.5\\n15479\\t15550\\ttRNA\\t\\ntRNA-Thr\\xa0(+)\\n\\t1.20E-11\\n15553\\t15620\\ttRNA\\t\\ntRNA-Pro\\xa0(-)\\n\\t2.60E-09\\n15663\\t16674\\trep-origin\\t\\nDloop\\xa0(+)\\n\\t3240692.3\\n',\n",
       "   'offset': 15843,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab1',\n",
       "    'File': 'Tab1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Annotation of the complete mitochondrial sequence using annotation tools followed by manual annotation of the D-Loop region in Geneious® yielded recovery of all coding and non-coding regions of the mitochondrial genome including thirteen protein coding regions, two rRNA coding regions, twenty-two tRNA coding regions, a replication region and a control region (Fig.\\xa04, Table\\xa01). These results demonstrate the efficiency of genome skimming over conventional single barcode PCR based assays. A single sequencing run on the MinION sequencer provided the entire mitochondrial sequence equivalent of multiple mitochondrial barcodes and multiple nuclear genetic loci. These data were acquired in a very brief time frame, without any amplification bias, and with minimal infrastructure compared to PCR based studies.',\n",
       "   'offset': 17397,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Species-level taxonomic resolution using phylogenetic analyses of multiple barcode regions',\n",
       "   'offset': 18211,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(A) Bayesian estimate of relationships among Carcharhinus spp. using complete mitogenomes. Bayesian phylogenetic estimate for complete mitochondrial genomes from 13 Carcharhinus species and Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis. Numbers at nodes are posterior probabilities. (B) Bayesian estimate of relationships among Carcharhinus spp. using protein coding genes. Bayesian phylogenetic estimate from 13 protein-coding mitochondrial genes from 13 Carcharhinus species and, Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis.\\xa0Numbers at nodes are posterior probabilities.',\n",
       "   'offset': 18302,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig5',\n",
       "    'file': '41598_2019_40940_Fig5_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We constructed phylogenetic trees based on the full mitochondrial genomes (Fig.\\xa05A) and 13 protein coding regions (Fig.\\xa05B) from the mitogenome of 13 shark species of the genus Carcharhinus, and Prionace glauca, and used Galeocerdo cuvier as an outgroup (family Carcharhinidae). Genbank accession numbers for reference sequences used are listed in Supplementary Tables\\xa0S1 and S2. Our phylogenies based on multiple mitochondrial markers are congruent with previous phylogenetic studies in the Carcharhinus genus conducted using a single NADH2 locus albeit ours have better statistical support (Fig.\\xa05A). Phylogenetic estimates showed robust support (>0.95 posterior probability (Fig.\\xa05A,B), >90% bootstrapping (Supplementary Fig.\\xa0S2) toward the tips of the phylogeny and lowered support at some deeper nodes. Phylogenetic estimates based on the full mitochondrial genome (Fig.\\xa05A) showed higher support throughout than those based on protein-coding genes only (Fig.\\xa05B). Using both analyses, based on full or protein-coding mitochondrial sequences, we recovered a sister relationship between the unknown sample and C. falciformis (Fig.\\xa05A,B). Our analyses estimated identical phylogenetic relationships among these 15 species, including support for the placement of P. glauca within Carcharhinus (Fig.\\xa05A,B).',\n",
       "   'offset': 18992,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bayesian estimate of relationships among Carcharhinus spp. using mitochondrial and nuclear genes. Bayesian phylogenetic estimate from concatenated alignment of three mitochondrial protein coding loci (NADH2, COI, NADH4) and one nuclear locus (ITS2) from 32 Carcharhinus species, Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis. Numbers at nodes are posterior probabilities.',\n",
       "   'offset': 20310,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig6',\n",
       "    'file': '41598_2019_40940_Fig6_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We also constructed phylogenetic trees based on a concatenated sequence of three mitochondrial protein-coding sequences (NADH2, COI, NADH4) and one nuclear intergenic region (ITS2) of 33 shark species of the genus Carcharhinus (of 35 currently described species), with P. glauca, and G. cuvier as an outgroup (Fig.\\xa06). Genbank accession numbers for references sequences used are listed in Supplementary Table\\xa0S3. Both Maximum Likelihood and Bayesian analyses using these concatenated genes strongly supported a sister relationship between the unknown sample and C. falciformis (Fig.\\xa06). The concatenated analyses showed robust support (>0.95 posterior probability (Fig.\\xa06), >90% bootstrapping, (Supplementary Fig.\\xa0S2) toward the tips of the phylogeny and lowered support at many deeper nodes. These results corroborate relationships seen using mitochondrial markers (Fig.\\xa05A,B) and improve on our phylogenetic analyses of P. glauca using only mitochondrial genomes, which did not include C. amblyrhynchos due to unavailability of a complete mitogenome.',\n",
       "   'offset': 20739,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Using phylogenetic analyses based on multiple nuclear and mitochondrial regions we were able to identify an unknown shark specimen as Carcharhinus falciformis, a CITES Appendix II listed species.',\n",
       "   'offset': 21798,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '*Only one mitochondrial reference sequence for C. falciformis with 17, 639 base pairs exists. Length of mitochondrial sequence obtained for C. falciformis in this study is 16, 629.',\n",
       "   'offset': 21994,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Accuracy of Carcharhinus falciformis mitochondrial sequences based on comparison with Genbank reference sequences (Table\\xa0S4).',\n",
       "   'offset': 22175,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab2',\n",
       "    'file': 'Tab2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genetic Loci\\tAccuracy\\tDifferences/Total base pairs\\nCytochrome oxidase 1 (COI)\\t99.33–100%\\t0–2/600–654\\nNADH2\\t98.5%\\t16/1044\\n12s rRNA\\t98.5%\\t6–7/1357–1411\\n16s rRNA\\t98.5%\\t6–7/1357–1411\\nMitochondria\\t98.75%\\t207/16,629–17,639*\\n',\n",
       "   'offset': 22302,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab2',\n",
       "    'File': 'Tab2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparison of individual protein-coding regions of the sample sequence to Carcharhinus falciformis sequences in the Genbank database, demonstrated ≧99.3% accuracy of our specimen sequences (Table\\xa02) (Genbank reference sequences used for comparison are listed in Supplementary Table\\xa0S4). In addition, we improved the phylogenetic placement, improved mitochondrial sequence data and increased the amount of nuclear genomic data on this species by five-fold (1.1 million bp).',\n",
       "   'offset': 22537,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Discussion',\n",
       "   'offset': 23014,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes are data deficient globally, making them highly vulnerable to fishing related exploitation. Molecular ecological assessments in areas of high data deficiency are often inefficient at addressing chondrichthyan deficiencies due to their reliance on molecular genetic tools that have a restrictive data output and because pan-genomic methods are inaccessible due to complex infrastructure and instrumentation requirements. As apex predators, Chondrichthyes are important for top-down regulation of oceanic ecosystems. Life history traits such as slow generation times, small litters, and stringent habitat requirements, make it challenging for chondrichthyan species populations to recover from overfishing practices. Due to overexploitation from unregulated fishing, of the 1,178 chondrichthyan species, approximately 25% face extinction and ~46% are data deficient, meaning information is insufficient for species conservation and management. To reduce these data deficiencies, the current manuscript describes methods that increased the taxonomic resolution of Carcharhinus falciformis, a Near Threatened CITES Appendix II listed species, in addition to providing genomic data for population genetic and evolutionary studies. The current study has increased the number of sequenced Chondrichthyes by 20%. The genomic data obtained, improved the genomic information for Carcharhinus falciformis, reduced data deficiencies through increased resolution of phylogenetic relationships within the Carcharhinus genus and in future studies will provide genomic data to understand population structure, population size and biogeography, which are important in management of priority species of concern such as the silky shark. Last, we described two different workflows for sequence analyses which will allow capacity building for taxonomic identification and population and phylogenetic studies among researchers with basic and advanced bioinformatic skills.',\n",
       "   'offset': 23025,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparison of Sequencing methods',\n",
       "   'offset': 24991,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The MinION portable sequencer obtained multiple barcode sequences, along with complete mitochondrial (Fig.\\xa03A, Fig.\\xa04) and partial nuclear genomes (Fig.\\xa03B) of a previously un-sequenced shark using genomic DNA collected from the market place. This method enabled accurate taxonomic identification of a shark tissue specimen by using multiple genetic barcodes, without the need for PCR amplification and without a priori sequence information.',\n",
       "   'offset': 25024,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We demonstrate the use of MinION sequencers to sequence larger genome sizes of 1–6.7\\u2009Gb, more typical of shark species. Nuclear genome size in sharks varies between 1–3 Gigabases and our method achieved optimal coverage (Fig.\\xa03A,B) of high copy number regions, such as mitochondrial and nuclear marker sequences often used for taxonomic identification and phylogenetic inference. Our methods expand on studies by Parker et al., who used the MinION sequencer to sequence relatively small Arabidopsis (~135 Mbp) genomes. Genome skimming using the MinION is amplification independent making it less affected by sequence biases. GC content is an important determinant of sequencing and computational bias in NGS. The mean GC content for mitochondrial genomes of Carcharhinid sharks is 38.44% (±0.006 STDEV), similar to the mitochondria of C. falciformis at a GC content of 38.4%. The range of GC content for all 673 chondrichthyan NADH2 barcodes sequenced to date is 32–47%. The range of GC content that the MinION sequenced in the C. falciformis genome was 29–64%, suggesting a wide capability of the sequencer for genomic assessment of a variety of Chondrichthyes. The method thus has potential to aide in taxonomic identification, ecological, evolutionary and population level studies of new and understudied species. Last, although the method relies on state of the art NGS technology, it only requires gDNA extraction from the specimen which takes about three and a half hours, a 15–30\\u2009minutes library preparation/loading time and finally sequencing with the MinION and a laptop. Basecalling can be done locally on the laptop computer in parallel or post-sequencing using ONT software. Analyses for the current study did not require internet connectivity as local databases were curated in advance onto the Geneious software on laptop computers. In the future, this method is adaptable to field-based ecological studies and will facilitate capacity building among conservation managers, wildlife researchers and forensics officials.',\n",
       "   'offset': 25469,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The higher cost of MinION flowcells compared to PCR amplification and sequencing of single barcodes remains a limiting factor to the wide adoption of this technology. Barcode sequencing using PCR requires gDNA extraction (~3.5 hrs), followed by PCR optimization for the species under consideration (hours to days), PCR amplification of a single barcode and possibly multiple species (~2 hrs) and last, PCR clean up (~30\\u2009minutes) and Sanger sequencing (~24 hrs) turnaround time. For the suggested timeline, the location should have access to a Sanger sequencer, which in case of the sampling site for the current study was a 7\\u2009hour drive away (403\\u2009km), suggesting that a 24\\u2009hr turnaround time is most unlikely. At the end of the PCR process a single barcode is sequenced for multiple species. In comparison, genome skimming on a MinION sequencer requires gDNA extraction (~3.5 hrs), followed by library preparation and flowcell loading (~15–30\\u2009minutes), followed by sequencing (~8–36\\u2009hours), and\\xa0recovery of fastq sequences and BLAST alignment to a local or global database for taxonomic identification. Therefore, PCR-based bar coded derives ~ 150–500\\u2009bp compared with 1,152,211\\u2009bp of data from the MinION. The MinION can be monitored in real time and the sequencing stopped once sufficient data has been received, such as mitochondrial sequences for taxonomic identification, further reducing sequencing time. PCR based DNA barcoding is economical for taxonomic classification purposes of multiple species, however, genome skimming using the MinION sequencer provides genomic information for a single species which will facilitate taxonomic classification, determination of population structure and size as well as accurate phylogenetic assessment. In addition, MinION sequencing allows researchers to maintain custody of samples during the entire process, whereas Sanger sequencing may require the samples to be de-identified before being sent to a sequencing facility for high-stakes studies.',\n",
       "   'offset': 27519,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The ability to perform sequencing experiments, assessing multiple genomic regions and determining taxonomic identities, irrespective of species or sample type, in remote locations provides a significant advantage for wildlife research and could make MinION a preferred choice for sequencing dependent upon the application of sequence data. Future studies in our laboratory will determine whether first, additional chondrichthyan and wildlife taxa can be analyzed in a fashion similar to C. falciformis using methods described here on the MinION sequencer and second, whether multiple specimens can be analyzed using our methods on a single MinION flowcell by barcoding each specimen, to reduce cost and increase efficiency. However, when sequencing multiple specimens simultaneously, reduced sequencing coverage of large chondrichthyan genomes could limit analyses of multiple samples with the MinION.',\n",
       "   'offset': 29538,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequencing large genomic sections with long read capabilities of the MinION will increase the size and diversity of sequence databases and improve annotation of sparsely annotated loci through gene mapping, and annotation. This will improve application of current and future genomic data for taxonomic, phylogenetic, metabolic or population studies. For example, currently the COI and NADH2 barcodes are well curated within the Genbank and BOLD databases for chondrichthyan species, but availability is sparse for other mitochondrial loci such as the 12s and 16s rRNA genes and many nuclear loci including ITS2. Therefore, taxonomic identification, phylogenetic and population studies in Chondrichthyes often rely on COI and NADH2 assessments, limiting the accuracy and depth of these analyses. Increased ease of Genome skimming or whole genome sequencing, using our methods will make larger regions of chondrichthyan genomes accessible and will help populate genetic databases. This approach will increase accuracy of phylogenetic, population and kinship assessments and will significantly reduce data deficiencies in Chondrichthyes for conservation and management purposes.',\n",
       "   'offset': 30440,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Analyses',\n",
       "   'offset': 31616,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We compared efficiency and expediency of two different workflows (Fig.\\xa02), one with a GUI (Workflow I) and the other command line based (Workflow II), for sequence analyses of MinION data. Mitochondrial sequences obtained from both workflows were 98% identical, differing by 400\\u2009bp in length. Phylogenetic analyses with both sequences resulted in an identical taxonomic classification of the unknown sample. Minor differences in mitochondrial sequences could be attributed to differences in the mapping and assembly algorithms underlying the assembly tools used in the two workflows. Workflow II also used more rigorous quality controls and trimmed sequences before assembly which may have led to the loss of some sequence base pairs.',\n",
       "   'offset': 31625,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The MinION Workflow I we developed allowed quick access to mitochondrial genomes, while Workflow II allowed access to nuclear genetic loci, phylogenetic analyses of which corroborated results from the mitochondrial analyses and allowed further investigation into phylogeny of the genus. Since mitochondrial genomes from Workflow I and II were comparable for taxonomic identification, we recommend using Workflow 1 (Fig.\\xa02) for field studies with an urgent timeline. Here a database of chondrichthyan sequences from BOLD and Genbank is first curated in Geneious for offline access. DNA sequences from unknown specimens are obtained from the MinION and directly mapped to reference mitochondrial sequences using Geneious software. The mapped mitochondrial sequences are then aligned with reference sequences in the offline databases for taxonomic identification making it as streamlined as upload and align sequence to identify species. The use of multiple mitochondrial barcodes ensures the highest level of accuracy. These analyses are straightforward and do not require internet connectivity, or extensive bioinformatics capabilities or computing power. Workflow I can be used to train researchers in accurate taxonomic identification of species without using computer programing.',\n",
       "   'offset': 32363,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Workflow II can be used in advanced stages of analyses not limited by computing power, bioinformatics expertise and time. The nuclear and mitochondrial sequences obtained from Workflow II can be used to corroborate results obtained using just mitochondrial makers in Workflow I and to obtain finer phylogenetic resolution among species. Population genetic studies may also be undertaken at this stage.',\n",
       "   'offset': 33646,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Phylogenetics',\n",
       "   'offset': 34048,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Our phylogenetic analyses using mitochondrial genomes and nuclear loci of the genus Carcharhinus consistently support placement of the unknown sample as sister to C. falciformis (Fig.\\xa05A,B) and demonstrate the utility of genome skimming techniques for taxonomic identification of unknown samples using multiple genetic loci. Additionally, analyses of mitochondrial and nuclear data are congruent with previous work and strongly support sister species relationships of sampled taxa throughout the genus (Fig.\\xa06). Albeit our analyses allow assessment of sister species relationship with denser sampling of taxa throughout the Carcharhinus genus.',\n",
       "   'offset': 34062,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The inclusion of multiple mitochondrial genes and nuclear loci provided higher statistical support and further evidence that P. glauca is nested well within Carcharhinus, advancing previous work in the genus that used fewer markers and subsequently had weaker statistical support. Our nuclear and mitochondrial data support a sister relationship between P. glauca and C. amblyrhynchos with C. falciformis as outgroup. These data exemplify the statistical power achieved using diverse genetic and intergenic markers which were accessible in the current study from using a genome-wide sequencing approach.',\n",
       "   'offset': 34708,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Deeper nodes relating four taxa-C. falciformis, P. glauca, C. amblyrhynchos and C. wheeleri are not well supported in previous studies or in our analyses (<0.95 PP), because of mito-nuclear discordance, missing data, and lack of coverage of mitochondrial loci. As Bayesian phylogenetic estimation techniques are often robust to missing data in a concatenated alignment, we think that a substantial amount of missing data, as seen with Chondrichthyes, has led to poor support for these phylogenetic analyses. This once again, highlights the need for further genomic analysis. In addition, Carcharhinus spp. are known to contain cryptic species and thus increasing the genomic databases for these species to include regions such as Ultra Conservative Elements (UCE) will be beneficial to understanding the genetic diversity of the genus. In our concatenated analysis, the unknown sample was consistently represented on a very long branch (Fig.\\xa06). This stems from difficulties in aligning the long segment of the unknown sample ITS2 sequence obtained from the MinION to the ITS2 sequences available on GenBank which were much shorter and thus had limited species’ information. In the final concatenated alignment, the unknown sample ITS2 sequence showed numerous insertions not present in other samples. These insertions in the MinION derived sequences are potential evidence of sequence information that is missing in the shorter PCR derived Genbank sequences which may be restricted by polymerase processing rates, amplicon length restrictions or PCR artifacts resulting from repeat regions of template sequences such as the ITS2 region. MinION derived sequences are contiguous long reads of actual genomic DNA which does not go through an amplification cycle but is simply denatured and sequenced, and hence the unknown sample ITS2 sequences is not affected by amplification biases or restrictions of amplicon size, thus providing a higher quality analysis.',\n",
       "   'offset': 35312,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus falciformis',\n",
       "   'offset': 37274,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus falciformis or silky shark is a pelagic species found in tropical waters in the Indian, Pacific, and Atlantic oceans. This species is highly migratory, and has low fecundity, placing it at risk from incidental take in high seas fisheries. Threats to the species are exacerbated by demand for meat and fins in the international market. Despite some regional prohibitions, silky shark mortality is under-reported and largely unmanaged. Inclusion of the silky shark in CITES Appendix II is likely to bolster compliance with existing protections, complement existing commitments under the Convention on Migratory Species (CMS) and facilitate international cooperation toward more comprehensive international conservation measures, thereby enhancing the chances for sustainable use. However, CITES listing and other measures are ineffective if the trade in this species cannot be detected in international fin exports. Methods and results presented here increase our ability to study silky sharks and increase traceability of this species in the fin trade.',\n",
       "   'offset': 37299,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A single complete mitochondrial genome sequence of 17,639\\u2009bp for Carcharhinus falciformis has been published. A suspicious insertion of ~939\\u2009bp is evident when the sequence is compared to all Carcharinus spp. mitogenomes, which are at ~16,700\\u2009bp. The mitochondrial sequence for Carcharhinus falciformis presented in the current report (Fig.\\xa04, Genbank accession #MK092088) provides the first complete and accurate mitochondrial sequence for the species. Further, the investigation into the nuclear and mitochondrial sequences presented in the current study provided better phylogenetic resolution for the Carcharhinus genus. The immunoglobulin and endocrine function encoding nuclear sequences will provide an understanding of evolutionary relationships, immune function development and physiology of the species in future studies.',\n",
       "   'offset': 38364,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Conclusions',\n",
       "   'offset': 39203,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We have bridged substantial knowledge gaps in chondrichthyan genetics through work presented in this report. Use of the MinION in the field is a potential game-changer because it can obtain vast amounts of data in areas with high data deficiency and high fishing effort with respect to Chondrichthyes. It can facilitate accurate identification and ecological investigation of existing as well as new and rare species, without the need for a priori’ sequence information. These methods will improve the accuracy of biodiversity estimates in areas with minimal infrastructure, thus addressing the data deficiency and knowledge gaps affecting Chondrichthyes where data is most needed. Sequencing of larger regions of chondrichthyan genomes as opposed to single barcode sequencing will allow wider ecological investigations for all Chondrichthyes such as population genetic studies, stock assessments, biogeographic and kinship studies, thus increasing the depth of our understanding of chondrichthyan populations. The whole genome provides insights into shark metabolism and health status. Data regarding population structure within geographically isolated populations of the same species can allow assessment of population stocks and inform conservation and management of the species. Last, the genome sequencing and analyses methods presented in the current report are independent of PCR bias, and have sequencing capacity across a wide GC %, making them amenable to ecological investigations on a variety of non-model species. The workflows presented here, can be used effectively to build capacity from molecular ecology research among wildlife researchers and fisheries scientists in a variety of infrastructure settings.',\n",
       "   'offset': 39215,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Methods',\n",
       "   'offset': 40941,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sampling and DNA extraction',\n",
       "   'offset': 40949,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark samples (n\\u2009=\\u200930) consisting of fin tissue ~5\\u2009mm3 in size were collected from dismembered shark specimen at a fish market in Veraval, Gujarat, India in April 2017 (Fig.\\xa01). Samples were stored in DNA/RNA GuardTM Zymoresearch at ambient temperature during collection and then stored at −20\\u2009°C until processing. DNA extraction was conducted using Quick-DNATM Plus extraction kit (ZymoresearchTM) with an extended 3-hour proteinase K digestion at 55\\u2009°C. Of these 30 samples, one was prioritized for analyses with the MinION sequencer and the rest were stored for analyses at a later time.',\n",
       "   'offset': 40977,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Morphological identification',\n",
       "   'offset': 41583,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Photo identification of the specimen was performed using protocols described by Ebert et al. and based on intact features of the dismembered specimen, including head, gills, and fins.',\n",
       "   'offset': 41612,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'PCR Amplification and Analyses',\n",
       "   'offset': 41796,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'PCR amplification of three gene fragments from the Cytochrome Oxidase 1 and NADH2 genetic loci was performed to resolve the taxonomic identity of the specimen at the species level using protocols described previously. PCR amplification of a 633\\u2009bp and 160\\u2009bp fragment of the COI regions was performed using methods described by Ward et al. and Fields et al., and that of a 1044\\u2009bp NADH2 fragment was done using protocols described by Naylor et al.. PCR amplification was followed by PCR clean up with ZymoresearchTM DNA Clean and Concentrator Kit and uni-directional sequencing using BigDye Terminator chemistry on an ABI 3730xl genetic analyzer (Applied Biosystems, Life Technologies). The resulting sequence was used to determine the taxonomic identity of the chondrichthyan specimen using the Barcode of Life Database (BOLD).',\n",
       "   'offset': 41827,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Library preparation and Sequencing on the MinION',\n",
       "   'offset': 42662,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to access the entire length of multiple nuclear and mitochondrial genomes in a PCR-independent fashion, genome skimming using the MinION device was performed. The extracted DNA was measured using a QubitTM spectrophotometer. Genomic DNA (~400\\u2009ng) was used directly for library preparation with the Rapid Sequencing kit (SQK-RAD003) from ONT using the manufacturer’s protocols. The library preparation and sample loading took 15–30\\u2009minutes in total and required access to a thermocycler or hot water bath incubator, pipettes and appropriate tips (1000\\u2009μl, 200\\u2009μl, and 10\\u2009μl), and molecular grade water. Bound DNA was loaded directly into the sample loading port of the FLO-MIN106 (R9.4 SpotON) flowcells after priming of the flowcells. Sequencing was continued for 36\\u2009hours, however, sequences acquired were monitored throughout the 36\\xa0\\u2009hours and mitochondrial barcodes could be obtained starting at 8\\u2009hours from start of sequencing.',\n",
       "   'offset': 42711,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bioinformatics Analyses',\n",
       "   'offset': 43677,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequence data were analyzed using two parallel workflows (Fig.\\xa02). For the first workflow, FASTQ files were uploaded into Geneious® software version 11.1.2. Mitochondrial sequences were mapped directly to reference chondrichthyan mitochondrial sequences from Genbank® using Geneious Read mapper and the resulting contig was used to obtain a contiguous mitochondrial sequence for the sample analyzed.',\n",
       "   'offset': 43701,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'For the second workflow, reads were assembled using CANU (Version 1.7) with flags that set the genome size as 16\\u2009kb, trimmed the minimum read length to 500\\u2009bp, and set the reads as coming from the MinION. Contigs were aligned using BLAST against the nt nucleotide database to identify mitochondrial and nuclear sequences. Using CANU reads from a circular genome assemble into a linear contig larger than the original genome with repeated sequences at both ends. Therefore, the mitochondrial genome was circularized using Samtools (version 1.8) and Nucmer (version 4.0.0 beta2) was used to identify repeated sequences in the mitochondrial contigs. Gepard (version 1.4) was used to produce dotplots and verify the circularization. This workflow is available in our associated Jupyter notebook: https://github.com/Adrian-Cantu/git-presentations/blob/master/shark_mito/Mito_mapping.ipynb.',\n",
       "   'offset': 44104,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mitochondrial contigs from both workflows were annotated using Mitos, a web-based server for de novo annotation of mitochondrial genomes, and Mito-annotator (Mitofish database), a tool for annotation of fish mitochondrial genomes. tRNA identification was conducted by ARWEN — a program to detect tRNAs in metazoan mitochondrial genomes. CGView was used to determine GC rich regions and GC skew.',\n",
       "   'offset': 44993,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Taxonomic identification and Phylogenetics',\n",
       "   'offset': 45390,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We investigated the taxonomic identity of our sample using whole mitochondrial genomes and nuclear DNA (ITS2) comparisons to previously sequenced Carcharhinus from the NCBI database. We reconstructed phylogenetic relationships within the genus Carcharhinus in parallel, first using complete mitochondrial genomes when available (n\\u2009=\\u200914), and second, using concatenated sequences for three protein-coding mtDNA genes (COX1, NADH2, and ND4), and one nDNA locus (ITS2), (n\\u2009=\\u200933 species within Carcharhinus). Galeocerdo cuvier and Prionace glauca were used as the outgroup in the two phylogenetic analyses. Sequences were acquired from GenBank (Supplementary Tables\\xa0S1, S2 and S3) and aligned using MUSCLE 3.8.31. The mitochondrial alignments were done in two different ways, using the 13 protein-coding regions of the mitochondrial genome or the complete mitogenome including coding and non-coding regions. Alignments were partitioned by gene and codon position. We used PartitionFinder 2.1.1 on XSEDE tool on CIPRES to identify the appropriate partitioning schemes and models of substitution using the greedy search algorithm. Maximum-likelihood phylogenies were estimated in RAxML-HPC v8.2.10 on XSEDE run on CIPRES Scientific Gateway using a separate GTR\\u2009+\\u2009Γ model specified for all partitions. We ran 10 searches beginning with different random stepwise addition parsimony trees with the default RAxML algorithm. Support values for the best tree were estimated using non-parametric bootstrapping implemented using the auto-MRE stopping criterion. We estimated Bayesian phylogenies in MrBayes v3.2.6, on XSEDE run on CIPRES Scientific Gateway using the partitioning scheme specified by PartitionFinder2. We used unlinked GTR substitution model across partitions and accommodated rate heterogeneity among sites by using a Γ distribution. We assessed convergence in Tracer v1.6.',\n",
       "   'offset': 45433,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information',\n",
       "   'offset': 47325,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Publisher’s note: Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.',\n",
       "   'offset': 47351,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'footnote', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information',\n",
       "   'offset': 47490,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information accompanies this paper at 10.1038/s41598-019-40940-9.',\n",
       "   'offset': 47516,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Author Contributions',\n",
       "   'offset': 47596,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'S.J. contributed to concept, sampling, sequencing, analyses and wrote the manuscript, J.S. contributed to sampling, DNA extractions and sequencing, A.C. and R.E. designed and performed Bioinformatic analyses, S.F. did the phylogenetic analyses, I.M. contributed to PCR and BOLD sequence alignments, A.V. helped with establishing social networks, understanding fisheries practices and sampling logistics and E.D. contributed to concept and editing of the manuscript.',\n",
       "   'offset': 47617,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Data Availability',\n",
       "   'offset': 48083,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'All data generated or analyzed during this study are included in this published article (and its Supplementary Information files).',\n",
       "   'offset': 48101,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Competing Interests',\n",
       "   'offset': 48232,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'COMP_INT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The authors declare no competing interests.',\n",
       "   'offset': 48252,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'COMP_INT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'References',\n",
       "   'offset': 48296,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Dent, F. & Clarke, S. State of the global market for shark products. 196 (2015).',\n",
       "   'offset': 48307,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Extinction risk and conservation of the world’s sharks and rays',\n",
       "   'offset': 48388,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Dulvy;given-names:NK',\n",
       "    'source': 'eLife',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '3',\n",
       "    'pub-id_pmid': '24448405',\n",
       "    'fpage': 'e00590',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.7554/eLife.00590'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Jabado, R. W. et al. Troubled waters: Threats and extinction risk of the sharks, rays and chimaeras of the Arabian Sea and adjacent waters. Fish Fish. 19, 1043–1062 (2018).',\n",
       "   'offset': 48454,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lack, M. & Sant, G. Trends in global shark catch and recent developments in management. TRAFFIC Int. 33 (2009).',\n",
       "   'offset': 48629,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A review of the application of molecular genetics for fisheries management and conservation of sharks and rays',\n",
       "   'offset': 48741,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '80',\n",
       "    'name_0': 'surname:Dudgeon;given-names:CL',\n",
       "    'source': 'J. Fish Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1843',\n",
       "    'pub-id_pmid': '22497408',\n",
       "    'fpage': '1789',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/j.1095-8649.2012.03265.x'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DNA barcoding Australia’s fish species',\n",
       "   'offset': 48852,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Innes;given-names:BH',\n",
       "    'name_3': 'surname:Last;given-names:PR',\n",
       "    'name_0': 'surname:Ward;given-names:RD',\n",
       "    'name_1': 'surname:Zemlak;given-names:TS',\n",
       "    'name_4': 'surname:Hebert;given-names:PD',\n",
       "    'volume': '360',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1857',\n",
       "    'source': 'Philos. Trans. R. Soc. B Biol. Sci.',\n",
       "    'fpage': '1847',\n",
       "    'year': '2005',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1098/rstb.2005.1716'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A multiplex PCR mini-barcode assay to identify processed shark products in the global trade',\n",
       "   'offset': 48893,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Cardeñosa;given-names:D',\n",
       "    'source': 'PLOS ONE',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '12',\n",
       "    'pub-id_pmid': '29020095',\n",
       "    'fpage': 'e0185368',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0185368'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A Novel Mini-DNA Barcoding Assay to Identify Processed Fins from Internationally Protected Shark Species',\n",
       "   'offset': 48985,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Eng;given-names:R',\n",
       "    'name_3': 'surname:Feldheim;given-names:K',\n",
       "    'name_0': 'surname:Fields;given-names:AT',\n",
       "    'name_1': 'surname:Abercrombie;given-names:DL',\n",
       "    'name_4': 'surname:Chapman;given-names:DD',\n",
       "    'source': 'PLOS ONE',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '10',\n",
       "    'pub-id_pmid': '25646789',\n",
       "    'fpage': 'e0114844',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0114844'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Naylor, G. J. P. et al. A DNA sequence-based approach to the identification of shark and ray species and its implications for global elasmobranch diversity and parasitology. (Bulletin of the American Museum of Natural History, no. 367). DNA identification of sharks and rays (2012).',\n",
       "   'offset': 49090,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hoelzel, A. Conservation Genetics2 69, 10.1023/A:1011590517389 (2001).',\n",
       "   'offset': 49373,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Identification and phylogenetic inferences on stocks of sharks affected by the fishing industry off the Northern coast of Brazil',\n",
       "   'offset': 49444,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Rodrigues-Filho;given-names:LF',\n",
       "    'source': 'Genet. Mol. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '413',\n",
       "    'pub-id_pmid': '21637699',\n",
       "    'fpage': '405',\n",
       "    'year': '2009',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1590/S1415-47572009005000039'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequence analysis of mitochondrial 12S rRNA gene can identify meat species',\n",
       "   'offset': 49573,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '66',\n",
       "    'name_0': 'surname:Girish;given-names:PS',\n",
       "    'source': 'Meat Sci.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '556',\n",
       "    'pub-id_pmid': '22060864',\n",
       "    'fpage': '551',\n",
       "    'year': '2004',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/S0309-1740(03)00158-X'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Phylogenetic relationships among the major lineages of modern elasmobranchs',\n",
       "   'offset': 49648,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Fedrigo;given-names:O',\n",
       "    'name_3': 'surname:Lopez;given-names:JA',\n",
       "    'name_0': 'surname:Naylor;given-names:GJ',\n",
       "    'name_1': 'surname:Ryburn;given-names:JA',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '3',\n",
       "    'source': 'Reprod. Biol. Phylogeny',\n",
       "    'fpage': '25',\n",
       "    'year': '2005',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A streamlined DNA tool for global identification of heavily exploited coastal shark species (genus Rhizoprionodon)',\n",
       "   'offset': 49724,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Nachtigall;given-names:PG',\n",
       "    'name_3': 'surname:Chapman;given-names:DD',\n",
       "    'name_0': 'surname:Pinhal;given-names:D',\n",
       "    'name_1': 'surname:Shivji;given-names:MS',\n",
       "    'name_4': 'surname:Martins;given-names:C',\n",
       "    'source': 'PloS One',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '7',\n",
       "    'pub-id_pmid': '22496864',\n",
       "    'fpage': 'e34797',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0034797'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Accuracy of species identification by fisheries observers in a north Australian shark fishery',\n",
       "   'offset': 49839,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '127–128',\n",
       "    'name_0': 'surname:Tillett;given-names:BJ',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '115',\n",
       "    'source': 'Fish. Res.',\n",
       "    'fpage': '109',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.fishres.2012.04.007'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Checklist of Chondrichthyans in Indian waters',\n",
       "   'offset': 49933,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '56',\n",
       "    'name_0': 'surname:Akhilesh;given-names:KV',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '120',\n",
       "    'source': 'J. Mar. Biol. Assoc. India',\n",
       "    'fpage': '109',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.6024/jmbai.2014.56.1.01750s-17'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Environmental DNA reveals tropical shark diversity in contrasting levels of anthropogenic impact',\n",
       "   'offset': 49979,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Bakker;given-names:J',\n",
       "    'source': 'Sci. Rep.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '7',\n",
       "    'pub-id_pmid': '29203793',\n",
       "    'fpage': '16886',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/s41598-017-17150-2'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mitochondrial cytochrome b variation in sleeper sharks (Squaliformes: Somniosidae)',\n",
       "   'offset': 50076,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '153',\n",
       "    'name_0': 'surname:Murray;given-names:BW',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1022',\n",
       "    'source': 'Mar. Biol.',\n",
       "    'fpage': '1015',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1007/s00227-007-0871-1'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Benz, G. W. et al. First Record of a Sleeper Shark in the Western Gulf of Mexico and Comments on Taxonomic Uncertainty Within Somniosus (Somniosus). Bulletin of Marine Science80, 343–351 (2007).',\n",
       "   'offset': 50159,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DRI busts illegal exports of shark fin from Maharashtra and Gujarat. The Indian Express (2018).',\n",
       "   'offset': 50356,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'van. Genome-wide SNPs reveal low effective population size within confined management units of the highly vagile Galapagos shark (Carcharhinus galapagensis)',\n",
       "   'offset': 50452,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Simpfendorfer;given-names:CA',\n",
       "    'name_3': 'surname:Salinas-de-León;given-names:P',\n",
       "    'name_0': 'surname:Pazmiño;given-names:DA',\n",
       "    'name_1': 'surname:Maes;given-names:GE',\n",
       "    'name_4': 'surname:Herwerden;given-names:L',\n",
       "    'volume': '18',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1163',\n",
       "    'source': 'Conserv. Genet.',\n",
       "    'fpage': '1151',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1007/s10592-017-0967-1'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Pazmiño, D. et al. Strong trans-Pacific break and local conservation units in the Galapagos shark (Carcharhinus galapagensis) revealed by genome-wide cytonuclear markers. Heredity120 (2018).',\n",
       "   'offset': 50609,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Selection and sex-biased dispersal in a coastal shark: The influence of philopatry on adaptive variation',\n",
       "   'offset': 50801,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Portnoy;given-names:D',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '24',\n",
       "    'source': 'Mol. Ecol.',\n",
       "    'fpage': '5885',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/mec.13441'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DNA capture reveals transoceanic gene flow in endangered river sharks',\n",
       "   'offset': 50906,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '112',\n",
       "    'name_0': 'surname:Li;given-names:C',\n",
       "    'source': 'Proc. Natl. Acad. Sci.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '13307',\n",
       "    'pub-id_pmid': '26460025',\n",
       "    'fpage': '13302',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1073/pnas.1508735112'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Harnessing the Power of Genomics to Secure the Future of Seafood',\n",
       "   'offset': 50976,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Bernatchez;given-names:L',\n",
       "    'source': 'Trends Ecol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '680',\n",
       "    'pub-id_pmid': '28818341',\n",
       "    'fpage': '665',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.tree.2017.06.010'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus - Nucleotide - NCBI. Available at, https://www.ncbi.nlm.nih.gov/nuccore (Accessed: 8th September 2018).',\n",
       "   'offset': 51041,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The complete mitochondrial DNA of the silky shark (Carcharhinus falciformis)',\n",
       "   'offset': 51158,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Diaz-Jaimes;given-names:P',\n",
       "    'name_3': 'surname:Marcet-Houben;given-names:M',\n",
       "    'name_0': 'surname:Galván-Tirado;given-names:C',\n",
       "    'name_1': 'surname:Hinojosa-Alvarez;given-names:S',\n",
       "    'name_4': 'surname:García-De-León;given-names:FJ',\n",
       "    'volume': '27',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '158',\n",
       "    'source': 'Mitochondrial DNA Part DNA Mapp. Seq. Anal.',\n",
       "    'fpage': '157',\n",
       "    'year': '2016',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MITOS: improved de novo metazoan mitochondrial genome annotation',\n",
       "   'offset': 51235,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '69',\n",
       "    'name_0': 'surname:Bernt;given-names:M',\n",
       "    'source': 'Mol. Phylogenet. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '319',\n",
       "    'pub-id_pmid': '22982435',\n",
       "    'fpage': '313',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.ympev.2012.08.023'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MitoFish and MitoAnnotator: a mitochondrial genome database of fish with an accurate and automatic annotation pipeline',\n",
       "   'offset': 51300,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '30',\n",
       "    'name_0': 'surname:Iwasaki;given-names:W',\n",
       "    'source': 'Mol. Biol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2540',\n",
       "    'pub-id_pmid': '23955518',\n",
       "    'fpage': '2531',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/molbev/mst141'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'ARWEN: a program to detect tRNA genes in metazoan mitochondrial nucleotide sequences',\n",
       "   'offset': 51419,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '24',\n",
       "    'name_0': 'surname:Laslett;given-names:D',\n",
       "    'name_1': 'surname:Canbäck;given-names:B',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '175',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '172',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btm573'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Geneious Basic: an integrated and extendable desktop software platform for the organization and analysis of sequence data',\n",
       "   'offset': 51504,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '28',\n",
       "    'name_0': 'surname:Kearse;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1649',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '1647',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/bts199'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The effects of fishing on sharks, rays, and chimaeras (chondrichthyans), and the implications for marine ecosystems',\n",
       "   'offset': 51626,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '57',\n",
       "    'name_0': 'surname:Stevens;given-names:J',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '494',\n",
       "    'source': 'ICES J. Mar. Sci.',\n",
       "    'fpage': '476',\n",
       "    'year': '2000',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1006/jmsc.2000.0724'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The importance of habitat and life history to extinction risk in sharks, skates, rays and chimaeras',\n",
       "   'offset': 51742,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Myers;given-names:RA',\n",
       "    'volume': '275',\n",
       "    'name_0': 'surname:García;given-names:VB',\n",
       "    'name_1': 'surname:Lucifora;given-names:LO',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '89',\n",
       "    'source': 'Proc. R. Soc. Lond. B Biol. Sci.',\n",
       "    'fpage': '83',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1098/rspb.2007.1295'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Elephant shark genome provides unique insights into gnathostome evolution',\n",
       "   'offset': 51842,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '505',\n",
       "    'name_0': 'surname:Venkatesh;given-names:B',\n",
       "    'source': 'Nature',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '179',\n",
       "    'pub-id_pmid': '24402279',\n",
       "    'fpage': '174',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/nature12826'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Draft sequencing and assembly of the genome of the world’s largest fish, the whale shark: Rhincodon typus Smith 1828',\n",
       "   'offset': 51916,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Read;given-names:TD',\n",
       "    'source': 'BMC Genomics',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '18',\n",
       "    'pub-id_pmid': '28709399',\n",
       "    'fpage': '532',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1186/s12864-017-3926-9'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark genomes provide insights into elasmobranch evolution and the origin of vertebrates',\n",
       "   'offset': 52035,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Hara;given-names:Y',\n",
       "    'source': 'Nat. Ecol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '2',\n",
       "    'pub-id_pmid': '30297745',\n",
       "    'fpage': '1761',\n",
       "    'year': '2018',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/s41559-018-0673-5'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Parker, J., Helmstetter, A. J., Devey, D., Wilkinson, T. & Papadopulos, A. S. T. Field-based species identification of closely-related plants using real-time nanopore sequencing. Sci. Rep. 7 (2017).',\n",
       "   'offset': 52124,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Characterizing and measuring bias in sequence data',\n",
       "   'offset': 52323,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Ross;given-names:MG',\n",
       "    'source': 'Genome Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '14',\n",
       "    'pub-id_pmid': '23718773',\n",
       "    'fpage': 'R51',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1186/gb-2013-14-5-r51'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Structure, function and diversity of the healthy human microbiome',\n",
       "   'offset': 52374,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '486',\n",
       "    'name_0': 'surname:Consortium;given-names:THMP',\n",
       "    'source': 'Nature',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '214',\n",
       "    'pub-id_pmid': '22699609',\n",
       "    'fpage': '207',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/nature11234'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequencing at sea: challenges and experiences in Ion Torrent PGM sequencing during the 2013 Southern Line Islands Research Expedition',\n",
       "   'offset': 52440,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Lim;given-names:YW',\n",
       "    'source': 'PeerJ',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '2',\n",
       "    'pub-id_pmid': '25177534',\n",
       "    'fpage': 'e520',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.7717/peerj.520'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': ' NCBI Resource Coordinators. Database Resources of the National Center for Biotechnology Information. Nucleic Acids Res. 45, D12–D17 (2017).',\n",
       "   'offset': 52574,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bold: The Barcode of Life Data System, (http://www.barcodinglife.org). Available at, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1890991/. (Accessed: 28th April 2018).',\n",
       "   'offset': 52717,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark tales: a molecular species-level phylogeny of sharks (Selachimorpha, Chondrichthyes)',\n",
       "   'offset': 52886,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '58',\n",
       "    'name_0': 'surname:Vélez-Zuazo;given-names:X',\n",
       "    'name_1': 'surname:Agnarsson;given-names:I',\n",
       "    'source': 'Mol. Phylogenet. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '217',\n",
       "    'pub-id_pmid': '21129490',\n",
       "    'fpage': '207',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.ympev.2010.11.018'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Missing data in phylogenetic analysis: reconciling results from simulations and empirical data',\n",
       "   'offset': 52977,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '60',\n",
       "    'name_0': 'surname:Wiens;given-names:JJ',\n",
       "    'name_1': 'surname:Morrill;given-names:MC',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '731',\n",
       "    'pub-id_pmid': '21447483',\n",
       "    'fpage': '719',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/syr025'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A novel field method to distinguish between cryptic carcharhinid sharks, Australian blacktip shark Carcharhinus tilstoni and common blacktip shark C. limbatus, despite the presence of hybrids',\n",
       "   'offset': 53072,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '90',\n",
       "    'name_0': 'surname:Johnson;given-names:GJ',\n",
       "    'source': 'J. Fish Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '60',\n",
       "    'pub-id_pmid': '27774596',\n",
       "    'fpage': '39',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/jfb.13102'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Towards better management of Australia’s shark fishery: genetic analyses reveal unexpected ratios of cryptic blacktip species Carcharhinus tilstoni and C. limbatus',\n",
       "   'offset': 53264,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kashiwagi;given-names:T',\n",
       "    'name_3': 'surname:Broderick;given-names:D',\n",
       "    'name_0': 'surname:Ovenden;given-names:JR',\n",
       "    'name_1': 'surname:Morgan;given-names:JAT',\n",
       "    'name_4': 'surname:Salini;given-names:J',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '61',\n",
       "    'source': 'Mar. Freshw. Res.',\n",
       "    'fpage': '253',\n",
       "    'year': '2010',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1071/MF09151'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Ultraconserved elements anchor thousands of genetic markers spanning multiple evolutionary timescales',\n",
       "   'offset': 53430,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '61',\n",
       "    'name_0': 'surname:Faircloth;given-names:BC',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '726',\n",
       "    'pub-id_pmid': '22232343',\n",
       "    'fpage': '717',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/sys004'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IUCN 2017. The IUCN Red List of Threatened Species. Version 2017-2, http://www.iucnredlist.org. Downloaded on 25 November 2017. Available at, http://www.iucnredlist.org/ (Accessed: 26th November 2017).',\n",
       "   'offset': 53532,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'History of CITES listing of sharks (Elasmobranchii) | CITES. Available at, https://www.cites.org/eng/prog/shark/history.php (Accessed: 21st August 2018).',\n",
       "   'offset': 53734,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lack, M. & Sant, G. The Future of Sharks: A Review of Action and Inaction (2011).',\n",
       "   'offset': 53888,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Ebert, D., Fowler, S. & Compagno, L. Sharks of the World: A Fully Illustrated Guide (2013).',\n",
       "   'offset': 53970,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Kearse, M., Kearse, M., Sturrock, S. & Meintjes, P. The Geneious 6.0.3 Read Mapper. 16 (2012).',\n",
       "   'offset': 54062,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation',\n",
       "   'offset': 54157,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '27',\n",
       "    'name_0': 'surname:Koren;given-names:S',\n",
       "    'source': 'Genome Res.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '736',\n",
       "    'pub-id_pmid': '28298431',\n",
       "    'fpage': '722',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1101/gr.215087.116'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Madden, T. The BLAST Sequence Analysis Tool. (National Center for Biotechnology Information (US), 2003).',\n",
       "   'offset': 54255,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The Sequence Alignment/Map format and SAMtools',\n",
       "   'offset': 54360,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '25',\n",
       "    'name_0': 'surname:Li;given-names:H',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2079',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '2078',\n",
       "    'year': '2009',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btp352'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MUMmer4: A fast and versatile genome alignment system. Available at, http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005944. (Accessed: 21st August 2018).',\n",
       "   'offset': 54407,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Gepard: a rapid and sensitive tool for creating dotplots on genome scale',\n",
       "   'offset': 54585,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Rattei;given-names:T',\n",
       "    'volume': '23',\n",
       "    'name_0': 'surname:Krumsiek;given-names:J',\n",
       "    'name_1': 'surname:Arnold;given-names:R',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1028',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '1026',\n",
       "    'year': '2007',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btm039'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Circular genome visualization and exploration using CGView',\n",
       "   'offset': 54658,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '21',\n",
       "    'name_0': 'surname:Stothard;given-names:P',\n",
       "    'name_1': 'surname:Wishart;given-names:DS',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '539',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '537',\n",
       "    'year': '2005',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/bti054'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MUSCLE: multiple sequence alignment with high accuracy and high throughput',\n",
       "   'offset': 54717,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Edgar;given-names:RC',\n",
       "    'source': 'Nucleic Acids Res.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1797',\n",
       "    'pub-id_pmid': '15034147',\n",
       "    'fpage': '1792',\n",
       "    'year': '2004',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/nar/gkh340'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lanfear, R., et al. PartitionFinder 2: New methods for selecting partitioned models of evolution for molecular and morphological phylogenetic analyses. Mol. Biol. Evol. 34 (2016).',\n",
       "   'offset': 54792,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Creating the CIPRES Science Gateway for Inference of Large Phylogenetic Trees',\n",
       "   'offset': 54972,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Schwartz;given-names:T',\n",
       "    'volume': '14',\n",
       "    'name_0': 'surname:Miller;given-names:M',\n",
       "    'name_1': 'surname:Pfeiffer;given-names:WT',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '8',\n",
       "    'source': 'In Proceedings of the Gateway Computing Environments Workshop: 14 Nov 2010',\n",
       "    'fpage': '1',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'RAxML-VI-HPC: maximum likelihood-based phylogenetic analyses with thousands of taxa and mixed models',\n",
       "   'offset': 55050,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '22',\n",
       "    'name_0': 'surname:Stamatakis;given-names:A',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2690',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '2688',\n",
       "    'year': '2006',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btl446'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MRBAYES: Bayesian inference of phylogenetic trees',\n",
       "   'offset': 55151,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '17',\n",
       "    'name_0': 'surname:Huelsenbeck;given-names:JP',\n",
       "    'name_1': 'surname:Ronquist;given-names:F',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '755',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '754',\n",
       "    'year': '2001',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/17.8.754'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MrBayes 3.2: Efficient Bayesian Phylogenetic Inference and Model Choice Across a Large Model Space',\n",
       "   'offset': 55201,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '61',\n",
       "    'name_0': 'surname:Ronquist;given-names:F',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '542',\n",
       "    'pub-id_pmid': '22357727',\n",
       "    'fpage': '539',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/sys029'},\n",
       "   'sentences': [],\n",
       "   'annotations': []}],\n",
       " 'infons': {'license': 'CC BY'},\n",
       " 'id': '6418218',\n",
       " 'relations': []}"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmcid_to_doc['6418218']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Images were processed and analyzed with ImageJ Fiji v1.50c. Demographs and line plots (Fig.\\xa01 and Fig.\\xa0S6) were constructed by manually measuring the fluorescence intensity profiles in Fiji and processing the data in R (version 3.3.1 [http://www.r-project.org]), with the cell profiles script ( [http://github.com/ta-cameron/cell-profiles]) and ggplot2 package (version 2.1.0; Hadley Wickham, Department of Statistics, Rice University [https://ggplot2.tidyverse.org/]). Intensity profiles either were oriented with the fluorescent cell pole toward the left (E. coli and R. sphaeroides), or random sorting was applied (M. gryphiswaldense and R. rubrum). In the case of C. crescentus, the stalk was used as the old pole reference. To confidently and accurately identify stalks, FM4-64 staining (which in our hands did not always stain all cells) and DAPI staining (which we found to unspecifically stain stalks) were used, in addition to DIC microscopy.']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t['text'] for t in pmcid_to_doc['6414705']['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The fMRI imaging analysis steps are outlined in Fig. 1 The first 5 time frames (15\\u200as) were removed to allow the MR signal to achieve T1 equilibrium. Time frames were slice-timing corrected, realigned to the mean echo-planar image using SPM12 (http://www.fil.ion.ucl.ac.uk/spm/), co-registered to the subject T1 space and then normalized to the standard MNI-152 2mm-template using Advanced Normalization Tools software (http://stnava.github.io/ANTs/). Signals from subject white matter and cerebrospinal fluid (3-mm cubes centered at Montreal Neurological Institute coordinates (MNI) (26, –12, 35) and (19, –33, 18) as these locations are away from the grey matter), as well as six head motion parameters were regressed out from each dataset. Global signal regression was not performed as it is controversial in resting state fMRI data preprocessing. Correction for global signal fluctuations with regression has a significant impact on resting state functional connectivity results and can further introduce “artifactual” anti-correlations. Furthermore, global signal includes both neural and non-neural variabilities and therefore makes the result difficult to interpret. FMRI data were spatially smoothed using an 8\\u200amm 3D-Gaussian filter. All voxel time courses were band pass filtered (0.008\\u200aHz<f<0.1\\u200aHz) and variance normalized. The mean signal noise ratio, an indicator of fMRI data quality, was not different between groups (F(3,120)\\u200a=\\u200a2.04, p\\u200a=\\u200a0.11).']"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t['text'] for t in pmcid_to_doc['6294587']['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## See if unique URLs are author repos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "url_counts = Counter(urls_in_papes.url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('broadinstitute.github.io/picard', 1106),\n",
       " ('github.com/najoshi/sickle', 275),\n",
       " ('github.com/jstjohn/SeqPrep', 163),\n",
       " ('github.com', 141),\n",
       " ('github.com/lh3/seqtk', 139),\n",
       " ('transdecoder.github.io', 131),\n",
       " ('cole-trapnell-lab.github.io/cufflinks', 116),\n",
       " ('github.com/alexdobin/STAR', 106),\n",
       " ('stnava.github.io/ANTs', 96),\n",
       " ('github.com/taoliu/MACS', 93),\n",
       " ('github.com/lh3/wgsim', 86),\n",
       " ('trinotate.github.io', 83),\n",
       " ('github.com/vsbuffalo/scythe', 65),\n",
       " ('github.com/broadinstitute/picard', 55),\n",
       " ('networkx.github.io', 54),\n",
       " ('github.com/korseby/container-mtbls520', 49),\n",
       " ('github.com/bulik/ldsc', 47),\n",
       " ('github.com/tseemann/snippy', 40),\n",
       " ('trinityrnaseq.github.io', 38),\n",
       " ('github.com/ekg/freebayes', 36),\n",
       " ('github.com/jdstorey/qvalue', 35),\n",
       " ('github.com/trinityrnaseq/trinityrnaseq/wiki', 34),\n",
       " ('github.com/tseemann/abricate', 34),\n",
       " ('picrust.github.io/picrust', 33),\n",
       " ('github.com/arq5x/bedtools2', 33),\n",
       " ('github.com/genome/bam-readcount', 32),\n",
       " ('gephi.github.io', 31),\n",
       " ('github.com/marcelm/cutadapt', 30),\n",
       " ('github.com/pysam-developers/pysam', 30),\n",
       " ('github.com/fchollet/keras', 29),\n",
       " ('deweylab.github.io/RSEM', 28),\n",
       " ('expressionanalysis.github.io/ea-utils', 27),\n",
       " ('github.com/tseemann/mlst', 27),\n",
       " ('github.com/lh3/bwa', 26),\n",
       " ('github.com/torognes/vsearch', 25),\n",
       " ('github.com/iontorrent/TMAP', 25),\n",
       " ('github.com/rrwick/Porechop', 25),\n",
       " ('tgennorth.github.io/NASP', 23),\n",
       " ('github.com/vcflib/vcflib', 23),\n",
       " ('github.com/widdowquinn/pyani', 23),\n",
       " ('github.com/FelixKrueger/TrimGalore', 22),\n",
       " ('github.com/PacificBiosciences/GenomicConsensus', 21),\n",
       " ('github.com/tanghaibao/goatools', 21),\n",
       " ('github.com/chapmanb/bcbio-nextgen', 21),\n",
       " ('cole-trapnell-lab.github.io/cufflinks/cuffdiff', 21),\n",
       " ('www.github.com', 21),\n",
       " ('samtools.github.io/hts-specs/SAMv1.pdf', 21),\n",
       " ('samtools.github.io/bcftools', 20),\n",
       " ('github.com/ucdavis-bioinformatics/sickle', 20),\n",
       " ('github.com/samtools/samtools', 20)]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "url_counts.most_common(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'github.com/lh3/wgsim': 86,\n",
       " 'github.com/JAvRZ/andi-dataprocessing': 1,\n",
       " 'github.com/AharoniLab/MatchWeiz': 1,\n",
       " 'broadinstitute.github.io/picard': 1106,\n",
       " 'github.com/ijmarshall/picotron': 1,\n",
       " 'github.com/awilfert/PSAP-pipeline': 1,\n",
       " 'gist.github.com/darmitage/3179407': 1,\n",
       " 'github.com/confunguido/PokemonDataAnalysis': 1,\n",
       " 'uhkniazi.github.io/dismiss': 1,\n",
       " 'github.com/alexandrovteam/spatial-corals': 1,\n",
       " 'uubram.github.io/RTCR/along': 1,\n",
       " 'github.com/bulik/ldsc': 47,\n",
       " 'github.com/tanghaibao/treecut/blob/master/scripts/eisen_to_newick.py': 1,\n",
       " 'github.com/aboyle/F-seq': 3,\n",
       " 'www.github.com/rhenley/Pythion': 1,\n",
       " 'github.com/Caleydo/caleydo_clue': 1,\n",
       " 'github.com/Caleydo/pathfinder': 1,\n",
       " 'github.com/GauravPandeyLab': 1,\n",
       " 'github.com/tdsmith/eleven': 1,\n",
       " 'github.com/nmcnulty/COPRO-Seq': 1,\n",
       " 'github.com/rhenley/Pyth-Ion': 4,\n",
       " 'github.com/rgcgithub/clamms': 2,\n",
       " 'github.com/ZWang-Lab/LSKAT': 1,\n",
       " 'github.com/markus-nilsson/md-dmri': 2,\n",
       " 'github.com/divara01/PSB2017_ReproducibilityOfBNs': 1,\n",
       " 'github.com/torognes/swarm': 2,\n",
       " 'github.com/oscar-franzen/oclust': 3,\n",
       " 'github.com/fangly/AmpliCopyrighter': 1,\n",
       " 'joey711.github.io/phyloseq': 8,\n",
       " 'github.com/jasonsahl/wgfast': 3,\n",
       " 'github.com/snayfach/PhyloCNV': 1,\n",
       " 'github.com/voutcn/megahit': 5,\n",
       " 'github.com/BinPro/CONCOCT': 8,\n",
       " 'ecogenomics.github.io/GroopM': 2,\n",
       " 'github.com/CK7/abawaca': 5,\n",
       " 'ecogenomics.github.io/CheckM': 4,\n",
       " 'github.com/snayfach/MicrobeCensus': 3,\n",
       " 'github.com/abhyudaynj/LSTM-CRF-models': 1,\n",
       " 'gist.github.com/svigneau/8846527': 1,\n",
       " 'github.com/ImprovementPathSystems/Measuring_Abnormality_in_High_Dimensional_Spaces': 1,\n",
       " 'github.com/charmasaur/M-LOOP': 1,\n",
       " 'github.com/BenJWard/Supertypes_RedQueen_TSE': 1,\n",
       " 'github.com/open-ephys/plugin-GUI': 1,\n",
       " 'github.com/NTCColumbia/moco': 2,\n",
       " 'github.com/cjfields/readfq': 1,\n",
       " 'baohongz.github.io/Lupus_eQTL': 1,\n",
       " 'github.com/vkozareva/geneCNV': 1,\n",
       " 'drpowell.github.io/vennt': 7,\n",
       " 'github.com/slipguru/palladio': 1,\n",
       " 'github.com/AnnaGuan/GISMMA/tree/AnnaGuan-patch-1': 1,\n",
       " 'github.com/torognes/vsearch/wiki/VSEARCH-pipeline': 2,\n",
       " 'github.com/reconhub/outbreaker2': 1,\n",
       " 'github.com/NabaviLab/CNV-Sim': 1,\n",
       " 'github.com/najoshi/sickle': 275,\n",
       " 'github.com/vsbuffalo/scythe': 65,\n",
       " 'github.com/topherconley/noleaven': 1,\n",
       " 'github.com/jtamames/squeezeM': 1,\n",
       " 'github.com/tseemann/barrnap': 15,\n",
       " 'github.com/bongsongkim/HH-CCDF': 1,\n",
       " 'github.com/rongjiewang/BdBG': 1,\n",
       " 'github.com/sdchandra/CNAclinic': 2,\n",
       " 'github.com/coulet/PredictDoseReduction': 1,\n",
       " 'github.com/cromulo/IWS': 1,\n",
       " 'github.com/opencobra/cobratoolbox': 8,\n",
       " 'github.com/danbang/article-adolescence-optimal': 1,\n",
       " 'github.com/edickie/icarus': 1,\n",
       " 'github.com/jenniesuz/tsetse_climate_change': 1,\n",
       " 'veg.github.io/hyphy-site': 2,\n",
       " 'github.com/linneas/condetri': 1,\n",
       " 'github.com/janelia-flyem/NeuTu': 5,\n",
       " 'github.com/ThomasDOtto/IPA': 3,\n",
       " 'jlizier.github.io/jidt': 1,\n",
       " 'github.com/kilicogluh/Bio-SCoRes': 3,\n",
       " 'github.com/brianyu2010/Mini-Metagenomic_Analyses': 1,\n",
       " 'github.com/elifesciences-publications/Mini-Metagenomic_Analyses': 1,\n",
       " 'github.com/hallamlab/metapathways2/wiki': 1,\n",
       " 'neuro.github.io': 1,\n",
       " 'github.com/jtladner/Scripts/tree/master/chimeric_reads': 1,\n",
       " 'github.com/lh3/seqtk': 139,\n",
       " 'github.com/BGI-shenzhen/PopLDdecay': 7,\n",
       " 'github.com/saschajung/ChromAccPrediction': 1,\n",
       " 'github.com/dizak/mothulity': 1,\n",
       " 'github.com/tkotani/ecalj': 1,\n",
       " 'github.com/rapidftr/RapidFTR/wiki': 1,\n",
       " 'github.com/marcopessoa/bioinfo-scripts': 1,\n",
       " 'github.com/LJI-Bioinformatics/HLATyphon': 4,\n",
       " 'github.com/GildasLepennetier/GildasSources': 2,\n",
       " 'github.com/UD-CBCB/RedRep': 1,\n",
       " 'github.com/jdstorey/qvalue': 35,\n",
       " 'github.com/xu1912/SMON.git': 1,\n",
       " 'github.com/airanmehr/clear': 1,\n",
       " 'yjx1217.github.io/Yeast_PacBio_2016/data': 4,\n",
       " 'github.com/biocompibens/SME': 1,\n",
       " 'github.com/KarenBell/rbcL-dual-index-metabarcoding': 2,\n",
       " 'github.com/LucksLab/Cotrans_SHAPE-Seq_Tools/releases': 2,\n",
       " 'github.com/LucksLab/spats/releases': 4,\n",
       " 'github.com/leschzinerlab/Relion': 1,\n",
       " 'github.com/parklab/NGSCheckMate': 1,\n",
       " 'transdecoder.github.io': 131,\n",
       " 'github.com/mikessh/migec': 2,\n",
       " 'github.com/milaboratory/mixcr': 5,\n",
       " 'github.com/imminfo/tcr': 1,\n",
       " 'nbviewer.ipython.org/urls/raw.github.com/cschin/Write_A_Genome_Assembler_With_IPython/master/Write_An_Assembler.ipynb': 1,\n",
       " 'github.com/drasmuss/nhrlmodel': 1,\n",
       " 'github.com/MITEL-UNIUD/SlideJ': 1,\n",
       " 'github.com/msoos/cryptominisat': 1,\n",
       " 'github.com/dsilvestro/PyRate': 5,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cuffdiff/index.html': 4,\n",
       " 'github.com/annajxli/scaredy-rat': 1,\n",
       " 'github.com/PacificBiosciences/SMRT-Analysis': 18,\n",
       " 'github.com/TACC/Lmod': 1,\n",
       " 'omicsoftdocs.github.io/ArraySuiteDoc': 1,\n",
       " 'github.com/ecjoliver/marineHeatWaves': 2,\n",
       " 'github.com/fchollet/keras': 29,\n",
       " 'github.com/ta-cameron/cell-profiles': 1,\n",
       " 'github.com/OLC-Bioinformatics/COWBAT/tree/v0.1.5': 1,\n",
       " 'github.com/JoanaGrah/MitosisAnalyser': 1,\n",
       " 'github.com/StewartLab/Stewart_TEDDY_Microbiome_Analysis': 1,\n",
       " 'github.com/Huanglab-ucsf/Deformed-Alignment': 1,\n",
       " 'github.com/foenpeng/Controller-of-experiment-platform.git': 1,\n",
       " 'github.com/mhaft/DeepVess': 1,\n",
       " 'github.com/GlobalFishingWatch/vessel-classification': 1,\n",
       " 'github.com/easybuilders/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/c/cutadapt': 1,\n",
       " 'github.com/USDA-ARS-GBRU/crossword': 1,\n",
       " 'github.com/USDA-ARS-GBRU/crossword/examples': 1,\n",
       " 'github.com/QijingZheng/VaspBandUnfolding': 1,\n",
       " 'github.com/pmoriano/Community-Based-Event-Detection': 1,\n",
       " 'github.com/lhting/Trauma_Study_GUI': 1,\n",
       " 'github.com/nskvir/RepEnrich14': 1,\n",
       " 'github.com/emreg00/toolbox': 4,\n",
       " 'github.com/michathoma/flywalkr': 1,\n",
       " 'stnava.github.com/WHopkinsNHP': 1,\n",
       " 'github.com/aleaverfay/swiftlib_javascript': 1,\n",
       " 'github.com/esling/mistagging': 1,\n",
       " 'github.com/UniversityofWarwick/ews_fli1_and_e2f_nar_article': 1,\n",
       " 'github.com/benliemory/Methylphet': 1,\n",
       " 'github.com/Victorian-Bioinformatics-Consortium/VelvetOptimiser': 4,\n",
       " 'github.com/sanger-pathogens/Bio-Tradis': 9,\n",
       " 'github.com/meren/illumina-utils': 9,\n",
       " 'github.com/csbl/platereader': 1,\n",
       " 'github.com/GenomicsCoreLeuven/GBSX': 3,\n",
       " 'github.com/crosslinks/XlinkAnalyzer': 1,\n",
       " 'github.com/minseven/mForensics.git': 1,\n",
       " 'github.com/jacobtennessen/HOLDRS': 1,\n",
       " 'github.com/slzhao/heatmap3': 1,\n",
       " 'github.com/jjberg2/PolygenicAdaptationCode': 1,\n",
       " 'github.com/fcchou/helixmc': 1,\n",
       " 'www.github.com/mathii/f2': 1,\n",
       " 'github.com/NeuroEngineering-UniMelb/cortical-units': 1,\n",
       " 'picrust.github.io/picrust': 33,\n",
       " 'github.com/Ruleby/ruleby': 1,\n",
       " 'picrust.github.com': 8,\n",
       " 'picrust.github.io/picrust/scripts/metagenome_contributions.html': 1,\n",
       " 'yanlilab.github.io/fastcov': 1,\n",
       " 'github.com/torognes/vsearch': 25,\n",
       " 'gephi.github.io': 31,\n",
       " 'github.com/maryreginaboland/SeaWAS': 1,\n",
       " 'github.com/luslab/multiregion-cnv-phylogenetics': 1,\n",
       " 'github.com/broadinstitute/picard/releases/tag/1.128': 2,\n",
       " 'github.com/GreallyLab/PE_Suzuki_et_al_2016': 1,\n",
       " 'github.com/ysu001/PUP': 4,\n",
       " 'github.com/justinjee/MDS': 1,\n",
       " 'github.com/susinmotion/barcode_tries': 1,\n",
       " 'github.com/ctorney/wildCount': 1,\n",
       " 'github.com/PacificBiosciences/DevNet/wiki/Drosophila-sequence-and-assembly': 2,\n",
       " 'github.com/Lakens/perfect-t-test': 2,\n",
       " 'github.com/meegpipe/meegpipe': 3,\n",
       " 'github.com/tsudalab/bpbi': 1,\n",
       " 'github.com/ijmarshall/robotsearch': 1,\n",
       " 'github.com/shka/STRTprep': 5,\n",
       " 'github.com/popgenomics/popPhylABC': 1,\n",
       " 'github.com/iontorrent': 3,\n",
       " 'github.com': 141,\n",
       " 'github.com/loghindo/NDPI_counting_MC1_bacteria': 1,\n",
       " 'github.com/FunctionLab/ExPecto': 1,\n",
       " 'broadinstitute.github.io/picard/index.html': 12,\n",
       " 'www.github.com/dspeth': 3,\n",
       " 'github.com/shengqh/TIGER': 1,\n",
       " 'github.com/ColumbusCollaboratory/cognizer': 1,\n",
       " 'github.com/robince/gcmi': 3,\n",
       " 'github.com/robince/partial-info-decomp': 1,\n",
       " 'github.com/biocore-ntnu/epic': 4,\n",
       " 'github.com/GRousselet/blog/tree/master/comp2dcorr': 1,\n",
       " 'github.com/mui-icbi/Immunophenogram': 2,\n",
       " 'github.com/Neurorobtics/ACOREL': 1,\n",
       " 'github.com/PacificBiosciences/pbdagcon.git': 1,\n",
       " 'github.com/rhodricusack/automaticanalysis': 7,\n",
       " 'chamber.github.io/fibernavigator_single': 1,\n",
       " 'github.com/dials/dxtbx_ED_formats': 1,\n",
       " 'github.com/ANGSD/asaMap': 1,\n",
       " 'github.com/ESPoppelaars/Cross-frequency-coupling': 1,\n",
       " 'github.com/gephi/gephi/wiki/OpenOrd': 1,\n",
       " 'github.com/LABSS/PeerReviewGPS': 1,\n",
       " 'github.com/Oshlack/Farlie_emu_wing_RNASeq_analysis': 1,\n",
       " 'samtools.github.io/bcftools': 20,\n",
       " 'github.com/EuropePMC/Biojs.Annotator/tree/Biojs.Annotator_1.0': 1,\n",
       " 'github.com/a3cel2/bc_pca_git': 1,\n",
       " 'github.com/statonlab/disculaSSRs': 1,\n",
       " 'github.com/PeeperLab/CopywriteR': 2,\n",
       " 'github.com/evanfeinberg/conformation/blob/master/grids.py': 1,\n",
       " 'interactivereport.github.io/ZSF1': 1,\n",
       " 'github.com/DavidCunefare/CNN-Cone-Detection': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks': 116,\n",
       " 'github.com/smfleming/HMM': 1,\n",
       " 'github.com/gollischlab/STNMFanalysis': 1,\n",
       " 'github.com/nylander': 1,\n",
       " 'github.com/RPeterson4/Baseline_Covariate_Files': 1,\n",
       " 'github.com/bcgsc/nanosim': 1,\n",
       " 'github.com/BGI-flexlab/SOAPnuke': 5,\n",
       " 'github.com/schae234/pypeline': 1,\n",
       " 'github.com/sliu2/BPAC': 1,\n",
       " 'github.com/jcaledo/PTM_sites_coevolution': 1,\n",
       " 'github.com/jstjohn/SeqPrep': 163,\n",
       " 'github.com/bdearlove/treeImbalance': 1,\n",
       " 'github.com/SEEG-Oxford/seegSDM': 4,\n",
       " 'github.com/flauro/3tck_comparative': 2,\n",
       " 'github.com/jimhester/knitrBootstrapFastQC': 1,\n",
       " 'github.com/iontorrent/TMAP': 25,\n",
       " 'sanger-pathogens.github.io/iva': 1,\n",
       " 'github.com/mattmattmattmatt/VASP': 1,\n",
       " 'github.com/bhi-kimlab/BiSpark': 1,\n",
       " 'github.com/Christensen-Lab-Dartmouth/CF_Epigenetics': 1,\n",
       " 'github.com/rcsb/PDB-Outlier-Analysis.git': 1,\n",
       " 'pasapipeline.github.io': 8,\n",
       " 'github.com/mgalardini/pyseer': 1,\n",
       " 'github.com/broadinstitute/gatk': 2,\n",
       " 'github.com/sbg/VBT-TrioAnalysis.git': 1,\n",
       " 'github.com/leesael/GIFT': 1,\n",
       " 'github.com/ICB-DCM/PESTO': 1,\n",
       " 'github.com/TheOwenLab/2018-Wild-et-al-Sleep-and-Cognition': 1,\n",
       " 'github.com/rowaniskandar/CM_ME_plosone': 1,\n",
       " 'github.com/TransDecoder/TransDecoder/wiki': 12,\n",
       " 'github.com/csamstag/mito-mutations': 1,\n",
       " 'github.com/mattlevine22/emerge2ohdsi_information_loss_CURATED.git': 1,\n",
       " 'github.com/elifesciences-publications/CATMAID-to-Blender': 1,\n",
       " 'nbisweden.github.io/MrBayes': 1,\n",
       " 'github.com/davidrasm/Marula': 1,\n",
       " 'github.com/cculianu/StimulateOpenGL_II': 1,\n",
       " 'github.com/pbnjay/lollipops/releases': 2,\n",
       " 'github.com/nci/drishti/wiki': 2,\n",
       " 'github.com/VafaeeLab/glioblastoma_exosomal_miR_markers': 1,\n",
       " 'benjjneb.github.io/dada2/index.html': 2,\n",
       " 'www.github.com/ctross/emberasearch': 1,\n",
       " 'github.com/prafols/rMSIproc': 1,\n",
       " 'cj-chen.github.io/tbtools': 4,\n",
       " 'atztogo.github.io/phonopy': 3,\n",
       " 'github.com/tseemann/snippy': 40,\n",
       " 'github.com/BUCK-LCO-NCI/VirConTaxa': 1,\n",
       " 'github.com/amnona/calour': 1,\n",
       " 'github.com/phac-nml/imid_microbiome': 1,\n",
       " 'github.com/mirnylab/cooler': 4,\n",
       " 'github.com/alexdobin/STAR': 106,\n",
       " 'github.com/fredericlemoine/goalign': 2,\n",
       " 'github.com/mingjunzheng/ceRNA_code': 1,\n",
       " 'github.com/ajvilleg/extractUpStreamDNA': 1,\n",
       " 'api.github.com': 1,\n",
       " 'github.com/FelixKrueger/TrimGalore': 22,\n",
       " 'github.com/bitcoin/bitcoin': 1,\n",
       " 'github.com/RamanLab/nbfpeg': 1,\n",
       " 'github.com/ENCODE-DCC/chip-seq-pipeline': 1,\n",
       " 'www.github.com/ElectricDinoLab': 1,\n",
       " 'github.com/taiyun/corrplot': 14,\n",
       " 'github.com/Sun-Yazhou/MDAD': 1,\n",
       " 'github.com/sydneycytometry/tSNEplots': 1,\n",
       " 'github.com/spyking-circus/spyking-circus': 1,\n",
       " 'github.com/Sage-Bionetworks/MoleMapper': 1,\n",
       " 'github.com/researchkit/researchkit': 3,\n",
       " 'github.com/Sage-Bionetworks/Bridge-iOS-SDK': 3,\n",
       " 'github.com/pid-ub/pid-UB': 1,\n",
       " 'dkoboldt.github.io/varscan': 9,\n",
       " 'stnava.github.io/ANTs': 96,\n",
       " 'github.com/alexandrebarachant/pyRiemann': 1,\n",
       " 'github.com/alexandrebarachant/WMFT_JBHI': 1,\n",
       " 'github.com/careylab/LocoMouse': 3,\n",
       " 'github.com/alanrice/paper-dosage-sensitivity-copy-number-variation': 1,\n",
       " 'github.com/ocisse/Neolecta_genome_project': 1,\n",
       " 'github.com/joanam/scripts': 1,\n",
       " 'homeveg.github.io/nuctools': 1,\n",
       " 'github.com/tseemann/wombac': 3,\n",
       " 'github.com/karpathy/char-rnn': 1,\n",
       " 'github.com/pimbongaerts/radseq': 1,\n",
       " 'github.com/pimbongaerts/bermuda-rad': 1,\n",
       " 'github.com/peteskene': 1,\n",
       " 'github.com/joey711/phyloseq/issues/143': 1,\n",
       " 'github.com/pillailab': 1,\n",
       " 'tgennorth.github.io/NASP': 23,\n",
       " 'github.com/ImperialStatGen/MARV': 2,\n",
       " 'github.com/colford/nbgw-plant-illumina-pipeline': 3,\n",
       " 'github.com/scarrazza/smpdf': 1,\n",
       " 'github.com/ericminikel/prnp_penetrance': 3,\n",
       " 'github.com/PerinatalLab/MatHeight-GestAge': 1,\n",
       " 'github.com/kalpathy/tacticsRT': 3,\n",
       " 'github.com/alvason/identifying_neurite_by_RRS': 1,\n",
       " 'github.com/tomotto/RSE-box': 1,\n",
       " 'github.com/bettytijms/Single_Subject_Grey_Matter_Networks': 3,\n",
       " 'trinotate.github.io': 83,\n",
       " 'github.com/merijnbeeksma/predict-EoL': 1,\n",
       " 'github.com/gfilaci/GridNSPT': 1,\n",
       " 'github.com/jgx65/quantinemo': 1,\n",
       " 'github.com/chlubba/PyPNS': 1,\n",
       " 'github.com/chlubba/PyPNS-PaperFigureshttp://github.com/chlubba/PyPNS-PaperFigures': 1,\n",
       " 'github.com/cestaroa/tab_blast_parser': 1,\n",
       " 'github.com/yonghee91/NGS_error_validation.git': 1,\n",
       " 'github.com/mrirecon/enlive': 1,\n",
       " 'github.com/btlindert/actant-1': 1,\n",
       " 'github.com/sarmistha123/TightClust_Large/blob/master/TightCluster_Large.R': 1,\n",
       " 'github.com/sensorimotorgroupdpz/MaCaQuE': 1,\n",
       " 'github.com/mariodosreis/tai': 1,\n",
       " 'github.com/osuchanglab/autoANI': 1,\n",
       " 'github.com/ctSkennerton/minced': 2,\n",
       " 'github.com/Bioconductor/GenomicRanges': 1,\n",
       " 'github.com/wangjingke/madresGpsClient': 1,\n",
       " 'github.com/VGalata/16S_Sarcoidosis': 1,\n",
       " 'github.com/chingtoe365/metaUnion': 1,\n",
       " 'github.com/srcbs/GenoBox': 1,\n",
       " 'github.com/MatthieuPalayret/vlsSMLM': 1,\n",
       " 'github.com/igraph/xdata-igraph': 1,\n",
       " 'github.com/jovo/FastApproximateQAP/tree/master/code/FAQ': 1,\n",
       " 'github.com/gersteinlab/loregic': 1,\n",
       " 'github.com/morganrcu/TTT': 1,\n",
       " 'biossa.github.io': 1,\n",
       " 'github.com/xutaodeng/EnsembleAssembler': 1,\n",
       " 'github.com/atmartens/ribosome_profiling': 1,\n",
       " 'github.com/lanagarmire/epiPredictor': 1,\n",
       " 'github.com/chrisamiller/copycat': 5,\n",
       " 'github.com/genome/gms-core/blob/master/lib/perl/Genome/Model/Tools/Sv/SquareDancer.pl': 2,\n",
       " 'github.com/genome/sciclone': 10,\n",
       " 'ntncmch.github.io/ebola/liberia_sitrep.html': 1,\n",
       " 'github.com/SachaEpskamp/semPlot': 1,\n",
       " 'libspatialindex.github.io': 1,\n",
       " 'github.com/StamLab/network-motifs': 1,\n",
       " 'github.com/widdowquinn/scripts/blob/master/bioinformatics/calculate_ani.py': 4,\n",
       " 'github.com/mazerj/pype3': 2,\n",
       " 'github.com/RavelLab/GuineaPigVMB': 1,\n",
       " 'github.com/abria/TeraStitcher': 1,\n",
       " 'github.com/xies/pulse_finding': 1,\n",
       " 'github.com/KTgreenham/TRiP': 1,\n",
       " 'github.com/janelia-flyem/NeuroProof.git': 1,\n",
       " 'github.com/SynBioDex/libSBOLj': 2,\n",
       " 'github.com/cni/widgets/tree/master/touch': 1,\n",
       " 'github.com/JaneliaSciComp/Muse': 1,\n",
       " 'github.com/DanRosauer/phylospatial': 1,\n",
       " 'github.com/sjackman/uniqtag': 1,\n",
       " 'github.com/sjackman/uniqtag-paper': 1,\n",
       " 'OrgMassSpec.github.io': 3,\n",
       " 'github.com/CovertLab/WholeCell/tree/parameter-estimation-DREAM-challenge-2013': 1,\n",
       " 'github.com/paulspychala/CMAPdart': 1,\n",
       " 'github.com/klusta-team': 3,\n",
       " 'idekerlab.github.io/cy-net-share': 3,\n",
       " 'dleelab.github.io/jepegmix': 1,\n",
       " 'turkunlp.github.io/Cell-line-recognition': 1,\n",
       " 'www.github.com/dsurujon/motifs': 1,\n",
       " 'github.com/iPlantCollaborativeOpenSource': 1,\n",
       " 'github.com/curtislarimer/Biofilm-Growth-Intensity': 1,\n",
       " 'github.com/IEDB/MRO': 1,\n",
       " 'github.com/raivokolde/seqlm': 1,\n",
       " 'github.com/wilkelab/influenza_pH1N1_timecourse': 1,\n",
       " 'github.com/owenrandlett/Z-Brain': 1,\n",
       " 'trinotate.github.io/TrinotateWeb.html': 1,\n",
       " 'github.com/DmitryVel/CANEapp': 1,\n",
       " 'github.com/sarahannison': 1,\n",
       " 'github.com/itseez/opencv': 4,\n",
       " 'github.com/zeochoy/tcga-embedding': 1,\n",
       " 'github.com/r-syst/databases/tree/master/r-syst::fungi': 1,\n",
       " 'github.com/soedinglab/b-lore': 1,\n",
       " 'sashagusev.github.io/2017-10/twas-vulnerabilities.html': 1,\n",
       " 'github.com/systemsmedicine/adjuvanted-vaccine': 1,\n",
       " 'github.com/ben-lerch/BaseMod-3.0': 1,\n",
       " 'github.com/fl16180/argonet': 1,\n",
       " 'github.com/jeremymsimon/MouseCortex': 1,\n",
       " 'github.com/dewyman/TranscriptClean': 1,\n",
       " 'github.com/soumitag/SLIDE': 1,\n",
       " 'github.com/jkbonfield/crumble': 1,\n",
       " 'github.com/davidjakubec/3DPatch': 1,\n",
       " 'github.com/b2du/bth': 1,\n",
       " 'github.com/alexjau/uqsa': 1,\n",
       " 'github.com/yunwilliamyu/opal': 1,\n",
       " 'github.com/mapircamera/QGIS': 1,\n",
       " 'github.com/chail/NMF_neurodevelopment': 1,\n",
       " 'github.com/vcflib/vcflib': 23,\n",
       " 'github.com/tseemann/nullarbor': 16,\n",
       " 'github.com/federicomarini/pcaExplorer': 1,\n",
       " 'github.com/linbingqing/stableDE': 1,\n",
       " 'github.com/ben-lerch/IsoSeq-3.0/blob/master/README.md': 1,\n",
       " 'github.com/Sunhh/NGS_data_processing': 1,\n",
       " 'github.com/bminkenberg/CRISPR-PLANTv2': 1,\n",
       " 'github.com/obigriffith/biostar-tutorials/blob/master/Heatmaps/heatmap.3.R': 2,\n",
       " 'github.com/NWPU-903PR/Deepm6A.git': 1,\n",
       " 'github.com/Romain-Laine/MiLeSIM': 1,\n",
       " 'github.com/elifesciences-publications/MiLeSIM': 1,\n",
       " 'github.com/sanger-pathogens/mlst_check': 12,\n",
       " 'github.com/marbl/Parsnp': 1,\n",
       " 'benjjneb.github.io/dada2/tutorial.html': 8,\n",
       " 'github.com/jfaberha/lastz_lav_expansion': 1,\n",
       " 'transdecoder.github.io.com/TransDecoder': 1,\n",
       " 'github.com/wg-perception/people': 1,\n",
       " 'lmfit.github.io/lmfit-py': 6,\n",
       " 'github.com/kthorn/particletrack': 1,\n",
       " 'github.com/KeithBurghardt/Stack-Exchange-User-Behavior': 1,\n",
       " 'github.com/cole-trapnell-lab/cufflinks': 12,\n",
       " 'github.com/babelomics/hipathia': 1,\n",
       " 'github.com/ding-lab/squaredancer': 1,\n",
       " 'github.com/spoilt333/onco-aae': 1,\n",
       " 'github.com/ruudvdent/WAM2layersPython': 1,\n",
       " 'github.com/OPIC-Oxford/localrec': 1,\n",
       " 'github.com/EdaEhler/bioutils_python': 1,\n",
       " 'github.com/mulin0424/cepip': 1,\n",
       " 'github.com/broadinstitute/mutect': 5,\n",
       " 'github.com/MKaptein/streamingbandit': 1,\n",
       " 'github.com/broadinstitute/picard': 55,\n",
       " 'github.com/AltschulerWu-Lab/simucell': 1,\n",
       " 'cloud.github.com/downloads/faircloth-lab/edittag/edit_metric_tags.txt': 1,\n",
       " 'github.com/nitishsrivastava/deepnet': 4,\n",
       " 'github.com/jouyun': 5,\n",
       " 'github.com/xnuttle/mips_cnv_typer': 1,\n",
       " 'github.com/scapella/markers_genes': 1,\n",
       " 'github.com/MichaelRiss/FTSPlot': 1,\n",
       " 'github.com/koldogarcia/metagenomics': 1,\n",
       " 'github.com/andymckenzie/AtxA': 1,\n",
       " 'github.com/selimonat/published_code/tree/master/condprob': 1,\n",
       " 'github.com/YeatmanLab/Speech_contrasts_public': 1,\n",
       " 'github.com/rrwick/Porechop': 25,\n",
       " 'github.com/fhcrc/taxtastic': 2,\n",
       " 'github.com/bjbarrett/bequeathal': 1,\n",
       " 'github.com/metacoglab/VaccaroFleming': 1,\n",
       " 'github.com/wajidarshad/LUPI-SVM': 1,\n",
       " 'github.com/BradnerLab/netseq': 3,\n",
       " 'github.com/chapmanb/bcbb/blob/master/align/adaptor_trim.py': 1,\n",
       " 'github.com/srkesler/bnets': 1,\n",
       " 'github.com/matterport/Mask_RCNN': 1,\n",
       " 'github.com/orzechoj/circRNA_finder.git': 2,\n",
       " 'github.com/PacificBiosciences/GenomicConsensus': 21,\n",
       " 'github.com/kendomaniac/docker4seq': 2,\n",
       " 'github.com/zcshinee/Pol-TIRF': 1,\n",
       " 'github.com/dbrvs/reservoir_persistence': 1,\n",
       " 'github.com/trinityrnaseq/trinityrnaseq/wiki': 34,\n",
       " 'pachterlab.github.io/kallisto': 17,\n",
       " 'github.com/aspen-shen/RefCell': 1,\n",
       " 'github.com/SoniMohapatra/MPCC': 1,\n",
       " 'deweylab.github.io/RSEM': 28,\n",
       " 'github.com/ENCODE-DCC/long-rna-seq-pipeline/blob/master/DAC/STAR_RSEM.sh': 2,\n",
       " 'tdwg.github.io/dwc/terms': 2,\n",
       " 'pypl.github.io/PYPL.html': 2,\n",
       " 'github.com/najoshi/sabre': 17,\n",
       " 'github.com/ADAPTABLETRIAL/PHENOTYPE': 1,\n",
       " 'github.com/TransDecoder/TransDecoder': 11,\n",
       " 'github.com/sinhrks/ggfortify': 1,\n",
       " 'github.com/mjobin/Imputor': 1,\n",
       " 'github.com/debbiemarkslab/EVcouplings': 3,\n",
       " 'github.com/zyndagj/BSMAPz': 1,\n",
       " 'github.com/al2na/methylKit': 5,\n",
       " 'github.com/taoliu/MACS': 93,\n",
       " 'github.com/rbslandau/schreiner_simstress': 1,\n",
       " 'github.com/molnarb14/analog-maxSAT-solver': 1,\n",
       " 'github.com/jakobrunge/tigramite': 1,\n",
       " 'github.com/hzi-bifo/FrechetTreeDistance': 1,\n",
       " 'github.com/kylepquinn/Code-and-Example-Data': 1,\n",
       " 'github.com/kharyuk/chemfin-plasp': 1,\n",
       " 'github.com/nakae-k/glia-neuron': 1,\n",
       " 'github.com/PlantSimLab': 1,\n",
       " 'dongjunchung.github.io/GPA': 2,\n",
       " 'github.com/reidpr/quac': 2,\n",
       " 'github.com/takaho/snpexp': 1,\n",
       " 'josephryan.github.com/estimate_genome_size.pl': 1,\n",
       " 'github.com/tanghaibao/goatools': 21,\n",
       " 'github.com/aherbert/GDSC-SMLM': 2,\n",
       " 'github.com/aramette/otu2ot': 1,\n",
       " 'github.com/bvilhjal/mixmogam': 7,\n",
       " 'github.com/kylebittinger/airway-fungi-paper': 1,\n",
       " 'github.com/gmpgis/distribution': 1,\n",
       " 'github.com/omegaicl/mcpp': 1,\n",
       " 'github.com/enormandeau': 1,\n",
       " 'github.com/mackerman44/idfgen': 1,\n",
       " 'github.com/michaelfrance/evol_app_2018_biofilms': 1,\n",
       " 'github.com/cwrowley/dmdtools/tree/master': 1,\n",
       " 'github.com/MariaICL/Paper_analysis_scripts': 1,\n",
       " 'trinityrnaseq.github.io': 38,\n",
       " 'github.com/lihong1github/Image-analysis': 1,\n",
       " 'github.com/cokelaer/spectrum': 1,\n",
       " 'github.com/pmpsa-hpc/NPA': 1,\n",
       " 'github.com/pmpsa-hpc/NPAModels': 1,\n",
       " 'github.com/maranasgroup/optstoic-python': 1,\n",
       " 'github.com/kwikteam/klustakwik2': 2,\n",
       " 'github.com/jyosa/CYP2D6': 1,\n",
       " 'github.com/maufrais/WHGDCA_ES': 1,\n",
       " 'github.com/myl446/BRWHNHA': 1,\n",
       " 'github.com/msettles/dbcAmplicons': 8,\n",
       " 'github.com/macarenadiaz/FAZ_Extraction': 1,\n",
       " 'github.com/gradinetal2018/BARA': 1,\n",
       " 'www.github.com/PublicHealthEngland/bayesint': 1,\n",
       " 'github.com/QingtianZhang/SHMAX_AuditoryPathway': 1,\n",
       " 'github.com/chop-dbhi/sepsis_01': 1,\n",
       " 'github.com/Emergent-Behaviors-in-Biology/community-simulator': 1,\n",
       " 'github.com/groupschoof/AHRD': 9,\n",
       " 'github.com/shenlab-sinai/ngsplot': 14,\n",
       " 'github.com/Zhong-Lab-UCSD/iMARGI_methods': 1,\n",
       " 'github.com/mirnylab/pairtools': 1,\n",
       " 'github.com/agarcor21/IdentityCoefficients': 1,\n",
       " 'github.com/chevrm/kopilot': 1,\n",
       " 'github.com/mmadsen/lipomadsen2015-idss-seriation-paper': 1,\n",
       " 'github.com/RodgerG/Hemiptera-of-Canada-DNA-Barcode-Library-workflow.git': 1,\n",
       " 'github.com/ripple-neuro/mksort': 5,\n",
       " 'github.com/genericworkflownodes': 2,\n",
       " 'ericmjonas.github.io/connectodiscovery': 1,\n",
       " 'github.com/dch33/Quad-Sim': 1,\n",
       " 'github.com/wangsy11/InfMod3DGen': 2,\n",
       " 'github.com/arq5x/piledriver': 5,\n",
       " 'github.com/gbif/checklistbank': 2,\n",
       " 'github.com/google/guice': 1,\n",
       " 'github.com/jts/sga/zipball/master': 2,\n",
       " 'github.com/meizhan/SVMelegans': 2,\n",
       " 'github.com/sina-cb/Tn-seqExplorer': 1,\n",
       " 'github.com/obophenotype/uberon/wiki/Editor-tools-guide': 1,\n",
       " 'github.com/akastrin/kastrin2014large': 1,\n",
       " 'dyerlab.github.io/popgraph': 1,\n",
       " 'github.com/najoshi/sickle.git': 3,\n",
       " 'networkx.github.io': 54,\n",
       " 'github.com/janelia-flyem/gala': 1,\n",
       " 'github.com/ged-lab/khmer/tree/2012-paper-diginorm/sandbox': 2,\n",
       " 'github.com/ged-lab/khmer': 3,\n",
       " 'github.com/ged-lab/screed': 2,\n",
       " 'github.com/bwrc/WishGLD': 1,\n",
       " 'github.com/fbarranco/eventVision-evbench': 1,\n",
       " 'github.com/davidssmith/csphantom': 1,\n",
       " 'github.com/bernardosabatinilab/SabalabSoftware_Nov2009.git': 2,\n",
       " 'github.com/usc-tlab/LinearFineContrast.git': 2,\n",
       " 'github.com/sofroniewn/tactile-coding': 1,\n",
       " 'github.com/stephenfloor/tripseq-analysis': 2,\n",
       " 'github.com/akrizhevsky/cuda-convnet2': 1,\n",
       " 'github.com/peterjfoster/eLife': 1,\n",
       " 'github.com/jeongchans/smrfArchived': 1,\n",
       " 'dgkontopoulos.github.io/Structuprint': 1,\n",
       " 'github.com/lsgh/ismb15': 1,\n",
       " 'github.com/frizfealer/IMS_project': 1,\n",
       " 'github.com/chengyuan/reago': 1,\n",
       " 'github.com/gemtools/gemtools': 4,\n",
       " 'wurmlab.github.io/data': 1,\n",
       " 'github.com/tylerhether/Flip2BeRAD': 1,\n",
       " 'github.com/samtools/BCFtools': 3,\n",
       " 'github.com/gc5k/GEAR/wiki': 1,\n",
       " 'github.com/DenisCarriere/geocoder': 1,\n",
       " 'github.com/cmdoret/chalcid_comparative_analysis': 1,\n",
       " 'github.com/QCaudron/timing_malaria_transmission': 1,\n",
       " 'github.com/thomasp85/patchwork': 1,\n",
       " 'github.com/thackl/ggworldmap': 1,\n",
       " 'github.com/babicknell/NeuriteGrowth': 1,\n",
       " 'github.com/TulsaBioinformaticsToolsmith/HebbPlot': 1,\n",
       " 'github.com/3DGenomes/tadbit': 4,\n",
       " 'github.com/shmohammadi86/Kavosh': 1,\n",
       " 'github.com/OneAngstrom/DeepTrajectory': 1,\n",
       " 'github.com/BrainDynamicsUSYD/nftsim': 1,\n",
       " 'github.com/gulyasmarton/PetriPrinter': 1,\n",
       " 'github.com/bicanski/HBPcollab/tree/master/SpatialEpisodicMemoryModel': 1,\n",
       " 'github.com/elifesciences-publications/HBPcollab/tree/master/SpatialEpisodicMemoryModel': 1,\n",
       " 'github.com/marsdeck/AFMech-Suite/archive/v1.0.zip': 1,\n",
       " 'github.com/marsdeck/AFMech-Suite': 1,\n",
       " 'github.com/pimbongaerts/monitoring': 1,\n",
       " 'github.com/weiliu123/RRSF-package': 1,\n",
       " 'githubcom/sccn/labstreaminglayer': 1,\n",
       " 'github.com/ItzikMalkiel/DeepNanoDesign': 1,\n",
       " 'github.com/princello/GePMI': 1,\n",
       " 'github.com/Sung-Huan/ANNOgesic': 1,\n",
       " 'github.com/Shoombuatong2527/anticancer-peptides-review': 1,\n",
       " 'github.com/AlanFreeman/anaMotion': 1,\n",
       " 'github.com/keitaroyam/yamtbx': 4,\n",
       " 'github.com/fmrireplicability/NCB_code': 1,\n",
       " 'github.com/jfmeadow/dualingTrees-pkg': 1,\n",
       " 'github.com/Jincheng2009/igseqanalysis': 1,\n",
       " 'alitvteam.github.io/AliTV/d3/AliTV.html': 1,\n",
       " 'github.com/macraild/6D8-CommsBiol': 1,\n",
       " 'github.com/jpcolella/Ermine_WGS_2018': 1,\n",
       " 'github.com/broadinstitute/pilon': 4,\n",
       " 'github.com/ranganawarshamanage/mxt': 1,\n",
       " 'github.com/annalam/foxa1-utr-manuscript-code': 1,\n",
       " 'github.com/annalam/breakfast': 2,\n",
       " 'github.com/annalam/seqkit': 1,\n",
       " 'github.com/abhik/crispr–analysis': 1,\n",
       " 'github.com/manutamminen/epride': 1,\n",
       " 'github.com/lh3/bwa': 26,\n",
       " 'github.com/broadgsa/gatk': 6,\n",
       " 'samtools.github.io': 7,\n",
       " 'github.com/arq5x/bedtools2': 33,\n",
       " 'github.com/Ensembl/ensembl-vep': 6,\n",
       " 'github.com/kuz/Human-Intracranial-Recordings-and-DCNN-to-Compare-Biological-and-Artificial-Mechanisms-of-Vision': 1,\n",
       " 'github.com/josephbarry/iQTL': 1,\n",
       " 'raw.githubusercontent.com/obigriffith/biostar-tutorials/master/Heatmaps/heatmap.3.R': 4,\n",
       " 'github.com/jkrijthe/Rtsne': 12,\n",
       " 'github.com/tpq/balance': 1,\n",
       " 'github.com/SharonLutz/Umediation': 3,\n",
       " 'github.com/WSpiller/RadialMR': 1,\n",
       " 'github.com/bmelinden/uncertainSPT': 2,\n",
       " 'github.com/roschkoenig/Zebrafish_Seizure': 1,\n",
       " 'ccraddock.github.io/cluster_roi/atlases.html': 2,\n",
       " 'github.com/stschiff/sequenceTools/tree/master/src-pileupCaller': 3,\n",
       " 'github.com/olavolav/TE-Causality': 1,\n",
       " 'github.com/contact-contagion/mrsa-c': 1,\n",
       " 'github.com/mcintyrelab': 1,\n",
       " 'github.com/theis188/theisen-plos-comp-bio': 1,\n",
       " 'kingaa.github.io/pomp': 3,\n",
       " 'github.com/jimhester/primerTree': 1,\n",
       " 'github.com/SamStudio8/goldilocks': 2,\n",
       " 'github.com/wikiselev/rnaseq.mcf10a': 1,\n",
       " 'middle-down.github.io/Software': 12,\n",
       " 'github.com/brwnj/umitools': 3,\n",
       " 'github.com/hesselberthlab/5OH': 1,\n",
       " 'github.com/hesselberthlab/trackhub': 1,\n",
       " 'github.com/JialiUMassWengLab/laSV/tree/master': 1,\n",
       " 'github.com/gyronaut/tomography': 1,\n",
       " 'workflowconversion.github.io': 1,\n",
       " 'github.com/kevinrue/shiny-MDM': 1,\n",
       " 'github.com/chiuyc/MAGIC': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/manual': 6,\n",
       " 'github.com/jnw29/AvrPto_Transcriptome': 1,\n",
       " 'github.com/joelewnard/choleraHaiti.git': 1,\n",
       " 'github.com/araksm/ceRNA': 1,\n",
       " 'github.com/l-modolo/TEtools': 4,\n",
       " 'github.com/hallowkm/RxODE': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cufflinks': 2,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cuffmerge': 3,\n",
       " 'github.com/FRED-2/OptiVac': 1,\n",
       " 'github.com/genome/bam-readcount': 32,\n",
       " 'github.com/aplested/aligator': 1,\n",
       " 'ComputationalProteomicsUnit.github.io/pRolocGUI': 1,\n",
       " 'github.com/alfbenpa/16S_MinION': 1,\n",
       " '1dl3.github.io/graph': 1,\n",
       " 'github.com/NDCcodes/Daut_ND-matlab-codes/tree/master': 1,\n",
       " 'github.com/GabeAl/NINJA-OPS': 1,\n",
       " 'github.com/nphynimhans/ANGEL': 1,\n",
       " 'github.com/BenKaehler/gapped': 2,\n",
       " 'github.com/acg-team/ProPIP': 1,\n",
       " 'github.com/LeonYenLeeChan/RNAdecay': 1,\n",
       " 'github.com/elifesciences-publications/RNAdecay': 1,\n",
       " 'github.com/aufrank/R-hacks/blob/master/mer-utils.R': 5,\n",
       " 'github.com/daniel-temko/CNVTiming': 1,\n",
       " 'github.com/schatzlab/genomescope': 2,\n",
       " 'www.github.com/tblazey/ogiMeta': 1,\n",
       " 'github.com/ChristianLabUIUC/BurstAnalysis': 1,\n",
       " 'github.com/bronsonweston/GMP-Modeling': 1,\n",
       " 'github.com/ebov/space-time/tree/master/Data/phyml_raxml_ML.sh': 1,\n",
       " 'github.com/alesssia/MAP': 1,\n",
       " 'github.com/krumsieklab/MissingValues': 1,\n",
       " 'github.com/waghsk/lvef-paper': 1,\n",
       " 'github.com/kmiller324/KymoFlow': 1,\n",
       " 'github.com/fcalafell/weighted_rho': 1,\n",
       " 'expressionanalysis.github.io/ea-utils': 27,\n",
       " 'openseadragon.github.io': 7,\n",
       " 'github.com/lme4/lme4/http://lme4.r-forge.r-project.org': 1,\n",
       " 'github.com/MikeAxtell/ShortStack/releases': 2,\n",
       " 'github.com/MikeAxtell/CleaveLand4/releases': 1,\n",
       " 'github.com/zhangtongda/Syntool': 1,\n",
       " 'github.com/gephi/gephi': 2,\n",
       " 'github.com/jinweiwu/perl/blob/master/extract': 1,\n",
       " 'github.com/owlcollab/owltools': 4,\n",
       " 'github.com/ekg/freebayes': 36,\n",
       " 'github.com/mdhumphries/AplysiaAttractorAnalysis': 1,\n",
       " 'github.com/elifesciences-publications/AplysiaAttractorAnalysis': 1,\n",
       " 'github.com/martcous/ppmi-study': 1,\n",
       " 'github.com/hyperic/sigar': 1,\n",
       " 'github.com/vivekbhr/Subread_to_DEXSeq': 1,\n",
       " 'github.com/denovogear': 1,\n",
       " 'github.com/ozagordi/VirMet/releases/tag/v0.3.3': 1,\n",
       " 'github.com/mestato/lab_code/tree/master/hwg_gssr_scripts': 3,\n",
       " 'github.com/vegandevs/vegan': 14,\n",
       " 'github.com/vcftools/vcftools': 2,\n",
       " 'github.com/microbiome/microbiome': 3,\n",
       " 'github.com/PacificBiosciences/Bioinformatics-Training/blob/master/barcoding/pacbio_384_barcodes.fasta': 2,\n",
       " 'github.com/chapmanb/bcbio-nextgen': 21,\n",
       " 'github.com/embruna/Costa-Atta-Seedlings': 1,\n",
       " 'github.com/lianaan/Perc_Var': 1,\n",
       " 'github.com/daquang/YAMDA': 1,\n",
       " 'github.com/PECAplus': 1,\n",
       " 'github.com/machenslab/dPCA': 6,\n",
       " 'github.com/gamaleldin/TME': 2,\n",
       " 'github.com/zalf-rpm/monica': 1,\n",
       " 'github.com/DSSAT/dssat-csm': 1,\n",
       " 'github.com/Methylflower/DMR-scan': 1,\n",
       " 'github.com/keras-team/keras': 7,\n",
       " 'github.com/mberacochea/sorghum-bicolor-M81E-16S': 1,\n",
       " 'github.com/jannahastings/mental-functioning-ontology': 1,\n",
       " 'github.com/jbowensruc/PerlScriptFrontiers': 1,\n",
       " 'github.com/pchrapka/phasereset': 1,\n",
       " 'github.com/lxchong/perimetry-custom-grids': 1,\n",
       " 'cole-trapnellab.github.io/cufflinks/papers': 1,\n",
       " 'github.com/OpenSourceBrain/GranCellSolinasEtAl10': 1,\n",
       " 'github.com/croessert/AnalyseGranCellRoessertEtAl14': 1,\n",
       " 'github.com/bastodian/shed/blob/master/Python/q-trim.py': 2,\n",
       " 'picrust.github.io': 3,\n",
       " 'github.com/FedericoComocglio/wavClusteR': 1,\n",
       " 'github.com/QBRC/PIPE-CLIP': 3,\n",
       " 'github.com/arq5x/lumpy-sv': 11,\n",
       " 'github.com/nickloman/entrezajax': 1,\n",
       " 'github.com/stamatak/Parsimonator-1.0.2': 3,\n",
       " 'cancerit.github.io/CaVEMan': 8,\n",
       " 'github.com/cogbio/Experimenter': 1,\n",
       " 'github.com/ThorntonLab/ESMtest': 1,\n",
       " 'github.com/MorrellLAB/Soja_Env_Association': 1,\n",
       " 'github.com/konradjk/loftee': 10,\n",
       " 'github.com/molgenis/systemsgenetics/tree/master/eqtl-mapping-pipeline': 1,\n",
       " 'github.com/ramhiser/datamicroarray/blob/master/README.md': 1,\n",
       " 'github.org/AEBilgrau/effadj': 1,\n",
       " 'github.com/danielfrg/tsne': 1,\n",
       " 'gist.github.com/coleoguy/6886683': 1,\n",
       " 'github.com/ucdavis-bioinformatics': 4,\n",
       " 'github.com/ibest/GRC_Scripts/blob/master/screen_duplicates_PE.py': 1,\n",
       " 'github.com/pintotm/PLoSOne2014': 1,\n",
       " 'github.com/dkoes/shapedb': 1,\n",
       " 'github.com/ComputationalMedicineLab/egpmrp': 2,\n",
       " 'github.com/micahgearhart/exotools': 1,\n",
       " 'github.com/wrightrc/flowTime': 1,\n",
       " 'github.com/charlesfarber/GWAS_networks': 1,\n",
       " 'github.com/Mark-Kramer/Spike-Ripple-Detector-Method': 1,\n",
       " 'github.com/PNNL-Comp-Mass-Spec/LIQUID': 1,\n",
       " 'github.com/shahmoradi/cordiv': 1,\n",
       " 'compomics.github.io/projects/peptide-shaker.html': 6,\n",
       " 'compomics.github.io/projects/searchgui.html': 3,\n",
       " 'github.com/Preparation-Publication-BD2K/db_compress': 1,\n",
       " 'github.com/LiuBioinfo/iMapSplice': 1,\n",
       " 'github.com/craigyk/emtools': 1,\n",
       " 'github.com/elifesciences-publications/emtools': 1,\n",
       " 'github.com/lauralewis/emergenceLPs': 1,\n",
       " 'github.com/elifesciences-publications/emergenceLPs': 1,\n",
       " 'github.com/toddleroux/health_prof_educ': 1,\n",
       " '/github.com/EACcodes/EmbeddingIntegralGenerator': 1,\n",
       " 'github.com/syubukun/Sp34': 1,\n",
       " 'github.com/FoodSafeCookCo/TwitterStream-': 1,\n",
       " 'github.com/rpremraj/mailR': 1,\n",
       " 'github.com/PacificBiosciences/Bioinformatics-Training/wiki/Circularizing-and-trimming': 4,\n",
       " 'uwpr.github.io/Lorikeet': 1,\n",
       " 'github.com/francescojm/CRISPRcleanR': 1,\n",
       " 'github.com/cancerit/pyCRISPRcleanR': 1,\n",
       " 'github.com/raerose01/deconstructSigs': 3,\n",
       " 'github.com/gcyuan/SCUBA': 2,\n",
       " 'github.com/jw156605/SLICER': 3,\n",
       " 'github.com/hubert-pop/signasel': 1,\n",
       " 'github.com/vickytorrance/uAUGs': 1,\n",
       " 'github.com/jordan-g/Irrelevance-by-Inhibition': 1,\n",
       " 'samtools.github.io/hts-specs/VCFv4.3.pdf': 3,\n",
       " 'github.com/therneau/survival': 5,\n",
       " 'github.com/epam/ketcher': 1,\n",
       " 'github.com/eiriniar/gleason_CNN': 1,\n",
       " 'github.com/MihChi/SIMELeD.git': 1,\n",
       " 'github.com/biopython/biopython.github.io': 3,\n",
       " 'czarrar.github.io/connectir': 3,\n",
       " 'wiki.github.com/asad/VFLib': 1,\n",
       " 'github.com/mailund/IMCoalHMM': 2,\n",
       " 'github.com/caulilin/lncRNA_Finder': 1,\n",
       " 'github.com/hilldr/Tsai_LGR5_endoderm_2015': 1,\n",
       " 'github.com/Victorian-Bioinformatics-Consortium/nesoni': 18,\n",
       " 'github.com/zacf/UM-gradient-metagenome-databases': 2,\n",
       " 'github.com/lskatz/lskScripts': 1,\n",
       " 'github.com/diazlab/scell': 1,\n",
       " 'github.com/Teichlab/celloline': 1,\n",
       " 'pachterlab.github.io/kallisto/about.html': 2,\n",
       " 'github.com/lengning/OEFinder': 3,\n",
       " 'github.com/PMBio/scLVM': 2,\n",
       " 'github.com/brentp/combat.py': 1,\n",
       " 'github.com/shka/R-SAMstrt': 2,\n",
       " 'github.com/MarioniLab/Deconvolution2016': 2,\n",
       " 'github.com/JustinaZ/pcaReduce': 2,\n",
       " 'lvdmaaten.github.io/tsne': 10,\n",
       " 'github.com/epierson9/ZIFA': 5,\n",
       " 'hms-dbmi.github.io/scde': 6,\n",
       " 'github.com/RGLab/MAST': 7,\n",
       " 'github.com/lanjiangboston/GiniClust': 1,\n",
       " 'github.com/govinda-kamath/clustering_on_transcript_compatibility_counts': 1,\n",
       " 'github.com/BatzoglouLabSU/SIMLR': 2,\n",
       " 'github.com/linnarsson-lab/BackSPIN': 5,\n",
       " 'cole-trapnell-lab.github.io/monocle-release': 7,\n",
       " 'github.com/kieranrcampbell/embeddr': 2,\n",
       " 'markummitchell.github.io/engauge-digitizer': 14,\n",
       " 'github.com/scipy/scipy/blob/v0.15.1/scipy/stats/stats.py': 1,\n",
       " 'github.com/hverbruggen/PMT': 1,\n",
       " 'wurmlab.github.io/tools/genevalidator': 1,\n",
       " 'github.com/FunctionLab/giant-api': 1,\n",
       " 'github.com/whcsu/RRotSF': 1,\n",
       " 'github.com/dg13/rasqual': 1,\n",
       " 'github.com/ChildMindInstitute/Biblio_Reader/blob/165ddc56779a5e55149184a0f95b7c14874cf0c5/biblio_reader/text_tools/text_tools.py': 1,\n",
       " 'github.com/abreschi/shinySpecClust': 1,\n",
       " 'github.com/isb-cgc/ISB-CGC-pipelines': 1,\n",
       " 'github.com/FernandoLab/2016_Paz_et_al_Dairy_Breeds': 1,\n",
       " 'github.com/pzylstra/ffm_cpp/tree/Zylstra2016': 1,\n",
       " 'github.com/VeraPancaldi/Heterosis-GRN-simulation': 1,\n",
       " 'github.com/sclamons/PED': 1,\n",
       " 'github.com/caballero/RepeatLandscape': 1,\n",
       " 'philogb.github.io/jit': 3,\n",
       " 'github.com/nh13/DWGSIM': 15,\n",
       " 'github.com/sbillin/IMLP': 1,\n",
       " 'github.com/aflaxman/dismod_pde': 1,\n",
       " 'github.com/craddm/microDetect': 1,\n",
       " 'github.com/minillinim/sam2fpkg': 1,\n",
       " 'github.com/dpessoaIGC/Dose-Invariant-Susceptibility-Estimator': 1,\n",
       " 'github.com/camwebb/tree-of-trees/blob/master/megatrees/R20120829.new': 2,\n",
       " 'github.com/brwnj/seqanno': 1,\n",
       " 'mustache.github.io': 1,\n",
       " 'github.com/biomedinfo/sugar': 1,\n",
       " 'github.com/OHDSI/KnowledgeBase': 2,\n",
       " 'github.com/hallamlab/MetaPathways': 1,\n",
       " 'github.com/UH-Bioinformatics/hBayeSSC': 1,\n",
       " 'github.com/penamiller/mmJ': 1,\n",
       " 'github.com/EliMor/LabScripts': 1,\n",
       " 'rdrr.io/github/ameztegui/Plasticity': 1,\n",
       " 'github.com/KiddLab/insertion-genotype': 1,\n",
       " 'github.com/MarcoDiSalvo90/RhoTermPredict': 1,\n",
       " 'github.com/LabForComputationalVision/matlabPyrTools': 1,\n",
       " 'github.com/santa-dev/santa-sim': 1,\n",
       " 'bbolker.github.io/mixedmodels-misc/glmmFAQ.html': 2,\n",
       " 'github.com/dcnieho/UDPMultiCast': 1,\n",
       " 'gemtools.github.io': 6,\n",
       " 'github.com/legrandp/xdsme': 6,\n",
       " 'github.com/reconhub/branchr': 2,\n",
       " 'github.com/OceanParcels/AntarcticPeninsulaPlastic': 1,\n",
       " 'github.com/roidtman/NatComm_dengue_China': 1,\n",
       " 'github.com/WashburnLab/Topological-score-TopS': 1,\n",
       " 'stnava.github.io/ANTsR': 7,\n",
       " 'github.com/linyuwangPHD/RNA-Secondary-Structure-Database': 1,\n",
       " 'github.com/benjamincjackson/dsim_variant_pipeline_ref_v2.02.git': 1,\n",
       " 'github.com/Dazzid/DataToRepovizz/tree/myo_to_repovizz/myo_recordings': 1,\n",
       " 'github.com/mitmedialab/BEEP': 2,\n",
       " 'github.com/rraadd88/beditor': 1,\n",
       " 'github.com/plaurent/mcpstdp': 1,\n",
       " 'github.com/PacificBiosciences': 6,\n",
       " 'github.com/deweylab/RSEM': 8,\n",
       " 'github.com/ChenFengling/HiCDB': 1,\n",
       " 'github.com/ChenFengling/RHiCDB': 1,\n",
       " 'github.com/weng-lab/Diff_signal_pred': 1,\n",
       " 'github.com/GiugnoLab/PanDelos': 1,\n",
       " 'github.com/ekg/vcflib': 13,\n",
       " 'github.com/Donders-Institute/Stereo-gaze-tracking': 1,\n",
       " 'github.com/bieneSchwarze/PaperModelingSelectionBias': 1,\n",
       " 'github.com/knowlabUnimelb/SMALLN': 1,\n",
       " 'github.com/gkichaev/PAINTOR_V3.0': 1,\n",
       " 'github.com/FormentoEmanuele/MuscleSpindleCircuitsModel.git': 1,\n",
       " 'github.com/plhdb/plhdb': 1,\n",
       " 'github.com/tgouhier/biwavelet': 7,\n",
       " 'github.com/lsy917/chenlimin': 1,\n",
       " 'github.com/obarquero/ECG_quality': 1,\n",
       " 'samtools.github.io/htsjdk': 5,\n",
       " 'github.com/gbouvignies/chemex': 5,\n",
       " 'github.com/sandain/pigeon': 3,\n",
       " 'github.com/harrelfe/Hmisc': 5,\n",
       " 'gist.github.com/kkmehta/0a1dbe272d7ff2aed011': 1,\n",
       " 'github.com/pezmaster31/bamtools': 18,\n",
       " 'github.com/mlizhangx/TCR-3D': 1,\n",
       " 'lp2.github.io/yeast-crispri': 4,\n",
       " 'spineml.github.io': 3,\n",
       " 'github.com/dvbrown/FijiMacros': 1,\n",
       " 'github.com/m-fofana/TB-PZA-model.git': 1,\n",
       " 'www.github.com/nedelec/cytosim': 1,\n",
       " 'github.com/sylvainforet/libngs': 5,\n",
       " 'github.com/sccn/SNAP': 3,\n",
       " 'pachterlab.github.io/sleuth': 4,\n",
       " 'github.com/jakejh/metapredict': 1,\n",
       " 'github.com/jdreyf/fat-exosome-microrna': 1,\n",
       " 'github.com/kdeck/TTVFast': 1,\n",
       " 'github.com/ericagol/TTVFaster': 1,\n",
       " 'github.com/smirik/mercury': 1,\n",
       " 'github.com/dylan-jayatilaka/tonto': 2,\n",
       " 'github.com/guigolab/chip-nf': 1,\n",
       " 'github.com/adwanner/PyKNOSSOS': 6,\n",
       " 'github.com/cole-trapnell-lab/monocle-release': 2,\n",
       " 'github.com/VRam142/combinatorialHiC': 1,\n",
       " 'opencobra.github.io/cobratoolbox': 5,\n",
       " 'github.com/maanrachid/Codes/blob/master/AAPSP.tar': 1,\n",
       " 'github.com/biochem-fan/eiger2cbf': 1,\n",
       " 'geodacenter.github.io': 4,\n",
       " 'github.com/owensgl/sunflower_heterosis_2018': 1,\n",
       " 'github.com/TransDecoder': 7,\n",
       " 'github.com/NCBI-Hackathons/NovoGraph': 1,\n",
       " 'github.com/LipengNing/RBF-Propagator': 3,\n",
       " 'github.com/CCBR/Pipeliner': 3,\n",
       " 'github.com/phagewebufpa/API': 1,\n",
       " 'github.com/lichen-lab/SICS': 1,\n",
       " 'github.com/mariFelix/deconvoNorm': 1,\n",
       " 'rescience.github.io': 2,\n",
       " 'github.com/bslaby/scripts': 3,\n",
       " 'github.com/zhangyimc/DeepSplice': 1,\n",
       " 'github.com/bcbio/bcbio-nextgen': 3,\n",
       " 'github.com/stamatak/standard-RAxML': 17,\n",
       " 'github.com/Pbdas/epa-ng': 1,\n",
       " 'github.com/CshlSiepelLab/LINSIGHT': 3,\n",
       " 'github.com/cancersysbio/BreastCancerITH': 1,\n",
       " 'github.com/alexandrebarachant/covariancetoolbox': 2,\n",
       " 'github.com/philipwijesinghe/Acoustic-trap-LSFM': 1,\n",
       " 'github.com/wkool/demandavoidance': 1,\n",
       " 'github.com/dunfieldlab/mg_wrapser': 2,\n",
       " 'github.com/jsalojar/DLCpar_visualize': 1,\n",
       " 'github.com/jaak-s/BayesianDataFusion.jl': 1,\n",
       " 'github.com/smithng1215': 1,\n",
       " 'github.com/jacoblevine/PhenoGraph': 3,\n",
       " 'github.com/AllonKleinLab/SPRING': 2,\n",
       " '4github.com/CMU-Perceptual-Computing-Lab/openpose': 1,\n",
       " 'github.com/Onogi/MultivariateFtest': 1,\n",
       " 'github.com/Xiaomu-Shan/Experiments-of-iFBEL': 1,\n",
       " 'github.com/BioinfoCP/visual-features-soybean-vigor': 1,\n",
       " 'github.com/dlrice/evoker-lite': 1,\n",
       " 'github.com/uricchio/PASTEL': 1,\n",
       " 'github.com/gadluru/Multiview-myocardial-perfusion-with-radial-SMS': 1,\n",
       " 'github.com/svijayakumar32/multi-objective-optimization-of-gut-community-metabolic-model': 1,\n",
       " 'github.com/akiyamalab/NRLMFb': 1,\n",
       " 'github.com/BHSAI/DNN-liverTox': 1,\n",
       " 'github.com/guofei-tju/Cancer-subtypes': 1,\n",
       " 'mzmine.github.io': 12,\n",
       " 'github.com/JoseAlanis/Supplementary_Soc_ERN': 1,\n",
       " 'github.com/eshoko/COMET': 1,\n",
       " 'github.com/madeluis/GENIST/commits/master': 1,\n",
       " 'github.com/YinLiLin/R-CMplot': 3,\n",
       " 'github.com/costlysignalling/Mate-choice-consistency-2': 1,\n",
       " 'github.com/stonezwr/SpiKL-IP': 1,\n",
       " 'github.com/JeromeAmbroise/Pathogenomics': 1,\n",
       " 'pachterlab.github.io/eXpress/index.html': 1,\n",
       " 'github.com/RausellLab/NCBoost': 1,\n",
       " 'xia2.github.io/index.html': 1,\n",
       " 'github.com/KrisJanssen/QuickCLEM': 1,\n",
       " 'github.com/delibasanta/Mexican-salamanders.git': 1,\n",
       " 'github.com/jeffeaton/currpreg-hiv-trends': 1,\n",
       " 'github.com/ANGSD': 1,\n",
       " 'github.com/mourisl/Lighter': 3,\n",
       " 'github.com/aestrivex/cvu/wiki': 1,\n",
       " 'cytoscape.github.io/cytoscape.js': 16,\n",
       " 'github.com/OscarKrijgsman/focalCall': 1,\n",
       " 'github.com/ahhurlbert/species-energy-simulation': 1,\n",
       " 'github.com/popeb/MCP05': 1,\n",
       " 'demotu.github.io/posts/prediction-ellipse-ellipsoid.html': 1,\n",
       " 'github.com/emarcos/beta_sheet': 1,\n",
       " 'github.com/leonfrench/HumanHabenula': 1,\n",
       " 'ccbr.github.io/Pipeliner/Tools/NGS_QC_Toolkit.html': 2,\n",
       " 'github.com/LCCN': 1,\n",
       " '1dstolz.github.io/epsych': 1,\n",
       " 'github.com/BlueFern/CoupledCells/releases/tag/v-4.0': 1,\n",
       " 'github.com/seaaroundus': 2,\n",
       " 'github.com/marliekevk/Integrating-Educational-Knowledge': 1,\n",
       " 'pachterlab.github.io/eXpress': 1,\n",
       " 'github.com/mlloyd23/SSWD_16S_analysis': 1,\n",
       " 'github.com/nbluoc/kk': 1,\n",
       " 'github.com/bioinformatics-sannio/lncrna-homologs': 1,\n",
       " 'github.com/OHDSI/StudyProtocols/tree/master/AhasHfBkleAmputation': 1,\n",
       " 'github.com/FernandoHartwig/AssortativeMating_Scripts': 1,\n",
       " 'github.com/hms-dbmi/spp': 2,\n",
       " 'github.com/nci/drishti': 1,\n",
       " 'github.com/perishky/meffil': 4,\n",
       " 'github.com/willyrv/nssc-tools': 1,\n",
       " 'github.com/dvera/ksift': 1,\n",
       " 'github.com/zippav/Jellyfish-2': 1,\n",
       " 'github.com/laurentnoe/yass': 1,\n",
       " 'github.com/shimbalama/screen_assembly': 1,\n",
       " 'github.com/katholt/srst2/tree/master/data': 2,\n",
       " 'github.com/TheWrightonLab': 1,\n",
       " 'github.com/joegage/GWAS_AUC': 1,\n",
       " 'github.com/danoreper/mnp2018.git': 1,\n",
       " 'github.com/Zhongyun-Huang/G3-2018-200605': 1,\n",
       " 'github.com/anainesvs/glioblastoma': 1,\n",
       " 'github.com/drh-uth/DataFakehouse': 1,\n",
       " 'github.com/jacobhurst/definetherain': 1,\n",
       " 'github.com/qichao1984/GSMer': 1,\n",
       " 'github.com/JulienLajugie/ReplicationTimingSimulation': 1,\n",
       " 'github.com/ofedrigo/pyEnrichment': 2,\n",
       " 'dschrimpf.github.io/randi3': 1,\n",
       " 'github.com/mlupien/ABC': 1,\n",
       " 'github.com/jonsonfox/HaploShare': 1,\n",
       " 'www.github.com/molgenis': 1,\n",
       " 'github.com/PF2-pasteur-fr/SARTools': 4,\n",
       " 'github.com//tumorml/TumorML-schema': 1,\n",
       " 'cytoscape.github.io': 1,\n",
       " 'madsalbertsen.github.io/mmgenome': 6,\n",
       " 'github.com/jstjohn/SimSeq': 12,\n",
       " 'github.com/srnas/barnaba': 7,\n",
       " 'github.com/SEEG-Oxford/prioritisation': 1,\n",
       " 'github.com/StoreyLab/gcat': 1,\n",
       " 'github.com/rachelss/SISRS/releases': 1,\n",
       " 'github.com/GriesbeckLab/ColonySelection': 1,\n",
       " 'github.com/GraceYoon/ISIS_EN': 1,\n",
       " 'github.com/cubocube/website': 1,\n",
       " 'github.com/svazzole/sparsevar': 1,\n",
       " 'github.com/DynaSim': 1,\n",
       " 'github.com/jsherfey/ACd_model': 1,\n",
       " 'github.com/knightlab-analyses/bloom-analyses': 1,\n",
       " 'github.com/tseemann/wombac/blob/master/bin/wombac-shred_fasta': 1,\n",
       " 'github.com/tonomartin2/MITOCLASS.1': 2,\n",
       " 'github.com/DenefLab/microcystis-oligotypes': 1,\n",
       " 'github.com/NanoSPD/NanoSPD': 1,\n",
       " 'github.com/iontorrent/TS/tree/master/Analysis/TMAP': 9,\n",
       " 'mi-sim.github.io': 1,\n",
       " 'github.com/ClandininLab/2pstim': 1,\n",
       " 'github.com/Ensembl/treebest': 4,\n",
       " 'github.com/apalania/consMoment': 1,\n",
       " 'github.com/dsarov/MGAP---Microbial-Genome-Assembler-Pipeline': 3,\n",
       " 'github.com/tseemann/VelvetOptimiser': 10,\n",
       " 'github.com/grev-uv/hpg-methyl': 1,\n",
       " 'github.com/PacificBiosciences/pbdagcon': 8,\n",
       " 'github.com/PacificBiosciences/FALCON-integrate': 6,\n",
       " 'github.com/grisoniFr/whales': 1,\n",
       " 'github.com/dekkerlab/cworld-dekker': 10,\n",
       " 'github.com/drpowell/degust': 3,\n",
       " 'github.com/SUNCAT-Center/catmap': 1,\n",
       " 'www.github.com/nextgenusfs/funannotate': 1,\n",
       " 'github.com/EricFournier3/PythonGenomicTools': 2,\n",
       " 'github.com/mw55309/ideel': 2,\n",
       " 'github.com/toreerdmann/edar': 1,\n",
       " 'github.com/alanavc/rodtracker': 1,\n",
       " 'github.com/AgreementMakerLight': 1,\n",
       " ...}"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict(url_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_urls = [k for k,v in url_counts.items() if v==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_url_filt = np.isin(urls_in_papes.url, unique_urls)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "#urls_in_papes.loc[unique_url_filt, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids_to_check = np.sort(np.unique(urls_in_papes.pmcid[unique_url_filt].values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['2712337', '2795760', '2820491', ..., '6484778', '6485367',\n",
       "       '6485654'], dtype=object)"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids_to_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+\n",
    "\n",
    "All our code is hosted on Github (http://github.com/intermine/) and contributions are welcome.\n",
    "\n",
    "The raw data and complete source code to reproduce the results of the analysis is freely available at https://github.com/akastrin/kastrin2014large.\n",
    "\n",
    "```\n",
    "{'text': '2Software available at https://github.com/vellamike/optimizer, online documentation at http://optimizer.readthedocs.org/',\n",
    "   'offset': 67424,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'footnote', 'section_type': 'CONCL'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}\n",
    "   \n",
    "{'text': '– To handle the RRF syntax we have developed the UMLS2RDF project.1 UMLS2RDF is the set of scripts that connect to the UMLS MySQL release and transforms its content into RDF triples.',\n",
    "   'offset': 17722,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}   \n",
    "{'text': '           https://github.com/ncbo/umls2rdf         ',\n",
    "   'offset': 20138,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'footnote', 'section_type': 'CONCL'},\n",
    "   'sentences': [],\n",
    "   'annotations': []},\n",
    "   \n",
    "{'passages': [{'text': 'Open-Source Syringe Pump Library',\n",
    "   'offset': 0,\n",
    "   'relations': [],\n",
    "   'infons': {'name_2': 'surname:Anzalone;given-names:Gerald C.',\n",
    "    'license': 'This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.',\n",
    "    'name_0': 'surname:Wijnen;given-names:Bas',\n",
    "    'name_1': 'surname:Hunt;given-names:Emily J.',\n",
    "    'alt-title': 'Open-Source Syringe Pump Library',\n",
    "    'article-id_publisher-id': 'PONE-D-14-11249',\n",
    "    'name_3': 'surname:Pearce;given-names:Joshua M.',\n",
    "    'type': 'front',\n",
    "    'elocation-id': 'e107216',\n",
    "    'article-id_doi': '10.1371/journal.pone.0107216',\n",
    "    'section_type': 'TITLE',\n",
    "    'article-id_pmid': '25229451',\n",
    "    'volume': '9',\n",
    "    'article-id_pmc': '4167991',\n",
    "    'year': '2014',\n",
    "    'title': 'Data Availability',\n",
    "    'name_4': 'surname:Gilestro;given-names:Giorgio F.',\n",
    "    'issue': '9',\n",
    "    'notes': 'The authors confirm that all data underlying the findings are fully available without restriction. All files needed to construct the device are held at https://github.com/mtu-most/linear-actuator All data for the paper presented within it.'},\n",
    "```\n",
    "\n",
    "To foster reproducibility and fast development of future work based upon these results, we have published functional graph analysis tools under an open source, GPLv3 license, available here: https://github.com/ssgrn/GraphInvariantsNeocortex.'\n",
    "\n",
    "Software and user manual are available at http://bioinformaticstools.mayo.edu/research/patterncnv/, and R package at https://github.com/topsoil/patternCNV/.\n",
    "\n",
    "\n",
    "Availability and implementation: The software for PatternCNV is implemented using Perl and R, and can be used in Mac or Linux environments. Software and user manual are available at http://bioinformaticstools.mayo.edu/research/patterncnv/, and R package at https://github.com/topsoil/patternCNV/.\n",
    "\n",
    "\n",
    "To quantify the number of c-Fos positive and β-gal positive cells, the red (c-Fos) and green (β-gal) color channels in each image were first filtered with a stringent threshold (mean +2 standard deviations of background pixel intensity level), and each color channel was then converted to a binary, black/white (BW) image using ImageBWconvertGUI- a custom-made program running in the 2013a Matlab Computing Environment with the Image Processing Toolbox (The MathWorks, Natick, MA; program available on Github: https://github.com/neuropil/ImageBWconvert/).\n",
    "\n",
    "In addition to quantification of the number of c-Fos and β-gal positive cell bodies, we also quantified, across different experimental conditions, the number and percentage of pixels within each nTS polygon with a pixel intensity value that exceeded 2 standard deviations for each stain, which includes both histological-positive cell bodies as well as associated cellular morphology (e.g., dendrites and axons) using ROIImageAnalysis - another custom-made MATLAB program (available on Github: https://github.com/neuropil/ROIImageAnalysis/).\n",
    "\n",
    "The complete R scripts to reproduce the analysis can be downloaded from http://sgibb.github.io/Culicoides/.\n",
    "\n",
    "Project home page:ftp://ftp.ebi.ac.uk/pub/databases/chembl/text-mining, and https://github.com/chembl/chembl_literature_classifier\n",
    "\n",
    "Proteome clustering results and matrix, scripts, and sequences alignments are available from GitHub (https://github.com/ocisse/Pneumocystis_comparative, last accessed July 27, 2014).\n",
    "\n",
    "All the figures for this article were produced using matplotlib, and figure scripts are available from https://github.com/rougier/ten-rules.\n",
    "\n",
    "New scripts written for this protocol, an example data set, and any future updates are available at https://github.com/listonlab/.\n",
    "\n",
    "A copy of the macro code is available as open source freeware; https://github.com/graemeball/ij_scripts/blob/master/Macros/Sum_Masked_Signal.ijm.\n",
    "\n",
    "The code is also available online at http://github.com/uygarsumbul/rgc.\n",
    "\n",
    "Source code repository URL: http://github.com/CovertLab/WholeCellSimDB\n",
    "\n",
    "The raw count data, script, script output, addition sample metadata, and saved Cytoscape session are available (https://github.com/brwnj/interaction_network).\n",
    "\n",
    "The R code used for all of the analysis has been made available on an open source basis (https://github.com/SEEG-Oxford/ebola_zoonotic).\n",
    "\n",
    "The data and source code used to generate this table are provided through http://jbloom.github.io/phyloExpCM/example_2014Analysis_lactamase.html (last accessed July 28, 2014).\n",
    "\n",
    "Code developed for this project, as well as issue trackers and full documentation for deploying the software are all available on github at https://github.com/apfejes/epigenetics-software.\n",
    "\n",
    "Source code can be freely obtained from https://github.com/bps10/color/tree/JOSA.\n",
    "\n",
    "The R package can be downloaded from our website (http://genemed.uchicago.edu/~pgeeleher/pRRophetic) or GitHub (https://github.com/paulgeeleher/pRRophetic).\n",
    "\n",
    "Phages genes were analyzed using Phamerator , an open-source program (GNU general public license) designed to compare phage genes and genomes. For this study, Phamerator was adapted and stored in a GitHub repository ( http://github.com/byuphamerator/phamerator-dev) separate from the original version.\n",
    "\n",
    "A reference implementation of SILP2 is provided at https://github.com/jim-bo/silp2.\n",
    "\n",
    "The code used for generating most of the reported results, including the implementation of the ALN50 metric, is available at https://github.com/jim-bo/scafathon.\n",
    "\n",
    "ARYANA with complete source code can be obtained from http://github.com/aryana-aligner\n",
    "\n",
    "\n",
    "\n",
    "-\n",
    "\n",
    "To process huge numbers of queries in parallel in the background, we use the Resque library (https://github.com/resque/resque).\n",
    "\n",
    "Networks were constructed using the online application Population Graphs v2 (http://dyerlab.github.io/popgraph/. Accessed 2014 June 16), and the analyses were performed with the software Genetic Studio.\n",
    "\n",
    "The phylogenetic analyses were performed using the software package phyloExpCM (phylogenetic analyses with experimental codon models, https://github.com/jbloom/phyloExpCM, last accessed July 28, 2014), which primarily serves as an interface to run HYPHY.\n",
    "\n",
    "The PyJade module (https://github.com/SyrusAkbary/pyjade) is used to render Jade templates into HTML to generate the interface.\n",
    "\n",
    "```\n",
    "{'text': 'MGFLI. count_genes_per_transcript.pl. https://github.com/MGFLI/RNAseq/tree/master',\n",
    "   'offset': 25770,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ARYANA with complete source code can be obtained from http://github.com/aryana-aligner',\n",
       " 'The experiments were performed on a platform with 48 AMD Opteron Processor 6174 CPUs each having 12 cores with clock speed of 2.2 GHz. The hg19 human genome assembly was used as the reference for all test cases. We used dwgsirn (https://github.com/nh13/DWGSIM/wiki) to simulate data sets similar to real reads produced by Illumina NGS platforms.']"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t['text'] for t in pmcid_to_doc[ids_to_check[521]]['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4159173'"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids_to_check[505]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'passages': [{'text': 'ILP-based maximum likelihood genome scaffolding',\n",
       "   'offset': 0,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Măndoiu;given-names:Ion',\n",
       "    'name_3': 'surname:Zelikovsky;given-names:Alex',\n",
       "    'name_0': 'surname:Lindsay;given-names:James',\n",
       "    'name_1': 'surname:Salooti;given-names:Hamed',\n",
       "    'license': 'This is an Open Access article distributed under the terms of the Creative Commons Attribution License (http://creativecommons.org/licenses/by/4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original work is properly cited. The Creative Commons Public Domain Dedication waiver (http://creativecommons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated.',\n",
       "    'article-id_publisher-id': '1471-2105-15-S9-S9',\n",
       "    'issue': 'Suppl 9',\n",
       "    'article-id_doi': '10.1186/1471-2105-15-S9-S9',\n",
       "    'section_type': 'TITLE',\n",
       "    'article-id_pmid': '25253180',\n",
       "    'lpage': 'S9',\n",
       "    'article-id_pmc': '4168704',\n",
       "    'fpage': 'S9',\n",
       "    'year': '2014',\n",
       "    'type': 'front',\n",
       "    'volume': '15'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Background',\n",
       "   'offset': 48,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract_title_1', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interest in de novo genome assembly has been renewed in the past decade due to rapid advances in high-throughput sequencing (HTS) technologies which generate relatively short reads resulting in highly fragmented assemblies consisting of contigs. Additional long-range linkage information is typically used to orient, order, and link contigs into larger structures referred to as scaffolds. Due to library preparation artifacts and erroneous mapping of reads originating from repeats, scaffolding remains a challenging problem. In this paper, we provide a scalable scaffolding algorithm (SILP2) employing a maximum likelihood model capturing read mapping uncertainty and/or non-uniformity of contig coverage which is solved using integer linear programming. A Non-Serial Dynamic Programming (NSDP) paradigm is applied to render our algorithm useful in the processing of larger mammalian genomes. To compare scaffolding tools, we employ novel quantitative metrics in addition to the extant metrics in the field. We have also expanded the set of experiments to include scaffolding of low-complexity metagenomic samples.',\n",
       "   'offset': 59,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Results',\n",
       "   'offset': 1176,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract_title_1', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'SILP2 achieves better scalability throughg a more efficient NSDP algorithm than previous release of SILP. The results show that SILP2 compares favorably to previous methods OPERA and MIP in both scalability and accuracy for scaffolding single genomes of up to human size, and significantly outperforms them on scaffolding low-complexity metagenomic samples.',\n",
       "   'offset': 1184,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Conclusions',\n",
       "   'offset': 1542,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract_title_1', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Equipped with NSDP, SILP2 is able to scaffold large mammalian genomes, resulting in the longest and most accurate scaffolds. The ILP formulation for the maximum likelihood model is shown to be flexible enough to handle metagenomic samples.',\n",
       "   'offset': 1554,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Introduction',\n",
       "   'offset': 1794,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'De novo genome assembly is one of the best studied problems in bioinformatics. Interest in the problem has been renewed in the past decade due to rapid advances in high-throughput sequencing (HTS) technologies, which have orders of magnitude higher throughput and lower cost compared to classic Sanger sequencing. Indeed, top-of-the-line instruments from Illumina and Life Technologies are currently able to generate in a single run billions of reads with an aggregate length of hundreds of gigabases, at a cost of mere cents per megabase. However, most HTS technologies generate relatively short reads, significantly increasing the computational difficulty of the assembly problem. Despite much work on improved assembly algorithms for HTS shotgun reads, de novo assembly remains challenging, often resulting in highly fragmented assemblies, see for recent reviews and benchmarking results. For example, the recent Assemblathon 2 community effort to benchmark de novo genome assemblers shows that the performance of evaluated assemblers is highly variable from dataset to dataset and generally degrades with the complexity of the sample.',\n",
       "   'offset': 1807,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'To increase the utility of such fragmented assemblies, additional long-range linkage information is typically used to orient, order, and link contigs into larger structures referred to as scaffolds. Although long-range linkage information can be generated using a variety of technologies, including Sanger sequencing of both ends of cloned DNA fragments of up to hundreds of kilobases, Pacific Biosciences reads of up to tens of kilobases, and optical maps, the most common type of data used in scaffolding are HTS read pairs generated from DNA fragments with length ranging between hundreds of bases to tens of kilobases. While HTS read pairs are relatively easy to generate, the linkage information they provide is noisy due to library preparation artifacts and erroneous mapping of reads originating from repeats. The general scaffolding problem is known to be computationally NP-hard when linkage data contains errors. Moreover, the associated contig orientation and contig ordering problems are intractable as well: the orientation problem is equivalent to finding a maximum bipartite subgraph, whereas the ordering problem is similar to the Optimal Linear Arrangement problem, both of which are NP-hard. Due to the intractability of the problem, greedy heuristics have been employed in practical scaffolding methods such as. Scaffolding methods such as SOPRA reduce the size of the problem by iteratively removing inconsistent links and contigs, while MIP heuristically partitions the biconnected components of the scaffolding graph when they are too large to scaffold optimally by mixed integer programming. In SLIQ, inequalities are derived from the geometry of contigs to predict the orientation and ordering of adjacent contigs. To find a feasible solution with minimum read pair inconsistency, OPERA provides a novel dynamic programming algorithm.',\n",
       "   'offset': 2946,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Algorithms based on explicit statistical models are currently gaining popularity in the area of genome assembly, with notable advances in the use of maximum likelihood (ML) methods for both contig assembly and assembly evaluation. In this paper we introduce a highly scalable algorithm based on likelihood maximization for the scaffolding problem. The key step in our algorithm is the selection of contig orientations and a set of read pairs consistent with these orientations (and locally consistent with each other) such that the overall likelihood of selected pairs is maximized. As in previous works, the likelihood model we employ assumes independence of the HTS read pairs. The currently implemented model takes into account read mapping uncertainty due to overlap with annotated contig repeats as well as variations in contig coverage. The model can be easily extended to incorporate sequencing errors and the distribution of insert lengths; currently we only use the latter to eliminate read pairs with highly discordant insert length lower-bounds and to compute ML estimates for the final gap lengths. Likelihood maximization is formulated as an integer linear program (ILP). Unlike MIP, our ILP formulation selects contig orientations and a set of locally consistent read pairs but neither explicitly orders the contigs nor fully guarantees global consistency of selected pairs. The latter are achieved by decomposing the set of selected read pairs into linear paths via bipartite matching.',\n",
       "   'offset': 4805,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Scalability of our algorithm, referred to as SILP2, is achieved by adopting a non-serial dynamical programming (NSDP) approach. Rather than solving one large ILP, several smaller ILPs can be solved seperately and composed to find the complete and optimal solution. The order in which the smaller ILPs are solved is determined by the 3-connected components of the underlying scaffolding graph, which can be efficiently identified in linear time via the SPQR-tree data structure.',\n",
       "   'offset': 6306,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Compared to the preliminary version of the algorithm published in, referred to as SILP1, SILP2 is based on explicit formalization of likelihood maximization as the optimization objective. We present experiments with several likelihood models capturing read mapping uncertainty and/or non-uniformity of contig coverage. SILP2 also achieves higher scalability by using a more efficient NSDP algorithm than SILP1. This greatly reduces the need for heuristics such as the hierarchical scaffolding approach of SILP1, whereby scaffolding is performed by progressively decreasing the minimum bound on the size of read pair bundles. We have also expanded the set of experiments to include scaffolding of low-complexity metagenomic samples. The results show that SILP2 compares favorably to previous methods OPERA and MIP in both scalability and accuracy for scaffolding single genomes of up to human size, and significantly outperforms them on scaffolding low-complexity metagenomic samples.',\n",
       "   'offset': 6784,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Methods',\n",
       "   'offset': 7768,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Given a set of contigs C and a set of read pairs R, the scaffolding problem asks for the most likely orientation of the contigs along with a partition of the contigs into ordered sets connected by read pairs of R. The main steps of the SILP2 algorithm are as follows (see Figure 1 for a high-level flowchart). We first map the read onto contigs using Bowtie2, disregarding pairing information in the mapping process. Alignments are processed to extract read pairs for which both reads have unique alignments, and the alignments are onto distinct contigs. A scaffolding graph is then constructed with nodes corresponding to contigs and edges corresponding to extracted read pairs. The scaffolding graph is partitioned into 3-connected components using the SPQR tree data structure implemented in OGDF. The maximum-likelihood contig orientation is formulated as an ILP that is efficiently solved by applying non-serial dynamic programming based the SPQR tree data structure. Next, scaffold chains are extracted from the ILP solution by using bipartite matching and breaking remaining cycles (see Section S1 in Additional file 1). Finally, maximum likelihood estimates for the gap lengths are obtained using quadratic programming (see Section S2 of Additional file 1). Below we detail the key steps of the algorithm, including scaffolding graph construction, the maximum likelihood models used for contig orientation and mapped read pair probability estimation, then we briefly overview the orientation, the ILP formulation and the improved NSDP algorithm for efficiently solving the ILP.',\n",
       "   'offset': 7776,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'SILP2 Flowchart.',\n",
       "   'offset': 9362,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F1',\n",
       "    'file': '1471-2105-15-S9-S9-1.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Scaffolding graph. The scaffolding problem is modeled with a scaffolding graph G = (V, E), where each node i ε V represents a contig and each edge (i, j) ε E represents all read pairs whose two individual reads are mapped to the contigs i and j, respectively. Each read in a pair is aligned either to the forward or reverse strand of corresponding contig sequence, and this results in 4 possible configurations for a read pair (denoted A, B, C, or D, see Figure S1 in Additional file 1) which can be modeled as a bidirected edge. Orientation of contigs and the bidirected orientation of edges should agree (be concordant) with each other and should not result in any directed cycles for linear genomes (e.g. eukaryotes).',\n",
       "   'offset': 9379,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Maximum likelihood scaffold graph orientation. As an intermediate step towards solving the scaffolding problem, we consider the problem of determining an orientation of the scaffolding graph, which includes choosing one of the two possible orientations for each node (contig) i ε V as well as choosing for each edge (i, j) ε E one of the four bidirections that is concordant with the orientations of i and j. A common way to reduce an inference problem to an optimization problem is to seek a feasible solution with maximum likelihood. Let each observation, i.e., aligned read pair r ε R, have a probability pr of being correct. Any feasible contig orientation O = O(C) either agrees or disagrees with the read pair r. Let RO be the set of read pairs agreeing with O. Assuming independence of observations, the likelihood of an orientation O can be written as',\n",
       "   'offset': 10112,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'and hence its log-likelihood is . Since the first sum does not depend on the orientation O, maximizing the log-likelihood is equivalent to maximizing',\n",
       "   'offset': 10990,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'over all contig orientations O.',\n",
       "   'offset': 11140,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"Mapping probability estimation. If pr 's are assumed to be the same for all read pairs, then the objective (1) reduces to maximization of the number of read pairs that agree with the contig orientation O. We consider the following factors that reduce the probability pr that read pair r is aligned correctly:\",\n",
       "   'offset': 11172,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '1. Overlap with repeats. As noted above, only pairs for which both reads map uniquely to the set of contigs are used for scaffolding. Still, a read that fully or partially overlaps a genomic repeat may be uniquely mapped to the incorrect location in case repeat copies are collapsed. We preprocess contigs to annotate repeats from known repeat families and by recording the location of multimapped reads. An estimate of the repeat-based mapping probability  is found by taking the percentage of bases of r aligned to non-repetitive portions of the contigs.',\n",
       "   'offset': 11481,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '2. Contig coverage dissimilarity. Although sequencing coverage can have significant departures from uniformity due to biases introduced in library preparation and sequencing, the average coverage of adjacent contigs is expected to be similarly affected by such biases (all read alignments, including randomly allocated non-unique alignments, are used for estimating computing average contig coverages). If the two reads of r map to contigs i, respectively j, the coverage-based mapping probability of r, , is defined as 1 − |coveragei − coveragej |/(coveragei + coveragej).',\n",
       "   'offset': 12038,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Note that factors such as repeat content of the sequenced genome and sequencing depth will determine how informative repeat-based and coverage-based mapping probabilities are. Depending on these factors, either , , or their product may provide the most accurate estimate for pr. Mismatches and indels in read alignments, that can be caused by sequencing errors or polymorphisms in the sequenced sample, can easily be incorporated in the estimation of mapping probabilities.',\n",
       "   'offset': 12616,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Integer linear program. Our integer linear program maximizes the log-likelihood of scaffold orientation using the following boolean variables:',\n",
       "   'offset': 13090,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"- a binary variable Si for each contig i, with Si equal to 0 if the contig's orientation remains the same and Si = 1 if the contig's orientation is flipped w.r.t. default orientation in the final scaffold.\",\n",
       "   'offset': 13233,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '- a binary variable Sij for each edge (i, j) ε E, which equals 0 if none or both ith and jth contigs are flipped, and equals 1 if only one of them is flipped.',\n",
       "   'offset': 13439,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '- binary variables Aij (respectively, Bij , Cij , and Dij) which are set to 1 if and only if an edge in state A (respectively, B, C, or D) is used to connect contigs i and j (see Figure S1 in Additional file 1). For any contig pair i and j, at most one of these variables can be one.',\n",
       "   'offset': 13604,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Let  (respectively, ,  or ) denote the set of read pairs supporting state A (respectively, B, C, or D), between the ith and jth contig. Define the constant  by',\n",
       "   'offset': 13888,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'with ,  and  defined analogously.',\n",
       "   'offset': 14048,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We now ready to formulate the ILP for maximizing the log-likelihood of a scaffold orientation:',\n",
       "   'offset': 14082,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'where',\n",
       "   'offset': 14177,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"In this ILP, constraints (3-5) enforce agreement between contig orientation variables Si's and edge orientation variables Sij's, Aij's, Bij's, Cij's, and Dij's.\",\n",
       "   'offset': 14183,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Since eukaryotic genomes are linear, a valid scaffold orientation should not contain any cycles. The constraints (5) already forbid 2-cycles. Additionally, 3-cycles are forbidden with the constraints shown in Figure S2 in Additional file 1. Larger cycles generated in the ILP solution are broken heuristically because it is infeasible to forbid all of them using explicit constraints.',\n",
       "   'offset': 14344,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Non-serial dynamic programming. For large mammalian genomes, the number of variables and constraints is too large for solving the ILP (2)-(5) via standard solvers (SILP2 uses CPLEX which is available free of charge for academic institutions). We adopt the non-serial dynamic programming (NSDP) paradigm to overcome this barrier and to optimally solve the problem. NSDP is based on the interaction graph with nodes corresponding to ILP variables and edges corresponding to the ILP constraints - two nodes are adjacent in the interaction graph if their associated variables appear together in the same constraint. Through the NSDP process, variables are removed in the way that adjacent vertices can be merged together. The first step in NSDP is identifying weakly connected components of the interaction graph. We find the 2- and 3-connected components of the interaction graph with efficient algorithms and then we solve each component independently in such a way that the solutions can be merged together to find the global solution.',\n",
       "   'offset': 14729,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"All constraints (3-5) as well 3-cycle constraints connect Si's following the edges of the scaffolding graph. Therefore, the Si-nodes of the interaction graph for our ILP will have the same connectivity structure as the scaffolding graph G = (V, E). As it has been noticed in, the scaffolding graph is a bounded-width graph and should be well decomposable in 2- and 3-connected components. The SPQR-tree data structure is employed to determine the decomposition order for 3-connected components the scaffolding graph. The solution to each component of the scaffolding graph is found using a bottom up traversal through which each component is solved 2 times: for similar and opposite orientations of the common nodes. The objective value of each case is then entered into the objective of the parental component. Having the solution of all components, top down DFS starting from the same root is performed to apply the chosen solution for each component.\",\n",
       "   'offset': 15764,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Below we illustrate the way how the solution is computed in stages through each of which the results of the previous stage are combined to dynamically solve the problem. Obviously, an isolated connected component will not influence other components. Moreover, it has been shown in that 2-connected components can be solved independently. As it can be seen in Figure 2(a), after removing the articulation point (1-cut) to decompose the graph into 2-connected components, each component is solved with the same arbitrary direction assigned to the common node, and then the resulting solutions are collapsed into the parent solution. The pre-assigned direction will never affect the parent solution since all contigs in the scaffold can be flipped at the same time.',\n",
       "   'offset': 16718,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Solving the maximum likelihood ILP via graph decomposition. (a) Graph decomposition into 2-connected components: Red (1-cut) node splits the graph into two 2-connected components A and B. The ILP is solved for each component separately. If the direction of the cut node in the ILP solution for B is opposite to the one in the solution for A, then the solution of B is inverted. Then ILP solutions for A and B are collapsed into the parent solution. (b) Graph decomposition into two 3-connected components: Red and yellow (2-cut) nodes split the graph into two 3-connected components A and B. The ILP is solved for component A twice - for the same and the opposite directions assigned to two 2-cut nodes. Then these two solutions are used in the objective for the ILP of component B. Finally, ILP solutions for A and B are collapsed into the parent solution.',\n",
       "   'offset': 17481,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F2',\n",
       "    'file': '1471-2105-15-S9-S9-2.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Still, 2-connected components can be very large, so we look for 2-cuts in order to decompose the graph into significantly smaller 3-connected components. Figure 2(b) shows that splitting the two 2-cut nodes i and j decomposes the graph into 3-connected components A and B. The ILP for component A is solved twice to obtain',\n",
       "   'offset': 18339,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(1) the ILP solution sol00 in which the 2-cut nodes i and j are constrained to both have default orientations;',\n",
       "   'offset': 18662,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(2) the ILP solution sol01 in which the 2-cut nodes i and j are constrained to have opposite orientations.',\n",
       "   'offset': 18773,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The two solutions are combined to solve the ILP for component B. The ILP objective for component B should be updated by adding the term of sol00+(sol01-sol00) · Sij or, equivalently, the value sol00 should be added to  and  and the value sol01 should be added to  and . The overall solution is obtained by identifying the common nodes of the components. In the example on Figure 2(b), the optimal solution happens when 2-cut nodes have opposite directions. The corresponding solution of ILP for the component A should be incorporated in the overall solution. When the scaffolding graph has 3-connected components too large to handle, 3-cuts could also be used for decomposition.',\n",
       "   'offset': 18880,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The pseudo-code of the SILP2 NSDP algorithm for processing 3-connected components is given in Figure 3 SILP2 is different from SILP1 in the else clause - instead of solving ILP for each of four possible combinations of assignments for Si and Sj as in SILP1, ILP is solved only two times for combinations Si = 0 & Sj = 0 and Si = 0 & Sj = 1.',\n",
       "   'offset': 19560,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"SILP2's NSDP algorithm for processing 3-connected components.\",\n",
       "   'offset': 19901,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F3',\n",
       "    'file': '1471-2105-15-S9-S9-3.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Thinning heuristic. Unfortunately the largest tri-connected component may still induce an ILP too large for CPLEX to solve in a reasonable amount of time. In order to address this problem a thinning heuristic is applied to the scaffolding graph. This scenario can be detected by setting a threshold on the maximum number of contigs allowed in a tri-connected component. When a component exceeds the threshold the number of read pairs necessary to induce an edge is increased by one and decomposition recomputed until there is no component above the threshold.',\n",
       "   'offset': 19963,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Results and discussion',\n",
       "   'offset': 20523,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Datasets and quality measures',\n",
       "   'offset': 20546,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to asses the quality and scalability of our scaffolding tool we developed a testing framework which closely mimics real world scaffolding problems. We utilized the Staphylococcus aureus (staph), Rhodobacter sphaeroides (rhodo) genomes and chromosome 14 of HapMap individual NA12878 (chr14) from the GAGE assembly comparison. Finally, in a test case designed to stress scalability, contigs from a draft assembly of individual NA12878 (NA) created by were scaffolded using short-read data.',\n",
       "   'offset': 20576,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In all test cases the read pairs used for scaffolding are aligned against the contigs using bowtie2. Each read in a pair was required to be aligned uniquely according to the default scoring scheme, for the pair to be considered valid. Each scaffolder was given the same set of valid read pairs. Two of the leading external scaffolding tools MIP and OPERA are used in this comparison. Although many other tools do exist, these two are widely utilized and actively maintained.',\n",
       "   'offset': 21073,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The three small test cases are used to test both correctness and scalability of the scaffolding tool. In order to test correctness, contigs simulating a draft assembly were created by placing gaps in the genome. The contig and gap sizes were sampled uniformly at random from the collection of all the assemblies used in the GAGE comparison. The procedure to generate the contigs was to alternatively sample with replacement from the set of all contig sizes, and gap sizes. In this way a simulated scaffold can be generated so that the position and relative orientation of all contigs and all gap sizes are known. The orientation of the simulated contigs was randomized to prevent biases.',\n",
       "   'offset': 21548,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'For each genome 10 replicates were created, all subsequent results are the average of the 10 replicates. By creating simulated contigs with no assembly error, the accuracy of subsequent scaffolds can be evaluated exactly. Although the contigs were simulated, real read pairs were aligned against them and used as input. Table S1 in Additional File 1 describes the characteristics of each dataset.',\n",
       "   'offset': 22236,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The NA12878 test case was produced by simply using the contigs created in the SGA assembler publication. The read pairs were obtained from a different lab, however they were generated using the same biological source material (ERP002490). Although more read pairs were available a random subset of approximately 2x coverage was used.',\n",
       "   'offset': 22633,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Finally a simulated metagenomics test case was created to explore the feasibility of utilizing SILP2 to scaffold metagenomes. This was created by artificially mixing the staph and rhodo contigs and reads at varying proportions.',\n",
       "   'offset': 22967,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A natural and common parameter present in all scaffolding algorithms in the bundle size, or the number of read pairs spanning two contigs. This parameter is a natural control of sensitivity and specificity; requiring more support increases specificity at the price of sensitivity and vice-versa. It should be noted that every scaffolding tool tested, including SILP2 does not abide by the set parameter absolutely. Each method raises it in order to ensure efficient operation. The simulated test cases were evaluated at several bundle sizes to asses its effect on accuracy and scalability. The NA12878 test case was only evaluated at the minimum feasible value due to resource constraints.',\n",
       "   'offset': 23195,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Accuracy',\n",
       "   'offset': 23885,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Calculating the accuracy of de novo assemblies or scaffolds is quite difficult. One of the key challenges is deciding on the appropriate measure. In this comparison we elect to present several metrics which will likely have different weight depending on the background and intention of the reader.',\n",
       "   'offset': 23894,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'For the simulated contigs we treat scaffolding as a binary classification problem where methods attempt to predict true adjacencies in the test dataset. The accuracy and sensitivity can be directly measured by computing true positive, and false positive rates. One common summary is MCC, or Mathews Correlation coefficient. This measure assess sensitivity and specificity simultaneously. In the context of scaffolding, this measure illustrates how many correctly ordered and oriented scaffolds were created.',\n",
       "   'offset': 24192,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'An alternative measure, commonly utilized in genome assembly comparison publications is the notion of corrected N50. Where N50 is the weighted mean scaffold size, the corrected N50 is the same statistic after errors are removed. This can be computed exactly on simulated data, however an alignment based approximation must be used on real test cases.',\n",
       "   'offset': 24700,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Finally the usefulness of a genome can also be measured by the number of identifiable biological features captured. Here we capture this measure by recording the percentage of known genes that are found contiguous in the scaffolds.',\n",
       "   'offset': 25051,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MCC',\n",
       "   'offset': 25283,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The MCC metric, as seen in Figure 4, indicates that SILP2 is able to correctly join the most contigs, followed by OPERA and finally MIP. This order holds for all three simulated test cases. Interestingly all three methods see a decrease in MCC on staph, but some have increases on rhodo and chr14. This trend illustrates the difficult to define variables such as genome uniqueness, assembly and read error which can make smaller genomes more challenging that larger genomes.',\n",
       "   'offset': 25287,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MCC comparison. MCC of SILP2, OPERA, and MIP across bundle sizes for staph, rhodo, and chr14 datasets. At bundle size 1 OPERA exceeded the allowed runtime of 2 days for the chr14 dataset.',\n",
       "   'offset': 25762,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F4',\n",
       "    'file': '1471-2105-15-S9-S9-4.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'While MCC is natural to a computer scientist its useful to a biologist is lacking because the content of the contigs is ignored. A biologist typically asses a scaffold by the N50, Unfortunately this measure does not reflect the accuracy of the scaffolds and rewards aggressive merging. Using MCC or its constituent components as metrics gives greater clarity to the researcher comparing different tools.',\n",
       "   'offset': 25950,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'N50',\n",
       "   'offset': 26354,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The most common metric found in genome assembly and scaffolding is N50. The most recent iteration of benchmark projects have transformed this descriptive number into an accuracy measure by introducing alignment based corrections. Here the scaffolds are aligned against a reference and miss-alignments are interpreted as orientation, or placement errors. We have developed a more efficient implementation of the correcting method developed by. This enables the tool to be utilized on the NA12878 test case at the cost of accuracy.',\n",
       "   'offset': 26358,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The true N50 value can be determined when using simulated contigs by breaking incorrect scaffolds, this measure is denoted as TPN50. An analog to the TPN50 measure can be obtained by aligning the scaffolds against the known reference. Scaffolds (and contigs) are broken at mis-assembled or mis-scaffolded regions. This post-alignment metrics can be obtained from the assembly evaluation tool called QUAST and it is denoted as NA50.',\n",
       "   'offset': 26888,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Unfortunately the implementation of QUAST required more than 128GB of RAM to evaluate the NA12878 test case, and therefore could not be run. We wrote an alternative implementation of NA50 called ALN50 which is more efficient, but follows a similar framework. Both NA50 and ALN50 are found in Figure 5. Although NA50 and ALN50 do not agree, they do indicate similar trends between methods. Therefore ALN50 will be used henceforth. In the staph genome, OPERA is clearly the best performing tool, followed by SILP2 and then MIP. However on the rhodo genome, SILP2 performs best, followed by OPERA then MIP.',\n",
       "   'offset': 27320,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TPN50, ALN50 and NA50 as a function of bundle size for single-genome datasets. TPN50 is obtained by breaking incorrect scaffolds, ALN50 is the post-alignment metric developed by us, and NA50 is the QUAST equivalent. The colors indicated in the legend correspond to the bundle size 1 through 7. OPERA was unable to complete on bundle size 1 for chr14 dataset.',\n",
       "   'offset': 27924,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F5',\n",
       "    'file': '1471-2105-15-S9-S9-5.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'First the highest ALN50 is always found at bundle size 3 or 5. If the intent of the assembly is to maximize N50 then clearly no algorithm should be run with bundle size less than 3. However, as it was pointed out in both GAGE and QUAST, N50 is a misleading metric and alternative measures may be a better judge.',\n",
       "   'offset': 28283,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Additionally it can be seen that both OPERA and SILP2 have approximately the same TPN50 in the staph and chr14 test cases, however in rhodo, SILP2 clearly out-performs OPERA and MIP at all bundle sizes. It is not clear why SILP2 performs much better on rhodo, and approximately equivalent on the others. For the complete genome SILP2, OPERA and MIP reported an N50 of 26,235, 39,366, 26,235 respectively. This is consistent with the observations from the synthetic data sets.',\n",
       "   'offset': 28595,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Gene reconstruction',\n",
       "   'offset': 29071,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'An alternative measure of the completeness of a scaffold is the number of genes aligned against the scaffold. For a given percentage of completeness the number of genes found in the corrected scaffold is an indicator of the usefulness of the genome.',\n",
       "   'offset': 29091,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'As seen in Table 1, SILP2 almost consistently equals or outperforms both OPERA and MIP at all bundle sizes and for each genome. The difference between SILP2 and MIP is often quite small.',\n",
       "   'offset': 29341,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order for a gene to be considered reconstructed 90% of its sequence must be found in a contiguous scaffold. Dashes (-) indicate the method was unable to complete and therefore the gene count could not be computed.',\n",
       "   'offset': 29528,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The number of reconstructed genes found in the corrected scaffolds for single-genome datasets.',\n",
       "   'offset': 29745,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T1',\n",
       "    'file': 'T1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'genome\\tbundle\\tSILP2\\tOPERA\\tMIP\\ttotal\\nstaph\\t1\\t1,727.70\\t1,168.50\\t1,545.00\\t2692\\n\\t2\\t1,727.70\\t1,168.50\\t1,559.50\\t\\n\\t3\\t1,727.70\\t1,210.60\\t1,575.30\\t\\n\\t5\\t1,727.70\\t1,262.70\\t1,584.60\\t\\n\\t7\\t1,727.40\\t1,280.40\\t1,588.50\\t\\n\\nrhodo\\t1\\t2022.7\\t1618.6\\t1897.3\\t3067\\n\\t2\\t2022.7\\t1618.6\\t1907\\t\\n\\t3\\t2022.6\\t1751.1\\t1894\\t\\n\\t5\\t2022.6\\t1834.2\\t1921.3\\t\\n\\t7\\t2022.6\\t1853.3\\t1933.3\\t\\n\\nchr14\\t1\\t350.9\\t-\\t349.6\\t529\\n\\t2\\t352.00\\t330.10\\t350.40\\t\\n\\t3\\t352.40\\t336.90\\t350.40\\t\\n\\t5\\t352.40\\t337.50\\t351.70\\t\\n\\t7\\t352.40\\t337.60\\t3.00\\t\\n\\nNA12878 2x\\t1\\t30817\\t-\\t30817\\t34039\\n\\t2\\t30850\\t30809\\t30849\\t\\n',\n",
       "   'offset': 29840,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T1',\n",
       "    'File': 'T1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Runtime',\n",
       "   'offset': 30353,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'One key advantage of SILP2 over other scaffolding tools is its speed and scalability. Table 2 gives the runtime of SILP1, SILP2, OPERA and MIP on single-genome testcases. All experiments were conducted on a Dell PowerEdge R815 server with quad 2.5GHz 16-core AMD Opteron 6380 processors and 256Gb RAM running under Ubuntu 12.04 LTS. IBM ILOG CPLEX 12.5.0.0 was used as ILP solver through the CPLEX Python API. Reported runtimes are only for the scaffolding portion of each program. Read alignment and pre-processing steps are not included, but it was observed that all methods had comparable pre-processing times.',\n",
       "   'offset': 30361,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'All timing was captured only during the scaffolding phase of each tool, all read alignment and formatting procedures were excluded from timing. The number is the average of 10 runs for each genome. A dash (-) indicated the tool was unable to complete in the allotted time of 2 days for staph,rhodo,chr14 and 3 days for NA12878.',\n",
       "   'offset': 30975,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Runtime (in seconds) for scaffolding single-genome datasets.',\n",
       "   'offset': 31303,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T2',\n",
       "    'file': 'T2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'genome\\tbundle\\tSILP1\\tSILP2\\tOPERA\\tMIP\\nstaph\\t1\\t1237\\t6.4\\t2538.1\\t35.8\\n\\t2\\t738\\t4.5\\t1456.5\\t17\\n\\t3\\t305\\t4\\t878.5\\t12.834\\n\\t5\\t142\\t3.9\\t386.9\\t10.54\\n\\t7\\t51\\t4.3\\t241\\t10.115\\n\\nrhodo\\t1\\t1134\\t10\\t2297\\t118.953\\n\\t2\\t632\\t4.1\\t455.2\\t25.3\\n\\t3\\t486\\t3.6\\t5.7\\t10.995\\n\\t5\\t86\\t3.4\\t2\\t8.778\\n\\t7\\t75\\t3\\t1.6\\t8.217\\n\\nchr14\\t1\\t-\\t64.7\\t-\\t706.3\\n\\t2\\t-\\t27.6\\t99.25\\t189.685\\n\\t3\\t629\\t25.5\\t11\\t137.67\\n\\t5\\t370\\t21.5\\t12\\t107.85\\n\\t7\\t400\\t19.25\\t10.75\\t94.9875\\n\\nNA12878 2x\\t1\\t-\\t55.2\\t-\\t89.3\\n\\t2\\t-\\t1670\\t76.49\\t53.28\\n\\t3\\t37751\\t3878\\t7875\\t121.61\\n\\t5\\t27341\\t3183\\t4270\\t134.6\\n\\t7\\t27470\\t3626\\t2180\\t125.66\\n',\n",
       "   'offset': 31364,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T2',\n",
       "    'File': 'T2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'On the staph, rhodo and chr14 datasets, it was observed that SILP2 was quicker at higher bundle sizes and no worse than OPERA or MIP at lower bundle sizes. The NA12878 testcase was extremely challenging for all methods and demonstrated the effect of heuristics on large test cases. It is clear from the reduced runtimes that all 3 methods activate some sort of heuristic at lower bundle sizes. The difference between SILP1 and SILP2 is evident at all bundle sizes.',\n",
       "   'offset': 31873,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The NA12878 genome was also scaffolded by SILP2 using 20x coverage reads, with a runtime of 18,205 seconds at bundle size 1. Negligible improvement in accuracy over the 2x dataset was observed. From Table 2 it is clear that runtime increases with the complexity of the genome more so than the number of read pairs.',\n",
       "   'offset': 32338,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Metagenomics',\n",
       "   'offset': 32653,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Metagenomics is the study of genetic material recovered from heterogeneous mixtures often found in nature. Just like in the de novo assembly of a single genome, the accuracy and size of the scaffolds is critical to subsequent analysis steps. Our ILP based solution is flexible enough to include new constraints and objectives to better serve this challenging scenario.',\n",
       "   'offset': 32666,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to test this hypothesis a simulated metagenomic dataset was created utilizing the staph and rhodo genomes from the GAGE dataset. The simulated contigs used previously were mixed, and both sets of reads were aligned with varying fractions (1.0, 9.5 0.25, 0.0) of staph reads present. Again all three of the major scaffolding tools were tested, however additional weighting scenarios were implemented in SILP2.',\n",
       "   'offset': 33035,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The runtime, MCC, SCFN50, TPN50 and ALN50 metrics are detailed for each of the compared methods in Table 3. Also an additional scaffolding tool BAMBUS2 was added to the comparison because it was previously shown to work well in the metagenomic scaffolding context.',\n",
       "   'offset': 33453,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The second column indicates the percentage of total read pairs used from the staph genome testcase, all of the rhodo pairs were used. SCFN50 is the uncorrected N50 reported by each scaffolding tool. The N50 of the contigs alone is 10,339 bp. The integer appended to SILP2 indicates the bundle weight; 0: none, 1: coverage, 2: repeat, 3:coverage * repeat. All methods were run at bundle size 1, the reported number is the average of 10 runs except for OPERA where 6 of the test cases exceeded runtime limits. TPN50 and MCC were unable to be computed for BAMBUS2 because the generated AGP had a non-standard format.',\n",
       "   'offset': 33718,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Combined runtime and accuracy results for the simulated low-complexity metagenome datasets.',\n",
       "   'offset': 34332,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T3',\n",
       "    'file': 'T3.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'METHOD\\tFRAC STAPH\\tRUNTIME\\tSCFN50\\tALN50\\tTPN50\\tMCC\\nSILP2 0\\t1.00\\t14.3\\t51,775.0\\t20,647\\t34,495\\t67.6\\nSILP2 0\\t0.50\\t13.3\\t50,450.0\\t20,103\\t36,356\\t69.1\\nSILP2 0\\t0.25\\t12.7\\t47,731.0\\t20,761\\t35,323\\t69.3\\nSILP2 0\\t0.00\\t11.5\\t21,753.0\\t13,948\\t15,649\\t39.9\\n\\nSILP2 1\\t1.00\\t14.3\\t52,557.0\\t20,655\\t33,683\\t67.5\\nSILP2 1\\t0.50\\t13.8\\t48,701.0\\t20,337\\t35,750\\t69.0\\nSILP2 1\\t0.25\\t13.4\\t49,766.0\\t20,752\\t35,146\\t69.0\\nSILP2 1\\t0.00\\t11.0\\t21,925.0\\t13,847\\t15,511\\t39.3\\n\\nSILP2 2\\t1.00\\t14.3\\t43,144.0\\t20,631\\t31,160\\t66.3\\nSILP2 2\\t0.50\\t13.5\\t42,244.0\\t20,198\\t32,477\\t67.5\\nSILP2 2\\t0.25\\t13.2\\t45,137.0\\t21,161\\t31,562\\t67.6\\nSILP2 2\\t0.00\\t10.8\\t22,190.0\\t13,813\\t16,205\\t41.6\\n\\nSILP2 3\\t1.00\\t14.1\\t43,646.0\\t19,998\\t28,856\\t65.3\\nSILP2 3\\t0.50\\t13.2\\t41,893.0\\t19,790\\t30,504\\t66.6\\nSILP2 3\\t0.25\\t13.0\\t42,188.0\\t19,945\\t30,449\\t66.4\\nSILP2 3\\t0.00\\t11.5\\t21,820.0\\t13,781\\t15,635\\t40.0\\n\\nOPERA\\t1.00\\t2247.2\\t15,573.0\\t13,082\\t10,386\\t10.1\\nOPERA\\t0.50\\t1567.6\\t13,928.0\\t12,006\\t10,440\\t10.7\\nOPERA\\t0.25\\t884.0\\t14,786.0\\t12,617\\t10,507\\t10.5\\nOPERA\\t0.00\\t544.3\\t11,121.0\\t10,720\\t10,273\\t4.9\\n\\nMIP\\t1.00\\t129.9\\t20,104.0\\t12,861\\t18,672\\t18.4\\nMIP\\t0.50\\t121.3\\t19,807.0\\t12,488\\t17,613\\t17.4\\nMIP\\t0.25\\t114.0\\t18,520.0\\t12,269\\t16,680\\t17.2\\nMIP\\t0.00\\t114.1\\t12,690.0\\t10,894\\t12,434\\t8.7\\n\\nBAMBUS2\\t1.00\\t1025.89\\t11,251.0\\t11,238\\t-\\t-\\nBAMBUS2\\t0.50\\t1452.75\\t10,781.0\\t10,822\\t-\\t-\\nBAMBUS2\\t0.25\\t1676.75\\t10,806.0\\t10,834\\t-\\t-\\nBAMBUS2\\t0.00\\t2272\\t11,526.0\\t11,698\\t-\\t-\\n',\n",
       "   'offset': 34424,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T3',\n",
       "    'File': 'T3.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interestingly all SILP2 variants fare much better than both OPERA, MIP and BAMBUS2 even with no staph reads present (this differs from results in Figure 5 because the rhodo reads were aligned to both staph and rhodo contigs). It is unclear is the different methodology used in SILP2 sets it apart, or if an implementation quirk throws off the other scaffolders. However across all metrics SILP2 variants perform the best.',\n",
       "   'offset': 35729,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In both SILP2 variants and MIP it is observed that the TPN50 decreases as fewer staph reads are utilized. This is expected since there are fewer opportunities to connect staph contigs and both staph and rhodo contigs are used in the calculation of N50. There is no major differences between the variants of SILP2. The coverage based weight seems to improve MCC at the cost of a slightly lowered TPN50 when compared to no weights.',\n",
       "   'offset': 36151,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'This highly simplified test scenario is not designed to fully explore metagenomic scaffolding, rather to point out an opportunity to further external genome scaffolding algorithms.',\n",
       "   'offset': 36581,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Conclusions',\n",
       "   'offset': 36762,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Scaffolding in an important step in the de novo assembly pipeline. Biologists rely on an accurate scaffold to perform many types of analysis. The larger the scaffold the more useful it will be to them. Recent advances in de novo assemblers has made it feasible to create draft assemblies for large mammalian genomes. We believe the SILP2, coupled with the most recent scalable assemblers will produce the largest and most complete assemblies. This is made possible utilizing non-serial dynamic programming approach to solve our robust ILP. The ILP formulation for the maximum likelihood model is shown to be flexible enough to handle metagenomic samples.',\n",
       "   'offset': 36774,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The future work includes more thorough experimental validation of SILP2 and comparison BAMBUS2 on metagenomic samples. Also we are going to validate SILP2 using methodology and benchmarks from recent paper.',\n",
       "   'offset': 37429,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'List of abbreviations',\n",
       "   'offset': 37636,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '• ILP - Integer Linear Programming',\n",
       "   'offset': 37658,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"• MCC - Matthew's correlation coefficient\",\n",
       "   'offset': 37695,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '• MIP - Mixed Integer Programming',\n",
       "   'offset': 37739,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '• NSDP - Non-serial Dynamic Programming',\n",
       "   'offset': 37775,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '• OGDF - Open Graph Drawing Framework',\n",
       "   'offset': 37817,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '• SQPR - a tree data structure used in graph algorithms to represent the 3-connected components of a graph ',\n",
       "   'offset': 37857,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Software and data availability',\n",
       "   'offset': 37967,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A reference implementation of SILP2 is provided at https://github.com/jim-bo/silp2. This implementation relies on the CPLEX optimization library. The code used for generating most of the reported results, including the implementation of the ALN50 metric, is available at https://github.com/jim-bo/scafathon. The SILP2 source code along with a small test dataset can also be downloaded from http://dna.engr.uconn.edu/software/SILP2.',\n",
       "   'offset': 37998,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"Authors' contributions\",\n",
       "   'offset': 38430,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'JL developed and implemented the algorithms, ran the experiments, and wrote result section, HS developed algorithms, participate in writing the paper, and implemented parts of the code, IM and AZ developed algorithms wrote the paper, conceived and supervised implementation.',\n",
       "   'offset': 38453,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary Material',\n",
       "   'offset': 38728,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Acknowledgements',\n",
       "   'offset': 38751,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'ACK_FUND'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'This work has been partially supported by Agriculture and Food Research Initiative Competitive Grant no. 201167016-30331 from the USDA National Institute of Food and Agriculture and NSF awards IIS-0916401 and IIS-0916948.',\n",
       "   'offset': 38768,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ACK_FUND'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Declarations',\n",
       "   'offset': 38990,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'ACK_FUND'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The authors declare that funding for publication of the article was by the research institute of the corresponding author.',\n",
       "   'offset': 39003,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ACK_FUND'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'This article has been published as part of BMC Bioinformatics Volume 15 Supplement 9, 2014: Proceedings of the Fourth Annual RECOMB Satellite Workshop on Massively Parallel Sequencing (RECOMB-Seq 2014). The full contents of the supplement are available online at http://www.biomedcentral.com/bmcbioinformatics/supplements/15/S9.',\n",
       "   'offset': 39126,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ACK_FUND'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'ABySS: a parallel assembler for short read sequence data',\n",
       "   'offset': 39455,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Jackman;given-names:SD',\n",
       "    'name_3': 'surname:Schein;given-names:JE',\n",
       "    'name_0': 'surname:Simpson;given-names:JT',\n",
       "    'name_1': 'surname:Wong;given-names:K',\n",
       "    'name_4': 'surname:Jones;given-names:SJ',\n",
       "    'name_5': 'surname:Birol;given-names:I',\n",
       "    'issue': '6',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1123',\n",
       "    'pub-id_pmid': '19251739',\n",
       "    'source': 'Genome research',\n",
       "    'fpage': '1117',\n",
       "    'year': '2009',\n",
       "    'type': 'ref',\n",
       "    'volume': '19'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'ALLPATHS: De novo assembly of whole-genome shotgun microreads',\n",
       "   'offset': 39512,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kleber;given-names:M',\n",
       "    'name_3': 'surname:Shlyakhter;given-names:IA',\n",
       "    'name_0': 'surname:Butler;given-names:J',\n",
       "    'name_1': 'surname:MacCallum;given-names:I',\n",
       "    'name_6': 'surname:Nusbaum;given-names:C',\n",
       "    'name_7': 'surname:Jaffe;given-names:DB',\n",
       "    'name_4': 'surname:Belmonte;given-names:MK',\n",
       "    'name_5': 'surname:Lander;given-names:ES',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '18',\n",
       "    'pub-id_pmid': '18340039',\n",
       "    'source': 'Genome Research',\n",
       "    'fpage': '810',\n",
       "    'year': '2008',\n",
       "    'lpage': '820',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'High-quality draft assemblies of mammalian genomes from massively parallel sequence data',\n",
       "   'offset': 39574,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Przybylski;given-names:D',\n",
       "    'name_3': 'surname:Ribeiro;given-names:FJ',\n",
       "    'name_0': 'surname:Gnerre;given-names:S',\n",
       "    'name_1': 'surname:MacCallum;given-names:I',\n",
       "    'name_6': 'surname:Sharpe;given-names:T',\n",
       "    'name_7': 'surname:Hall;given-names:G',\n",
       "    'name_4': 'surname:Burton;given-names:JN',\n",
       "    'name_5': 'surname:Walker;given-names:BJ',\n",
       "    'name_8': 'surname:Shea;given-names:TP',\n",
       "    'name_9': 'surname:Sykes;given-names:S',\n",
       "    'lpage': '1518',\n",
       "    'fpage': '1513',\n",
       "    'year': '2011',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'Proceedings of the National Academy of Sciences',\n",
       "    'type': 'ref',\n",
       "    'volume': '108',\n",
       "    'issue': '4',\n",
       "    'name_18': 'surname:Lander;given-names:ES',\n",
       "    'name_19': 'surname:Jaffe;given-names:DB',\n",
       "    'name_10': 'surname:Berlin;given-names:AM',\n",
       "    'name_11': 'surname:Aird;given-names:D',\n",
       "    'name_12': 'surname:Costello;given-names:M',\n",
       "    'name_13': 'surname:Daza;given-names:R',\n",
       "    'name_14': 'surname:Williams;given-names:L',\n",
       "    'name_15': 'surname:Nicol;given-names:R',\n",
       "    'name_16': 'surname:Gnirke;given-names:A',\n",
       "    'name_17': 'surname:Nusbaum;given-names:C'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'De novo fragment assembly with short mate-paired reads: Does the read length matter?',\n",
       "   'offset': 39663,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Pevzner;given-names:P',\n",
       "    'name_0': 'surname:Chaisson;given-names:M',\n",
       "    'name_1': 'surname:Brinza;given-names:D',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'Genome Res',\n",
       "    'year': '2008',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'De novo assembly of human genomes with massively parallel short read sequencing',\n",
       "   'offset': 39748,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1101/gr.097261.109',\n",
       "    'name_2': 'surname:Ruan;given-names:J',\n",
       "    'name_0': 'surname:Li;given-names:R',\n",
       "    'name_1': 'surname:Zhu;given-names:H',\n",
       "    'name_6': 'surname:Li;given-names:Y',\n",
       "    'name_7': 'surname:Li;given-names:S',\n",
       "    'name_4': 'surname:Fang;given-names:X',\n",
       "    'name_3': 'surname:Qian;given-names:W',\n",
       "    'name_8': 'surname:Shan;given-names:G',\n",
       "    'name_9': 'surname:Kristiansen;given-names:K',\n",
       "    'lpage': '272',\n",
       "    'fpage': '265',\n",
       "    'year': '2010',\n",
       "    'name_5': 'surname:Shi;given-names:Z',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'Genome Research',\n",
       "    'issue': '2',\n",
       "    'volume': '20',\n",
       "    'pub-id_pmid': '20019144',\n",
       "    'type': 'ref',\n",
       "    'name_10': 'surname:Li;given-names:S',\n",
       "    'name_11': 'surname:Yang;given-names:H',\n",
       "    'name_12': 'surname:Wang;given-names:J',\n",
       "    'name_13': 'surname:Wang;given-names:J'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Algorithms for de novo short read assembly using de Bruijn graphs Velvet',\n",
       "   'offset': 39828,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '829',\n",
       "    'name_0': 'surname:Zerbino;given-names:DR',\n",
       "    'name_1': 'surname:Birney;given-names:E',\n",
       "    'pub-id_pmid': '18349386',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '18',\n",
       "    'source': 'Genome Research',\n",
       "    'fpage': '821',\n",
       "    'year': '2008',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Assemblathon 2: evaluating de novo methods of genome assembly in three vertebrate species',\n",
       "   'offset': 39901,\n",
       "   'relations': [],\n",
       "   'infons': {'name_61': 'surname:Przybylski;given-names:D',\n",
       "    'name_60': 'surname:Place;given-names:M',\n",
       "    'name_63': 'surname:Qu;given-names:C',\n",
       "    'name_62': 'surname:Qin;given-names:X',\n",
       "    'name_65': 'surname:Richards;given-names:S',\n",
       "    'name_64': 'surname:Ribeiro;given-names:FJ',\n",
       "    'name_67': 'surname:Ruby;given-names:JG',\n",
       "    'name_66': 'surname:Rokhsar;given-names:DS',\n",
       "    'name_69': 'surname:Schatz;given-names:MC',\n",
       "    'name_68': 'surname:Scalabrin;given-names:S',\n",
       "    'name_78': 'surname:Tsarev;given-names:F',\n",
       "    'name_79': 'surname:Vezzi;given-names:F',\n",
       "    'name_76': 'surname:Simpson;given-names:JT',\n",
       "    'name_77': 'surname:Song;given-names:H',\n",
       "    'name_74': 'surname:Shendure;given-names:J',\n",
       "    'name_75': 'surname:Shi;given-names:Y',\n",
       "    'name_72': 'surname:Sharpe;given-names:T',\n",
       "    'name_73': 'surname:Shaw;given-names:TI',\n",
       "    'name_70': 'surname:Schwartz;given-names:DC',\n",
       "    'name_71': 'surname:Sergushichev;given-names:A',\n",
       "    'name_2': 'surname:Alexandrov;given-names:A',\n",
       "    'name_3': 'surname:Baranay;given-names:P',\n",
       "    'name_0': 'surname:Bradnam;given-names:KR',\n",
       "    'name_1': 'surname:Fass;given-names:JN',\n",
       "    'name_6': 'surname:Boisvert;given-names:S',\n",
       "    'name_7': 'surname:Chapman;given-names:JA',\n",
       "    'name_4': 'surname:Bechner;given-names:M',\n",
       "    'name_5': 'surname:Birol;given-names:I',\n",
       "    'name_8': 'surname:Chapuis;given-names:G',\n",
       "    'name_9': 'surname:Chikhi;given-names:R',\n",
       "    'name_83': 'surname:Worley;given-names:KC',\n",
       "    'name_82': 'surname:Wang;given-names:J',\n",
       "    'name_81': 'surname:Vieira;given-names:BM',\n",
       "    'name_80': 'surname:Vicedomini;given-names:R',\n",
       "    'name_87': 'surname:Zhang;given-names:G',\n",
       "    'name_86': 'surname:Yuan;given-names:J',\n",
       "    'fpage': '10+',\n",
       "    'year': '2013',\n",
       "    'name_89': 'surname:Zhou;given-names:S',\n",
       "    'name_88': 'surname:Zhang;given-names:H',\n",
       "    'name_90': 'surname:Korf;given-names:IF',\n",
       "    'name_22': 'surname:Gnerre;given-names:S',\n",
       "    'name_18': 'surname:Fedotov;given-names:P',\n",
       "    'name_19': 'surname:Fonseca;given-names:NA',\n",
       "    'name_10': 'surname:Chitsaz;given-names:H',\n",
       "    'name_11': 'surname:Chou;given-names:WCC',\n",
       "    'name_12': 'surname:Corbeil;given-names:J',\n",
       "    'name_13': 'surname:Del Fabbro;given-names:C',\n",
       "    'name_14': 'surname:Docking;given-names:TR',\n",
       "    'name_15': 'surname:Durbin;given-names:R',\n",
       "    'name_16': 'surname:Earl;given-names:D',\n",
       "    'name_17': 'surname:Emrich;given-names:S',\n",
       "    'section_type': 'REF',\n",
       "    'name_25': 'surname:Haimel;given-names:M',\n",
       "    'name_24': 'surname:Goldstein;given-names:S',\n",
       "    'name_27': 'surname:Haussler;given-names:D',\n",
       "    'name_26': 'surname:Hall;given-names:G',\n",
       "    'name_21': 'surname:Gibbs;given-names:RA',\n",
       "    'name_20': 'surname:Ganapathy;given-names:G',\n",
       "    'name_23': 'surname:Godzaridis;given-names:E',\n",
       "    'source': 'GigaScience',\n",
       "    'name_29': 'surname:Ho;given-names:IY',\n",
       "    'name_28': 'surname:Hiatt;given-names:JB',\n",
       "    'type': 'ref',\n",
       "    'volume': '2',\n",
       "    'name_32': 'surname:Jackman;given-names:SD',\n",
       "    'name_33': 'surname:Jaffe;given-names:DB',\n",
       "    'name_30': 'surname:Howard;given-names:J',\n",
       "    'name_31': 'surname:Hunt;given-names:M',\n",
       "    'name_36': 'surname:Kazakov;given-names:S',\n",
       "    'name_37': 'surname:Kersey;given-names:PJ',\n",
       "    'name_34': 'surname:Jarvis;given-names:E',\n",
       "    'name_35': 'surname:Jiang;given-names:H',\n",
       "    'name_38': 'surname:Kitzman;given-names:JO',\n",
       "    'name_39': 'surname:Knight;given-names:JR',\n",
       "    'name_85': 'surname:Yiu;given-names:SMM',\n",
       "    'name_84': 'surname:Yin;given-names:S',\n",
       "    'name_47': 'surname:Liu;given-names:Y',\n",
       "    'name_46': 'surname:Liu;given-names:B',\n",
       "    'name_45': 'surname:Li;given-names:Z',\n",
       "    'name_44': 'surname:Li;given-names:Y',\n",
       "    'name_43': 'surname:Laviolette;given-names:F',\n",
       "    'name_42': 'surname:Lavenier;given-names:D',\n",
       "    'name_41': 'surname:Lam;given-names:TWW',\n",
       "    'name_40': 'surname:Koren;given-names:S',\n",
       "    'name_49': 'surname:Maccallum;given-names:I',\n",
       "    'name_48': 'surname:Luo;given-names:R',\n",
       "    'pub-id_pmid': '23870653',\n",
       "    'name_54': 'surname:Ning;given-names:Z',\n",
       "    'name_55': 'surname:Otto;given-names:TD',\n",
       "    'name_56': 'surname:Paten;given-names:B',\n",
       "    'name_57': 'surname:Paulo;given-names:OS',\n",
       "    'name_50': 'surname:Macmanes;given-names:MD',\n",
       "    'name_51': 'surname:Maillet;given-names:N',\n",
       "    'name_52': 'surname:Melnikov;given-names:S',\n",
       "    'name_53': 'surname:Naquin;given-names:D',\n",
       "    'name_58': 'surname:Phillippy;given-names:AM',\n",
       "    'name_59': 'surname:Pina-Martins;given-names:F'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sense from sequence reads: methods for alignment and assembly',\n",
       "   'offset': 39991,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': 'S12',\n",
       "    'name_0': 'surname:Flicek;given-names:P',\n",
       "    'name_1': 'surname:Birney;given-names:E',\n",
       "    'pub-id_pmid': '19844229',\n",
       "    'issue': '11s',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '6',\n",
       "    'source': 'Nature Methods',\n",
       "    'fpage': 'S6',\n",
       "    'year': '2009',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparative Studies of de novo Assembly Tools for Next-generation Sequencing Technologies',\n",
       "   'offset': 40053,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Shen;given-names:H',\n",
       "    'name_3': 'surname:Zhang;given-names:L',\n",
       "    'name_0': 'surname:Lin;given-names:Y',\n",
       "    'name_1': 'surname:Li;given-names:J',\n",
       "    'name_4': 'surname:Papasian;given-names:CJ',\n",
       "    'name_5': 'surname:Deng;given-names:HW',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'Bioinformatics',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Assembly algorithms for next-generation sequencing data',\n",
       "   'offset': 40143,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Sutton;given-names:G',\n",
       "    'lpage': '327',\n",
       "    'name_0': 'surname:Miller;given-names:JR',\n",
       "    'name_1': 'surname:Koren;given-names:S',\n",
       "    'pub-id_pmid': '20211242',\n",
       "    'issue': '6',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '95',\n",
       "    'source': 'Genomics',\n",
       "    'fpage': '315',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'De novo assembly of short sequence reads',\n",
       "   'offset': 40199,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '472',\n",
       "    'name_0': 'surname:Paszkiewicz;given-names:KH',\n",
       "    'name_1': 'surname:Studholme;given-names:DJ',\n",
       "    'pub-id_pmid': '20724458',\n",
       "    'issue': '5',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '11',\n",
       "    'source': 'Briefings in Bioinformatics',\n",
       "    'fpage': '457',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genome assembly reborn: recent computational challenges',\n",
       "   'offset': 40240,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '366',\n",
       "    'name_0': 'surname:Pop;given-names:M',\n",
       "    'pub-id_pmid': '19482960',\n",
       "    'issue': '4',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '10',\n",
       "    'source': 'Briefings in Bioinformatics',\n",
       "    'fpage': '354',\n",
       "    'year': '2009',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Assembly of large genomes using second-generation sequencing',\n",
       "   'offset': 40296,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Salzberg;given-names:SL',\n",
       "    'lpage': '1173',\n",
       "    'name_0': 'surname:Schatz;given-names:MC',\n",
       "    'name_1': 'surname:Delcher;given-names:AL',\n",
       "    'pub-id_pmid': '20508146',\n",
       "    'issue': '9',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '20',\n",
       "    'source': 'Genome Research',\n",
       "    'fpage': '1165',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'GAGE: A critical evaluation of genome assemblies and assembly algorithms',\n",
       "   'offset': 40357,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Zimin;given-names:A',\n",
       "    'name_3': 'surname:Puiu;given-names:D',\n",
       "    'name_0': 'surname:Salzberg;given-names:SL',\n",
       "    'name_1': 'surname:Phillippy;given-names:AM',\n",
       "    'name_6': 'surname:Treangen;given-names:TJ',\n",
       "    'name_7': 'surname:Schatz;given-names:MC',\n",
       "    'name_4': 'surname:Magoc;given-names:T',\n",
       "    'name_5': 'surname:Koren;given-names:S',\n",
       "    'name_8': 'surname:Delcher;given-names:AL',\n",
       "    'name_9': 'surname:Roberts;given-names:M',\n",
       "    'lpage': '567',\n",
       "    'fpage': '557',\n",
       "    'year': '2012',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'Genome Research',\n",
       "    'issue': '3',\n",
       "    'volume': '22',\n",
       "    'pub-id_pmid': '22147368',\n",
       "    'type': 'ref',\n",
       "    'name_10': 'surname:Mar¸cais;given-names:G',\n",
       "    'name_11': 'surname:Pop;given-names:M',\n",
       "    'name_12': 'surname:Yorke;given-names:JA'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A hybrid approach for the automated finishing of bacterial genomes',\n",
       "   'offset': 40430,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Robins;given-names:WP',\n",
       "    'name_3': 'surname:Chin;given-names:CS',\n",
       "    'name_0': 'surname:Bashir;given-names:A',\n",
       "    'name_1': 'surname:Klammer;given-names:AA',\n",
       "    'name_6': 'surname:Hsu;given-names:D',\n",
       "    'name_7': 'surname:Ashby;given-names:M',\n",
       "    'name_4': 'surname:Webster;given-names:D',\n",
       "    'name_5': 'surname:Paxinos;given-names:E',\n",
       "    'issue': '7',\n",
       "    'name_8': 'surname:Wang;given-names:S',\n",
       "    'name_9': 'surname:Peluso;given-names:P',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '707',\n",
       "    'source': 'Nature Biotechnology',\n",
       "    'fpage': '701',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Single-molecule-based methods for mapping genomes Optical mapping of DNA',\n",
       "   'offset': 40497,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Hofkens;given-names:J',\n",
       "    'lpage': '311',\n",
       "    'name_0': 'surname:Neely;given-names:RK',\n",
       "    'name_1': 'surname:Deen;given-names:J',\n",
       "    'pub-id_pmid': '21207457',\n",
       "    'issue': '5',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '95',\n",
       "    'source': 'Biopolymers',\n",
       "    'fpage': '298',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The greedy path-merging algorithm for contig scaffolding',\n",
       "   'offset': 40570,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Myers;given-names:EW',\n",
       "    'lpage': '615',\n",
       "    'name_0': 'surname:Huson;given-names:DH',\n",
       "    'name_1': 'surname:Reinert;given-names:K',\n",
       "    'issue': '5',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '49',\n",
       "    'source': 'J ACM',\n",
       "    'fpage': '603',\n",
       "    'year': '2002',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Some simplified NP-complete problems',\n",
       "   'offset': 40627,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1145/800119.803884',\n",
       "    'name_2': 'surname:Stockmeyer;given-names:L',\n",
       "    'name_0': 'surname:Garey;given-names:MR',\n",
       "    'name_1': 'surname:Johnson;given-names:DS',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '63',\n",
       "    'source': 'Proceedings of the sixth annual ACM symposium on Theory of computing',\n",
       "    'fpage': '47',\n",
       "    'year': '1974',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hierarchical scaffolding with Bambus',\n",
       "   'offset': 40664,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Salzberg;given-names:SL',\n",
       "    'lpage': '159',\n",
       "    'name_0': 'surname:Pop;given-names:M',\n",
       "    'name_1': 'surname:Kosack;given-names:DS',\n",
       "    'pub-id_pmid': '14707177',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '14',\n",
       "    'source': 'Genome research',\n",
       "    'fpage': '149',\n",
       "    'year': '2004',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'SOPRA: Scaffolding algorithm for paired reads via statistical optimization',\n",
       "   'offset': 40701,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Sengupta;given-names:A',\n",
       "    'name_0': 'surname:Dayarian;given-names:A',\n",
       "    'name_1': 'surname:Michael;given-names:T',\n",
       "    'pub-id_pmid': '20576136',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '11',\n",
       "    'source': 'BMC Bioinformatics',\n",
       "    'fpage': '345',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fast scaffolding with small independent mixed integer programs',\n",
       "   'offset': 40776,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1093/bioinformatics/btr562',\n",
       "    'name_2': 'surname:V¨alim¨aki;given-names:N',\n",
       "    'name_0': 'surname:Salmela;given-names:L',\n",
       "    'name_1': 'surname:M¨akinen;given-names:V',\n",
       "    'name_4': 'surname:Ukkonen;given-names:E',\n",
       "    'name_3': 'surname:Ylinen;given-names:J',\n",
       "    'issue': '23',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '3265',\n",
       "    'pub-id_pmid': '21998153',\n",
       "    'source': 'Bioinformatics',\n",
       "    'fpage': '3259',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'volume': '27'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'SLIQ: Simple Linear Inequalities for Efficient Contig Scaffolding',\n",
       "   'offset': 40839,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://doi.acm.org/10.1145/6462.6502',\n",
       "    'name_2': 'surname:Segupta;given-names:AM',\n",
       "    'name_0': 'surname:Roy;given-names:RS',\n",
       "    'name_1': 'surname:Chen;given-names:KC',\n",
       "    'name_3': 'surname:Schliep;given-names:A',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'arXiv:1111.1426v2[q-bio.GN]',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Opera: reconstructing optimal genomic scaffolds with high-throughput paired-end sequences',\n",
       "   'offset': 40905,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Sung;given-names:WK',\n",
       "    'name_0': 'surname:Gao;given-names:S',\n",
       "    'name_1': 'surname:Nagarajan;given-names:N',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '451',\n",
       "    'source': 'Proc 15th Annual international conference on Research in computational molecular biology',\n",
       "    'fpage': '437',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Toward a statistically explicit understanding of de novo sequence assembly',\n",
       "   'offset': 40995,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Dunn;given-names:CW',\n",
       "    'lpage': '2963',\n",
       "    'name_0': 'surname:Howison;given-names:M',\n",
       "    'name_1': 'surname:Zapata;given-names:F',\n",
       "    'pub-id_pmid': '24021385',\n",
       "    'issue': '23',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '29',\n",
       "    'source': 'Bioinformatics',\n",
       "    'fpage': '2959',\n",
       "    'year': '2013',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Maximum Likelihood Genome Assembly',\n",
       "   'offset': 41070,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '1116',\n",
       "    'name_0': 'surname:Medvedev;given-names:P',\n",
       "    'name_1': 'surname:Brudno;given-names:M',\n",
       "    'pub-id_pmid': '19645596',\n",
       "    'issue': '8',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '16',\n",
       "    'source': 'Journal of Computational Biology',\n",
       "    'fpage': '1101',\n",
       "    'year': '2009',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'CGAL: computing genome assembly likelihoods',\n",
       "   'offset': 41105,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Rahman;given-names:A',\n",
       "    'name_1': 'surname:Pachter;given-names:L',\n",
       "    'pub-id_pmid': '23360652',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '14',\n",
       "    'source': 'Genome Biology',\n",
       "    'fpage': 'R8',\n",
       "    'year': '2013',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Nonserial Dynamic Programming and Tree Decomposition in Discrete Optimization',\n",
       "   'offset': 41149,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Shcherbina;given-names:O',\n",
       "    'pub-id_pmid': '16773195',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '160',\n",
       "    'source': 'OR',\n",
       "    'fpage': '155',\n",
       "    'year': '2006',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Dividing a Graph into Triconnected Components',\n",
       "   'offset': 41227,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://link.aip.org/link/?SMJ/2/135/1',\n",
       "    'lpage': '158',\n",
       "    'name_0': 'surname:Hopcroft;given-names:JE',\n",
       "    'name_1': 'surname:Tarjan;given-names:RE',\n",
       "    'issue': '3',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '2',\n",
       "    'source': 'SIAM Journal on Computing',\n",
       "    'fpage': '135',\n",
       "    'year': '1973',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'On-line graph algorithms with SPQR-trees',\n",
       "   'offset': 41273,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Di Battista;given-names:G',\n",
       "    'name_1': 'surname:Tamassia;given-names:R',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '611',\n",
       "    'source': 'Automata, Languages and Programming',\n",
       "    'fpage': '598',\n",
       "    'year': '1990',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Scalable Genome Scaffolding Using Integer Linear Programming',\n",
       "   'offset': 41314,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://doi.acm.org/10.1145/2382936.2382984',\n",
       "    'name_2': 'surname:Zelikovsky;given-names:A',\n",
       "    'name_0': 'surname:Lindsay;given-names:J',\n",
       "    'name_1': 'surname:Salooti;given-names:H',\n",
       "    'name_3': 'surname:Măndoiu;given-names:I',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '383',\n",
       "    'source': 'Proceedings of the ACM Conference on Bioinformatics, Computational Biology and Biomedicine',\n",
       "    'fpage': '377',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fast gapped-read alignment with Bowtie 2',\n",
       "   'offset': 41375,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Langmead;given-names:B',\n",
       "    'name_1': 'surname:Salzberg;given-names:SL',\n",
       "    'pub-id_pmid': '22388286',\n",
       "    'issue': '4',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '359',\n",
       "    'source': 'Nature Methods',\n",
       "    'fpage': '357',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The open graph drawing framework',\n",
       "   'offset': 41416,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Jünger;given-names:M',\n",
       "    'name_3': 'surname:Klein;given-names:K',\n",
       "    'name_0': 'surname:Chimani;given-names:M',\n",
       "    'name_1': 'surname:Gutwenger;given-names:C',\n",
       "    'name_4': 'surname:Mutzel;given-names:P',\n",
       "    'name_5': 'surname:Schulz;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '26',\n",
       "    'source': '15th International Symposium on Graph Drawing',\n",
       "    'fpage': '23',\n",
       "    'year': '2007',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fast scaffolding with small independent mixed integer programs',\n",
       "   'offset': 41449,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1093/bioinformatics/btr562',\n",
       "    'name_2': 'surname:Välimäki;given-names:N',\n",
       "    'name_0': 'surname:Salmela;given-names:L',\n",
       "    'name_1': 'surname:Mäkinen;given-names:V',\n",
       "    'name_4': 'surname:Ukkonen;given-names:E',\n",
       "    'name_3': 'surname:Ylinen;given-names:J',\n",
       "    'issue': '23',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '3265',\n",
       "    'source': 'Bioinformatics (Oxford, England)',\n",
       "    'fpage': '3259',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'volume': '27'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': \"V12. 1: User's Manual for CPLEX\",\n",
       "   'offset': 41512,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:CPLEX;given-names:II',\n",
       "    'issue': '53',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '46',\n",
       "    'source': 'International Business Machines Corporation',\n",
       "    'fpage': '157',\n",
       "    'year': '2009',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Efficient de novo assembly of large genomes using compressed data structures',\n",
       "   'offset': 41544,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '556',\n",
       "    'name_0': 'surname:Simpson;given-names:JT',\n",
       "    'name_1': 'surname:Durbin;given-names:R',\n",
       "    'pub-id_pmid': '22156294',\n",
       "    'issue': '3',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '22',\n",
       "    'source': 'Genome Research',\n",
       "    'fpage': '549',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fast gapped-read alignment with Bowtie 2',\n",
       "   'offset': 41621,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1038/nmeth.1923',\n",
       "    'lpage': '359',\n",
       "    'name_0': 'surname:Langmead;given-names:B',\n",
       "    'name_1': 'surname:Salzberg;given-names:SL',\n",
       "    'issue': '4',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '9',\n",
       "    'source': 'Nat Meth',\n",
       "    'fpage': '357',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Opera: Reconstructing Optimal Genomic Scaffolds with High-Throughput Paired-End Sequences Research in Computational Molecular Biology',\n",
       "   'offset': 41662,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1007/978-3-642-20036-6\\\\40',\n",
       "    'name_2': 'surname:Sung;given-names:WK',\n",
       "    'name_0': 'surname:Gao;given-names:S',\n",
       "    'name_1': 'surname:Nagarajan;given-names:N',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '451',\n",
       "    'source': 'Springer Berlin / Heidelberg, Volume 6577 of Lecture Notes in Computer Science',\n",
       "    'fpage': '437',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'QUAST: quality assessment tool for genome assemblies',\n",
       "   'offset': 41796,\n",
       "   'relations': [],\n",
       "   'infons': {'comment': 'http://dx.doi.org/10.1093/bioinformatics/btt086',\n",
       "    'name_2': 'surname:Vyahhi;given-names:N',\n",
       "    'name_0': 'surname:Gurevich;given-names:A',\n",
       "    'name_1': 'surname:Saveliev;given-names:V',\n",
       "    'name_3': 'surname:Tesler;given-names:G',\n",
       "    'issue': '8',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1075',\n",
       "    'pub-id_pmid': '23422339',\n",
       "    'source': 'Bioinformatics',\n",
       "    'fpage': '1072',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'volume': '29'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bambus 2: scaffolding metagenomes',\n",
       "   'offset': 41849,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Pop;given-names:M',\n",
       "    'lpage': '2971',\n",
       "    'name_0': 'surname:Koren;given-names:S',\n",
       "    'name_1': 'surname:Treangen;given-names:TJ',\n",
       "    'pub-id_pmid': '21926123',\n",
       "    'issue': '21',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '27',\n",
       "    'source': 'Bioinformatics',\n",
       "    'fpage': '2964',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A comprehensive evaluation of assembly scaffolding tools',\n",
       "   'offset': 41883,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Berriman;given-names:M',\n",
       "    'name_3': 'surname:Otto;given-names:TD',\n",
       "    'name_0': 'surname:Hunt;given-names:M',\n",
       "    'name_1': 'surname:Newbold;given-names:C',\n",
       "    'pub-id_pmid': '24581555',\n",
       "    'issue': '3',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '15',\n",
       "    'source': 'Genome Biology',\n",
       "    'fpage': 'R42',\n",
       "    'year': '2014',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []}],\n",
       " 'infons': {'license': 'CC BY'},\n",
       " 'id': '4168704',\n",
       " 'relations': []}"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmcid_to_doc[ids_to_check[520]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
