{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import json\n",
    "import github_api\n",
    "import search_for_url\n",
    "import numpy as np\n",
    "from urllib import parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "papes = glob.glob('../git_papes/*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "papes = sorted(papes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "176"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(papes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(papes[5]) as infile:\n",
    "    dat = json.load(infile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "def standardize_url(u):\n",
    "    pp = parse.urlparse(u)\n",
    "    u = pp.netloc + pp.path\n",
    "    if u[-1]=='/':\n",
    "        u = u[:-1]\n",
    "    return u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "pairs = []\n",
    "for p in papes:\n",
    "    with open(p) as infile:\n",
    "        dat = json.load(infile)\n",
    "        for d in dat:\n",
    "            #assert(len(d['documents'])==1)\n",
    "            pmcid = d['documents'][0]['id']\n",
    "            urls = search_for_url.search_for_url(' '.join([i['text'] for i in d['documents'][0]['passages']]))\n",
    "            for u in urls:\n",
    "                u = standardize_url(u)\n",
    "                if 'github' in u:\n",
    "                    pairs.append((pmcid, u))\n",
    "pairs = np.array(pairs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'github.com/hello'"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "standardize_url('https://github.com/hello/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'Series' object has no attribute 'as_string'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-251-f5c222a44426>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0murls_in_papes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'pmcid'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpairs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'url'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpairs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0murls_in_papes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pmcid'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murls_in_papes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pmcid'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mas_string\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/miniconda3/envs/gitpubs/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   5065\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_can_hold_identifiers_and_holds_name\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5066\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5067\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5068\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5069\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'Series' object has no attribute 'as_string'"
     ]
    }
   ],
   "source": [
    "urls_in_papes = pd.DataFrame({'pmcid': np.array(pairs)[:, 0], 'url': np.array(pairs)[:, 1]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmcid</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4910214</td>\n",
       "      <td>github.com/lh3/wgsim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5071354</td>\n",
       "      <td>github.com/JAvRZ/andi-dataprocessing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5013563</td>\n",
       "      <td>github.com/AharoniLab/MatchWeiz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4958917</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5107194</td>\n",
       "      <td>github.com/ijmarshall/picotron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5127779</td>\n",
       "      <td>github.com/awilfert/PSAP-pipeline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4928833</td>\n",
       "      <td>gist.github.com/darmitage/3179407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5140850</td>\n",
       "      <td>github.com/confunguido/PokemonDataAnalysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4966778</td>\n",
       "      <td>uhkniazi.github.io/dismiss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5045566</td>\n",
       "      <td>github.com/alexandrovteam/spatial-corals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5048062</td>\n",
       "      <td>uubram.github.io/RTCR/along</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5137812</td>\n",
       "      <td>github.com/bulik/ldsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5141584</td>\n",
       "      <td>github.com/tanghaibao/treecut/blob/master/scri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5142241</td>\n",
       "      <td>github.com/aboyle/F-seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5142633</td>\n",
       "      <td>www.github.com/rhenley/Pythion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5145274</td>\n",
       "      <td>github.com/Caleydo/caleydo_clue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5146994</td>\n",
       "      <td>github.com/Caleydo/pathfinder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>5147733</td>\n",
       "      <td>github.com/GauravPandeyLab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5148158</td>\n",
       "      <td>github.com/tdsmith/eleven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>5152673</td>\n",
       "      <td>github.com/nmcnulty/COPRO-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>5152685</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5154552</td>\n",
       "      <td>github.com/rhenley/Pyth-Ion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5154821</td>\n",
       "      <td>github.com/rgcgithub/clamms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>5154867</td>\n",
       "      <td>github.com/ZWang-Lab/LSKAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>5159287</td>\n",
       "      <td>github.com/markus-nilsson/md-dmri</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>5161348</td>\n",
       "      <td>github.com/divara01/PSB2017_ReproducibilityOfBNs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>5164934</td>\n",
       "      <td>github.com/bulik/ldsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/torognes/swarm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/oscar-franzen/oclust</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5164958</td>\n",
       "      <td>github.com/fangly/AmpliCopyrighter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29223</th>\n",
       "      <td>6168466</td>\n",
       "      <td>github.com/bulik/ldsc/wiki/Genetic-Correlation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29224</th>\n",
       "      <td>6168517</td>\n",
       "      <td>github.com/baj12/clean_ngs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29225</th>\n",
       "      <td>6168540</td>\n",
       "      <td>github.com/SantanderMetGroup/notebooks</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29226</th>\n",
       "      <td>6168587</td>\n",
       "      <td>github.com/CamachoDejay/FRET-calculations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29227</th>\n",
       "      <td>6168625</td>\n",
       "      <td>github.com/BigNeuron/Data/releases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29228</th>\n",
       "      <td>6168662</td>\n",
       "      <td>github.com/CuppenResearch/IAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29229</th>\n",
       "      <td>6168678</td>\n",
       "      <td>k-basinski.github.io/mia_music_prefs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29230</th>\n",
       "      <td>6168856</td>\n",
       "      <td>github.com/lh3/seqtk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29231</th>\n",
       "      <td>6169016</td>\n",
       "      <td>github.com/svkucheryavski/mdatoolsm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29232</th>\n",
       "      <td>6169378</td>\n",
       "      <td>github.com/arunsethuraman/inrelate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29233</th>\n",
       "      <td>6169389</td>\n",
       "      <td>github.com/chaozhi/RABBIT.git</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29234</th>\n",
       "      <td>6169393</td>\n",
       "      <td>github.com/danrdanny/Drosophila15GenomesProject</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29235</th>\n",
       "      <td>6169397</td>\n",
       "      <td>github.com/danrdanny/Nanopore_ISO1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29236</th>\n",
       "      <td>6169397</td>\n",
       "      <td>github.com/esolares/DMI1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29237</th>\n",
       "      <td>6169740</td>\n",
       "      <td>github.com/cancerit/CaVEMan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29238</th>\n",
       "      <td>6169915</td>\n",
       "      <td>github.com/ChusCarro/MatCardiacMLab/releases/t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29239</th>\n",
       "      <td>6169918</td>\n",
       "      <td>github.com/TheJacksonLaboratory/GDCSlicing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29240</th>\n",
       "      <td>6169937</td>\n",
       "      <td>github.com/VHIRHepatiques/QSutils</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29241</th>\n",
       "      <td>6170166</td>\n",
       "      <td>broadinstitute.github.io/picard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29242</th>\n",
       "      <td>6170167</td>\n",
       "      <td>github.com/satijalab/seurat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29243</th>\n",
       "      <td>5526259</td>\n",
       "      <td>github.com/james-e-barrett</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29244</th>\n",
       "      <td>5526664</td>\n",
       "      <td>github.com/swiederm/predictive-gain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29245</th>\n",
       "      <td>5526664</td>\n",
       "      <td>github.com/elifesciences-publications/predicti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29246</th>\n",
       "      <td>5526665</td>\n",
       "      <td>github.com/kescobo/kvasir</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29247</th>\n",
       "      <td>5526665</td>\n",
       "      <td>github.com/elifesciences-publications/kvasir</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29248</th>\n",
       "      <td>5526666</td>\n",
       "      <td>github.com/JaneliaSciComp/CohenBolstadLee_eLif...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29249</th>\n",
       "      <td>5526896</td>\n",
       "      <td>cole-trapnell-lab.github.io/cufflinks</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29250</th>\n",
       "      <td>5526946</td>\n",
       "      <td>drpowell.github.io/vennt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29251</th>\n",
       "      <td>5526970</td>\n",
       "      <td>github.com/vaibhhav/phage_competition_paper_fr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29252</th>\n",
       "      <td>5526973</td>\n",
       "      <td>github.com/alexdobin/STAR/releases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29253 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         pmcid                                                url\n",
       "0      4910214                               github.com/lh3/wgsim\n",
       "1      5071354               github.com/JAvRZ/andi-dataprocessing\n",
       "2      5013563                    github.com/AharoniLab/MatchWeiz\n",
       "3      4958917                    broadinstitute.github.io/picard\n",
       "4      5107194                     github.com/ijmarshall/picotron\n",
       "5      5127779                  github.com/awilfert/PSAP-pipeline\n",
       "6      4928833                  gist.github.com/darmitage/3179407\n",
       "7      5140850         github.com/confunguido/PokemonDataAnalysis\n",
       "8      4966778                         uhkniazi.github.io/dismiss\n",
       "9      5045566           github.com/alexandrovteam/spatial-corals\n",
       "10     5048062                        uubram.github.io/RTCR/along\n",
       "11     5137812                              github.com/bulik/ldsc\n",
       "12     5141584  github.com/tanghaibao/treecut/blob/master/scri...\n",
       "13     5142241                            github.com/aboyle/F-seq\n",
       "14     5142633                     www.github.com/rhenley/Pythion\n",
       "15     5145274                    github.com/Caleydo/caleydo_clue\n",
       "16     5146994                      github.com/Caleydo/pathfinder\n",
       "17     5147733                         github.com/GauravPandeyLab\n",
       "18     5148158                          github.com/tdsmith/eleven\n",
       "19     5152673                      github.com/nmcnulty/COPRO-Seq\n",
       "20     5152685                    broadinstitute.github.io/picard\n",
       "21     5154552                        github.com/rhenley/Pyth-Ion\n",
       "22     5154821                        github.com/rgcgithub/clamms\n",
       "23     5154867                         github.com/ZWang-Lab/LSKAT\n",
       "24     5159287                  github.com/markus-nilsson/md-dmri\n",
       "25     5161348   github.com/divara01/PSB2017_ReproducibilityOfBNs\n",
       "26     5164934                              github.com/bulik/ldsc\n",
       "27     5164958                          github.com/torognes/swarm\n",
       "28     5164958                    github.com/oscar-franzen/oclust\n",
       "29     5164958                 github.com/fangly/AmpliCopyrighter\n",
       "...        ...                                                ...\n",
       "29223  6168466     github.com/bulik/ldsc/wiki/Genetic-Correlation\n",
       "29224  6168517                         github.com/baj12/clean_ngs\n",
       "29225  6168540             github.com/SantanderMetGroup/notebooks\n",
       "29226  6168587          github.com/CamachoDejay/FRET-calculations\n",
       "29227  6168625                 github.com/BigNeuron/Data/releases\n",
       "29228  6168662                      github.com/CuppenResearch/IAP\n",
       "29229  6168678               k-basinski.github.io/mia_music_prefs\n",
       "29230  6168856                               github.com/lh3/seqtk\n",
       "29231  6169016                github.com/svkucheryavski/mdatoolsm\n",
       "29232  6169378                 github.com/arunsethuraman/inrelate\n",
       "29233  6169389                      github.com/chaozhi/RABBIT.git\n",
       "29234  6169393    github.com/danrdanny/Drosophila15GenomesProject\n",
       "29235  6169397                 github.com/danrdanny/Nanopore_ISO1\n",
       "29236  6169397                           github.com/esolares/DMI1\n",
       "29237  6169740                        github.com/cancerit/CaVEMan\n",
       "29238  6169915  github.com/ChusCarro/MatCardiacMLab/releases/t...\n",
       "29239  6169918         github.com/TheJacksonLaboratory/GDCSlicing\n",
       "29240  6169937                  github.com/VHIRHepatiques/QSutils\n",
       "29241  6170166                    broadinstitute.github.io/picard\n",
       "29242  6170167                        github.com/satijalab/seurat\n",
       "29243  5526259                         github.com/james-e-barrett\n",
       "29244  5526664                github.com/swiederm/predictive-gain\n",
       "29245  5526664  github.com/elifesciences-publications/predicti...\n",
       "29246  5526665                          github.com/kescobo/kvasir\n",
       "29247  5526665       github.com/elifesciences-publications/kvasir\n",
       "29248  5526666  github.com/JaneliaSciComp/CohenBolstadLee_eLif...\n",
       "29249  5526896              cole-trapnell-lab.github.io/cufflinks\n",
       "29250  5526946                           drpowell.github.io/vennt\n",
       "29251  5526970  github.com/vaibhhav/phage_competition_paper_fr...\n",
       "29252  5526973                 github.com/alexdobin/STAR/releases\n",
       "\n",
       "[29253 rows x 2 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "urls_in_papes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "urls_in_papes.to_csv('../data/papes_and_github_urls.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [],
   "source": [
    "urls_in_papes = pd.read_csv('../data/papes_and_github_urls.csv')\n",
    "urls_in_papes['pmcid'] = urls_in_papes['pmcid'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16510"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(urls_in_papes.pmcid))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Diagnose some cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "pmcid_to_doc = {}\n",
    "for p in papes:\n",
    "    with open(p) as infile:\n",
    "        dat = json.load(infile)\n",
    "        for d in dat:\n",
    "            #assert(len(d['documents'])==1)\n",
    "            pmcid = d['documents'][0]['id']\n",
    "            pmcid_to_doc[pmcid] = d['documents'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'passages': [{'text': '‘Genome skimming’ with the MinION hand-held sequencer identifies CITES-listed shark species in India’s exports market',\n",
       "   'offset': 0,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Cantu;given-names:Vito Adrian',\n",
       "    'name_3': 'surname:Fellows;given-names:Sam R.',\n",
       "    'name_0': 'surname:Johri;given-names:Shaili',\n",
       "    'name_1': 'surname:Solanki;given-names:Jitesh',\n",
       "    'name_6': 'surname:Vyas;given-names:Asit',\n",
       "    'article-id_publisher-id': '40940',\n",
       "    'license': 'Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons license, and indicate if changes were made. The images or other third party material in this article are included in the article’s Creative Commons license, unless indicated otherwise in a credit line to the material. If material is not included in the article’s Creative Commons license and your intended use is not permitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this license, visit http://creativecommons.org/licenses/by/4.0/.',\n",
       "    'name_5': 'surname:Moreno;given-names:Isabel',\n",
       "    'elocation-id': '4476',\n",
       "    'article-id_doi': '10.1038/s41598-019-40940-9',\n",
       "    'section_type': 'TITLE',\n",
       "    'article-id_pmid': '30872700',\n",
       "    'volume': '9',\n",
       "    'article-id_pmc': '6418218',\n",
       "    'year': '2019',\n",
       "    'name_4': 'surname:Edwards;given-names:Robert A.',\n",
       "    'type': 'front',\n",
       "    'name_7': 'surname:Dinsdale;given-names:Elizabeth A.'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes - sharks, rays, skates, and chimeras, are among the most threatened and data deficient vertebrate species. Global demand for shark and ray derived products, drives unregulated and exploitative fishing practices, which are in turn facilitated by the lack of ecological data required for effective conservation of these species. Here, we describe a Next Generation Sequencing method (using the MinION, a hand-held portable sequencing device from Oxford Nanopore Technologies), and analyses pipeline for molecular ecological studies in Chondrichthyes. Using this method, the complete mitochondrial genome and nuclear intergenic and protein-coding sequences were obtained by direct sequencing of genomic DNA obtained from shark fin tissue. Recovered loci include mitochondrial barcode sequences- Cytochrome oxidase I, NADH2, 16S rRNA and 12S rRNA- and nuclear genetic loci such as 5.8S rRNA, Internal Transcribed Spacer 2, and 28S rRNA regions, which are commonly used for taxonomic identification. Other loci recovered were the nuclear protein-coding genes for antithrombin or SerpinC, Immunoglobulin lambda light chain, Preprogehrelin, selenium binding protein 1(SBP1), Interleukin-1\\xa0beta (IL-1β) and Recombination-Activating Gene 1 (RAG1). The median coverage across all genetic loci was 20x and sequence accuracy was ≥99.8% compared to reference sequences. Analyses of the nuclear ITS2 region and the mitochondrial protein-encoding loci allowed accurate taxonomic identification of the shark specimen as Carcharhinus falciformis, a CITES Appendix II species. MinION sequencing provided 1,152,211\\u2009bp of new shark genome, increasing the number of sequenced shark genomes to five. Phylogenetic analyses using both mitochondrial and nuclear loci provided evidence that Prionace glauca is nested within Carcharhinus, suggesting the need for taxonomic reassignment of P. glauca. We increased genomic information about a shark species for ecological and population genetic studies, enabled accurate identification of the shark tissue for biodiversity indexing and resolved phylogenetic relationships among multiple taxa. The method was independent of amplification bias, and adaptable for field assessments of other Chondrichthyes and wildlife species in the future.',\n",
       "   'offset': 124,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Introduction',\n",
       "   'offset': 2405,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes (sharks, rays, skates, and chimaeras) are among the oldest extant vertebrates. They are vital to top-down regulation and maintenance of oceanic ecosystems, and to sustain healthy commercial fisheries and ecosystem services derived from marine environments. The high demand for shark and ray fins and an emerging demand for shark meat and oil has made world trade in Chondrichthyes a >$US 800 million industry. Demand for shark fins and meat has caused overfishing and exploitation of Chondrichthyes worldwide, driving an unprecedented number of chondrichthyan species towards extinction. Of the ~1,178 chondrichthyan species known today, 25% are threatened with extinction globally, making Chondrichthyes one of the most endangered vertebrate species groups. The threat is even more severe in areas such as the Arabian Seas Region (ASR) with up to 50% of chondrichthyan species in the region facing extinction threat.',\n",
       "   'offset': 2418,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyan fisheries are driven by 20 nations and a vast majority of these countries experience the highest levels of data deficiency. An estimated 46% of Chondrichthyes are considered data deficient, suggesting that information is insufficient to determine their conservation status. These data deficiencies restrict design and implementation of efficient conservation measures and allow continued exploitation of the most vulnerable Chondrichthyes. Not surprisingly, nations with high data deficiency also have some of the highest percentages of threatened Chondrichthyes. For effective conservation of remaining chondrichthyan populations continued monitoring and assessment is essential in order to determine taxonomic units, population structure, population size and identify priority conservation units with respect to evolutionary and management significance. Once these data are available, they can be used to design effective conservation measures including but not limited to, the designation of protected areas, reduction of targeted and incidental fishing of priority concern units, and CITES listing of priority concern units to protect vulnerable populations.',\n",
       "   'offset': 3351,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Data deficiency of Chondrichthyes can be addressed by increasing the rate of ecological investigations in areas with a high density of threatened and data deficient species. In the current scenario, molecular ecological studies in Chondrichthyes are typically restricted to DNA barcoding and microsatellite analyses for taxonomic identification and population studies respectively. Molecular taxonomy is conducted by PCR amplification and sequencing of DNA barcodes such as Cytochrome oxidase 1(COI)– NADH2 16S rRNA and 12S rRNA, of the mitochondrial genome and the Recombination-Activating Gene 1(RAG1) and Internal transcribed spacer 2 (ITS) regions of the nuclear genome. Sampling each one of these barcodes from a specimen requires a priori sequence information, PCR optimization, PCR amplification, post PCR clean up and finally sequencing. Single barcode PCR based identification is not the most efficient method for taxonomic identification and can be limiting due to the following reasons. First, single barcodes do not always guarantee accurate taxonomic identification, due to non-amplification for certain species using universal primers, or identical barcode sequences for multiple species such as Carcharhinid sharks or different barcode sequences for geographically isolated populations of the same nominal species. Second, the laboratory required for amplification and sequencing of barcodes from chondrichthyan specimens is often inaccessible to researchers and fisheries scientists in areas with heavy chondrichthyan fishing and exports, requiring them to outsource detection, adding significantly to the processing time. Last, single barcode PCRs provides up to 1000\\u2009bp of data and restrict the amount of genetic data derived for the respective species. Microsatellites used for population studies among Chondrichthyes are also PCR based, only include neutral genetic loci, require standardization for use across different laboratories and do not provide genome-wide assessments of neutral as well as adaptive loci. Accurate estimates of population size and structure as well as understanding of evolutionary units are only obtained from screening multiple genomic loci such as single nucleotide polymorphisms (SNPs) from genomic data sets. Thus, PCR based approaches for taxonomic and population studies are limiting to molecular ecological studies in Chondrichthyes. Adopting a genomics approach could allow assessment of multiple genomic loci without amplification bias and without the need for a priori sequence information. Such an approach would be ideal to increase ecological and evolutionary investigations in Chondrichthyes, similar to those of other commercial fisheries. Increasing capacity for use of genome-wide data addresses taxonomic inadequacies and enables complex ecological and evolutionary questions to be answered by providing information about population size, structure, biogeography and priority conservation units. These data together will address chondrichthyan data deficiencies and assist conservation efforts.',\n",
       "   'offset': 4529,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genomic analyses have limited adoption among chondrichthyan researchers due to complex instrumentation and expensive sequencing for whole genome shotgun sequencing or reduced representation methods such as Restriction site-Associated-DNA-Sequencing (RADseq) and Hybridization based gene capture. A crucial aim of the current study was to reduce the burden of sample processing and analyses, and to increase accuracy, speed and breadth of ecological investigations of Chondrichthyes through genome sequencing with sophisticated yet accessible technology. We explored the use of multiple mitochondrial and nuclear barcode regions which form the high copy number fraction of the genome to increase accuracy of taxonomic identification. We explored direct sequencing of genomic DNA (gDNA) to reduce sample processing times and to avoid amplification biases, thus increasing the potential for application of the assay to multi- species assessments. ‘Genome skimming’ which allows deep sequencing of high copy number fractions of the genome was used to access multiple mitochondrial and nuclear barcode sequences from shark genomic DNA. Sequencing was done on the MinION portable sequencer from Oxford Nanopore Technologies (ONT) which requires library preparation and loading times of as little as thirty minutes, allows sequence acquisition as early as 2\\u2009hours into sequencing, and allows local sequencing and analyses of genomic DNA with a user-friendly software interface installed on a portable computer.',\n",
       "   'offset': 7592,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We present characterization of ‘genome skimming’ on the MinION portable sequencer for taxonomic identification and phylogenetic analyses of Chondrichthyes. We report identification of a shark fin specimen in India’s export market to Carcharhinus falciformis, based on the complete mitochondrial sequence and nuclear genetic regions of the specimen. In addition, we present two sequence analyses methods which can be adapted to other wildlife species and which are suitable for standalone analyses in the field or laboratory, respectively. Last we resolved the placement of P. glauca within the Carcharhinus genus and described its relationships with other taxa in the genus, with high statistical power using the entire mitochondrial genome and nuclear genes from C. falciformis. Our sequencing and analyses methods increase the available genomic information about C. falciformis and provide an accessible tool for genome-wide studies across Chondrichthyes.',\n",
       "   'offset': 9102,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Results',\n",
       "   'offset': 10066,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genus level taxonomic resolution through photo identification and COI fragment analyses',\n",
       "   'offset': 10074,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sampling location and sample morphology of a dismembered shark specimen at time of sampling.',\n",
       "   'offset': 10162,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig1',\n",
       "    'file': '41598_2019_40940_Fig1_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The shark specimen encountered in Veraval, Gujarat, was processed for international fin and meat exports, and the head and fins of the specimen could be sampled for tissue collection and photo-identification purposes (Fig.\\xa01). Based on features of the head and mouth cavities the specimen was identified to the Carcharhinidae family. In order to obtain a species-level taxonomic resolution, we used DNA extracted from the specimen to PCR amplify three different fragments from COI and NADH2 mitochondrial loci. We were successful in amplifying one short 160\\u2009bp fragment of the cytochrome oxidase 1 (COI) gene, while PCR amplification of two longer fragments from COI and NADH2 did not yield a PCR product. The 160\\u2009bp PCR fragment was cross-referenced with the BOLD database which resulted in 100% sequence similarity to six species of the Carcharhinus genus including Carcharhinus brevipinna, Carcharhinus falciformis, Carcharhinus melanopterus, Carcharhinus sorrah, Carcharhinus altimus and Carcharhinus limbatus. These results obtained using a single short barcode supported identification based on morphology but were inconclusive for species-level identification.',\n",
       "   'offset': 10255,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Assessment of complete mitochondrial sequence using ‘genome skimming’ on the MinION',\n",
       "   'offset': 11428,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to obtain higher taxonomic resolution by assessment of longer and multiple barcode sequences ‘genome skimming’ using genomic DNA of the specimen was conducted on the MinION sequencer. We obtained 74,536 sequences with read lengths between 5–98,997\\u2009bp within\\xa036\\u2009hours of a single sequencing run.',\n",
       "   'offset': 11516,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Two workflow designs used in this study one, for field or forensic applications with no connectivity and average laptop computing power and two, for detailed laboratory analyses with connectivity and high computing power.',\n",
       "   'offset': 11831,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig2',\n",
       "    'file': '41598_2019_40940_Fig2_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(A) Sequencing coverage for complete mitochondrial genome of the specimen. The mean coverage for 16, 675 bases of the mitogenome was 19.4 with a Std Dev of 2.0, and minimum and maximum coverage of 16x and 25x respectively. Coverage in the forward direction was 9.5 and 9.9 in the reverse. Sequencing coverage for regularly used individual mitochondrial barcodes is shown in Fig.\\xa0S1. (B) Sequencing coverage for mitochondrial and nuclear genes of the specimen. Sequencing coverage for nine annotated nuclear genome contigs and one mitochondrial contig recovered with Workflow II ranged from 1x to >50x.',\n",
       "   'offset': 12053,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig3',\n",
       "    'file': '41598_2019_40940_Fig3_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The sequences were analyzed using two parallel workflows (Fig.\\xa02). In the first workflow, raw reads from the MinION were mapped to reference mitogenomes of Carcharhinus spp. identified to match the shark specimen by PCR fragment alignment. Mitogenomes were available for four of six Carcharhinus spp. (Carcharhinus melanopterus Carcharhinus sorrah, Carcharhinus brevipinna and Carcharhinus falciformis (Supplementary Table\\xa0S1) that matched the\\xa0specimen using PCR. These 4 mitogenomes were used to account for biases in the database and increase the mapping\\xa0accuracy of MinION sequences. All of the 74,536 reads from the MionION were mapped onto each of the 4 mitogenomes, which identified the same set of 47 reads (~0.06%) from each mapping. Assembly of the 47 reads generated a complete mitochondrial contig of 16,675\\u2009bp (Fig.\\xa03A, Genbank accession #MK092088), a genome size similar to mitochondrial genomes reported for Carcharhinus spp., (eg. Carcharhinus sorrah (16,707\\u2009bp), Carcharhinus brevipinna (16,706\\u2009bp), Carcharhinus melanopterus (16,706\\u2009bp), Carcharhinus plumbeus (16,706\\u2009bp), and Carcharhinus obscurus (16,706\\u2009bp), but significantly different than the published mitogenome for Carcharhinus falciformis at 17,639\\u2009bp. Mean coverage for the complete mitochondrial genome was 19.4\\u2009×\\u2009±\\u20092\\u2009×\\u2009(Fig.\\xa03A, Supplementary Fig.\\xa0S1).',\n",
       "   'offset': 12656,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In the second workflow (Fig.\\xa02), all 74,536 sample reads were first assembled into contigs, quality controlled by trimming read ends and aligned against a database of chondrichthyan mitochondrial and nuclear sequences to annotate the contigs. This method yielded a mitochondrial sequence of 16,477\\u2009bp which was 98% identical to the sequence in workflow 1 and nuclear genetic loci such as the Internal transcribed spacer 2 (ITS2) region, 5.8S, 28S and 18S\\u2009 rRNA coding regions, and protein-coding regions for antithrombin or SerpinC, Immunoglobulin lambda light chain, Preprogehrelin, selenium binding protein 1(SBP1), Interleukin-1\\xa0beta (IL-1β) and Recombination-Activating ene 1 (RAG1) (SRA accession #PRJNA503780). With the second workflow, we obtained an almost complete mitochondrial sequence and in addition, nuclear coding and non-coding regions. Mean coverage for nuclear coding and non-coding genes recovered ranged from 1x to >50x\\xa0(Fig.\\xa03B, Supplementary Fig.\\xa0S1).',\n",
       "   'offset': 14023,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Annotated mitochondrial sequence indicating gene position and G/C richness regions. Outer circle represents aligned genes with outermost representing the heavy strand (H) and adjacent inner ring the light strand (L). Green represents protein coding regions, pink tRNA, blue rRNA, and grey is the D-loop. Middle circle in black represents GC content: Protruding outwards indicates GC rich while inward protrusion shows AT rich regions. The inner most circle is GC skew (G\\u2009−\\u2009C/G\\u2009+\\u2009C) with positive skew (between 0 and 1) represented by green and negative as purple (between 0 and −1).',\n",
       "   'offset': 15007,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig4',\n",
       "    'file': '41598_2019_40940_Fig4_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genes identified in the Carcharhinus falciformis mitochondrial sequence with their start and end positions on the forward (+) or reverse (−) strands\\xa0and a score representing the heuristic measure of the annotation content for each entry.',\n",
       "   'offset': 15602,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab1',\n",
       "    'file': 'Tab1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Start\\tStop\\tType\\tName\\xa0(Direction)\\tScore\\n1\\t70\\ttRNA\\t\\ntRNA-Phe\\xa0(+)\\n\\t1.50E-14\\n72\\t1031\\trRNA\\t\\n12s rRNA\\xa0(+)\\n\\t0\\n1029\\t1100\\ttRNA\\t\\ntRNA-VAL\\xa0(+)\\n\\t1.00E-14\\n1124\\t2764\\trRNA\\t\\n16s rRNA\\xa0(+)\\n\\t0\\n2764\\t2838\\ttRNA\\t\\ntRNA-Leu2\\xa0(+)\\n\\t1.40E-12\\n2839\\t3813\\tgene\\t\\nND1\\xa0(+)\\n\\t453393778\\n3814\\t3883\\ttRNA\\t\\ntRNA-Ile\\xa0(+)\\n\\t2.00E-14\\n3885\\t3956\\ttRNA\\t\\ntRNA-Gln\\xa0(-)\\n\\t5.00E-15\\n3956\\t4024\\ttRNA\\t\\ntRNA-Met\\xa0(+)\\n\\t6.10E-15\\n4025\\t5071\\tgene\\t\\nND2\\xa0(+)\\n\\t324312000.6\\n5070\\t5140\\ttRNA\\t\\ntRNA-Trp\\xa0(+)\\n\\t1.80E-15\\n5142\\t5210\\ttRNA\\t\\ntRNA-Ala\\xa0(-)\\n\\t9.80E-14\\n5211\\t5281\\ttRNA\\t\\ntRNA-Asn\\xa0(-)\\n\\t2.00E-12\\n5317\\t5384\\ttRNA\\t\\ntRNA-Cys\\xa0(-)\\n\\t1.50E-11\\n5386\\t5454\\ttRNA\\t\\ntRNA-Tyr\\xa0(-)\\n\\t2.60E-16\\n5456\\t7012\\tgene\\t\\nCOI\\xa0(+)\\n\\t1166241108\\n7013\\t7083\\ttRNA\\t\\ntRNA-Ser2\\xa0(-)\\n\\t6.40E-15\\n7087\\t7156\\ttRNA\\t\\ntRNA-Asp\\xa0(+)\\n\\t6.30E-13\\n7164\\t7854\\tgene\\t\\nCOII\\xa0(+)\\n\\t280357893.9\\n7855\\t7925\\ttRNA\\t\\ntRNA-Lys\\xa0(+)\\n\\t4.90E-13\\n7927\\t8094\\tgene\\t\\nATPase 8\\xa0(+)\\n\\t3849114.9\\n8085\\t8719\\tgene\\t\\nATPase 6\\xa0(+)\\n\\t135520179.4\\n8763\\t9548\\tgene\\t\\nCOIII\\xa0(+)\\n\\t361705805.6\\n9551\\t9619\\ttRNA\\t\\ntRNA-Gly\\xa0(+)\\n\\t1.90E-13\\n9620\\t9970\\tgene\\t\\nND3\\xa0(+)\\n\\t43327351.7\\n9969\\t10038\\ttRNA\\t\\ntRNA-Arg\\xa0(+)\\n\\t2.90E-12\\n10039\\t10335\\tgene\\t\\nND4L\\xa0(+)\\n\\t25640182.3\\n10329\\t11709\\tgene\\t\\nND4\\xa0(+)\\n\\t770745173.3\\n11710\\t11778\\ttRNA\\t\\ntRNA-His\\xa0(+)\\n\\t1.40E-11\\n11779\\t11843\\ttRNA\\t\\ntRNA-Ser1\\xa0(+)\\n\\t5.60E-08\\n11844\\t11915\\ttRNA\\t\\ntRNA-Leu1\\xa0(+)\\n\\t8.30E-22\\n11916\\t13745\\tgene\\t\\nND5\\xa0(+)\\n\\t1036110292\\n13741\\t14262\\tgene\\t\\nND6\\xa0(-)\\n\\t47160329.7\\n14263\\t14331\\ttRNA\\t\\ntRNA-Glu\\xa0(-)\\n\\t1.10E-11\\n14334\\t15479\\tgene\\t\\nCyt b\\xa0(+)\\n\\t760558631.5\\n15479\\t15550\\ttRNA\\t\\ntRNA-Thr\\xa0(+)\\n\\t1.20E-11\\n15553\\t15620\\ttRNA\\t\\ntRNA-Pro\\xa0(-)\\n\\t2.60E-09\\n15663\\t16674\\trep-origin\\t\\nDloop\\xa0(+)\\n\\t3240692.3\\n',\n",
       "   'offset': 15843,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab1',\n",
       "    'File': 'Tab1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Annotation of the complete mitochondrial sequence using annotation tools followed by manual annotation of the D-Loop region in Geneious® yielded recovery of all coding and non-coding regions of the mitochondrial genome including thirteen protein coding regions, two rRNA coding regions, twenty-two tRNA coding regions, a replication region and a control region (Fig.\\xa04, Table\\xa01). These results demonstrate the efficiency of genome skimming over conventional single barcode PCR based assays. A single sequencing run on the MinION sequencer provided the entire mitochondrial sequence equivalent of multiple mitochondrial barcodes and multiple nuclear genetic loci. These data were acquired in a very brief time frame, without any amplification bias, and with minimal infrastructure compared to PCR based studies.',\n",
       "   'offset': 17397,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Species-level taxonomic resolution using phylogenetic analyses of multiple barcode regions',\n",
       "   'offset': 18211,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '(A) Bayesian estimate of relationships among Carcharhinus spp. using complete mitogenomes. Bayesian phylogenetic estimate for complete mitochondrial genomes from 13 Carcharhinus species and Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis. Numbers at nodes are posterior probabilities. (B) Bayesian estimate of relationships among Carcharhinus spp. using protein coding genes. Bayesian phylogenetic estimate from 13 protein-coding mitochondrial genes from 13 Carcharhinus species and, Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis.\\xa0Numbers at nodes are posterior probabilities.',\n",
       "   'offset': 18302,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig5',\n",
       "    'file': '41598_2019_40940_Fig5_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We constructed phylogenetic trees based on the full mitochondrial genomes (Fig.\\xa05A) and 13 protein coding regions (Fig.\\xa05B) from the mitogenome of 13 shark species of the genus Carcharhinus, and Prionace glauca, and used Galeocerdo cuvier as an outgroup (family Carcharhinidae). Genbank accession numbers for reference sequences used are listed in Supplementary Tables\\xa0S1 and S2. Our phylogenies based on multiple mitochondrial markers are congruent with previous phylogenetic studies in the Carcharhinus genus conducted using a single NADH2 locus albeit ours have better statistical support (Fig.\\xa05A). Phylogenetic estimates showed robust support (>0.95 posterior probability (Fig.\\xa05A,B), >90% bootstrapping (Supplementary Fig.\\xa0S2) toward the tips of the phylogeny and lowered support at some deeper nodes. Phylogenetic estimates based on the full mitochondrial genome (Fig.\\xa05A) showed higher support throughout than those based on protein-coding genes only (Fig.\\xa05B). Using both analyses, based on full or protein-coding mitochondrial sequences, we recovered a sister relationship between the unknown sample and C. falciformis (Fig.\\xa05A,B). Our analyses estimated identical phylogenetic relationships among these 15 species, including support for the placement of P. glauca within Carcharhinus (Fig.\\xa05A,B).',\n",
       "   'offset': 18992,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bayesian estimate of relationships among Carcharhinus spp. using mitochondrial and nuclear genes. Bayesian phylogenetic estimate from concatenated alignment of three mitochondrial protein coding loci (NADH2, COI, NADH4) and one nuclear locus (ITS2) from 32 Carcharhinus species, Prionace glauca, and Galeocerdo cuvier as an outgroup. The unknown sample clusters with C. falciformis. Numbers at nodes are posterior probabilities.',\n",
       "   'offset': 20310,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'Fig6',\n",
       "    'file': '41598_2019_40940_Fig6_HTML.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We also constructed phylogenetic trees based on a concatenated sequence of three mitochondrial protein-coding sequences (NADH2, COI, NADH4) and one nuclear intergenic region (ITS2) of 33 shark species of the genus Carcharhinus (of 35 currently described species), with P. glauca, and G. cuvier as an outgroup (Fig.\\xa06). Genbank accession numbers for references sequences used are listed in Supplementary Table\\xa0S3. Both Maximum Likelihood and Bayesian analyses using these concatenated genes strongly supported a sister relationship between the unknown sample and C. falciformis (Fig.\\xa06). The concatenated analyses showed robust support (>0.95 posterior probability (Fig.\\xa06), >90% bootstrapping, (Supplementary Fig.\\xa0S2) toward the tips of the phylogeny and lowered support at many deeper nodes. These results corroborate relationships seen using mitochondrial markers (Fig.\\xa05A,B) and improve on our phylogenetic analyses of P. glauca using only mitochondrial genomes, which did not include C. amblyrhynchos due to unavailability of a complete mitogenome.',\n",
       "   'offset': 20739,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Using phylogenetic analyses based on multiple nuclear and mitochondrial regions we were able to identify an unknown shark specimen as Carcharhinus falciformis, a CITES Appendix II listed species.',\n",
       "   'offset': 21798,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '*Only one mitochondrial reference sequence for C. falciformis with 17, 639 base pairs exists. Length of mitochondrial sequence obtained for C. falciformis in this study is 16, 629.',\n",
       "   'offset': 21994,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Accuracy of Carcharhinus falciformis mitochondrial sequences based on comparison with Genbank reference sequences (Table\\xa0S4).',\n",
       "   'offset': 22175,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab2',\n",
       "    'file': 'Tab2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genetic Loci\\tAccuracy\\tDifferences/Total base pairs\\nCytochrome oxidase 1 (COI)\\t99.33–100%\\t0–2/600–654\\nNADH2\\t98.5%\\t16/1044\\n12s rRNA\\t98.5%\\t6–7/1357–1411\\n16s rRNA\\t98.5%\\t6–7/1357–1411\\nMitochondria\\t98.75%\\t207/16,629–17,639*\\n',\n",
       "   'offset': 22302,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'Tab2',\n",
       "    'File': 'Tab2.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparison of individual protein-coding regions of the sample sequence to Carcharhinus falciformis sequences in the Genbank database, demonstrated ≧99.3% accuracy of our specimen sequences (Table\\xa02) (Genbank reference sequences used for comparison are listed in Supplementary Table\\xa0S4). In addition, we improved the phylogenetic placement, improved mitochondrial sequence data and increased the amount of nuclear genomic data on this species by five-fold (1.1 million bp).',\n",
       "   'offset': 22537,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Discussion',\n",
       "   'offset': 23014,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chondrichthyes are data deficient globally, making them highly vulnerable to fishing related exploitation. Molecular ecological assessments in areas of high data deficiency are often inefficient at addressing chondrichthyan deficiencies due to their reliance on molecular genetic tools that have a restrictive data output and because pan-genomic methods are inaccessible due to complex infrastructure and instrumentation requirements. As apex predators, Chondrichthyes are important for top-down regulation of oceanic ecosystems. Life history traits such as slow generation times, small litters, and stringent habitat requirements, make it challenging for chondrichthyan species populations to recover from overfishing practices. Due to overexploitation from unregulated fishing, of the 1,178 chondrichthyan species, approximately 25% face extinction and ~46% are data deficient, meaning information is insufficient for species conservation and management. To reduce these data deficiencies, the current manuscript describes methods that increased the taxonomic resolution of Carcharhinus falciformis, a Near Threatened CITES Appendix II listed species, in addition to providing genomic data for population genetic and evolutionary studies. The current study has increased the number of sequenced Chondrichthyes by 20%. The genomic data obtained, improved the genomic information for Carcharhinus falciformis, reduced data deficiencies through increased resolution of phylogenetic relationships within the Carcharhinus genus and in future studies will provide genomic data to understand population structure, population size and biogeography, which are important in management of priority species of concern such as the silky shark. Last, we described two different workflows for sequence analyses which will allow capacity building for taxonomic identification and population and phylogenetic studies among researchers with basic and advanced bioinformatic skills.',\n",
       "   'offset': 23025,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparison of Sequencing methods',\n",
       "   'offset': 24991,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The MinION portable sequencer obtained multiple barcode sequences, along with complete mitochondrial (Fig.\\xa03A, Fig.\\xa04) and partial nuclear genomes (Fig.\\xa03B) of a previously un-sequenced shark using genomic DNA collected from the market place. This method enabled accurate taxonomic identification of a shark tissue specimen by using multiple genetic barcodes, without the need for PCR amplification and without a priori sequence information.',\n",
       "   'offset': 25024,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We demonstrate the use of MinION sequencers to sequence larger genome sizes of 1–6.7\\u2009Gb, more typical of shark species. Nuclear genome size in sharks varies between 1–3 Gigabases and our method achieved optimal coverage (Fig.\\xa03A,B) of high copy number regions, such as mitochondrial and nuclear marker sequences often used for taxonomic identification and phylogenetic inference. Our methods expand on studies by Parker et al., who used the MinION sequencer to sequence relatively small Arabidopsis (~135 Mbp) genomes. Genome skimming using the MinION is amplification independent making it less affected by sequence biases. GC content is an important determinant of sequencing and computational bias in NGS. The mean GC content for mitochondrial genomes of Carcharhinid sharks is 38.44% (±0.006 STDEV), similar to the mitochondria of C. falciformis at a GC content of 38.4%. The range of GC content for all 673 chondrichthyan NADH2 barcodes sequenced to date is 32–47%. The range of GC content that the MinION sequenced in the C. falciformis genome was 29–64%, suggesting a wide capability of the sequencer for genomic assessment of a variety of Chondrichthyes. The method thus has potential to aide in taxonomic identification, ecological, evolutionary and population level studies of new and understudied species. Last, although the method relies on state of the art NGS technology, it only requires gDNA extraction from the specimen which takes about three and a half hours, a 15–30\\u2009minutes library preparation/loading time and finally sequencing with the MinION and a laptop. Basecalling can be done locally on the laptop computer in parallel or post-sequencing using ONT software. Analyses for the current study did not require internet connectivity as local databases were curated in advance onto the Geneious software on laptop computers. In the future, this method is adaptable to field-based ecological studies and will facilitate capacity building among conservation managers, wildlife researchers and forensics officials.',\n",
       "   'offset': 25469,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The higher cost of MinION flowcells compared to PCR amplification and sequencing of single barcodes remains a limiting factor to the wide adoption of this technology. Barcode sequencing using PCR requires gDNA extraction (~3.5 hrs), followed by PCR optimization for the species under consideration (hours to days), PCR amplification of a single barcode and possibly multiple species (~2 hrs) and last, PCR clean up (~30\\u2009minutes) and Sanger sequencing (~24 hrs) turnaround time. For the suggested timeline, the location should have access to a Sanger sequencer, which in case of the sampling site for the current study was a 7\\u2009hour drive away (403\\u2009km), suggesting that a 24\\u2009hr turnaround time is most unlikely. At the end of the PCR process a single barcode is sequenced for multiple species. In comparison, genome skimming on a MinION sequencer requires gDNA extraction (~3.5 hrs), followed by library preparation and flowcell loading (~15–30\\u2009minutes), followed by sequencing (~8–36\\u2009hours), and\\xa0recovery of fastq sequences and BLAST alignment to a local or global database for taxonomic identification. Therefore, PCR-based bar coded derives ~ 150–500\\u2009bp compared with 1,152,211\\u2009bp of data from the MinION. The MinION can be monitored in real time and the sequencing stopped once sufficient data has been received, such as mitochondrial sequences for taxonomic identification, further reducing sequencing time. PCR based DNA barcoding is economical for taxonomic classification purposes of multiple species, however, genome skimming using the MinION sequencer provides genomic information for a single species which will facilitate taxonomic classification, determination of population structure and size as well as accurate phylogenetic assessment. In addition, MinION sequencing allows researchers to maintain custody of samples during the entire process, whereas Sanger sequencing may require the samples to be de-identified before being sent to a sequencing facility for high-stakes studies.',\n",
       "   'offset': 27519,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The ability to perform sequencing experiments, assessing multiple genomic regions and determining taxonomic identities, irrespective of species or sample type, in remote locations provides a significant advantage for wildlife research and could make MinION a preferred choice for sequencing dependent upon the application of sequence data. Future studies in our laboratory will determine whether first, additional chondrichthyan and wildlife taxa can be analyzed in a fashion similar to C. falciformis using methods described here on the MinION sequencer and second, whether multiple specimens can be analyzed using our methods on a single MinION flowcell by barcoding each specimen, to reduce cost and increase efficiency. However, when sequencing multiple specimens simultaneously, reduced sequencing coverage of large chondrichthyan genomes could limit analyses of multiple samples with the MinION.',\n",
       "   'offset': 29538,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequencing large genomic sections with long read capabilities of the MinION will increase the size and diversity of sequence databases and improve annotation of sparsely annotated loci through gene mapping, and annotation. This will improve application of current and future genomic data for taxonomic, phylogenetic, metabolic or population studies. For example, currently the COI and NADH2 barcodes are well curated within the Genbank and BOLD databases for chondrichthyan species, but availability is sparse for other mitochondrial loci such as the 12s and 16s rRNA genes and many nuclear loci including ITS2. Therefore, taxonomic identification, phylogenetic and population studies in Chondrichthyes often rely on COI and NADH2 assessments, limiting the accuracy and depth of these analyses. Increased ease of Genome skimming or whole genome sequencing, using our methods will make larger regions of chondrichthyan genomes accessible and will help populate genetic databases. This approach will increase accuracy of phylogenetic, population and kinship assessments and will significantly reduce data deficiencies in Chondrichthyes for conservation and management purposes.',\n",
       "   'offset': 30440,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Analyses',\n",
       "   'offset': 31616,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We compared efficiency and expediency of two different workflows (Fig.\\xa02), one with a GUI (Workflow I) and the other command line based (Workflow II), for sequence analyses of MinION data. Mitochondrial sequences obtained from both workflows were 98% identical, differing by 400\\u2009bp in length. Phylogenetic analyses with both sequences resulted in an identical taxonomic classification of the unknown sample. Minor differences in mitochondrial sequences could be attributed to differences in the mapping and assembly algorithms underlying the assembly tools used in the two workflows. Workflow II also used more rigorous quality controls and trimmed sequences before assembly which may have led to the loss of some sequence base pairs.',\n",
       "   'offset': 31625,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The MinION Workflow I we developed allowed quick access to mitochondrial genomes, while Workflow II allowed access to nuclear genetic loci, phylogenetic analyses of which corroborated results from the mitochondrial analyses and allowed further investigation into phylogeny of the genus. Since mitochondrial genomes from Workflow I and II were comparable for taxonomic identification, we recommend using Workflow 1 (Fig.\\xa02) for field studies with an urgent timeline. Here a database of chondrichthyan sequences from BOLD and Genbank is first curated in Geneious for offline access. DNA sequences from unknown specimens are obtained from the MinION and directly mapped to reference mitochondrial sequences using Geneious software. The mapped mitochondrial sequences are then aligned with reference sequences in the offline databases for taxonomic identification making it as streamlined as upload and align sequence to identify species. The use of multiple mitochondrial barcodes ensures the highest level of accuracy. These analyses are straightforward and do not require internet connectivity, or extensive bioinformatics capabilities or computing power. Workflow I can be used to train researchers in accurate taxonomic identification of species without using computer programing.',\n",
       "   'offset': 32363,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Workflow II can be used in advanced stages of analyses not limited by computing power, bioinformatics expertise and time. The nuclear and mitochondrial sequences obtained from Workflow II can be used to corroborate results obtained using just mitochondrial makers in Workflow I and to obtain finer phylogenetic resolution among species. Population genetic studies may also be undertaken at this stage.',\n",
       "   'offset': 33646,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Phylogenetics',\n",
       "   'offset': 34048,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Our phylogenetic analyses using mitochondrial genomes and nuclear loci of the genus Carcharhinus consistently support placement of the unknown sample as sister to C. falciformis (Fig.\\xa05A,B) and demonstrate the utility of genome skimming techniques for taxonomic identification of unknown samples using multiple genetic loci. Additionally, analyses of mitochondrial and nuclear data are congruent with previous work and strongly support sister species relationships of sampled taxa throughout the genus (Fig.\\xa06). Albeit our analyses allow assessment of sister species relationship with denser sampling of taxa throughout the Carcharhinus genus.',\n",
       "   'offset': 34062,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The inclusion of multiple mitochondrial genes and nuclear loci provided higher statistical support and further evidence that P. glauca is nested well within Carcharhinus, advancing previous work in the genus that used fewer markers and subsequently had weaker statistical support. Our nuclear and mitochondrial data support a sister relationship between P. glauca and C. amblyrhynchos with C. falciformis as outgroup. These data exemplify the statistical power achieved using diverse genetic and intergenic markers which were accessible in the current study from using a genome-wide sequencing approach.',\n",
       "   'offset': 34708,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Deeper nodes relating four taxa-C. falciformis, P. glauca, C. amblyrhynchos and C. wheeleri are not well supported in previous studies or in our analyses (<0.95 PP), because of mito-nuclear discordance, missing data, and lack of coverage of mitochondrial loci. As Bayesian phylogenetic estimation techniques are often robust to missing data in a concatenated alignment, we think that a substantial amount of missing data, as seen with Chondrichthyes, has led to poor support for these phylogenetic analyses. This once again, highlights the need for further genomic analysis. In addition, Carcharhinus spp. are known to contain cryptic species and thus increasing the genomic databases for these species to include regions such as Ultra Conservative Elements (UCE) will be beneficial to understanding the genetic diversity of the genus. In our concatenated analysis, the unknown sample was consistently represented on a very long branch (Fig.\\xa06). This stems from difficulties in aligning the long segment of the unknown sample ITS2 sequence obtained from the MinION to the ITS2 sequences available on GenBank which were much shorter and thus had limited species’ information. In the final concatenated alignment, the unknown sample ITS2 sequence showed numerous insertions not present in other samples. These insertions in the MinION derived sequences are potential evidence of sequence information that is missing in the shorter PCR derived Genbank sequences which may be restricted by polymerase processing rates, amplicon length restrictions or PCR artifacts resulting from repeat regions of template sequences such as the ITS2 region. MinION derived sequences are contiguous long reads of actual genomic DNA which does not go through an amplification cycle but is simply denatured and sequenced, and hence the unknown sample ITS2 sequences is not affected by amplification biases or restrictions of amplicon size, thus providing a higher quality analysis.',\n",
       "   'offset': 35312,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus falciformis',\n",
       "   'offset': 37274,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus falciformis or silky shark is a pelagic species found in tropical waters in the Indian, Pacific, and Atlantic oceans. This species is highly migratory, and has low fecundity, placing it at risk from incidental take in high seas fisheries. Threats to the species are exacerbated by demand for meat and fins in the international market. Despite some regional prohibitions, silky shark mortality is under-reported and largely unmanaged. Inclusion of the silky shark in CITES Appendix II is likely to bolster compliance with existing protections, complement existing commitments under the Convention on Migratory Species (CMS) and facilitate international cooperation toward more comprehensive international conservation measures, thereby enhancing the chances for sustainable use. However, CITES listing and other measures are ineffective if the trade in this species cannot be detected in international fin exports. Methods and results presented here increase our ability to study silky sharks and increase traceability of this species in the fin trade.',\n",
       "   'offset': 37299,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A single complete mitochondrial genome sequence of 17,639\\u2009bp for Carcharhinus falciformis has been published. A suspicious insertion of ~939\\u2009bp is evident when the sequence is compared to all Carcharinus spp. mitogenomes, which are at ~16,700\\u2009bp. The mitochondrial sequence for Carcharhinus falciformis presented in the current report (Fig.\\xa04, Genbank accession #MK092088) provides the first complete and accurate mitochondrial sequence for the species. Further, the investigation into the nuclear and mitochondrial sequences presented in the current study provided better phylogenetic resolution for the Carcharhinus genus. The immunoglobulin and endocrine function encoding nuclear sequences will provide an understanding of evolutionary relationships, immune function development and physiology of the species in future studies.',\n",
       "   'offset': 38364,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Conclusions',\n",
       "   'offset': 39203,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We have bridged substantial knowledge gaps in chondrichthyan genetics through work presented in this report. Use of the MinION in the field is a potential game-changer because it can obtain vast amounts of data in areas with high data deficiency and high fishing effort with respect to Chondrichthyes. It can facilitate accurate identification and ecological investigation of existing as well as new and rare species, without the need for a priori’ sequence information. These methods will improve the accuracy of biodiversity estimates in areas with minimal infrastructure, thus addressing the data deficiency and knowledge gaps affecting Chondrichthyes where data is most needed. Sequencing of larger regions of chondrichthyan genomes as opposed to single barcode sequencing will allow wider ecological investigations for all Chondrichthyes such as population genetic studies, stock assessments, biogeographic and kinship studies, thus increasing the depth of our understanding of chondrichthyan populations. The whole genome provides insights into shark metabolism and health status. Data regarding population structure within geographically isolated populations of the same species can allow assessment of population stocks and inform conservation and management of the species. Last, the genome sequencing and analyses methods presented in the current report are independent of PCR bias, and have sequencing capacity across a wide GC %, making them amenable to ecological investigations on a variety of non-model species. The workflows presented here, can be used effectively to build capacity from molecular ecology research among wildlife researchers and fisheries scientists in a variety of infrastructure settings.',\n",
       "   'offset': 39215,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'CONCL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Methods',\n",
       "   'offset': 40941,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sampling and DNA extraction',\n",
       "   'offset': 40949,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark samples (n\\u2009=\\u200930) consisting of fin tissue ~5\\u2009mm3 in size were collected from dismembered shark specimen at a fish market in Veraval, Gujarat, India in April 2017 (Fig.\\xa01). Samples were stored in DNA/RNA GuardTM Zymoresearch at ambient temperature during collection and then stored at −20\\u2009°C until processing. DNA extraction was conducted using Quick-DNATM Plus extraction kit (ZymoresearchTM) with an extended 3-hour proteinase K digestion at 55\\u2009°C. Of these 30 samples, one was prioritized for analyses with the MinION sequencer and the rest were stored for analyses at a later time.',\n",
       "   'offset': 40977,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Morphological identification',\n",
       "   'offset': 41583,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Photo identification of the specimen was performed using protocols described by Ebert et al. and based on intact features of the dismembered specimen, including head, gills, and fins.',\n",
       "   'offset': 41612,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'PCR Amplification and Analyses',\n",
       "   'offset': 41796,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'PCR amplification of three gene fragments from the Cytochrome Oxidase 1 and NADH2 genetic loci was performed to resolve the taxonomic identity of the specimen at the species level using protocols described previously. PCR amplification of a 633\\u2009bp and 160\\u2009bp fragment of the COI regions was performed using methods described by Ward et al. and Fields et al., and that of a 1044\\u2009bp NADH2 fragment was done using protocols described by Naylor et al.. PCR amplification was followed by PCR clean up with ZymoresearchTM DNA Clean and Concentrator Kit and uni-directional sequencing using BigDye Terminator chemistry on an ABI 3730xl genetic analyzer (Applied Biosystems, Life Technologies). The resulting sequence was used to determine the taxonomic identity of the chondrichthyan specimen using the Barcode of Life Database (BOLD).',\n",
       "   'offset': 41827,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Library preparation and Sequencing on the MinION',\n",
       "   'offset': 42662,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to access the entire length of multiple nuclear and mitochondrial genomes in a PCR-independent fashion, genome skimming using the MinION device was performed. The extracted DNA was measured using a QubitTM spectrophotometer. Genomic DNA (~400\\u2009ng) was used directly for library preparation with the Rapid Sequencing kit (SQK-RAD003) from ONT using the manufacturer’s protocols. The library preparation and sample loading took 15–30\\u2009minutes in total and required access to a thermocycler or hot water bath incubator, pipettes and appropriate tips (1000\\u2009μl, 200\\u2009μl, and 10\\u2009μl), and molecular grade water. Bound DNA was loaded directly into the sample loading port of the FLO-MIN106 (R9.4 SpotON) flowcells after priming of the flowcells. Sequencing was continued for 36\\u2009hours, however, sequences acquired were monitored throughout the 36\\xa0\\u2009hours and mitochondrial barcodes could be obtained starting at 8\\u2009hours from start of sequencing.',\n",
       "   'offset': 42711,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bioinformatics Analyses',\n",
       "   'offset': 43677,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequence data were analyzed using two parallel workflows (Fig.\\xa02). For the first workflow, FASTQ files were uploaded into Geneious® software version 11.1.2. Mitochondrial sequences were mapped directly to reference chondrichthyan mitochondrial sequences from Genbank® using Geneious Read mapper and the resulting contig was used to obtain a contiguous mitochondrial sequence for the sample analyzed.',\n",
       "   'offset': 43701,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'For the second workflow, reads were assembled using CANU (Version 1.7) with flags that set the genome size as 16\\u2009kb, trimmed the minimum read length to 500\\u2009bp, and set the reads as coming from the MinION. Contigs were aligned using BLAST against the nt nucleotide database to identify mitochondrial and nuclear sequences. Using CANU reads from a circular genome assemble into a linear contig larger than the original genome with repeated sequences at both ends. Therefore, the mitochondrial genome was circularized using Samtools (version 1.8) and Nucmer (version 4.0.0 beta2) was used to identify repeated sequences in the mitochondrial contigs. Gepard (version 1.4) was used to produce dotplots and verify the circularization. This workflow is available in our associated Jupyter notebook: https://github.com/Adrian-Cantu/git-presentations/blob/master/shark_mito/Mito_mapping.ipynb.',\n",
       "   'offset': 44104,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mitochondrial contigs from both workflows were annotated using Mitos, a web-based server for de novo annotation of mitochondrial genomes, and Mito-annotator (Mitofish database), a tool for annotation of fish mitochondrial genomes. tRNA identification was conducted by ARWEN — a program to detect tRNAs in metazoan mitochondrial genomes. CGView was used to determine GC rich regions and GC skew.',\n",
       "   'offset': 44993,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Taxonomic identification and Phylogenetics',\n",
       "   'offset': 45390,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We investigated the taxonomic identity of our sample using whole mitochondrial genomes and nuclear DNA (ITS2) comparisons to previously sequenced Carcharhinus from the NCBI database. We reconstructed phylogenetic relationships within the genus Carcharhinus in parallel, first using complete mitochondrial genomes when available (n\\u2009=\\u200914), and second, using concatenated sequences for three protein-coding mtDNA genes (COX1, NADH2, and ND4), and one nDNA locus (ITS2), (n\\u2009=\\u200933 species within Carcharhinus). Galeocerdo cuvier and Prionace glauca were used as the outgroup in the two phylogenetic analyses. Sequences were acquired from GenBank (Supplementary Tables\\xa0S1, S2 and S3) and aligned using MUSCLE 3.8.31. The mitochondrial alignments were done in two different ways, using the 13 protein-coding regions of the mitochondrial genome or the complete mitogenome including coding and non-coding regions. Alignments were partitioned by gene and codon position. We used PartitionFinder 2.1.1 on XSEDE tool on CIPRES to identify the appropriate partitioning schemes and models of substitution using the greedy search algorithm. Maximum-likelihood phylogenies were estimated in RAxML-HPC v8.2.10 on XSEDE run on CIPRES Scientific Gateway using a separate GTR\\u2009+\\u2009Γ model specified for all partitions. We ran 10 searches beginning with different random stepwise addition parsimony trees with the default RAxML algorithm. Support values for the best tree were estimated using non-parametric bootstrapping implemented using the auto-MRE stopping criterion. We estimated Bayesian phylogenies in MrBayes v3.2.6, on XSEDE run on CIPRES Scientific Gateway using the partitioning scheme specified by PartitionFinder2. We used unlinked GTR substitution model across partitions and accommodated rate heterogeneity among sites by using a Γ distribution. We assessed convergence in Tracer v1.6.',\n",
       "   'offset': 45433,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information',\n",
       "   'offset': 47325,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Publisher’s note: Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.',\n",
       "   'offset': 47351,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'footnote', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information',\n",
       "   'offset': 47490,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Supplementary information accompanies this paper at 10.1038/s41598-019-40940-9.',\n",
       "   'offset': 47516,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Author Contributions',\n",
       "   'offset': 47596,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'S.J. contributed to concept, sampling, sequencing, analyses and wrote the manuscript, J.S. contributed to sampling, DNA extractions and sequencing, A.C. and R.E. designed and performed Bioinformatic analyses, S.F. did the phylogenetic analyses, I.M. contributed to PCR and BOLD sequence alignments, A.V. helped with establishing social networks, understanding fisheries practices and sampling logistics and E.D. contributed to concept and editing of the manuscript.',\n",
       "   'offset': 47617,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'AUTH_CONT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Data Availability',\n",
       "   'offset': 48083,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'All data generated or analyzed during this study are included in this published article (and its Supplementary Information files).',\n",
       "   'offset': 48101,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Competing Interests',\n",
       "   'offset': 48232,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'COMP_INT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The authors declare no competing interests.',\n",
       "   'offset': 48252,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'COMP_INT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'References',\n",
       "   'offset': 48296,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Dent, F. & Clarke, S. State of the global market for shark products. 196 (2015).',\n",
       "   'offset': 48307,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Extinction risk and conservation of the world’s sharks and rays',\n",
       "   'offset': 48388,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Dulvy;given-names:NK',\n",
       "    'source': 'eLife',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '3',\n",
       "    'pub-id_pmid': '24448405',\n",
       "    'fpage': 'e00590',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.7554/eLife.00590'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Jabado, R. W. et al. Troubled waters: Threats and extinction risk of the sharks, rays and chimaeras of the Arabian Sea and adjacent waters. Fish Fish. 19, 1043–1062 (2018).',\n",
       "   'offset': 48454,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lack, M. & Sant, G. Trends in global shark catch and recent developments in management. TRAFFIC Int. 33 (2009).',\n",
       "   'offset': 48629,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A review of the application of molecular genetics for fisheries management and conservation of sharks and rays',\n",
       "   'offset': 48741,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '80',\n",
       "    'name_0': 'surname:Dudgeon;given-names:CL',\n",
       "    'source': 'J. Fish Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1843',\n",
       "    'pub-id_pmid': '22497408',\n",
       "    'fpage': '1789',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/j.1095-8649.2012.03265.x'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DNA barcoding Australia’s fish species',\n",
       "   'offset': 48852,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Innes;given-names:BH',\n",
       "    'name_3': 'surname:Last;given-names:PR',\n",
       "    'name_0': 'surname:Ward;given-names:RD',\n",
       "    'name_1': 'surname:Zemlak;given-names:TS',\n",
       "    'name_4': 'surname:Hebert;given-names:PD',\n",
       "    'volume': '360',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1857',\n",
       "    'source': 'Philos. Trans. R. Soc. B Biol. Sci.',\n",
       "    'fpage': '1847',\n",
       "    'year': '2005',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1098/rstb.2005.1716'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A multiplex PCR mini-barcode assay to identify processed shark products in the global trade',\n",
       "   'offset': 48893,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Cardeñosa;given-names:D',\n",
       "    'source': 'PLOS ONE',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '12',\n",
       "    'pub-id_pmid': '29020095',\n",
       "    'fpage': 'e0185368',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0185368'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A Novel Mini-DNA Barcoding Assay to Identify Processed Fins from Internationally Protected Shark Species',\n",
       "   'offset': 48985,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Eng;given-names:R',\n",
       "    'name_3': 'surname:Feldheim;given-names:K',\n",
       "    'name_0': 'surname:Fields;given-names:AT',\n",
       "    'name_1': 'surname:Abercrombie;given-names:DL',\n",
       "    'name_4': 'surname:Chapman;given-names:DD',\n",
       "    'source': 'PLOS ONE',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '10',\n",
       "    'pub-id_pmid': '25646789',\n",
       "    'fpage': 'e0114844',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0114844'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Naylor, G. J. P. et al. A DNA sequence-based approach to the identification of shark and ray species and its implications for global elasmobranch diversity and parasitology. (Bulletin of the American Museum of Natural History, no. 367). DNA identification of sharks and rays (2012).',\n",
       "   'offset': 49090,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hoelzel, A. Conservation Genetics2 69, 10.1023/A:1011590517389 (2001).',\n",
       "   'offset': 49373,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Identification and phylogenetic inferences on stocks of sharks affected by the fishing industry off the Northern coast of Brazil',\n",
       "   'offset': 49444,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Rodrigues-Filho;given-names:LF',\n",
       "    'source': 'Genet. Mol. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '413',\n",
       "    'pub-id_pmid': '21637699',\n",
       "    'fpage': '405',\n",
       "    'year': '2009',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1590/S1415-47572009005000039'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequence analysis of mitochondrial 12S rRNA gene can identify meat species',\n",
       "   'offset': 49573,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '66',\n",
       "    'name_0': 'surname:Girish;given-names:PS',\n",
       "    'source': 'Meat Sci.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '556',\n",
       "    'pub-id_pmid': '22060864',\n",
       "    'fpage': '551',\n",
       "    'year': '2004',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/S0309-1740(03)00158-X'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Phylogenetic relationships among the major lineages of modern elasmobranchs',\n",
       "   'offset': 49648,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Fedrigo;given-names:O',\n",
       "    'name_3': 'surname:Lopez;given-names:JA',\n",
       "    'name_0': 'surname:Naylor;given-names:GJ',\n",
       "    'name_1': 'surname:Ryburn;given-names:JA',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '3',\n",
       "    'source': 'Reprod. Biol. Phylogeny',\n",
       "    'fpage': '25',\n",
       "    'year': '2005',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A streamlined DNA tool for global identification of heavily exploited coastal shark species (genus Rhizoprionodon)',\n",
       "   'offset': 49724,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Nachtigall;given-names:PG',\n",
       "    'name_3': 'surname:Chapman;given-names:DD',\n",
       "    'name_0': 'surname:Pinhal;given-names:D',\n",
       "    'name_1': 'surname:Shivji;given-names:MS',\n",
       "    'name_4': 'surname:Martins;given-names:C',\n",
       "    'source': 'PloS One',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '7',\n",
       "    'pub-id_pmid': '22496864',\n",
       "    'fpage': 'e34797',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1371/journal.pone.0034797'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Accuracy of species identification by fisheries observers in a north Australian shark fishery',\n",
       "   'offset': 49839,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '127–128',\n",
       "    'name_0': 'surname:Tillett;given-names:BJ',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '115',\n",
       "    'source': 'Fish. Res.',\n",
       "    'fpage': '109',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.fishres.2012.04.007'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Checklist of Chondrichthyans in Indian waters',\n",
       "   'offset': 49933,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '56',\n",
       "    'name_0': 'surname:Akhilesh;given-names:KV',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '120',\n",
       "    'source': 'J. Mar. Biol. Assoc. India',\n",
       "    'fpage': '109',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.6024/jmbai.2014.56.1.01750s-17'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Environmental DNA reveals tropical shark diversity in contrasting levels of anthropogenic impact',\n",
       "   'offset': 49979,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Bakker;given-names:J',\n",
       "    'source': 'Sci. Rep.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '7',\n",
       "    'pub-id_pmid': '29203793',\n",
       "    'fpage': '16886',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/s41598-017-17150-2'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mitochondrial cytochrome b variation in sleeper sharks (Squaliformes: Somniosidae)',\n",
       "   'offset': 50076,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '153',\n",
       "    'name_0': 'surname:Murray;given-names:BW',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1022',\n",
       "    'source': 'Mar. Biol.',\n",
       "    'fpage': '1015',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1007/s00227-007-0871-1'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Benz, G. W. et al. First Record of a Sleeper Shark in the Western Gulf of Mexico and Comments on Taxonomic Uncertainty Within Somniosus (Somniosus). Bulletin of Marine Science80, 343–351 (2007).',\n",
       "   'offset': 50159,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DRI busts illegal exports of shark fin from Maharashtra and Gujarat. The Indian Express (2018).',\n",
       "   'offset': 50356,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'van. Genome-wide SNPs reveal low effective population size within confined management units of the highly vagile Galapagos shark (Carcharhinus galapagensis)',\n",
       "   'offset': 50452,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Simpfendorfer;given-names:CA',\n",
       "    'name_3': 'surname:Salinas-de-León;given-names:P',\n",
       "    'name_0': 'surname:Pazmiño;given-names:DA',\n",
       "    'name_1': 'surname:Maes;given-names:GE',\n",
       "    'name_4': 'surname:Herwerden;given-names:L',\n",
       "    'volume': '18',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1163',\n",
       "    'source': 'Conserv. Genet.',\n",
       "    'fpage': '1151',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1007/s10592-017-0967-1'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Pazmiño, D. et al. Strong trans-Pacific break and local conservation units in the Galapagos shark (Carcharhinus galapagensis) revealed by genome-wide cytonuclear markers. Heredity120 (2018).',\n",
       "   'offset': 50609,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Selection and sex-biased dispersal in a coastal shark: The influence of philopatry on adaptive variation',\n",
       "   'offset': 50801,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Portnoy;given-names:D',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '24',\n",
       "    'source': 'Mol. Ecol.',\n",
       "    'fpage': '5885',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/mec.13441'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DNA capture reveals transoceanic gene flow in endangered river sharks',\n",
       "   'offset': 50906,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '112',\n",
       "    'name_0': 'surname:Li;given-names:C',\n",
       "    'source': 'Proc. Natl. Acad. Sci.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '13307',\n",
       "    'pub-id_pmid': '26460025',\n",
       "    'fpage': '13302',\n",
       "    'year': '2015',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1073/pnas.1508735112'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Harnessing the Power of Genomics to Secure the Future of Seafood',\n",
       "   'offset': 50976,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Bernatchez;given-names:L',\n",
       "    'source': 'Trends Ecol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '680',\n",
       "    'pub-id_pmid': '28818341',\n",
       "    'fpage': '665',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.tree.2017.06.010'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Carcharhinus - Nucleotide - NCBI. Available at, https://www.ncbi.nlm.nih.gov/nuccore (Accessed: 8th September 2018).',\n",
       "   'offset': 51041,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The complete mitochondrial DNA of the silky shark (Carcharhinus falciformis)',\n",
       "   'offset': 51158,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Diaz-Jaimes;given-names:P',\n",
       "    'name_3': 'surname:Marcet-Houben;given-names:M',\n",
       "    'name_0': 'surname:Galván-Tirado;given-names:C',\n",
       "    'name_1': 'surname:Hinojosa-Alvarez;given-names:S',\n",
       "    'name_4': 'surname:García-De-León;given-names:FJ',\n",
       "    'volume': '27',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '158',\n",
       "    'source': 'Mitochondrial DNA Part DNA Mapp. Seq. Anal.',\n",
       "    'fpage': '157',\n",
       "    'year': '2016',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MITOS: improved de novo metazoan mitochondrial genome annotation',\n",
       "   'offset': 51235,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '69',\n",
       "    'name_0': 'surname:Bernt;given-names:M',\n",
       "    'source': 'Mol. Phylogenet. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '319',\n",
       "    'pub-id_pmid': '22982435',\n",
       "    'fpage': '313',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.ympev.2012.08.023'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MitoFish and MitoAnnotator: a mitochondrial genome database of fish with an accurate and automatic annotation pipeline',\n",
       "   'offset': 51300,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '30',\n",
       "    'name_0': 'surname:Iwasaki;given-names:W',\n",
       "    'source': 'Mol. Biol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2540',\n",
       "    'pub-id_pmid': '23955518',\n",
       "    'fpage': '2531',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/molbev/mst141'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'ARWEN: a program to detect tRNA genes in metazoan mitochondrial nucleotide sequences',\n",
       "   'offset': 51419,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '24',\n",
       "    'name_0': 'surname:Laslett;given-names:D',\n",
       "    'name_1': 'surname:Canbäck;given-names:B',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '175',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '172',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btm573'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Geneious Basic: an integrated and extendable desktop software platform for the organization and analysis of sequence data',\n",
       "   'offset': 51504,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '28',\n",
       "    'name_0': 'surname:Kearse;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1649',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '1647',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/bts199'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The effects of fishing on sharks, rays, and chimaeras (chondrichthyans), and the implications for marine ecosystems',\n",
       "   'offset': 51626,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '57',\n",
       "    'name_0': 'surname:Stevens;given-names:J',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '494',\n",
       "    'source': 'ICES J. Mar. Sci.',\n",
       "    'fpage': '476',\n",
       "    'year': '2000',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1006/jmsc.2000.0724'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The importance of habitat and life history to extinction risk in sharks, skates, rays and chimaeras',\n",
       "   'offset': 51742,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Myers;given-names:RA',\n",
       "    'volume': '275',\n",
       "    'name_0': 'surname:García;given-names:VB',\n",
       "    'name_1': 'surname:Lucifora;given-names:LO',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '89',\n",
       "    'source': 'Proc. R. Soc. Lond. B Biol. Sci.',\n",
       "    'fpage': '83',\n",
       "    'year': '2008',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1098/rspb.2007.1295'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Elephant shark genome provides unique insights into gnathostome evolution',\n",
       "   'offset': 51842,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '505',\n",
       "    'name_0': 'surname:Venkatesh;given-names:B',\n",
       "    'source': 'Nature',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '179',\n",
       "    'pub-id_pmid': '24402279',\n",
       "    'fpage': '174',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/nature12826'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Draft sequencing and assembly of the genome of the world’s largest fish, the whale shark: Rhincodon typus Smith 1828',\n",
       "   'offset': 51916,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Read;given-names:TD',\n",
       "    'source': 'BMC Genomics',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '18',\n",
       "    'pub-id_pmid': '28709399',\n",
       "    'fpage': '532',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1186/s12864-017-3926-9'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark genomes provide insights into elasmobranch evolution and the origin of vertebrates',\n",
       "   'offset': 52035,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Hara;given-names:Y',\n",
       "    'source': 'Nat. Ecol. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '2',\n",
       "    'pub-id_pmid': '30297745',\n",
       "    'fpage': '1761',\n",
       "    'year': '2018',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/s41559-018-0673-5'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Parker, J., Helmstetter, A. J., Devey, D., Wilkinson, T. & Papadopulos, A. S. T. Field-based species identification of closely-related plants using real-time nanopore sequencing. Sci. Rep. 7 (2017).',\n",
       "   'offset': 52124,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Characterizing and measuring bias in sequence data',\n",
       "   'offset': 52323,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Ross;given-names:MG',\n",
       "    'source': 'Genome Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '14',\n",
       "    'pub-id_pmid': '23718773',\n",
       "    'fpage': 'R51',\n",
       "    'year': '2013',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1186/gb-2013-14-5-r51'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Structure, function and diversity of the healthy human microbiome',\n",
       "   'offset': 52374,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '486',\n",
       "    'name_0': 'surname:Consortium;given-names:THMP',\n",
       "    'source': 'Nature',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '214',\n",
       "    'pub-id_pmid': '22699609',\n",
       "    'fpage': '207',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1038/nature11234'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Sequencing at sea: challenges and experiences in Ion Torrent PGM sequencing during the 2013 Southern Line Islands Research Expedition',\n",
       "   'offset': 52440,\n",
       "   'relations': [],\n",
       "   'infons': {'name_0': 'surname:Lim;given-names:YW',\n",
       "    'source': 'PeerJ',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '2',\n",
       "    'pub-id_pmid': '25177534',\n",
       "    'fpage': 'e520',\n",
       "    'year': '2014',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.7717/peerj.520'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': ' NCBI Resource Coordinators. Database Resources of the National Center for Biotechnology Information. Nucleic Acids Res. 45, D12–D17 (2017).',\n",
       "   'offset': 52574,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Bold: The Barcode of Life Data System, (http://www.barcodinglife.org). Available at, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1890991/. (Accessed: 28th April 2018).',\n",
       "   'offset': 52717,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Shark tales: a molecular species-level phylogeny of sharks (Selachimorpha, Chondrichthyes)',\n",
       "   'offset': 52886,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '58',\n",
       "    'name_0': 'surname:Vélez-Zuazo;given-names:X',\n",
       "    'name_1': 'surname:Agnarsson;given-names:I',\n",
       "    'source': 'Mol. Phylogenet. Evol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '217',\n",
       "    'pub-id_pmid': '21129490',\n",
       "    'fpage': '207',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1016/j.ympev.2010.11.018'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Missing data in phylogenetic analysis: reconciling results from simulations and empirical data',\n",
       "   'offset': 52977,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '60',\n",
       "    'name_0': 'surname:Wiens;given-names:JJ',\n",
       "    'name_1': 'surname:Morrill;given-names:MC',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '731',\n",
       "    'pub-id_pmid': '21447483',\n",
       "    'fpage': '719',\n",
       "    'year': '2011',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/syr025'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A novel field method to distinguish between cryptic carcharhinid sharks, Australian blacktip shark Carcharhinus tilstoni and common blacktip shark C. limbatus, despite the presence of hybrids',\n",
       "   'offset': 53072,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '90',\n",
       "    'name_0': 'surname:Johnson;given-names:GJ',\n",
       "    'source': 'J. Fish Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '60',\n",
       "    'pub-id_pmid': '27774596',\n",
       "    'fpage': '39',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1111/jfb.13102'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Towards better management of Australia’s shark fishery: genetic analyses reveal unexpected ratios of cryptic blacktip species Carcharhinus tilstoni and C. limbatus',\n",
       "   'offset': 53264,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kashiwagi;given-names:T',\n",
       "    'name_3': 'surname:Broderick;given-names:D',\n",
       "    'name_0': 'surname:Ovenden;given-names:JR',\n",
       "    'name_1': 'surname:Morgan;given-names:JAT',\n",
       "    'name_4': 'surname:Salini;given-names:J',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '61',\n",
       "    'source': 'Mar. Freshw. Res.',\n",
       "    'fpage': '253',\n",
       "    'year': '2010',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1071/MF09151'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Ultraconserved elements anchor thousands of genetic markers spanning multiple evolutionary timescales',\n",
       "   'offset': 53430,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '61',\n",
       "    'name_0': 'surname:Faircloth;given-names:BC',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '726',\n",
       "    'pub-id_pmid': '22232343',\n",
       "    'fpage': '717',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/sys004'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IUCN 2017. The IUCN Red List of Threatened Species. Version 2017-2, http://www.iucnredlist.org. Downloaded on 25 November 2017. Available at, http://www.iucnredlist.org/ (Accessed: 26th November 2017).',\n",
       "   'offset': 53532,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'History of CITES listing of sharks (Elasmobranchii) | CITES. Available at, https://www.cites.org/eng/prog/shark/history.php (Accessed: 21st August 2018).',\n",
       "   'offset': 53734,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lack, M. & Sant, G. The Future of Sharks: A Review of Action and Inaction (2011).',\n",
       "   'offset': 53888,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Ebert, D., Fowler, S. & Compagno, L. Sharks of the World: A Fully Illustrated Guide (2013).',\n",
       "   'offset': 53970,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Kearse, M., Kearse, M., Sturrock, S. & Meintjes, P. The Geneious 6.0.3 Read Mapper. 16 (2012).',\n",
       "   'offset': 54062,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation',\n",
       "   'offset': 54157,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '27',\n",
       "    'name_0': 'surname:Koren;given-names:S',\n",
       "    'source': 'Genome Res.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '736',\n",
       "    'pub-id_pmid': '28298431',\n",
       "    'fpage': '722',\n",
       "    'year': '2017',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1101/gr.215087.116'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Madden, T. The BLAST Sequence Analysis Tool. (National Center for Biotechnology Information (US), 2003).',\n",
       "   'offset': 54255,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The Sequence Alignment/Map format and SAMtools',\n",
       "   'offset': 54360,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '25',\n",
       "    'name_0': 'surname:Li;given-names:H',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2079',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '2078',\n",
       "    'year': '2009',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btp352'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MUMmer4: A fast and versatile genome alignment system. Available at, http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005944. (Accessed: 21st August 2018).',\n",
       "   'offset': 54407,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Gepard: a rapid and sensitive tool for creating dotplots on genome scale',\n",
       "   'offset': 54585,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Rattei;given-names:T',\n",
       "    'volume': '23',\n",
       "    'name_0': 'surname:Krumsiek;given-names:J',\n",
       "    'name_1': 'surname:Arnold;given-names:R',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1028',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '1026',\n",
       "    'year': '2007',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btm039'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Circular genome visualization and exploration using CGView',\n",
       "   'offset': 54658,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '21',\n",
       "    'name_0': 'surname:Stothard;given-names:P',\n",
       "    'name_1': 'surname:Wishart;given-names:DS',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '539',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '537',\n",
       "    'year': '2005',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/bti054'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MUSCLE: multiple sequence alignment with high accuracy and high throughput',\n",
       "   'offset': 54717,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '32',\n",
       "    'name_0': 'surname:Edgar;given-names:RC',\n",
       "    'source': 'Nucleic Acids Res.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '1797',\n",
       "    'pub-id_pmid': '15034147',\n",
       "    'fpage': '1792',\n",
       "    'year': '2004',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/nar/gkh340'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Lanfear, R., et al. PartitionFinder 2: New methods for selecting partitioned models of evolution for molecular and morphological phylogenetic analyses. Mol. Biol. Evol. 34 (2016).',\n",
       "   'offset': 54792,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Creating the CIPRES Science Gateway for Inference of Large Phylogenetic Trees',\n",
       "   'offset': 54972,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Schwartz;given-names:T',\n",
       "    'volume': '14',\n",
       "    'name_0': 'surname:Miller;given-names:M',\n",
       "    'name_1': 'surname:Pfeiffer;given-names:WT',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '8',\n",
       "    'source': 'In Proceedings of the Gateway Computing Environments Workshop: 14 Nov 2010',\n",
       "    'fpage': '1',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'RAxML-VI-HPC: maximum likelihood-based phylogenetic analyses with thousands of taxa and mixed models',\n",
       "   'offset': 55050,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '22',\n",
       "    'name_0': 'surname:Stamatakis;given-names:A',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '2690',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '2688',\n",
       "    'year': '2006',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/btl446'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MRBAYES: Bayesian inference of phylogenetic trees',\n",
       "   'offset': 55151,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '17',\n",
       "    'name_0': 'surname:Huelsenbeck;given-names:JP',\n",
       "    'name_1': 'surname:Ronquist;given-names:F',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '755',\n",
       "    'source': 'Bioinforma. Oxf. Engl.',\n",
       "    'fpage': '754',\n",
       "    'year': '2001',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/bioinformatics/17.8.754'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MrBayes 3.2: Efficient Bayesian Phylogenetic Inference and Model Choice Across a Large Model Space',\n",
       "   'offset': 55201,\n",
       "   'relations': [],\n",
       "   'infons': {'volume': '61',\n",
       "    'name_0': 'surname:Ronquist;given-names:F',\n",
       "    'source': 'Syst. Biol.',\n",
       "    'section_type': 'REF',\n",
       "    'lpage': '542',\n",
       "    'pub-id_pmid': '22357727',\n",
       "    'fpage': '539',\n",
       "    'year': '2012',\n",
       "    'type': 'ref',\n",
       "    'pub-id_doi': '10.1093/sysbio/sys029'},\n",
       "   'sentences': [],\n",
       "   'annotations': []}],\n",
       " 'infons': {'license': 'CC BY'},\n",
       " 'id': '6418218',\n",
       " 'relations': []}"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmcid_to_doc['6418218']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'Images were processed and analyzed with ImageJ Fiji v1.50c. Demographs and line plots (Fig.\\xa01 and Fig.\\xa0S6) were constructed by manually measuring the fluorescence intensity profiles in Fiji and processing the data in R (version 3.3.1 [http://www.r-project.org]), with the cell profiles script ( [http://github.com/ta-cameron/cell-profiles]) and ggplot2 package (version 2.1.0; Hadley Wickham, Department of Statistics, Rice University [https://ggplot2.tidyverse.org/]). Intensity profiles either were oriented with the fluorescent cell pole toward the left (E. coli and R. sphaeroides), or random sorting was applied (M. gryphiswaldense and R. rubrum). In the case of C. crescentus, the stalk was used as the old pole reference. To confidently and accurately identify stalks, FM4-64 staining (which in our hands did not always stain all cells) and DAPI staining (which we found to unspecifically stain stalks) were used, in addition to DIC microscopy.',\n",
       "  'offset': 64911,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []}]"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t for t in pmcid_to_doc['6414705']['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The fMRI imaging analysis steps are outlined in Fig. 1 The first 5 time frames (15\\u200as) were removed to allow the MR signal to achieve T1 equilibrium. Time frames were slice-timing corrected, realigned to the mean echo-planar image using SPM12 (http://www.fil.ion.ucl.ac.uk/spm/), co-registered to the subject T1 space and then normalized to the standard MNI-152 2mm-template using Advanced Normalization Tools software (http://stnava.github.io/ANTs/). Signals from subject white matter and cerebrospinal fluid (3-mm cubes centered at Montreal Neurological Institute coordinates (MNI) (26, –12, 35) and (19, –33, 18) as these locations are away from the grey matter), as well as six head motion parameters were regressed out from each dataset. Global signal regression was not performed as it is controversial in resting state fMRI data preprocessing. Correction for global signal fluctuations with regression has a significant impact on resting state functional connectivity results and can further introduce “artifactual” anti-correlations. Furthermore, global signal includes both neural and non-neural variabilities and therefore makes the result difficult to interpret. FMRI data were spatially smoothed using an 8\\u200amm 3D-Gaussian filter. All voxel time courses were band pass filtered (0.008\\u200aHz<f<0.1\\u200aHz) and variance normalized. The mean signal noise ratio, an indicator of fMRI data quality, was not different between groups (F(3,120)\\u200a=\\u200a2.04, p\\u200a=\\u200a0.11).']"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t['text'] for t in pmcid_to_doc['6294587']['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What are papers with multiple hits?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2837031\n",
      "2867826\n",
      "2906491\n",
      "3288584\n",
      "3398141\n",
      "3415573\n",
      "3465669\n",
      "3508300\n",
      "3733738\n",
      "3733808\n",
      "3734164\n",
      "3738158\n",
      "3741535\n",
      "3750055\n",
      "3767577\n",
      "3767578\n",
      "3769542\n",
      "3771214\n",
      "3771231\n",
      "3777708\n",
      "3778014\n",
      "3778324\n",
      "3787254\n",
      "3789805\n",
      "3792112\n",
      "3794442\n",
      "3797637\n",
      "3810855\n",
      "3817376\n",
      "3825257\n",
      "3825639\n",
      "3827500\n",
      "3831347\n",
      "3834793\n",
      "3836225\n",
      "3840589\n",
      "3840732\n",
      "3843372\n",
      "3846741\n",
      "3847652\n",
      "3848435\n",
      "3851838\n",
      "3852983\n",
      "3856601\n",
      "3857637\n",
      "3866249\n",
      "3866560\n",
      "3867689\n",
      "3868217\n",
      "3870320\n",
      "3871669\n",
      "3871959\n",
      "3872296\n",
      "3872335\n",
      "3874632\n",
      "3875239\n",
      "3877988\n",
      "3879085\n",
      "3880260\n",
      "3881492\n",
      "3890649\n",
      "3892064\n",
      "3892334\n",
      "3892923\n",
      "3894454\n",
      "3897414\n",
      "3897846\n",
      "3898676\n",
      "3898926\n",
      "3901538\n",
      "3903476\n",
      "3905898\n",
      "3906813\n",
      "3907392\n",
      "3913709\n",
      "3915074\n",
      "3915617\n",
      "3916389\n",
      "3918504\n",
      "3921152\n",
      "3922334\n",
      "3922690\n",
      "3924332\n",
      "3924952\n",
      "3927368\n",
      "3930095\n",
      "3930868\n",
      "3931231\n",
      "3938395\n",
      "3940440\n",
      "3941972\n",
      "3942789\n",
      "3944147\n",
      "3945519\n",
      "3945768\n",
      "3946168\n",
      "3949292\n",
      "3949323\n",
      "3953487\n",
      "3953750\n",
      "3954160\n",
      "3957070\n",
      "3957075\n",
      "3960092\n",
      "3961385\n",
      "3962008\n",
      "3964938\n",
      "3965009\n",
      "3965049\n",
      "3965399\n",
      "3967918\n",
      "3967922\n",
      "3970007\n",
      "3971626\n",
      "3973741\n",
      "3974646\n",
      "3983936\n",
      "3985171\n",
      "3995771\n",
      "3998141\n",
      "4000208\n",
      "4001068\n",
      "4004357\n",
      "4004789\n",
      "4005938\n",
      "4005939\n",
      "4009425\n",
      "4009432\n",
      "4010726\n",
      "4010757\n",
      "4012149\n",
      "4014419\n",
      "4014436\n",
      "4016541\n",
      "4016703\n",
      "4016711\n",
      "4018560\n",
      "4019564\n",
      "4019854\n",
      "4021553\n",
      "4021573\n",
      "4022496\n",
      "4023661\n",
      "4027178\n",
      "4029974\n",
      "4030211\n",
      "4030251\n",
      "4032127\n",
      "4033128\n",
      "4033259\n",
      "4034701\n",
      "4038925\n",
      "4039476\n",
      "4040986\n",
      "4041412\n",
      "4051631\n",
      "4053734\n",
      "4056372\n",
      "4057744\n",
      "4058940\n",
      "4061050\n",
      "4063228\n",
      "4064277\n",
      "4065201\n",
      "4066757\n",
      "4067307\n",
      "4067548\n",
      "4072515\n",
      "4080447\n",
      "4082612\n",
      "4085032\n",
      "4086947\n",
      "4086997\n",
      "4091312\n",
      "4091703\n",
      "4092319\n",
      "4094564\n",
      "4097353\n",
      "4099271\n",
      "4102429\n",
      "4103476\n",
      "4103489\n",
      "4103490\n",
      "4103495\n",
      "4104314\n",
      "4104320\n",
      "4105478\n",
      "4106841\n",
      "4107982\n",
      "4109307\n",
      "4110736\n",
      "4111116\n",
      "4111482\n",
      "4113289\n",
      "4115634\n",
      "4117414\n",
      "4118779\n",
      "4123383\n",
      "4125065\n",
      "4125409\n",
      "4130510\n",
      "4132706\n",
      "4132753\n",
      "4133763\n",
      "4134206\n",
      "4134471\n",
      "4136786\n",
      "4137082\n",
      "4138049\n",
      "4138190\n",
      "4138493\n",
      "4141107\n",
      "4141640\n",
      "4141981\n",
      "4142861\n",
      "4144660\n",
      "4146706\n",
      "4147884\n",
      "4147885\n",
      "4147925\n",
      "4148538\n",
      "4150124\n",
      "4151372\n",
      "4152123\n",
      "4152273\n",
      "4152597\n",
      "4152885\n",
      "4153048\n",
      "4153563\n",
      "4154443\n",
      "4154734\n",
      "4154752\n",
      "4155097\n",
      "4155926\n",
      "4156027\n",
      "4156431\n",
      "4159173\n",
      "4161295\n",
      "4164236\n",
      "4165746\n",
      "4165886\n",
      "4166924\n",
      "4166927\n",
      "4166999\n",
      "4168712\n",
      "4168749\n",
      "4168754\n",
      "4169225\n",
      "4172099\n",
      "4172526\n",
      "4173023\n",
      "4174579\n",
      "4174883\n",
      "4176157\n",
      "4176421\n",
      "4176422\n",
      "4176427\n",
      "4177049\n",
      "4177199\n",
      "4177597\n",
      "4177671\n",
      "4180802\n",
      "4180832\n",
      "4180961\n",
      "4184106\n",
      "4184317\n",
      "4191382\n",
      "4193140\n",
      "4195908\n",
      "4197741\n",
      "4197745\n",
      "4197993\n",
      "4199626\n",
      "4199702\n",
      "4201578\n",
      "4203760\n",
      "4203834\n",
      "4207244\n",
      "4207245\n",
      "4207426\n",
      "4208745\n",
      "4212056\n"
     ]
    }
   ],
   "source": [
    "for pmcid in np.unique(urls_in_papes.pmcid.values)[:1000]:\n",
    "    texts = [t['text'] for t in pmcid_to_doc[pmcid]['passages'] if 'github' in t['text']]\n",
    "    if len(texts) > 1:\n",
    "        print(pmcid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'BTK is governed by the CeCILL-B license under French law and abiding by the rules of distribution of free software. You can use, modify and/or redistribute the software under the terms of the CeCILL-B license as circulated by CEA, CNRS and INRIA at the following URL: www.cecill.info. The sources of BTK can be downloaded at this address: https://github.com/rousseau/fbrain',\n",
       "  'offset': 4964,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'BTK is an on-going project and we expect to improve the robustness of the algorithms through the feedback of the community. Moreover, BTK is hosted on github (github.com) which combines some features of social networking sites with distributed source-control Git (git-scm.com). It makes this system highly suited to collaborative development projects.',\n",
       "  'offset': 16502,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'see the BTK documentation available at this address: https://github.com/rousseau/fbrain/blob/master/Documentation/Latex/btk.pdf',\n",
       "  'offset': 18039,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'footnote', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []}]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t for t in pmcid_to_doc['3508300']['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## See if unique URLs are author repos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "url_counts = Counter(urls_in_papes.url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('broadinstitute.github.io/picard', 1106),\n",
       " ('github.com/najoshi/sickle', 275),\n",
       " ('github.com/jstjohn/SeqPrep', 163),\n",
       " ('github.com', 141),\n",
       " ('github.com/lh3/seqtk', 139),\n",
       " ('transdecoder.github.io', 131),\n",
       " ('cole-trapnell-lab.github.io/cufflinks', 116),\n",
       " ('github.com/alexdobin/STAR', 106),\n",
       " ('stnava.github.io/ANTs', 96),\n",
       " ('github.com/taoliu/MACS', 93),\n",
       " ('github.com/lh3/wgsim', 86),\n",
       " ('trinotate.github.io', 83),\n",
       " ('github.com/vsbuffalo/scythe', 65),\n",
       " ('github.com/broadinstitute/picard', 55),\n",
       " ('networkx.github.io', 54),\n",
       " ('github.com/korseby/container-mtbls520', 49),\n",
       " ('github.com/bulik/ldsc', 47),\n",
       " ('github.com/tseemann/snippy', 40),\n",
       " ('trinityrnaseq.github.io', 38),\n",
       " ('github.com/ekg/freebayes', 36),\n",
       " ('github.com/jdstorey/qvalue', 35),\n",
       " ('github.com/trinityrnaseq/trinityrnaseq/wiki', 34),\n",
       " ('github.com/tseemann/abricate', 34),\n",
       " ('picrust.github.io/picrust', 33),\n",
       " ('github.com/arq5x/bedtools2', 33),\n",
       " ('github.com/genome/bam-readcount', 32),\n",
       " ('gephi.github.io', 31),\n",
       " ('github.com/marcelm/cutadapt', 30),\n",
       " ('github.com/pysam-developers/pysam', 30),\n",
       " ('github.com/fchollet/keras', 29),\n",
       " ('deweylab.github.io/RSEM', 28),\n",
       " ('expressionanalysis.github.io/ea-utils', 27),\n",
       " ('github.com/tseemann/mlst', 27),\n",
       " ('github.com/lh3/bwa', 26),\n",
       " ('github.com/torognes/vsearch', 25),\n",
       " ('github.com/iontorrent/TMAP', 25),\n",
       " ('github.com/rrwick/Porechop', 25),\n",
       " ('tgennorth.github.io/NASP', 23),\n",
       " ('github.com/vcflib/vcflib', 23),\n",
       " ('github.com/widdowquinn/pyani', 23),\n",
       " ('github.com/FelixKrueger/TrimGalore', 22),\n",
       " ('github.com/PacificBiosciences/GenomicConsensus', 21),\n",
       " ('github.com/tanghaibao/goatools', 21),\n",
       " ('github.com/chapmanb/bcbio-nextgen', 21),\n",
       " ('cole-trapnell-lab.github.io/cufflinks/cuffdiff', 21),\n",
       " ('www.github.com', 21),\n",
       " ('samtools.github.io/hts-specs/SAMv1.pdf', 21),\n",
       " ('samtools.github.io/bcftools', 20),\n",
       " ('github.com/ucdavis-bioinformatics/sickle', 20),\n",
       " ('github.com/samtools/samtools', 20)]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "url_counts.most_common(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'github.com/lh3/wgsim': 86,\n",
       " 'github.com/JAvRZ/andi-dataprocessing': 1,\n",
       " 'github.com/AharoniLab/MatchWeiz': 1,\n",
       " 'broadinstitute.github.io/picard': 1106,\n",
       " 'github.com/ijmarshall/picotron': 1,\n",
       " 'github.com/awilfert/PSAP-pipeline': 1,\n",
       " 'gist.github.com/darmitage/3179407': 1,\n",
       " 'github.com/confunguido/PokemonDataAnalysis': 1,\n",
       " 'uhkniazi.github.io/dismiss': 1,\n",
       " 'github.com/alexandrovteam/spatial-corals': 1,\n",
       " 'uubram.github.io/RTCR/along': 1,\n",
       " 'github.com/bulik/ldsc': 47,\n",
       " 'github.com/tanghaibao/treecut/blob/master/scripts/eisen_to_newick.py': 1,\n",
       " 'github.com/aboyle/F-seq': 3,\n",
       " 'www.github.com/rhenley/Pythion': 1,\n",
       " 'github.com/Caleydo/caleydo_clue': 1,\n",
       " 'github.com/Caleydo/pathfinder': 1,\n",
       " 'github.com/GauravPandeyLab': 1,\n",
       " 'github.com/tdsmith/eleven': 1,\n",
       " 'github.com/nmcnulty/COPRO-Seq': 1,\n",
       " 'github.com/rhenley/Pyth-Ion': 4,\n",
       " 'github.com/rgcgithub/clamms': 2,\n",
       " 'github.com/ZWang-Lab/LSKAT': 1,\n",
       " 'github.com/markus-nilsson/md-dmri': 2,\n",
       " 'github.com/divara01/PSB2017_ReproducibilityOfBNs': 1,\n",
       " 'github.com/torognes/swarm': 2,\n",
       " 'github.com/oscar-franzen/oclust': 3,\n",
       " 'github.com/fangly/AmpliCopyrighter': 1,\n",
       " 'joey711.github.io/phyloseq': 8,\n",
       " 'github.com/jasonsahl/wgfast': 3,\n",
       " 'github.com/snayfach/PhyloCNV': 1,\n",
       " 'github.com/voutcn/megahit': 5,\n",
       " 'github.com/BinPro/CONCOCT': 8,\n",
       " 'ecogenomics.github.io/GroopM': 2,\n",
       " 'github.com/CK7/abawaca': 5,\n",
       " 'ecogenomics.github.io/CheckM': 4,\n",
       " 'github.com/snayfach/MicrobeCensus': 3,\n",
       " 'github.com/abhyudaynj/LSTM-CRF-models': 1,\n",
       " 'gist.github.com/svigneau/8846527': 1,\n",
       " 'github.com/ImprovementPathSystems/Measuring_Abnormality_in_High_Dimensional_Spaces': 1,\n",
       " 'github.com/charmasaur/M-LOOP': 1,\n",
       " 'github.com/BenJWard/Supertypes_RedQueen_TSE': 1,\n",
       " 'github.com/open-ephys/plugin-GUI': 1,\n",
       " 'github.com/NTCColumbia/moco': 2,\n",
       " 'github.com/cjfields/readfq': 1,\n",
       " 'baohongz.github.io/Lupus_eQTL': 1,\n",
       " 'github.com/vkozareva/geneCNV': 1,\n",
       " 'drpowell.github.io/vennt': 7,\n",
       " 'github.com/slipguru/palladio': 1,\n",
       " 'github.com/AnnaGuan/GISMMA/tree/AnnaGuan-patch-1': 1,\n",
       " 'github.com/torognes/vsearch/wiki/VSEARCH-pipeline': 2,\n",
       " 'github.com/reconhub/outbreaker2': 1,\n",
       " 'github.com/NabaviLab/CNV-Sim': 1,\n",
       " 'github.com/najoshi/sickle': 275,\n",
       " 'github.com/vsbuffalo/scythe': 65,\n",
       " 'github.com/topherconley/noleaven': 1,\n",
       " 'github.com/jtamames/squeezeM': 1,\n",
       " 'github.com/tseemann/barrnap': 15,\n",
       " 'github.com/bongsongkim/HH-CCDF': 1,\n",
       " 'github.com/rongjiewang/BdBG': 1,\n",
       " 'github.com/sdchandra/CNAclinic': 2,\n",
       " 'github.com/coulet/PredictDoseReduction': 1,\n",
       " 'github.com/cromulo/IWS': 1,\n",
       " 'github.com/opencobra/cobratoolbox': 8,\n",
       " 'github.com/danbang/article-adolescence-optimal': 1,\n",
       " 'github.com/edickie/icarus': 1,\n",
       " 'github.com/jenniesuz/tsetse_climate_change': 1,\n",
       " 'veg.github.io/hyphy-site': 2,\n",
       " 'github.com/linneas/condetri': 1,\n",
       " 'github.com/janelia-flyem/NeuTu': 5,\n",
       " 'github.com/ThomasDOtto/IPA': 3,\n",
       " 'jlizier.github.io/jidt': 1,\n",
       " 'github.com/kilicogluh/Bio-SCoRes': 3,\n",
       " 'github.com/brianyu2010/Mini-Metagenomic_Analyses': 1,\n",
       " 'github.com/elifesciences-publications/Mini-Metagenomic_Analyses': 1,\n",
       " 'github.com/hallamlab/metapathways2/wiki': 1,\n",
       " 'neuro.github.io': 1,\n",
       " 'github.com/jtladner/Scripts/tree/master/chimeric_reads': 1,\n",
       " 'github.com/lh3/seqtk': 139,\n",
       " 'github.com/BGI-shenzhen/PopLDdecay': 7,\n",
       " 'github.com/saschajung/ChromAccPrediction': 1,\n",
       " 'github.com/dizak/mothulity': 1,\n",
       " 'github.com/tkotani/ecalj': 1,\n",
       " 'github.com/rapidftr/RapidFTR/wiki': 1,\n",
       " 'github.com/marcopessoa/bioinfo-scripts': 1,\n",
       " 'github.com/LJI-Bioinformatics/HLATyphon': 4,\n",
       " 'github.com/GildasLepennetier/GildasSources': 2,\n",
       " 'github.com/UD-CBCB/RedRep': 1,\n",
       " 'github.com/jdstorey/qvalue': 35,\n",
       " 'github.com/xu1912/SMON.git': 1,\n",
       " 'github.com/airanmehr/clear': 1,\n",
       " 'yjx1217.github.io/Yeast_PacBio_2016/data': 4,\n",
       " 'github.com/biocompibens/SME': 1,\n",
       " 'github.com/KarenBell/rbcL-dual-index-metabarcoding': 2,\n",
       " 'github.com/LucksLab/Cotrans_SHAPE-Seq_Tools/releases': 2,\n",
       " 'github.com/LucksLab/spats/releases': 4,\n",
       " 'github.com/leschzinerlab/Relion': 1,\n",
       " 'github.com/parklab/NGSCheckMate': 1,\n",
       " 'transdecoder.github.io': 131,\n",
       " 'github.com/mikessh/migec': 2,\n",
       " 'github.com/milaboratory/mixcr': 5,\n",
       " 'github.com/imminfo/tcr': 1,\n",
       " 'nbviewer.ipython.org/urls/raw.github.com/cschin/Write_A_Genome_Assembler_With_IPython/master/Write_An_Assembler.ipynb': 1,\n",
       " 'github.com/drasmuss/nhrlmodel': 1,\n",
       " 'github.com/MITEL-UNIUD/SlideJ': 1,\n",
       " 'github.com/msoos/cryptominisat': 1,\n",
       " 'github.com/dsilvestro/PyRate': 5,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cuffdiff/index.html': 4,\n",
       " 'github.com/annajxli/scaredy-rat': 1,\n",
       " 'github.com/PacificBiosciences/SMRT-Analysis': 18,\n",
       " 'github.com/TACC/Lmod': 1,\n",
       " 'omicsoftdocs.github.io/ArraySuiteDoc': 1,\n",
       " 'github.com/ecjoliver/marineHeatWaves': 2,\n",
       " 'github.com/fchollet/keras': 29,\n",
       " 'github.com/ta-cameron/cell-profiles': 1,\n",
       " 'github.com/OLC-Bioinformatics/COWBAT/tree/v0.1.5': 1,\n",
       " 'github.com/JoanaGrah/MitosisAnalyser': 1,\n",
       " 'github.com/StewartLab/Stewart_TEDDY_Microbiome_Analysis': 1,\n",
       " 'github.com/Huanglab-ucsf/Deformed-Alignment': 1,\n",
       " 'github.com/foenpeng/Controller-of-experiment-platform.git': 1,\n",
       " 'github.com/mhaft/DeepVess': 1,\n",
       " 'github.com/GlobalFishingWatch/vessel-classification': 1,\n",
       " 'github.com/easybuilders/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/c/cutadapt': 1,\n",
       " 'github.com/USDA-ARS-GBRU/crossword': 1,\n",
       " 'github.com/USDA-ARS-GBRU/crossword/examples': 1,\n",
       " 'github.com/QijingZheng/VaspBandUnfolding': 1,\n",
       " 'github.com/pmoriano/Community-Based-Event-Detection': 1,\n",
       " 'github.com/lhting/Trauma_Study_GUI': 1,\n",
       " 'github.com/nskvir/RepEnrich14': 1,\n",
       " 'github.com/emreg00/toolbox': 4,\n",
       " 'github.com/michathoma/flywalkr': 1,\n",
       " 'stnava.github.com/WHopkinsNHP': 1,\n",
       " 'github.com/aleaverfay/swiftlib_javascript': 1,\n",
       " 'github.com/esling/mistagging': 1,\n",
       " 'github.com/UniversityofWarwick/ews_fli1_and_e2f_nar_article': 1,\n",
       " 'github.com/benliemory/Methylphet': 1,\n",
       " 'github.com/Victorian-Bioinformatics-Consortium/VelvetOptimiser': 4,\n",
       " 'github.com/sanger-pathogens/Bio-Tradis': 9,\n",
       " 'github.com/meren/illumina-utils': 9,\n",
       " 'github.com/csbl/platereader': 1,\n",
       " 'github.com/GenomicsCoreLeuven/GBSX': 3,\n",
       " 'github.com/crosslinks/XlinkAnalyzer': 1,\n",
       " 'github.com/minseven/mForensics.git': 1,\n",
       " 'github.com/jacobtennessen/HOLDRS': 1,\n",
       " 'github.com/slzhao/heatmap3': 1,\n",
       " 'github.com/jjberg2/PolygenicAdaptationCode': 1,\n",
       " 'github.com/fcchou/helixmc': 1,\n",
       " 'www.github.com/mathii/f2': 1,\n",
       " 'github.com/NeuroEngineering-UniMelb/cortical-units': 1,\n",
       " 'picrust.github.io/picrust': 33,\n",
       " 'github.com/Ruleby/ruleby': 1,\n",
       " 'picrust.github.com': 8,\n",
       " 'picrust.github.io/picrust/scripts/metagenome_contributions.html': 1,\n",
       " 'yanlilab.github.io/fastcov': 1,\n",
       " 'github.com/torognes/vsearch': 25,\n",
       " 'gephi.github.io': 31,\n",
       " 'github.com/maryreginaboland/SeaWAS': 1,\n",
       " 'github.com/luslab/multiregion-cnv-phylogenetics': 1,\n",
       " 'github.com/broadinstitute/picard/releases/tag/1.128': 2,\n",
       " 'github.com/GreallyLab/PE_Suzuki_et_al_2016': 1,\n",
       " 'github.com/ysu001/PUP': 4,\n",
       " 'github.com/justinjee/MDS': 1,\n",
       " 'github.com/susinmotion/barcode_tries': 1,\n",
       " 'github.com/ctorney/wildCount': 1,\n",
       " 'github.com/PacificBiosciences/DevNet/wiki/Drosophila-sequence-and-assembly': 2,\n",
       " 'github.com/Lakens/perfect-t-test': 2,\n",
       " 'github.com/meegpipe/meegpipe': 3,\n",
       " 'github.com/tsudalab/bpbi': 1,\n",
       " 'github.com/ijmarshall/robotsearch': 1,\n",
       " 'github.com/shka/STRTprep': 5,\n",
       " 'github.com/popgenomics/popPhylABC': 1,\n",
       " 'github.com/iontorrent': 3,\n",
       " 'github.com': 141,\n",
       " 'github.com/loghindo/NDPI_counting_MC1_bacteria': 1,\n",
       " 'github.com/FunctionLab/ExPecto': 1,\n",
       " 'broadinstitute.github.io/picard/index.html': 12,\n",
       " 'www.github.com/dspeth': 3,\n",
       " 'github.com/shengqh/TIGER': 1,\n",
       " 'github.com/ColumbusCollaboratory/cognizer': 1,\n",
       " 'github.com/robince/gcmi': 3,\n",
       " 'github.com/robince/partial-info-decomp': 1,\n",
       " 'github.com/biocore-ntnu/epic': 4,\n",
       " 'github.com/GRousselet/blog/tree/master/comp2dcorr': 1,\n",
       " 'github.com/mui-icbi/Immunophenogram': 2,\n",
       " 'github.com/Neurorobtics/ACOREL': 1,\n",
       " 'github.com/PacificBiosciences/pbdagcon.git': 1,\n",
       " 'github.com/rhodricusack/automaticanalysis': 7,\n",
       " 'chamber.github.io/fibernavigator_single': 1,\n",
       " 'github.com/dials/dxtbx_ED_formats': 1,\n",
       " 'github.com/ANGSD/asaMap': 1,\n",
       " 'github.com/ESPoppelaars/Cross-frequency-coupling': 1,\n",
       " 'github.com/gephi/gephi/wiki/OpenOrd': 1,\n",
       " 'github.com/LABSS/PeerReviewGPS': 1,\n",
       " 'github.com/Oshlack/Farlie_emu_wing_RNASeq_analysis': 1,\n",
       " 'samtools.github.io/bcftools': 20,\n",
       " 'github.com/EuropePMC/Biojs.Annotator/tree/Biojs.Annotator_1.0': 1,\n",
       " 'github.com/a3cel2/bc_pca_git': 1,\n",
       " 'github.com/statonlab/disculaSSRs': 1,\n",
       " 'github.com/PeeperLab/CopywriteR': 2,\n",
       " 'github.com/evanfeinberg/conformation/blob/master/grids.py': 1,\n",
       " 'interactivereport.github.io/ZSF1': 1,\n",
       " 'github.com/DavidCunefare/CNN-Cone-Detection': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks': 116,\n",
       " 'github.com/smfleming/HMM': 1,\n",
       " 'github.com/gollischlab/STNMFanalysis': 1,\n",
       " 'github.com/nylander': 1,\n",
       " 'github.com/RPeterson4/Baseline_Covariate_Files': 1,\n",
       " 'github.com/bcgsc/nanosim': 1,\n",
       " 'github.com/BGI-flexlab/SOAPnuke': 5,\n",
       " 'github.com/schae234/pypeline': 1,\n",
       " 'github.com/sliu2/BPAC': 1,\n",
       " 'github.com/jcaledo/PTM_sites_coevolution': 1,\n",
       " 'github.com/jstjohn/SeqPrep': 163,\n",
       " 'github.com/bdearlove/treeImbalance': 1,\n",
       " 'github.com/SEEG-Oxford/seegSDM': 4,\n",
       " 'github.com/flauro/3tck_comparative': 2,\n",
       " 'github.com/jimhester/knitrBootstrapFastQC': 1,\n",
       " 'github.com/iontorrent/TMAP': 25,\n",
       " 'sanger-pathogens.github.io/iva': 1,\n",
       " 'github.com/mattmattmattmatt/VASP': 1,\n",
       " 'github.com/bhi-kimlab/BiSpark': 1,\n",
       " 'github.com/Christensen-Lab-Dartmouth/CF_Epigenetics': 1,\n",
       " 'github.com/rcsb/PDB-Outlier-Analysis.git': 1,\n",
       " 'pasapipeline.github.io': 8,\n",
       " 'github.com/mgalardini/pyseer': 1,\n",
       " 'github.com/broadinstitute/gatk': 2,\n",
       " 'github.com/sbg/VBT-TrioAnalysis.git': 1,\n",
       " 'github.com/leesael/GIFT': 1,\n",
       " 'github.com/ICB-DCM/PESTO': 1,\n",
       " 'github.com/TheOwenLab/2018-Wild-et-al-Sleep-and-Cognition': 1,\n",
       " 'github.com/rowaniskandar/CM_ME_plosone': 1,\n",
       " 'github.com/TransDecoder/TransDecoder/wiki': 12,\n",
       " 'github.com/csamstag/mito-mutations': 1,\n",
       " 'github.com/mattlevine22/emerge2ohdsi_information_loss_CURATED.git': 1,\n",
       " 'github.com/elifesciences-publications/CATMAID-to-Blender': 1,\n",
       " 'nbisweden.github.io/MrBayes': 1,\n",
       " 'github.com/davidrasm/Marula': 1,\n",
       " 'github.com/cculianu/StimulateOpenGL_II': 1,\n",
       " 'github.com/pbnjay/lollipops/releases': 2,\n",
       " 'github.com/nci/drishti/wiki': 2,\n",
       " 'github.com/VafaeeLab/glioblastoma_exosomal_miR_markers': 1,\n",
       " 'benjjneb.github.io/dada2/index.html': 2,\n",
       " 'www.github.com/ctross/emberasearch': 1,\n",
       " 'github.com/prafols/rMSIproc': 1,\n",
       " 'cj-chen.github.io/tbtools': 4,\n",
       " 'atztogo.github.io/phonopy': 3,\n",
       " 'github.com/tseemann/snippy': 40,\n",
       " 'github.com/BUCK-LCO-NCI/VirConTaxa': 1,\n",
       " 'github.com/amnona/calour': 1,\n",
       " 'github.com/phac-nml/imid_microbiome': 1,\n",
       " 'github.com/mirnylab/cooler': 4,\n",
       " 'github.com/alexdobin/STAR': 106,\n",
       " 'github.com/fredericlemoine/goalign': 2,\n",
       " 'github.com/mingjunzheng/ceRNA_code': 1,\n",
       " 'github.com/ajvilleg/extractUpStreamDNA': 1,\n",
       " 'api.github.com': 1,\n",
       " 'github.com/FelixKrueger/TrimGalore': 22,\n",
       " 'github.com/bitcoin/bitcoin': 1,\n",
       " 'github.com/RamanLab/nbfpeg': 1,\n",
       " 'github.com/ENCODE-DCC/chip-seq-pipeline': 1,\n",
       " 'www.github.com/ElectricDinoLab': 1,\n",
       " 'github.com/taiyun/corrplot': 14,\n",
       " 'github.com/Sun-Yazhou/MDAD': 1,\n",
       " 'github.com/sydneycytometry/tSNEplots': 1,\n",
       " 'github.com/spyking-circus/spyking-circus': 1,\n",
       " 'github.com/Sage-Bionetworks/MoleMapper': 1,\n",
       " 'github.com/researchkit/researchkit': 3,\n",
       " 'github.com/Sage-Bionetworks/Bridge-iOS-SDK': 3,\n",
       " 'github.com/pid-ub/pid-UB': 1,\n",
       " 'dkoboldt.github.io/varscan': 9,\n",
       " 'stnava.github.io/ANTs': 96,\n",
       " 'github.com/alexandrebarachant/pyRiemann': 1,\n",
       " 'github.com/alexandrebarachant/WMFT_JBHI': 1,\n",
       " 'github.com/careylab/LocoMouse': 3,\n",
       " 'github.com/alanrice/paper-dosage-sensitivity-copy-number-variation': 1,\n",
       " 'github.com/ocisse/Neolecta_genome_project': 1,\n",
       " 'github.com/joanam/scripts': 1,\n",
       " 'homeveg.github.io/nuctools': 1,\n",
       " 'github.com/tseemann/wombac': 3,\n",
       " 'github.com/karpathy/char-rnn': 1,\n",
       " 'github.com/pimbongaerts/radseq': 1,\n",
       " 'github.com/pimbongaerts/bermuda-rad': 1,\n",
       " 'github.com/peteskene': 1,\n",
       " 'github.com/joey711/phyloseq/issues/143': 1,\n",
       " 'github.com/pillailab': 1,\n",
       " 'tgennorth.github.io/NASP': 23,\n",
       " 'github.com/ImperialStatGen/MARV': 2,\n",
       " 'github.com/colford/nbgw-plant-illumina-pipeline': 3,\n",
       " 'github.com/scarrazza/smpdf': 1,\n",
       " 'github.com/ericminikel/prnp_penetrance': 3,\n",
       " 'github.com/PerinatalLab/MatHeight-GestAge': 1,\n",
       " 'github.com/kalpathy/tacticsRT': 3,\n",
       " 'github.com/alvason/identifying_neurite_by_RRS': 1,\n",
       " 'github.com/tomotto/RSE-box': 1,\n",
       " 'github.com/bettytijms/Single_Subject_Grey_Matter_Networks': 3,\n",
       " 'trinotate.github.io': 83,\n",
       " 'github.com/merijnbeeksma/predict-EoL': 1,\n",
       " 'github.com/gfilaci/GridNSPT': 1,\n",
       " 'github.com/jgx65/quantinemo': 1,\n",
       " 'github.com/chlubba/PyPNS': 1,\n",
       " 'github.com/chlubba/PyPNS-PaperFigureshttp://github.com/chlubba/PyPNS-PaperFigures': 1,\n",
       " 'github.com/cestaroa/tab_blast_parser': 1,\n",
       " 'github.com/yonghee91/NGS_error_validation.git': 1,\n",
       " 'github.com/mrirecon/enlive': 1,\n",
       " 'github.com/btlindert/actant-1': 1,\n",
       " 'github.com/sarmistha123/TightClust_Large/blob/master/TightCluster_Large.R': 1,\n",
       " 'github.com/sensorimotorgroupdpz/MaCaQuE': 1,\n",
       " 'github.com/mariodosreis/tai': 1,\n",
       " 'github.com/osuchanglab/autoANI': 1,\n",
       " 'github.com/ctSkennerton/minced': 2,\n",
       " 'github.com/Bioconductor/GenomicRanges': 1,\n",
       " 'github.com/wangjingke/madresGpsClient': 1,\n",
       " 'github.com/VGalata/16S_Sarcoidosis': 1,\n",
       " 'github.com/chingtoe365/metaUnion': 1,\n",
       " 'github.com/srcbs/GenoBox': 1,\n",
       " 'github.com/MatthieuPalayret/vlsSMLM': 1,\n",
       " 'github.com/igraph/xdata-igraph': 1,\n",
       " 'github.com/jovo/FastApproximateQAP/tree/master/code/FAQ': 1,\n",
       " 'github.com/gersteinlab/loregic': 1,\n",
       " 'github.com/morganrcu/TTT': 1,\n",
       " 'biossa.github.io': 1,\n",
       " 'github.com/xutaodeng/EnsembleAssembler': 1,\n",
       " 'github.com/atmartens/ribosome_profiling': 1,\n",
       " 'github.com/lanagarmire/epiPredictor': 1,\n",
       " 'github.com/chrisamiller/copycat': 5,\n",
       " 'github.com/genome/gms-core/blob/master/lib/perl/Genome/Model/Tools/Sv/SquareDancer.pl': 2,\n",
       " 'github.com/genome/sciclone': 10,\n",
       " 'ntncmch.github.io/ebola/liberia_sitrep.html': 1,\n",
       " 'github.com/SachaEpskamp/semPlot': 1,\n",
       " 'libspatialindex.github.io': 1,\n",
       " 'github.com/StamLab/network-motifs': 1,\n",
       " 'github.com/widdowquinn/scripts/blob/master/bioinformatics/calculate_ani.py': 4,\n",
       " 'github.com/mazerj/pype3': 2,\n",
       " 'github.com/RavelLab/GuineaPigVMB': 1,\n",
       " 'github.com/abria/TeraStitcher': 1,\n",
       " 'github.com/xies/pulse_finding': 1,\n",
       " 'github.com/KTgreenham/TRiP': 1,\n",
       " 'github.com/janelia-flyem/NeuroProof.git': 1,\n",
       " 'github.com/SynBioDex/libSBOLj': 2,\n",
       " 'github.com/cni/widgets/tree/master/touch': 1,\n",
       " 'github.com/JaneliaSciComp/Muse': 1,\n",
       " 'github.com/DanRosauer/phylospatial': 1,\n",
       " 'github.com/sjackman/uniqtag': 1,\n",
       " 'github.com/sjackman/uniqtag-paper': 1,\n",
       " 'OrgMassSpec.github.io': 3,\n",
       " 'github.com/CovertLab/WholeCell/tree/parameter-estimation-DREAM-challenge-2013': 1,\n",
       " 'github.com/paulspychala/CMAPdart': 1,\n",
       " 'github.com/klusta-team': 3,\n",
       " 'idekerlab.github.io/cy-net-share': 3,\n",
       " 'dleelab.github.io/jepegmix': 1,\n",
       " 'turkunlp.github.io/Cell-line-recognition': 1,\n",
       " 'www.github.com/dsurujon/motifs': 1,\n",
       " 'github.com/iPlantCollaborativeOpenSource': 1,\n",
       " 'github.com/curtislarimer/Biofilm-Growth-Intensity': 1,\n",
       " 'github.com/IEDB/MRO': 1,\n",
       " 'github.com/raivokolde/seqlm': 1,\n",
       " 'github.com/wilkelab/influenza_pH1N1_timecourse': 1,\n",
       " 'github.com/owenrandlett/Z-Brain': 1,\n",
       " 'trinotate.github.io/TrinotateWeb.html': 1,\n",
       " 'github.com/DmitryVel/CANEapp': 1,\n",
       " 'github.com/sarahannison': 1,\n",
       " 'github.com/itseez/opencv': 4,\n",
       " 'github.com/zeochoy/tcga-embedding': 1,\n",
       " 'github.com/r-syst/databases/tree/master/r-syst::fungi': 1,\n",
       " 'github.com/soedinglab/b-lore': 1,\n",
       " 'sashagusev.github.io/2017-10/twas-vulnerabilities.html': 1,\n",
       " 'github.com/systemsmedicine/adjuvanted-vaccine': 1,\n",
       " 'github.com/ben-lerch/BaseMod-3.0': 1,\n",
       " 'github.com/fl16180/argonet': 1,\n",
       " 'github.com/jeremymsimon/MouseCortex': 1,\n",
       " 'github.com/dewyman/TranscriptClean': 1,\n",
       " 'github.com/soumitag/SLIDE': 1,\n",
       " 'github.com/jkbonfield/crumble': 1,\n",
       " 'github.com/davidjakubec/3DPatch': 1,\n",
       " 'github.com/b2du/bth': 1,\n",
       " 'github.com/alexjau/uqsa': 1,\n",
       " 'github.com/yunwilliamyu/opal': 1,\n",
       " 'github.com/mapircamera/QGIS': 1,\n",
       " 'github.com/chail/NMF_neurodevelopment': 1,\n",
       " 'github.com/vcflib/vcflib': 23,\n",
       " 'github.com/tseemann/nullarbor': 16,\n",
       " 'github.com/federicomarini/pcaExplorer': 1,\n",
       " 'github.com/linbingqing/stableDE': 1,\n",
       " 'github.com/ben-lerch/IsoSeq-3.0/blob/master/README.md': 1,\n",
       " 'github.com/Sunhh/NGS_data_processing': 1,\n",
       " 'github.com/bminkenberg/CRISPR-PLANTv2': 1,\n",
       " 'github.com/obigriffith/biostar-tutorials/blob/master/Heatmaps/heatmap.3.R': 2,\n",
       " 'github.com/NWPU-903PR/Deepm6A.git': 1,\n",
       " 'github.com/Romain-Laine/MiLeSIM': 1,\n",
       " 'github.com/elifesciences-publications/MiLeSIM': 1,\n",
       " 'github.com/sanger-pathogens/mlst_check': 12,\n",
       " 'github.com/marbl/Parsnp': 1,\n",
       " 'benjjneb.github.io/dada2/tutorial.html': 8,\n",
       " 'github.com/jfaberha/lastz_lav_expansion': 1,\n",
       " 'transdecoder.github.io.com/TransDecoder': 1,\n",
       " 'github.com/wg-perception/people': 1,\n",
       " 'lmfit.github.io/lmfit-py': 6,\n",
       " 'github.com/kthorn/particletrack': 1,\n",
       " 'github.com/KeithBurghardt/Stack-Exchange-User-Behavior': 1,\n",
       " 'github.com/cole-trapnell-lab/cufflinks': 12,\n",
       " 'github.com/babelomics/hipathia': 1,\n",
       " 'github.com/ding-lab/squaredancer': 1,\n",
       " 'github.com/spoilt333/onco-aae': 1,\n",
       " 'github.com/ruudvdent/WAM2layersPython': 1,\n",
       " 'github.com/OPIC-Oxford/localrec': 1,\n",
       " 'github.com/EdaEhler/bioutils_python': 1,\n",
       " 'github.com/mulin0424/cepip': 1,\n",
       " 'github.com/broadinstitute/mutect': 5,\n",
       " 'github.com/MKaptein/streamingbandit': 1,\n",
       " 'github.com/broadinstitute/picard': 55,\n",
       " 'github.com/AltschulerWu-Lab/simucell': 1,\n",
       " 'cloud.github.com/downloads/faircloth-lab/edittag/edit_metric_tags.txt': 1,\n",
       " 'github.com/nitishsrivastava/deepnet': 4,\n",
       " 'github.com/jouyun': 5,\n",
       " 'github.com/xnuttle/mips_cnv_typer': 1,\n",
       " 'github.com/scapella/markers_genes': 1,\n",
       " 'github.com/MichaelRiss/FTSPlot': 1,\n",
       " 'github.com/koldogarcia/metagenomics': 1,\n",
       " 'github.com/andymckenzie/AtxA': 1,\n",
       " 'github.com/selimonat/published_code/tree/master/condprob': 1,\n",
       " 'github.com/YeatmanLab/Speech_contrasts_public': 1,\n",
       " 'github.com/rrwick/Porechop': 25,\n",
       " 'github.com/fhcrc/taxtastic': 2,\n",
       " 'github.com/bjbarrett/bequeathal': 1,\n",
       " 'github.com/metacoglab/VaccaroFleming': 1,\n",
       " 'github.com/wajidarshad/LUPI-SVM': 1,\n",
       " 'github.com/BradnerLab/netseq': 3,\n",
       " 'github.com/chapmanb/bcbb/blob/master/align/adaptor_trim.py': 1,\n",
       " 'github.com/srkesler/bnets': 1,\n",
       " 'github.com/matterport/Mask_RCNN': 1,\n",
       " 'github.com/orzechoj/circRNA_finder.git': 2,\n",
       " 'github.com/PacificBiosciences/GenomicConsensus': 21,\n",
       " 'github.com/kendomaniac/docker4seq': 2,\n",
       " 'github.com/zcshinee/Pol-TIRF': 1,\n",
       " 'github.com/dbrvs/reservoir_persistence': 1,\n",
       " 'github.com/trinityrnaseq/trinityrnaseq/wiki': 34,\n",
       " 'pachterlab.github.io/kallisto': 17,\n",
       " 'github.com/aspen-shen/RefCell': 1,\n",
       " 'github.com/SoniMohapatra/MPCC': 1,\n",
       " 'deweylab.github.io/RSEM': 28,\n",
       " 'github.com/ENCODE-DCC/long-rna-seq-pipeline/blob/master/DAC/STAR_RSEM.sh': 2,\n",
       " 'tdwg.github.io/dwc/terms': 2,\n",
       " 'pypl.github.io/PYPL.html': 2,\n",
       " 'github.com/najoshi/sabre': 17,\n",
       " 'github.com/ADAPTABLETRIAL/PHENOTYPE': 1,\n",
       " 'github.com/TransDecoder/TransDecoder': 11,\n",
       " 'github.com/sinhrks/ggfortify': 1,\n",
       " 'github.com/mjobin/Imputor': 1,\n",
       " 'github.com/debbiemarkslab/EVcouplings': 3,\n",
       " 'github.com/zyndagj/BSMAPz': 1,\n",
       " 'github.com/al2na/methylKit': 5,\n",
       " 'github.com/taoliu/MACS': 93,\n",
       " 'github.com/rbslandau/schreiner_simstress': 1,\n",
       " 'github.com/molnarb14/analog-maxSAT-solver': 1,\n",
       " 'github.com/jakobrunge/tigramite': 1,\n",
       " 'github.com/hzi-bifo/FrechetTreeDistance': 1,\n",
       " 'github.com/kylepquinn/Code-and-Example-Data': 1,\n",
       " 'github.com/kharyuk/chemfin-plasp': 1,\n",
       " 'github.com/nakae-k/glia-neuron': 1,\n",
       " 'github.com/PlantSimLab': 1,\n",
       " 'dongjunchung.github.io/GPA': 2,\n",
       " 'github.com/reidpr/quac': 2,\n",
       " 'github.com/takaho/snpexp': 1,\n",
       " 'josephryan.github.com/estimate_genome_size.pl': 1,\n",
       " 'github.com/tanghaibao/goatools': 21,\n",
       " 'github.com/aherbert/GDSC-SMLM': 2,\n",
       " 'github.com/aramette/otu2ot': 1,\n",
       " 'github.com/bvilhjal/mixmogam': 7,\n",
       " 'github.com/kylebittinger/airway-fungi-paper': 1,\n",
       " 'github.com/gmpgis/distribution': 1,\n",
       " 'github.com/omegaicl/mcpp': 1,\n",
       " 'github.com/enormandeau': 1,\n",
       " 'github.com/mackerman44/idfgen': 1,\n",
       " 'github.com/michaelfrance/evol_app_2018_biofilms': 1,\n",
       " 'github.com/cwrowley/dmdtools/tree/master': 1,\n",
       " 'github.com/MariaICL/Paper_analysis_scripts': 1,\n",
       " 'trinityrnaseq.github.io': 38,\n",
       " 'github.com/lihong1github/Image-analysis': 1,\n",
       " 'github.com/cokelaer/spectrum': 1,\n",
       " 'github.com/pmpsa-hpc/NPA': 1,\n",
       " 'github.com/pmpsa-hpc/NPAModels': 1,\n",
       " 'github.com/maranasgroup/optstoic-python': 1,\n",
       " 'github.com/kwikteam/klustakwik2': 2,\n",
       " 'github.com/jyosa/CYP2D6': 1,\n",
       " 'github.com/maufrais/WHGDCA_ES': 1,\n",
       " 'github.com/myl446/BRWHNHA': 1,\n",
       " 'github.com/msettles/dbcAmplicons': 8,\n",
       " 'github.com/macarenadiaz/FAZ_Extraction': 1,\n",
       " 'github.com/gradinetal2018/BARA': 1,\n",
       " 'www.github.com/PublicHealthEngland/bayesint': 1,\n",
       " 'github.com/QingtianZhang/SHMAX_AuditoryPathway': 1,\n",
       " 'github.com/chop-dbhi/sepsis_01': 1,\n",
       " 'github.com/Emergent-Behaviors-in-Biology/community-simulator': 1,\n",
       " 'github.com/groupschoof/AHRD': 9,\n",
       " 'github.com/shenlab-sinai/ngsplot': 14,\n",
       " 'github.com/Zhong-Lab-UCSD/iMARGI_methods': 1,\n",
       " 'github.com/mirnylab/pairtools': 1,\n",
       " 'github.com/agarcor21/IdentityCoefficients': 1,\n",
       " 'github.com/chevrm/kopilot': 1,\n",
       " 'github.com/mmadsen/lipomadsen2015-idss-seriation-paper': 1,\n",
       " 'github.com/RodgerG/Hemiptera-of-Canada-DNA-Barcode-Library-workflow.git': 1,\n",
       " 'github.com/ripple-neuro/mksort': 5,\n",
       " 'github.com/genericworkflownodes': 2,\n",
       " 'ericmjonas.github.io/connectodiscovery': 1,\n",
       " 'github.com/dch33/Quad-Sim': 1,\n",
       " 'github.com/wangsy11/InfMod3DGen': 2,\n",
       " 'github.com/arq5x/piledriver': 5,\n",
       " 'github.com/gbif/checklistbank': 2,\n",
       " 'github.com/google/guice': 1,\n",
       " 'github.com/jts/sga/zipball/master': 2,\n",
       " 'github.com/meizhan/SVMelegans': 2,\n",
       " 'github.com/sina-cb/Tn-seqExplorer': 1,\n",
       " 'github.com/obophenotype/uberon/wiki/Editor-tools-guide': 1,\n",
       " 'github.com/akastrin/kastrin2014large': 1,\n",
       " 'dyerlab.github.io/popgraph': 1,\n",
       " 'github.com/najoshi/sickle.git': 3,\n",
       " 'networkx.github.io': 54,\n",
       " 'github.com/janelia-flyem/gala': 1,\n",
       " 'github.com/ged-lab/khmer/tree/2012-paper-diginorm/sandbox': 2,\n",
       " 'github.com/ged-lab/khmer': 3,\n",
       " 'github.com/ged-lab/screed': 2,\n",
       " 'github.com/bwrc/WishGLD': 1,\n",
       " 'github.com/fbarranco/eventVision-evbench': 1,\n",
       " 'github.com/davidssmith/csphantom': 1,\n",
       " 'github.com/bernardosabatinilab/SabalabSoftware_Nov2009.git': 2,\n",
       " 'github.com/usc-tlab/LinearFineContrast.git': 2,\n",
       " 'github.com/sofroniewn/tactile-coding': 1,\n",
       " 'github.com/stephenfloor/tripseq-analysis': 2,\n",
       " 'github.com/akrizhevsky/cuda-convnet2': 1,\n",
       " 'github.com/peterjfoster/eLife': 1,\n",
       " 'github.com/jeongchans/smrfArchived': 1,\n",
       " 'dgkontopoulos.github.io/Structuprint': 1,\n",
       " 'github.com/lsgh/ismb15': 1,\n",
       " 'github.com/frizfealer/IMS_project': 1,\n",
       " 'github.com/chengyuan/reago': 1,\n",
       " 'github.com/gemtools/gemtools': 4,\n",
       " 'wurmlab.github.io/data': 1,\n",
       " 'github.com/tylerhether/Flip2BeRAD': 1,\n",
       " 'github.com/samtools/BCFtools': 3,\n",
       " 'github.com/gc5k/GEAR/wiki': 1,\n",
       " 'github.com/DenisCarriere/geocoder': 1,\n",
       " 'github.com/cmdoret/chalcid_comparative_analysis': 1,\n",
       " 'github.com/QCaudron/timing_malaria_transmission': 1,\n",
       " 'github.com/thomasp85/patchwork': 1,\n",
       " 'github.com/thackl/ggworldmap': 1,\n",
       " 'github.com/babicknell/NeuriteGrowth': 1,\n",
       " 'github.com/TulsaBioinformaticsToolsmith/HebbPlot': 1,\n",
       " 'github.com/3DGenomes/tadbit': 4,\n",
       " 'github.com/shmohammadi86/Kavosh': 1,\n",
       " 'github.com/OneAngstrom/DeepTrajectory': 1,\n",
       " 'github.com/BrainDynamicsUSYD/nftsim': 1,\n",
       " 'github.com/gulyasmarton/PetriPrinter': 1,\n",
       " 'github.com/bicanski/HBPcollab/tree/master/SpatialEpisodicMemoryModel': 1,\n",
       " 'github.com/elifesciences-publications/HBPcollab/tree/master/SpatialEpisodicMemoryModel': 1,\n",
       " 'github.com/marsdeck/AFMech-Suite/archive/v1.0.zip': 1,\n",
       " 'github.com/marsdeck/AFMech-Suite': 1,\n",
       " 'github.com/pimbongaerts/monitoring': 1,\n",
       " 'github.com/weiliu123/RRSF-package': 1,\n",
       " 'githubcom/sccn/labstreaminglayer': 1,\n",
       " 'github.com/ItzikMalkiel/DeepNanoDesign': 1,\n",
       " 'github.com/princello/GePMI': 1,\n",
       " 'github.com/Sung-Huan/ANNOgesic': 1,\n",
       " 'github.com/Shoombuatong2527/anticancer-peptides-review': 1,\n",
       " 'github.com/AlanFreeman/anaMotion': 1,\n",
       " 'github.com/keitaroyam/yamtbx': 4,\n",
       " 'github.com/fmrireplicability/NCB_code': 1,\n",
       " 'github.com/jfmeadow/dualingTrees-pkg': 1,\n",
       " 'github.com/Jincheng2009/igseqanalysis': 1,\n",
       " 'alitvteam.github.io/AliTV/d3/AliTV.html': 1,\n",
       " 'github.com/macraild/6D8-CommsBiol': 1,\n",
       " 'github.com/jpcolella/Ermine_WGS_2018': 1,\n",
       " 'github.com/broadinstitute/pilon': 4,\n",
       " 'github.com/ranganawarshamanage/mxt': 1,\n",
       " 'github.com/annalam/foxa1-utr-manuscript-code': 1,\n",
       " 'github.com/annalam/breakfast': 2,\n",
       " 'github.com/annalam/seqkit': 1,\n",
       " 'github.com/abhik/crispr–analysis': 1,\n",
       " 'github.com/manutamminen/epride': 1,\n",
       " 'github.com/lh3/bwa': 26,\n",
       " 'github.com/broadgsa/gatk': 6,\n",
       " 'samtools.github.io': 7,\n",
       " 'github.com/arq5x/bedtools2': 33,\n",
       " 'github.com/Ensembl/ensembl-vep': 6,\n",
       " 'github.com/kuz/Human-Intracranial-Recordings-and-DCNN-to-Compare-Biological-and-Artificial-Mechanisms-of-Vision': 1,\n",
       " 'github.com/josephbarry/iQTL': 1,\n",
       " 'raw.githubusercontent.com/obigriffith/biostar-tutorials/master/Heatmaps/heatmap.3.R': 4,\n",
       " 'github.com/jkrijthe/Rtsne': 12,\n",
       " 'github.com/tpq/balance': 1,\n",
       " 'github.com/SharonLutz/Umediation': 3,\n",
       " 'github.com/WSpiller/RadialMR': 1,\n",
       " 'github.com/bmelinden/uncertainSPT': 2,\n",
       " 'github.com/roschkoenig/Zebrafish_Seizure': 1,\n",
       " 'ccraddock.github.io/cluster_roi/atlases.html': 2,\n",
       " 'github.com/stschiff/sequenceTools/tree/master/src-pileupCaller': 3,\n",
       " 'github.com/olavolav/TE-Causality': 1,\n",
       " 'github.com/contact-contagion/mrsa-c': 1,\n",
       " 'github.com/mcintyrelab': 1,\n",
       " 'github.com/theis188/theisen-plos-comp-bio': 1,\n",
       " 'kingaa.github.io/pomp': 3,\n",
       " 'github.com/jimhester/primerTree': 1,\n",
       " 'github.com/SamStudio8/goldilocks': 2,\n",
       " 'github.com/wikiselev/rnaseq.mcf10a': 1,\n",
       " 'middle-down.github.io/Software': 12,\n",
       " 'github.com/brwnj/umitools': 3,\n",
       " 'github.com/hesselberthlab/5OH': 1,\n",
       " 'github.com/hesselberthlab/trackhub': 1,\n",
       " 'github.com/JialiUMassWengLab/laSV/tree/master': 1,\n",
       " 'github.com/gyronaut/tomography': 1,\n",
       " 'workflowconversion.github.io': 1,\n",
       " 'github.com/kevinrue/shiny-MDM': 1,\n",
       " 'github.com/chiuyc/MAGIC': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/manual': 6,\n",
       " 'github.com/jnw29/AvrPto_Transcriptome': 1,\n",
       " 'github.com/joelewnard/choleraHaiti.git': 1,\n",
       " 'github.com/araksm/ceRNA': 1,\n",
       " 'github.com/l-modolo/TEtools': 4,\n",
       " 'github.com/hallowkm/RxODE': 1,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cufflinks': 2,\n",
       " 'cole-trapnell-lab.github.io/cufflinks/cuffmerge': 3,\n",
       " 'github.com/FRED-2/OptiVac': 1,\n",
       " 'github.com/genome/bam-readcount': 32,\n",
       " 'github.com/aplested/aligator': 1,\n",
       " 'ComputationalProteomicsUnit.github.io/pRolocGUI': 1,\n",
       " 'github.com/alfbenpa/16S_MinION': 1,\n",
       " '1dl3.github.io/graph': 1,\n",
       " 'github.com/NDCcodes/Daut_ND-matlab-codes/tree/master': 1,\n",
       " 'github.com/GabeAl/NINJA-OPS': 1,\n",
       " 'github.com/nphynimhans/ANGEL': 1,\n",
       " 'github.com/BenKaehler/gapped': 2,\n",
       " 'github.com/acg-team/ProPIP': 1,\n",
       " 'github.com/LeonYenLeeChan/RNAdecay': 1,\n",
       " 'github.com/elifesciences-publications/RNAdecay': 1,\n",
       " 'github.com/aufrank/R-hacks/blob/master/mer-utils.R': 5,\n",
       " 'github.com/daniel-temko/CNVTiming': 1,\n",
       " 'github.com/schatzlab/genomescope': 2,\n",
       " 'www.github.com/tblazey/ogiMeta': 1,\n",
       " 'github.com/ChristianLabUIUC/BurstAnalysis': 1,\n",
       " 'github.com/bronsonweston/GMP-Modeling': 1,\n",
       " 'github.com/ebov/space-time/tree/master/Data/phyml_raxml_ML.sh': 1,\n",
       " 'github.com/alesssia/MAP': 1,\n",
       " 'github.com/krumsieklab/MissingValues': 1,\n",
       " 'github.com/waghsk/lvef-paper': 1,\n",
       " 'github.com/kmiller324/KymoFlow': 1,\n",
       " 'github.com/fcalafell/weighted_rho': 1,\n",
       " 'expressionanalysis.github.io/ea-utils': 27,\n",
       " 'openseadragon.github.io': 7,\n",
       " 'github.com/lme4/lme4/http://lme4.r-forge.r-project.org': 1,\n",
       " 'github.com/MikeAxtell/ShortStack/releases': 2,\n",
       " 'github.com/MikeAxtell/CleaveLand4/releases': 1,\n",
       " 'github.com/zhangtongda/Syntool': 1,\n",
       " 'github.com/gephi/gephi': 2,\n",
       " 'github.com/jinweiwu/perl/blob/master/extract': 1,\n",
       " 'github.com/owlcollab/owltools': 4,\n",
       " 'github.com/ekg/freebayes': 36,\n",
       " 'github.com/mdhumphries/AplysiaAttractorAnalysis': 1,\n",
       " 'github.com/elifesciences-publications/AplysiaAttractorAnalysis': 1,\n",
       " 'github.com/martcous/ppmi-study': 1,\n",
       " 'github.com/hyperic/sigar': 1,\n",
       " 'github.com/vivekbhr/Subread_to_DEXSeq': 1,\n",
       " 'github.com/denovogear': 1,\n",
       " 'github.com/ozagordi/VirMet/releases/tag/v0.3.3': 1,\n",
       " 'github.com/mestato/lab_code/tree/master/hwg_gssr_scripts': 3,\n",
       " 'github.com/vegandevs/vegan': 14,\n",
       " 'github.com/vcftools/vcftools': 2,\n",
       " 'github.com/microbiome/microbiome': 3,\n",
       " 'github.com/PacificBiosciences/Bioinformatics-Training/blob/master/barcoding/pacbio_384_barcodes.fasta': 2,\n",
       " 'github.com/chapmanb/bcbio-nextgen': 21,\n",
       " 'github.com/embruna/Costa-Atta-Seedlings': 1,\n",
       " 'github.com/lianaan/Perc_Var': 1,\n",
       " 'github.com/daquang/YAMDA': 1,\n",
       " 'github.com/PECAplus': 1,\n",
       " 'github.com/machenslab/dPCA': 6,\n",
       " 'github.com/gamaleldin/TME': 2,\n",
       " 'github.com/zalf-rpm/monica': 1,\n",
       " 'github.com/DSSAT/dssat-csm': 1,\n",
       " 'github.com/Methylflower/DMR-scan': 1,\n",
       " 'github.com/keras-team/keras': 7,\n",
       " 'github.com/mberacochea/sorghum-bicolor-M81E-16S': 1,\n",
       " 'github.com/jannahastings/mental-functioning-ontology': 1,\n",
       " 'github.com/jbowensruc/PerlScriptFrontiers': 1,\n",
       " 'github.com/pchrapka/phasereset': 1,\n",
       " 'github.com/lxchong/perimetry-custom-grids': 1,\n",
       " 'cole-trapnellab.github.io/cufflinks/papers': 1,\n",
       " 'github.com/OpenSourceBrain/GranCellSolinasEtAl10': 1,\n",
       " 'github.com/croessert/AnalyseGranCellRoessertEtAl14': 1,\n",
       " 'github.com/bastodian/shed/blob/master/Python/q-trim.py': 2,\n",
       " 'picrust.github.io': 3,\n",
       " 'github.com/FedericoComocglio/wavClusteR': 1,\n",
       " 'github.com/QBRC/PIPE-CLIP': 3,\n",
       " 'github.com/arq5x/lumpy-sv': 11,\n",
       " 'github.com/nickloman/entrezajax': 1,\n",
       " 'github.com/stamatak/Parsimonator-1.0.2': 3,\n",
       " 'cancerit.github.io/CaVEMan': 8,\n",
       " 'github.com/cogbio/Experimenter': 1,\n",
       " 'github.com/ThorntonLab/ESMtest': 1,\n",
       " 'github.com/MorrellLAB/Soja_Env_Association': 1,\n",
       " 'github.com/konradjk/loftee': 10,\n",
       " 'github.com/molgenis/systemsgenetics/tree/master/eqtl-mapping-pipeline': 1,\n",
       " 'github.com/ramhiser/datamicroarray/blob/master/README.md': 1,\n",
       " 'github.org/AEBilgrau/effadj': 1,\n",
       " 'github.com/danielfrg/tsne': 1,\n",
       " 'gist.github.com/coleoguy/6886683': 1,\n",
       " 'github.com/ucdavis-bioinformatics': 4,\n",
       " 'github.com/ibest/GRC_Scripts/blob/master/screen_duplicates_PE.py': 1,\n",
       " 'github.com/pintotm/PLoSOne2014': 1,\n",
       " 'github.com/dkoes/shapedb': 1,\n",
       " 'github.com/ComputationalMedicineLab/egpmrp': 2,\n",
       " 'github.com/micahgearhart/exotools': 1,\n",
       " 'github.com/wrightrc/flowTime': 1,\n",
       " 'github.com/charlesfarber/GWAS_networks': 1,\n",
       " 'github.com/Mark-Kramer/Spike-Ripple-Detector-Method': 1,\n",
       " 'github.com/PNNL-Comp-Mass-Spec/LIQUID': 1,\n",
       " 'github.com/shahmoradi/cordiv': 1,\n",
       " 'compomics.github.io/projects/peptide-shaker.html': 6,\n",
       " 'compomics.github.io/projects/searchgui.html': 3,\n",
       " 'github.com/Preparation-Publication-BD2K/db_compress': 1,\n",
       " 'github.com/LiuBioinfo/iMapSplice': 1,\n",
       " 'github.com/craigyk/emtools': 1,\n",
       " 'github.com/elifesciences-publications/emtools': 1,\n",
       " 'github.com/lauralewis/emergenceLPs': 1,\n",
       " 'github.com/elifesciences-publications/emergenceLPs': 1,\n",
       " 'github.com/toddleroux/health_prof_educ': 1,\n",
       " '/github.com/EACcodes/EmbeddingIntegralGenerator': 1,\n",
       " 'github.com/syubukun/Sp34': 1,\n",
       " 'github.com/FoodSafeCookCo/TwitterStream-': 1,\n",
       " 'github.com/rpremraj/mailR': 1,\n",
       " 'github.com/PacificBiosciences/Bioinformatics-Training/wiki/Circularizing-and-trimming': 4,\n",
       " 'uwpr.github.io/Lorikeet': 1,\n",
       " 'github.com/francescojm/CRISPRcleanR': 1,\n",
       " 'github.com/cancerit/pyCRISPRcleanR': 1,\n",
       " 'github.com/raerose01/deconstructSigs': 3,\n",
       " 'github.com/gcyuan/SCUBA': 2,\n",
       " 'github.com/jw156605/SLICER': 3,\n",
       " 'github.com/hubert-pop/signasel': 1,\n",
       " 'github.com/vickytorrance/uAUGs': 1,\n",
       " 'github.com/jordan-g/Irrelevance-by-Inhibition': 1,\n",
       " 'samtools.github.io/hts-specs/VCFv4.3.pdf': 3,\n",
       " 'github.com/therneau/survival': 5,\n",
       " 'github.com/epam/ketcher': 1,\n",
       " 'github.com/eiriniar/gleason_CNN': 1,\n",
       " 'github.com/MihChi/SIMELeD.git': 1,\n",
       " 'github.com/biopython/biopython.github.io': 3,\n",
       " 'czarrar.github.io/connectir': 3,\n",
       " 'wiki.github.com/asad/VFLib': 1,\n",
       " 'github.com/mailund/IMCoalHMM': 2,\n",
       " 'github.com/caulilin/lncRNA_Finder': 1,\n",
       " 'github.com/hilldr/Tsai_LGR5_endoderm_2015': 1,\n",
       " 'github.com/Victorian-Bioinformatics-Consortium/nesoni': 18,\n",
       " 'github.com/zacf/UM-gradient-metagenome-databases': 2,\n",
       " 'github.com/lskatz/lskScripts': 1,\n",
       " 'github.com/diazlab/scell': 1,\n",
       " 'github.com/Teichlab/celloline': 1,\n",
       " 'pachterlab.github.io/kallisto/about.html': 2,\n",
       " 'github.com/lengning/OEFinder': 3,\n",
       " 'github.com/PMBio/scLVM': 2,\n",
       " 'github.com/brentp/combat.py': 1,\n",
       " 'github.com/shka/R-SAMstrt': 2,\n",
       " 'github.com/MarioniLab/Deconvolution2016': 2,\n",
       " 'github.com/JustinaZ/pcaReduce': 2,\n",
       " 'lvdmaaten.github.io/tsne': 10,\n",
       " 'github.com/epierson9/ZIFA': 5,\n",
       " 'hms-dbmi.github.io/scde': 6,\n",
       " 'github.com/RGLab/MAST': 7,\n",
       " 'github.com/lanjiangboston/GiniClust': 1,\n",
       " 'github.com/govinda-kamath/clustering_on_transcript_compatibility_counts': 1,\n",
       " 'github.com/BatzoglouLabSU/SIMLR': 2,\n",
       " 'github.com/linnarsson-lab/BackSPIN': 5,\n",
       " 'cole-trapnell-lab.github.io/monocle-release': 7,\n",
       " 'github.com/kieranrcampbell/embeddr': 2,\n",
       " 'markummitchell.github.io/engauge-digitizer': 14,\n",
       " 'github.com/scipy/scipy/blob/v0.15.1/scipy/stats/stats.py': 1,\n",
       " 'github.com/hverbruggen/PMT': 1,\n",
       " 'wurmlab.github.io/tools/genevalidator': 1,\n",
       " 'github.com/FunctionLab/giant-api': 1,\n",
       " 'github.com/whcsu/RRotSF': 1,\n",
       " 'github.com/dg13/rasqual': 1,\n",
       " 'github.com/ChildMindInstitute/Biblio_Reader/blob/165ddc56779a5e55149184a0f95b7c14874cf0c5/biblio_reader/text_tools/text_tools.py': 1,\n",
       " 'github.com/abreschi/shinySpecClust': 1,\n",
       " 'github.com/isb-cgc/ISB-CGC-pipelines': 1,\n",
       " 'github.com/FernandoLab/2016_Paz_et_al_Dairy_Breeds': 1,\n",
       " 'github.com/pzylstra/ffm_cpp/tree/Zylstra2016': 1,\n",
       " 'github.com/VeraPancaldi/Heterosis-GRN-simulation': 1,\n",
       " 'github.com/sclamons/PED': 1,\n",
       " 'github.com/caballero/RepeatLandscape': 1,\n",
       " 'philogb.github.io/jit': 3,\n",
       " 'github.com/nh13/DWGSIM': 15,\n",
       " 'github.com/sbillin/IMLP': 1,\n",
       " 'github.com/aflaxman/dismod_pde': 1,\n",
       " 'github.com/craddm/microDetect': 1,\n",
       " 'github.com/minillinim/sam2fpkg': 1,\n",
       " 'github.com/dpessoaIGC/Dose-Invariant-Susceptibility-Estimator': 1,\n",
       " 'github.com/camwebb/tree-of-trees/blob/master/megatrees/R20120829.new': 2,\n",
       " 'github.com/brwnj/seqanno': 1,\n",
       " 'mustache.github.io': 1,\n",
       " 'github.com/biomedinfo/sugar': 1,\n",
       " 'github.com/OHDSI/KnowledgeBase': 2,\n",
       " 'github.com/hallamlab/MetaPathways': 1,\n",
       " 'github.com/UH-Bioinformatics/hBayeSSC': 1,\n",
       " 'github.com/penamiller/mmJ': 1,\n",
       " 'github.com/EliMor/LabScripts': 1,\n",
       " 'rdrr.io/github/ameztegui/Plasticity': 1,\n",
       " 'github.com/KiddLab/insertion-genotype': 1,\n",
       " 'github.com/MarcoDiSalvo90/RhoTermPredict': 1,\n",
       " 'github.com/LabForComputationalVision/matlabPyrTools': 1,\n",
       " 'github.com/santa-dev/santa-sim': 1,\n",
       " 'bbolker.github.io/mixedmodels-misc/glmmFAQ.html': 2,\n",
       " 'github.com/dcnieho/UDPMultiCast': 1,\n",
       " 'gemtools.github.io': 6,\n",
       " 'github.com/legrandp/xdsme': 6,\n",
       " 'github.com/reconhub/branchr': 2,\n",
       " 'github.com/OceanParcels/AntarcticPeninsulaPlastic': 1,\n",
       " 'github.com/roidtman/NatComm_dengue_China': 1,\n",
       " 'github.com/WashburnLab/Topological-score-TopS': 1,\n",
       " 'stnava.github.io/ANTsR': 7,\n",
       " 'github.com/linyuwangPHD/RNA-Secondary-Structure-Database': 1,\n",
       " 'github.com/benjamincjackson/dsim_variant_pipeline_ref_v2.02.git': 1,\n",
       " 'github.com/Dazzid/DataToRepovizz/tree/myo_to_repovizz/myo_recordings': 1,\n",
       " 'github.com/mitmedialab/BEEP': 2,\n",
       " 'github.com/rraadd88/beditor': 1,\n",
       " 'github.com/plaurent/mcpstdp': 1,\n",
       " 'github.com/PacificBiosciences': 6,\n",
       " 'github.com/deweylab/RSEM': 8,\n",
       " 'github.com/ChenFengling/HiCDB': 1,\n",
       " 'github.com/ChenFengling/RHiCDB': 1,\n",
       " 'github.com/weng-lab/Diff_signal_pred': 1,\n",
       " 'github.com/GiugnoLab/PanDelos': 1,\n",
       " 'github.com/ekg/vcflib': 13,\n",
       " 'github.com/Donders-Institute/Stereo-gaze-tracking': 1,\n",
       " 'github.com/bieneSchwarze/PaperModelingSelectionBias': 1,\n",
       " 'github.com/knowlabUnimelb/SMALLN': 1,\n",
       " 'github.com/gkichaev/PAINTOR_V3.0': 1,\n",
       " 'github.com/FormentoEmanuele/MuscleSpindleCircuitsModel.git': 1,\n",
       " 'github.com/plhdb/plhdb': 1,\n",
       " 'github.com/tgouhier/biwavelet': 7,\n",
       " 'github.com/lsy917/chenlimin': 1,\n",
       " 'github.com/obarquero/ECG_quality': 1,\n",
       " 'samtools.github.io/htsjdk': 5,\n",
       " 'github.com/gbouvignies/chemex': 5,\n",
       " 'github.com/sandain/pigeon': 3,\n",
       " 'github.com/harrelfe/Hmisc': 5,\n",
       " 'gist.github.com/kkmehta/0a1dbe272d7ff2aed011': 1,\n",
       " 'github.com/pezmaster31/bamtools': 18,\n",
       " 'github.com/mlizhangx/TCR-3D': 1,\n",
       " 'lp2.github.io/yeast-crispri': 4,\n",
       " 'spineml.github.io': 3,\n",
       " 'github.com/dvbrown/FijiMacros': 1,\n",
       " 'github.com/m-fofana/TB-PZA-model.git': 1,\n",
       " 'www.github.com/nedelec/cytosim': 1,\n",
       " 'github.com/sylvainforet/libngs': 5,\n",
       " 'github.com/sccn/SNAP': 3,\n",
       " 'pachterlab.github.io/sleuth': 4,\n",
       " 'github.com/jakejh/metapredict': 1,\n",
       " 'github.com/jdreyf/fat-exosome-microrna': 1,\n",
       " 'github.com/kdeck/TTVFast': 1,\n",
       " 'github.com/ericagol/TTVFaster': 1,\n",
       " 'github.com/smirik/mercury': 1,\n",
       " 'github.com/dylan-jayatilaka/tonto': 2,\n",
       " 'github.com/guigolab/chip-nf': 1,\n",
       " 'github.com/adwanner/PyKNOSSOS': 6,\n",
       " 'github.com/cole-trapnell-lab/monocle-release': 2,\n",
       " 'github.com/VRam142/combinatorialHiC': 1,\n",
       " 'opencobra.github.io/cobratoolbox': 5,\n",
       " 'github.com/maanrachid/Codes/blob/master/AAPSP.tar': 1,\n",
       " 'github.com/biochem-fan/eiger2cbf': 1,\n",
       " 'geodacenter.github.io': 4,\n",
       " 'github.com/owensgl/sunflower_heterosis_2018': 1,\n",
       " 'github.com/TransDecoder': 7,\n",
       " 'github.com/NCBI-Hackathons/NovoGraph': 1,\n",
       " 'github.com/LipengNing/RBF-Propagator': 3,\n",
       " 'github.com/CCBR/Pipeliner': 3,\n",
       " 'github.com/phagewebufpa/API': 1,\n",
       " 'github.com/lichen-lab/SICS': 1,\n",
       " 'github.com/mariFelix/deconvoNorm': 1,\n",
       " 'rescience.github.io': 2,\n",
       " 'github.com/bslaby/scripts': 3,\n",
       " 'github.com/zhangyimc/DeepSplice': 1,\n",
       " 'github.com/bcbio/bcbio-nextgen': 3,\n",
       " 'github.com/stamatak/standard-RAxML': 17,\n",
       " 'github.com/Pbdas/epa-ng': 1,\n",
       " 'github.com/CshlSiepelLab/LINSIGHT': 3,\n",
       " 'github.com/cancersysbio/BreastCancerITH': 1,\n",
       " 'github.com/alexandrebarachant/covariancetoolbox': 2,\n",
       " 'github.com/philipwijesinghe/Acoustic-trap-LSFM': 1,\n",
       " 'github.com/wkool/demandavoidance': 1,\n",
       " 'github.com/dunfieldlab/mg_wrapser': 2,\n",
       " 'github.com/jsalojar/DLCpar_visualize': 1,\n",
       " 'github.com/jaak-s/BayesianDataFusion.jl': 1,\n",
       " 'github.com/smithng1215': 1,\n",
       " 'github.com/jacoblevine/PhenoGraph': 3,\n",
       " 'github.com/AllonKleinLab/SPRING': 2,\n",
       " '4github.com/CMU-Perceptual-Computing-Lab/openpose': 1,\n",
       " 'github.com/Onogi/MultivariateFtest': 1,\n",
       " 'github.com/Xiaomu-Shan/Experiments-of-iFBEL': 1,\n",
       " 'github.com/BioinfoCP/visual-features-soybean-vigor': 1,\n",
       " 'github.com/dlrice/evoker-lite': 1,\n",
       " 'github.com/uricchio/PASTEL': 1,\n",
       " 'github.com/gadluru/Multiview-myocardial-perfusion-with-radial-SMS': 1,\n",
       " 'github.com/svijayakumar32/multi-objective-optimization-of-gut-community-metabolic-model': 1,\n",
       " 'github.com/akiyamalab/NRLMFb': 1,\n",
       " 'github.com/BHSAI/DNN-liverTox': 1,\n",
       " 'github.com/guofei-tju/Cancer-subtypes': 1,\n",
       " 'mzmine.github.io': 12,\n",
       " 'github.com/JoseAlanis/Supplementary_Soc_ERN': 1,\n",
       " 'github.com/eshoko/COMET': 1,\n",
       " 'github.com/madeluis/GENIST/commits/master': 1,\n",
       " 'github.com/YinLiLin/R-CMplot': 3,\n",
       " 'github.com/costlysignalling/Mate-choice-consistency-2': 1,\n",
       " 'github.com/stonezwr/SpiKL-IP': 1,\n",
       " 'github.com/JeromeAmbroise/Pathogenomics': 1,\n",
       " 'pachterlab.github.io/eXpress/index.html': 1,\n",
       " 'github.com/RausellLab/NCBoost': 1,\n",
       " 'xia2.github.io/index.html': 1,\n",
       " 'github.com/KrisJanssen/QuickCLEM': 1,\n",
       " 'github.com/delibasanta/Mexican-salamanders.git': 1,\n",
       " 'github.com/jeffeaton/currpreg-hiv-trends': 1,\n",
       " 'github.com/ANGSD': 1,\n",
       " 'github.com/mourisl/Lighter': 3,\n",
       " 'github.com/aestrivex/cvu/wiki': 1,\n",
       " 'cytoscape.github.io/cytoscape.js': 16,\n",
       " 'github.com/OscarKrijgsman/focalCall': 1,\n",
       " 'github.com/ahhurlbert/species-energy-simulation': 1,\n",
       " 'github.com/popeb/MCP05': 1,\n",
       " 'demotu.github.io/posts/prediction-ellipse-ellipsoid.html': 1,\n",
       " 'github.com/emarcos/beta_sheet': 1,\n",
       " 'github.com/leonfrench/HumanHabenula': 1,\n",
       " 'ccbr.github.io/Pipeliner/Tools/NGS_QC_Toolkit.html': 2,\n",
       " 'github.com/LCCN': 1,\n",
       " '1dstolz.github.io/epsych': 1,\n",
       " 'github.com/BlueFern/CoupledCells/releases/tag/v-4.0': 1,\n",
       " 'github.com/seaaroundus': 2,\n",
       " 'github.com/marliekevk/Integrating-Educational-Knowledge': 1,\n",
       " 'pachterlab.github.io/eXpress': 1,\n",
       " 'github.com/mlloyd23/SSWD_16S_analysis': 1,\n",
       " 'github.com/nbluoc/kk': 1,\n",
       " 'github.com/bioinformatics-sannio/lncrna-homologs': 1,\n",
       " 'github.com/OHDSI/StudyProtocols/tree/master/AhasHfBkleAmputation': 1,\n",
       " 'github.com/FernandoHartwig/AssortativeMating_Scripts': 1,\n",
       " 'github.com/hms-dbmi/spp': 2,\n",
       " 'github.com/nci/drishti': 1,\n",
       " 'github.com/perishky/meffil': 4,\n",
       " 'github.com/willyrv/nssc-tools': 1,\n",
       " 'github.com/dvera/ksift': 1,\n",
       " 'github.com/zippav/Jellyfish-2': 1,\n",
       " 'github.com/laurentnoe/yass': 1,\n",
       " 'github.com/shimbalama/screen_assembly': 1,\n",
       " 'github.com/katholt/srst2/tree/master/data': 2,\n",
       " 'github.com/TheWrightonLab': 1,\n",
       " 'github.com/joegage/GWAS_AUC': 1,\n",
       " 'github.com/danoreper/mnp2018.git': 1,\n",
       " 'github.com/Zhongyun-Huang/G3-2018-200605': 1,\n",
       " 'github.com/anainesvs/glioblastoma': 1,\n",
       " 'github.com/drh-uth/DataFakehouse': 1,\n",
       " 'github.com/jacobhurst/definetherain': 1,\n",
       " 'github.com/qichao1984/GSMer': 1,\n",
       " 'github.com/JulienLajugie/ReplicationTimingSimulation': 1,\n",
       " 'github.com/ofedrigo/pyEnrichment': 2,\n",
       " 'dschrimpf.github.io/randi3': 1,\n",
       " 'github.com/mlupien/ABC': 1,\n",
       " 'github.com/jonsonfox/HaploShare': 1,\n",
       " 'www.github.com/molgenis': 1,\n",
       " 'github.com/PF2-pasteur-fr/SARTools': 4,\n",
       " 'github.com//tumorml/TumorML-schema': 1,\n",
       " 'cytoscape.github.io': 1,\n",
       " 'madsalbertsen.github.io/mmgenome': 6,\n",
       " 'github.com/jstjohn/SimSeq': 12,\n",
       " 'github.com/srnas/barnaba': 7,\n",
       " 'github.com/SEEG-Oxford/prioritisation': 1,\n",
       " 'github.com/StoreyLab/gcat': 1,\n",
       " 'github.com/rachelss/SISRS/releases': 1,\n",
       " 'github.com/GriesbeckLab/ColonySelection': 1,\n",
       " 'github.com/GraceYoon/ISIS_EN': 1,\n",
       " 'github.com/cubocube/website': 1,\n",
       " 'github.com/svazzole/sparsevar': 1,\n",
       " 'github.com/DynaSim': 1,\n",
       " 'github.com/jsherfey/ACd_model': 1,\n",
       " 'github.com/knightlab-analyses/bloom-analyses': 1,\n",
       " 'github.com/tseemann/wombac/blob/master/bin/wombac-shred_fasta': 1,\n",
       " 'github.com/tonomartin2/MITOCLASS.1': 2,\n",
       " 'github.com/DenefLab/microcystis-oligotypes': 1,\n",
       " 'github.com/NanoSPD/NanoSPD': 1,\n",
       " 'github.com/iontorrent/TS/tree/master/Analysis/TMAP': 9,\n",
       " 'mi-sim.github.io': 1,\n",
       " 'github.com/ClandininLab/2pstim': 1,\n",
       " 'github.com/Ensembl/treebest': 4,\n",
       " 'github.com/apalania/consMoment': 1,\n",
       " 'github.com/dsarov/MGAP---Microbial-Genome-Assembler-Pipeline': 3,\n",
       " 'github.com/tseemann/VelvetOptimiser': 10,\n",
       " 'github.com/grev-uv/hpg-methyl': 1,\n",
       " 'github.com/PacificBiosciences/pbdagcon': 8,\n",
       " 'github.com/PacificBiosciences/FALCON-integrate': 6,\n",
       " 'github.com/grisoniFr/whales': 1,\n",
       " 'github.com/dekkerlab/cworld-dekker': 10,\n",
       " 'github.com/drpowell/degust': 3,\n",
       " 'github.com/SUNCAT-Center/catmap': 1,\n",
       " 'www.github.com/nextgenusfs/funannotate': 1,\n",
       " 'github.com/EricFournier3/PythonGenomicTools': 2,\n",
       " 'github.com/mw55309/ideel': 2,\n",
       " 'github.com/toreerdmann/edar': 1,\n",
       " 'github.com/alanavc/rodtracker': 1,\n",
       " 'github.com/AgreementMakerLight': 1,\n",
       " ...}"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict(url_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_urls = [k for k,v in url_counts.items() if v==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_url_filt = np.isin(urls_in_papes.url, unique_urls)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "#urls_in_papes.loc[unique_url_filt, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids_to_check = np.sort(np.unique(urls_in_papes.pmcid[unique_url_filt].values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['2712337', '2795760', '2820491', ..., '6484778', '6485367',\n",
       "       '6485654'], dtype=object)"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids_to_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Positive feature heuristic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if abstract section or title \n",
    "positive_feats = ['code', 'custom', 'develop', 'documentation', \n",
    "                  'download', 'host', 'license',\n",
    "       'manual', 'publish', 'raw', 'reproduc', 'script', 'source', 'update', 'present']\n",
    "positive_feats = positive_feats + [f.capitalize() for f in positive_feats]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [],
   "source": [
    "hit_table = {}\n",
    "\n",
    "for pmcid in np.unique(urls_in_papes.pmcid.values):\n",
    "    textt = ' '.join([t['text'] for t in pmcid_to_doc[pmcid]['passages'] if 'github' in t['text']])\n",
    "    hits_vec = [(f in textt) for f in positive_feats]\n",
    "    hit_table[pmcid] = np.sum(hits_vec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {},
   "outputs": [],
   "source": [
    "hit_table = pd.DataFrame.from_dict(hit_table, orient = 'index', columns = ['score'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [],
   "source": [
    "no_hits = hit_table.index[hit_table.score == 0]\n",
    "has_hits = hit_table.index[hit_table.score > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {},
   "outputs": [],
   "source": [
    "# [t for t in pmcid_to_doc[no_hits[6]]['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "metadata": {},
   "outputs": [],
   "source": [
    "# [t for t in pmcid_to_doc[has_hits[15]]['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3946"
      ]
     },
     "execution_count": 348,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(hit_table.score == 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "False Positives\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "False Negatives\n",
    "\n",
    "'3643894'\n",
    "[{'text': 'We next considered asymmetries in each of the regions. After adjusting alpha for multiple comparisons, one sample t-tests on the AQ scores revealed significant rightward asymmetries in cortical thickness for 10 of the 12 regions including orbital and dorsal prefrontal cortex, pre- and post-central gyri, superior parietal lobe, supramaginal gyrus, superior temporal gyrus, middle temporal gyrus, inferior frontal gyrus and posterior superior temporal gyrus (see Table 1). The rightward asymmetries in grey matter thickness based on the region of interest approach were largely confirmed when using a voxel-based approach (see http://stnava.github.com/WHopkinsNHP/).',\n",
    "  'offset': 22029,\n",
    "  'relations': [],\n",
    "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
    "  'sentences': [],\n",
    "  'annotations': []}]\n",
    "  \n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3643894'"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_hits[2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'Regional and Hemispheric Variation in Cortical Thickness in Chimpanzees (Pan troglodytes)',\n",
       "  'offset': 0,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '5248',\n",
       "   'article-id_manuscript': 'NIHMS458577',\n",
       "   'name_1': 'surname:Avants;given-names:Brian B.',\n",
       "   'license': '\\n          This file is available for text mining. It may also be used consistent with the principles of fair use under the copyright law.\\n        ',\n",
       "   'issue': '12',\n",
       "   'article-id_doi': '10.1523/JNEUROSCI.2996-12.2013',\n",
       "   'section_type': 'TITLE',\n",
       "   'name_0': 'surname:Hopkins;given-names:William D.',\n",
       "   'article-id_pmid': '23516289',\n",
       "   'volume': '33',\n",
       "   'article-id_pmc': '3643894',\n",
       "   'fpage': '5241',\n",
       "   'year': '2013',\n",
       "   'type': 'front'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Recent advances in structural magnetic resonance imaging technology and analysis now allows for accurate in vivo measurement of cortical thickness, an important aspect of cortical organization that has historically only been conducted on post-mortem brains. In this study, for the first time, we examined regional and lateralized cortical thickness in a sample of 71 chimpanzees for comparison to previously reported findings in humans. We also measured grey and white matter volumes for each subject. The results indicated that chimpanzees showed significant regional variation in cortical thickness with lower values in primary motor and sensory cortex compared to association cortex. Furthermore, chimpanzees showed significant rightward asymmetries in cortical thickness for a number of regions of interest throughout the cortex and leftward asymmetries in white but not grey matter volume. We also found that total and region specific cortical thickness was significantly negatively correlated with white matter volume. Thus, chimpanzees with greater white matter volumes had thinner cortical thickness. The collective findings are discussed within the context of previous findings in humans and theories on the evolution of cortical organization and lateralization in primates.',\n",
       "  'offset': 90,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'In modern structural brain imaging protocols, cortical volume (CV) is a main measure of interest from the standpoint of understanding disease progression, development, and aging. CV is made up of cortical thickness, surface area and the white matter that underlies connectivity between regions. In terms of brain evolution, the human brain is roughly three times larger than that of chimpanzees, the closest living relative to humans. In addition, humans have disproportionate increases in grey and particularly white matter volume compared to other primates. It has been suggested that the relative increase in white matter volume and gyrification are prominent in brain regions, such as the prefrontal cortex and temporal lobes, that are thought to underlie certain human cognitive specializations.',\n",
       "  'offset': 1374,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'An important aspect of cortical organization that has not been examined from a comparative primate perspective is cortical thickness. Studies of post-mortem human brains have shown that there are significant regional and lateralized differences in cortical thickness. For instance, in post-mortem material, cortical thickness varies from approximately 2 to 4 mm across the human cerebrum, with thinner cortex found in primary motor and sensory regions compared to thicker cortex in association cortex. Recent advances in in vivo imaging technologies employed with human brains have largely validated these findings, when derived fromT1-weighted structural magnetic resonance imaging (MRI) scans. In terms of lateralization, small but significant asymmetries in cortical thickness have been found in the human brain including leftward biases for the precentral gyrus, middle frontal, anterior temporal and superior parietal lobes. Significant rightward biases have been reported for inferior posterior temporal lobe and inferior frontal gyrus with some evidence that these effects are influenced by gender.',\n",
       "  'offset': 2175,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Cortical thickness is relatively conserved in primate brain organization. Whether thickness varies in relation to different cortical regions and lateralization in closely related primates, such as chimpanzees, is virtually unknown but may provide important information on primate brain evolution and the mechanisms that underlie human cognitive and motor specializations. The aim of this study was to quantify regional and lateralized variation in cortical thickness in chimpanzees. In addition, we also computed the surface area and cortical grey and white matter volumes. and others have reported that there is an inverse association between white matter volume and cortical thickness during human development. More recently, in humans from different age groups, has reported that there is a positive association between white matter volume and cortical surface and significant negative associations between these two measures and cortical thickness. has hypothesized that, during development, white matter grows outward and expands the outer surface of the cortex much like a balloon. The consequence of this expansion is that outer grey matter surface has to thin out in order to cover the expanding surface area. Based on the theory proposed by, we hypothesized that, like humans, chimpanzee subjects with either larger white matter volumes or cortical surface aras would have thinner cortical thickness values.',\n",
       "  'offset': 3281,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Methods',\n",
       "  'offset': 4698,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Subjects',\n",
       "  'offset': 4706,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Magnetic resonance images (MRI) were obtained in 71 captive chimpanzees (23 males, 48 females). The subjects ranged in age from 11 to 55 years (Mean = 22.34. s.d. = 11.11) and were all housed at the Yerkes National Primate Research Center. To evaluate the potential effect of handedness on asymmetries in cortical thickness, we also compared right-, left and ambiguously-handed individuals in the analyses of AQ scores. Handedness was measured on a task that requires coordinated bimanual actions, referred to as the tube task. Binomial z-scores based on the frequency of right and left hand use were used to classify the hand preference of the chimpanzees. Apes with z-scores > 1.95 or 1.95 were classified as right- and left-handed. Chimpanzees with z-scores >−1.96 or < 1.96 were classified as ambiguously-handed (i.e., did not have a significant hand preference) Within the sample, there were 24 left-,13 ambiguously-and 34 right-handed chimpanzees All scanning procedures were approved by the University Institutional Animal Care and Use committee and followed the guidelines outlined by the Institute of Medicine.',\n",
       "  'offset': 4715,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Magnetic Resonance Image Collection',\n",
       "  'offset': 5837,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'All chimpanzees were scanned during one of their yearly physical examinations. MRI scans followed standard procedures at the YNPRC. Subjects were first immobilized with ketamine (10 mg/kg) or telazol (3–5mg/kg) and subsequently anaesthetized with propofol (40–60 mg/(kg/h)). Each subject was then transported to the MRI scanner and placed in a supine position in the scanner with their head in a human-head coil. Upon completion of the MRI, chimpanzees were temporarily singly housed for 2–12 hours to permit close monitoring and safe recovery from the anesthesia prior to return to the home social group. Chimpanzees were scanned using a 3.0 Tesla scanner (Siemens Trio, Siemens Medical Solutions USA, Inc., Malvern, Pennsylvania, USA). T1-weighted images were collected using a three-dimensional gradient echo sequence (pulse repetition = 2300 ms, echo time = 4.4 ms, number of signals averaged = 3, matrix size = 320 × 320, with 0.6 × 0.6 × 0.6 resolution).',\n",
       "  'offset': 5873,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Image Processing and Cortical Thickness Measurement',\n",
       "  'offset': 6843,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The MRI scans were processed in a pipeline that required several steps including a) skull stripping using FSL, diffeomorphic registration to a symmetric population-specific template chimpanzee brain and subsequent probabilistic segmentation of the T1-weighted images into grey matter, white matter and CSF following procedures and software that have been described in detail elsewhere.',\n",
       "  'offset': 6895,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'After brain extraction, we used the openly available Advanced Normalization Tools (ANTS) (http://www.picsl.upenn.edu/ANTS/) to perform multivariate normalization and structure-specific processing of our data. ANTs encodes current best practices in image registration, optimal template construction and segmentation and is scalable to large-scale, distributed computing environments. ANTs cross-sectional studies deform each individual dataset into a standard local template space and/or a canonical stereotactic coordinate system. The core processing maps T1 structural MRI to an optimal template space, which is defined as the population-specific, unbiased average shape and appearance image derived from a representative population. The average template was constructed to optimally represent both the original and a flipped version of the dataset such that the final template is symmetric about the midsagittal plane. The coordinate deformations themselves are smooth and invertible, that is, diffeomorphic – neuroanatomical neighbors remain neighbors under the mapping. At the same time, the algorithms used to create these deformations are biased towards the reference space chosen to compute the mappings. Moreover, these topology-preserving maps capture the large deformation necessary to aggregate populations of images in a common space. Recent evaluation studies suggest that ANTs-based normalization is currently the most stable and reliable method available. After defining the template image to target image coordinate transformation, we employ template-based priors and N4 inhomogeneity filed correction to accurately segment cortical gray matter segmentation and perform cortical parcellation. After these initial processing steps, the diffeomorphic registration-based cortical thickness (DiReCT) method is used to compute cortical thickness.',\n",
       "  'offset': 7281,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'DiReCT uses the segmentation probability images to compute a continuous voxel-wise estimate of cortical thickness. DiReCT is unique in that it exploits tissue segmentation probability maps to identify a maximum likelihood correspondence between the white matter surface and the outer gray matter surface where the correspondence mapping is constrained to be spatially regular, differentiable and invertible, i.e. diffeomorphic. As a consequence, DiReCT thickness estimates incorporate both shape constraints and subtle probabilistic information about the likely position of sulci that may not be visible in a hard segmentation. Thus, DiReCT is a robust image-based technique for identifying voxel-wise and regional thickness information. An independent implementation and validation of this method showed that it is competitive with Freesurfer, a commonly used program for estimating cortical thickness. We note, however, that the implementation used by was never itself directly compared to that available in the ANTs toolkit and we suspect that differences between the Clarkson et al implementation and our own gold standard implementation may exist.',\n",
       "  'offset': 9141,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'As has been done in some studies of cortical thickness in humans and monkeys, we used a seed or region-of-interest approach to quantify GM thickness in 12 select areas of interest (see Table 1) including the a) dorsal, mesial, and orbital prefrontal cortex b) superior, middle and inferior temporal gyri c) pre- and post-central gyri d) inferior frontal gyrus e) posterior superior temporal gyrus f) supramarginal gyrus and g) superior parietal lobe. The landmarks used to define each region of interest are provided below and they were selected for theoretical and pragmatic reasons. Specifically, there has been considerable comparative interest in the evolution of the prefrontal, temporal and parietal cortex and therefore we focused on these regions. Similarly, there has been significant interest in the evolution of cortical organization and lateralization in the homologs to Broca’s and Wernicke’s area in chimpanzees and this was our reasoning for quantifying the inferior frontal gyrus and posterior superior temporal gyri. Finally, we also included the pre- and post-central gyri as regions of interest because they are primary motor and sensory cortex and we hypothesized that cortical thickness would be lower in these regions compared to the others. The region of interest masks were drawn on the chimpanzee template brain using a mouse-controlled pointer and then transformed back to the individual GM thickness maps for each subject, using the inverse matrix of the original registration. The masks were then applied to the individual subjects’ GM thickness map to derive average thickness measures for each of the 12 regions within each hemisphere. We also computed an average cortical thickness measure for each hemisphere. This was done by drawing a mask that covered the entire left or right hemisphere, excluding the brain stem and cerebellum. We note that, because the template is symmetric, the masks only had to be drawn on one hemisphere of the template brain and could then be transformed to label both left and right sides of the individual subject’s brain.',\n",
       "  'offset': 10294,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Regions of Interest',\n",
       "  'offset': 12387,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Prefrontal Cortex (PFC)',\n",
       "  'offset': 12407,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We adopted the landmarks used by in defining the orbital, dorsal and mesial prefrontal cortex in serial coronal images (see Figure 1). Beginning posteriorly when the frontal orbital (FO) sulcus was first visible, for the orbital PFC, a line was drawn from the surface of the FO sulcus to its most medial point. A second line was then drawn from the medial point of FO to the tip of the rectus gyrus, which was then followed along the inferior outer surface of the cortex until intersecting with the lateral point of the FO sulcus. If the caudate and putamen were visible on the images, the line extending from the medial FO sulcus to the rectrus gyrus excluded these regions. The dorsal PFC was outlined by laterally tracing on the surface of the brain from FO to the upper, dorsal most mesial point. From there, a line was drawn that connected the most medial point of all frontal sulci between the most mesial point and the FO sulcus. Finally, the mesial PFC was defined from the grey matter tip of the rectus gyrus to the upper dorsal end of the mesial surface, which included portions of the superior frontal gyrus but omitted any tissue belonging to the anterior cingulate cortex and subgenu region.',\n",
       "  'offset': 12431,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Temporal Lobe Cortex',\n",
       "  'offset': 13636,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We adopted similar landmarks to those used by and quantified the superior (STG), middle (MTG) and inferior temporal gyri (ITG) (see Figure 1). These three regions were primarily traced in the coronal plane. The superior and inferior borders of the STG were the sylvian fissure (SF) and superior temporal sulcus (STS). For all images in which the SF and STS were clearly visible, the entire gyrus was outlined with a mouse-driven cursor from the most medial point of the SF, laterally to the surface of the brain and then to the most medial point of the STS. The superior and inferior borders of the MTG were the STS and inferior temporal sulcus (ITS). Like the STG, for all images in which the STS and ITS were clearly visible, the entire gyrus was outlined with a mouse-driven cursor from the most medial point of the STS, laterally to the surface of the brain, and then to the most medial point of the ITS. The superior and inferior borders for the ITG were the ITS and the lateral occipital temporal sulcus. The entire gyrus between the medial border of the ITS to the lateral surface, then to the most medial point of the lateral occipital temporal sulcus were traced on images in which both sulci were visible.',\n",
       "  'offset': 13657,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Superior Parietal lobe (SPL)',\n",
       "  'offset': 14873,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'In the transverse plane, the lateral border was the intraparietal sulcus (IPS) and the medial border was the superior parietal sulcus (SPS) (see Figure 2). On serial transverse images in which the IPS and SPS could be identified, the gyrus between these two sulci was traced. On each image, the IPS sulcus was traced from its most anterior to posterior point. A line was then drawn between the most posterior points of the IPS and SPS. We then followed the SPS from its most posterior to most anterior point. Lastly, a line was drawn between the two most anterior points of the IPS and SPS.',\n",
       "  'offset': 14902,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Supramarginal Gyrus (SMG)',\n",
       "  'offset': 15493,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'This region was primarily traced in the transverse plane and included the entire gyrus between the IPS, the superior post central sulcus (SPCS) and the lateral surface of the cortex (see Figure 2). The inferior border of the SMG was the point at which the IPS connected to the SPCS. Moving dorsally, the SMG was traced on all slices in which the SPCS and IPS were visible. On each image, a line was drawn starting at the most lateral edge of SPCS to the most posterior point of the IPS. Next, a line was traced along the lateral surface of the cortex between the most posterior point of IPS and the lateral point of the SPCS. Tracing continued on serial images until the IPS was no longer visible.',\n",
       "  'offset': 15519,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Precentral gyrus (PreCG)',\n",
       "  'offset': 16217,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'This region was traced in the sagittal and transverse planes. The PreCG was the entire gyrus between the central sulcus (CS) and the precentral sulcus, including the inferior (PCI) and superior (SPC) portions (see Figure 2). The CS was the posterior border of the PreCG. Ventrally, we used PCI sulcus as the anterior border. Moving dorally, PCI terminates and at that point we used the SPC sulcus as the anterior border of the PreCG. On each image, we drew a line from the most medial point of PCI to the lateral edge of the cortex. We then drew a line on the lateral edge of the cortex between PCI and CS, and then drew a line to the most medial point of the CS.',\n",
       "  'offset': 16242,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Post-central Gyrus (PostCG)',\n",
       "  'offset': 16906,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The anterior border of the PostCG was the CS and the posterior border was the post-central sulcus (PCS) (see Figure 2). For all images in the transverse plan in which the CS and IPCS could be seen, the entire gyrus was traced. On each image, a line was drawn from the most medial point of CS to the outer cortex. We then drew a line along the outer cortex from the lateral tip of CS to PCS, then traced along PCS to its most medial point.',\n",
       "  'offset': 16934,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Inferior Frontal Gyrus (IFG)',\n",
       "  'offset': 17373,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The anterior border of the IFG in chimpanzees is the fronto-orbital sulcus (FO), the posterior border is the precentral inferior sulcus (PCI) and superior border is the inferior frontal sulcus (IFS) (see Figure 3). Following methods previously employed by, we created a mask by tracing the entire gyrus between PCI and FO on serial dorsal to ventral images in the transverse plane until PCI terminated. On each image, we drew a line from the most medial point of PCI to the lateral surface of the cortex. We then drew a line along the surface of the cortex between PCI and FO and then drew a line to the most lateral point of FO. Finally, a line was drawn between the medial points of the FO and PCI sulci.',\n",
       "  'offset': 17402,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Posterior Superior Temporal Gyrus (pSTG)',\n",
       "  'offset': 18109,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The pSTG was quantified using methods and landmarks that have been employed previously in human and nonhuman primate brains. The anterior border of the pSTG was the first slice in the coronal plane in which the inferior branch of insular sulcus terminated. From that point, as described above, we traced the entire gyrus using SF as the superior border and the inferior border was the superior temporal sulcus (STS). The entire gyrus between the SF and STS were traced on all slices until the SF terminated posteriorly (see Figure 3).',\n",
       "  'offset': 18150,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Surface Area, Grey and White Matter Volume',\n",
       "  'offset': 18685,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We employ a classic expectation maximization framework (Atropos,available within ANTs) and N4 bias field correction (available in ITK and ANTs) to find a maximum a posteriori brain segmentation guided by probabilistic tissue priors. The tissue priors provide both initialization and spatial guidance to the tissue classification and the same segmentation maps were used to estimate white matter volume, gray matter volume and cortical thickness. Object masks were drawn over the entire left and right hemispheres for each subject and subsequently applied to the segmented grey (GM) and white matter (WM) volumes to calculate volumes within each half of the brain. The cerebellum and related brain stem structures were excluded from calculations in total grey and white matter volume. To compute the surface area, we used BrainVisa (BV), an automated program that determines the outer cortical folding of the left and right cerebral cortex excluding the cerebellum and brain stem.',\n",
       "  'offset': 18728,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Data Analysis',\n",
       "  'offset': 19708,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Average thickness measures for each region were obtained by summing the thickness measures for left and right hemispheres and dividing by two. Asymmetry quotients (AQ) for each region were derived following the formula [AQ = (R − L) / R + L) where R and L indicate the cortical thickness for the left and right region. Unless otherwise noted, parametric statistics were used for all analyses with alpha set to p < .05. Post-hoc tests, when necessary, were conducted using Tukey’s Honestly Significant difference (p < .05).',\n",
       "  'offset': 19722,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Results',\n",
       "  'offset': 20249,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Global and Regional Variation in Cortical Thickness',\n",
       "  'offset': 20257,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We initially tested for regional variation in cortical thickness using a mixed model analysis of variance. Region was the repeated measure while sex was the between group factor. The chimpanzees showed significant variation in grey matter thickness throughout the cortex F(11, 759)=144.71, p < .001 (Figure 4 & Table 1). Post-hoc analysis indicated that, as is the case in human brain, cortical thickness was lower in primary sensory and motor regions including the pre- and post-central gyri compared to all other cortical regions (see Figure 4). In addition, a significant two-way interaction was found between sex and region F(11, 759)=2.28, p < .009 (Figure 4). Subsequent post-hoc analysis indicated no significant differences between males and females for any of the regions.',\n",
       "  'offset': 20309,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Adjusted Thickness Measures',\n",
       "  'offset': 21091,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': ' found that regional and sex differences in cortical thickness can be influenced when adjustments are made for variation in brain volume. To address this specific issue, we conducted a second analysis examining sex and regional variation after adjusting for total grey and white matter volume. For this analysis, we re-ran the mixed model analysis of variance using the thickness values as the repeated measure while sex was the between group factor but we included brain volume as a covariate. As before, we found a significant main effect for region F(11, 759)=242.18, p < .001 and a significant two-way interaction between sex and region F(11, 759)=3.57, p < .001. Males were found to have thicker cortex than females for the pre-central and post-central gyri as well as the superior parietal lobe. In contrast, females were found to have thicker dorsal PFC than males.',\n",
       "  'offset': 21119,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Lateralization in Cortical Thickness',\n",
       "  'offset': 21992,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We next considered asymmetries in each of the regions. After adjusting alpha for multiple comparisons, one sample t-tests on the AQ scores revealed significant rightward asymmetries in cortical thickness for 10 of the 12 regions including orbital and dorsal prefrontal cortex, pre- and post-central gyri, superior parietal lobe, supramaginal gyrus, superior temporal gyrus, middle temporal gyrus, inferior frontal gyrus and posterior superior temporal gyrus (see Table 1). The rightward asymmetries in grey matter thickness based on the region of interest approach were largely confirmed when using a voxel-based approach (see http://stnava.github.com/WHopkinsNHP/).',\n",
       "  'offset': 22029,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Because population-level asymmetries in cortical thickness were found for 10 regions and most regions showed some level of skewness, we opted to use non-parametric Mann-Whitney U-tests to assess sex differences in the AQ scores. Sex differences were found for 2 regions including the inferior frontal gyrus (z = −2.04, p < .05) and pre-central gyrus (z = −2.73, p < .007). Females showed greater rightward asymmetries than males for both the inferior frontal gyrus and pre-central gyrus. All of the sex differences in AQ scores were small and, after correcting alpha for multiple comparisons (p < .004), none of the differences remained significant, so the results should be interpreted with some caution. Kruskal-Wallis tests comparing the AQ scores for each region between left, ambiguously- and right-handed chimpanzees failed to reveal any significant differences.',\n",
       "  'offset': 22696,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Surface Area (SA), Grey (GM) and White Matter (WM) Volume',\n",
       "  'offset': 23569,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'For the SA, GM and WM volumes, three repeated measures analyses of variance were performed with hemisphere (left, right) as the repeated measure and sex as the between group factor. The mean SA, GM and WM volumes for the left and right hemisphere in males and females are shown in Table 2. For grey matter, significant main effects for sex F(1, 69)= 21.71, p < .001 and hemisphere F(1, 69)=92.52, p < .001 were found. Males had significantly higher grey matter volumes than females and a significant rightward asymmetry was found. No other significant main effects or interactions were found. For white matter, a significant main effect for hemisphere was found F(1, 69)=14.49, p < .001. White matter volume was larger in the left compared to right hemisphere. No other significant main effects or interactions were found. Finally, for SA, a significant sex difference was found F(1, 69)= 13.01, p < .001 with males having a larger surface area then females.',\n",
       "  'offset': 23627,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Correlations Between Cortical Thickness, Surface, Grey and White Matter Volume',\n",
       "  'offset': 24586,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'In this analysis, we correlated the average GM, WM and SA volumes with the average cortical thickness for each region of interest (see Table 3). In this initial analysis, we used the average cortical thickness for the entire cortex instead of the thickness measures for each region in order to minimize the number of analyses and reduce the likelihood of Type I error. The results of these analyses are shown in Table 3. As can be seen, significant negative associations were found between average cortical thickness and GM and WM volumes but not surface area. Chimpanzees with thinner cortex had larger GM and, particularly, WM volumes. We also found significant positive associations between cortical surface area, GM and WM volumes.',\n",
       "  'offset': 24665,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'We also performed correlation analyses between the AQ measures for GM, WM, SA and average thickness and these are also shown in Table 3. A significant positive association was found between SA and WM AQ scores. Subjects with greater rightward asymmetries in WM had similarly greater rightward asymmetries in SA. Significant negative correlations were found between the AQ scores for WM volume and average cortical thickness. Subjects with greater rightward asymmetries in cortical thickness had greater leftward asymmetries in WM volume.',\n",
       "  'offset': 25401,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Discussion',\n",
       "  'offset': 25939,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'title_1', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The results reported here are the first evidence of regional variation and lateralization in cortical thickness in chimpanzees. There were three significant findings. First, as is the case with humans, chimpanzees show regional variation in cortical thickness with thinner cortex in primary motor and sensory cortex and thicker grey matter in association cortex including prefrontal cortex and portions of the parietal and temporal lobes. We also found some evidence of regional sex differences in cortical thickness. Second, our findings indicate that chimpanzees show robust population-level rightward asymmetries in grey matter thickness and a concomitant leftward asymmetry in white but not grey matter volume. Third, white but not grey matter volume was negatively correlated with cortical thickness. Thus, subjects with greater white matter volume had thinner grey matter thickness.',\n",
       "  'offset': 25950,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'After adjustment for overall brain size, we found four significant regional sex differences in cortical thickness in the chimpanzees including the pre- and post-central gyri superior parietal lobe and dorsal prefrontal cortex. For all 3 of the 4 regions, males had higher cortical thickness measures than the females. It should also be noted that males were found to have significantly larger grey but not white matter volumes (see Table 2). We had more than twice as many females than males in the sample and the age ranges were quite variable within our chimpanzee sample. However, when we matched all 23 males with females of the same age and re-ran the analyses, all the significant sex differences found in the original analysis remained significant. Thus, the disparities in sample sizes between sexes and the within sex variability in age does not seem to explain the results. In short, the differences appear to be relatively small but not due to other factors such as age.',\n",
       "  'offset': 26839,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Sex differences in grey matter thickness have also been reported in humans and they appear to somewhat differ from those reported here in chimpanzees. reported that females had thicker cortex bilaterally in the pre- and post-central regions, the occipital lobe and the anterior tip of the temporal lobe. Males did not differ from females in cortical thickness for any regions and this was the case when these authors scaled or did not scale the thickness measures on the basis of differences in brain size. also reported sex difference in cortical thickness in humans and their results were not all that different from the findings reported by. Specifically, found that females were found to have thicker cortex than males, particularly in the frontal, parietal and occipital regions. Moreover, there was not a single brain region that was thicker in males compared to females. When the findings of these two studies are compared with those reported here in chimpanzees, it suggests that sex differences in grey matter thickness may differ between humans and chimpanzees. The functional and evolutionary significance of these differences are not clear but the results seem to be evident even considering some potential confounds within our sample as well as between sexes.',\n",
       "  'offset': 27821,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The chimpanzees showed prominent rightward asymmetries in cortical thickness and significant leftward asymmetries in white matter volume. Chimpanzees with greater rightward asymmetries in cortical thickness were also found to have greater leftward asymmetries in white matter. Furthermore, without consideration of asymmetries, chimpanzees with larger white matter volumes were found to have thinner overall cortical thickness (see Table 3). When considered together, the findings complement each other and suggest that that the rightward cortical thickness asymmetries may be attributable to the expanded white matter volumes in the left compared to right hemisphere. Significant population-level white matter asymmetries have previously been found in the perisylvian regions of the chimpanzees brain as well as with voxel-based morphometry. More recently, using diffusion tensor imaging, leftward asymmetries have been found in arcuate fasciculus of chimpanzees, the major white matter tract connecting the superior temporal gyrus with the inferior frontal gyrus. Thus, multiple methods have revealed leftward asymmetries in white matter volume, spatial location and connectivity in chimpanzees and this is likely linked to regional variation in lateralization in cortical thickness.',\n",
       "  'offset': 29094,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Alternatively, the rightward asymmetries found in cortical thickness may be attributable to a larger left hemisphere surface area, including within the cortical folds. Significant population-level leftward asymmetries have been found in the surface areas and depths of several cortical sulci in chimpanzees including the sylvian fissure, fronto-orbital, superior precentral, intraparietal and post-central inferior and lunate sulci. Recall that rightward population-level asymmetries were found in grey matter volume. Thus, in this explanation, the volume of grey matter in the left compared to right hemisphere has to be expanded over a larger surface area which results in increased cortical thinning. It should be noted that we did not find a significant association between surface area and cortical thickness and therefore this interpretation was not directly supported by the findings reported here. Nonetheless as has been reported in humans, cortical surface area was positively correlated with white matter volume in the chimpanzees (see Table 3). Thus, it is likely that both surface area and white matter volume have some influence on cortical thickness, though our results suggest that white matter plays a more significant role.',\n",
       "  'offset': 30380,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The underlying mechanisms that determine population-level asymmetries in humans and other primates remain unclear but the comparative data continue to challenge the long held belief that they are unique to humans and explicitly linked to the evolution of language and speech, as suggested by some. We would suggest that anatomical and functional asymmetries were present in the common ancestor of humans and chimpanzees and that each species developed their own patterns of asymmetry that reflect their unique ecological adaptations rather than the occurrence of a saltational event that resulted in the emergence of asymmetries after the split between chimpanzees and humans.',\n",
       "  'offset': 31622,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'In summary, this is the first report of regional and lateralized variation in cortical thickness in chimpanzees. Like human brains, chimpanzees show thinner cortex in the primary motor and somatosensory cortex compared to association cortex. Moreover, chimpanzees show small but significant population-level asymmetries in cortical thickness, as has been reported in human brains. The functional significance of individual differences in cortical thickness in chimpanzees remain unclear but future studies should focus on establishing whether variability in cognitive performance or other behavioral phenotypes are linked to cortical thickness.',\n",
       "  'offset': 32299,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The authors declare no competing financial interests.',\n",
       "  'offset': 32944,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'footnote', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                ',\n",
       "  'offset': 32998,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'footnote', 'section_type': 'DISCUSS'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Geodesic estimation for large deformation anatomical shape and intensity averaging',\n",
       "  'offset': 33015,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '150',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Gee;given-names:JC',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '1',\n",
       "   'source': 'Neuroimage Supplement',\n",
       "   'fpage': 'S139',\n",
       "   'year': '2004',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Symmetric diffeomorphic image registration with cross-correlation: Evaluating automated labeling of elderly and neurodegenerative brain',\n",
       "  'offset': 33098,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Grossman;given-names:M',\n",
       "   'name_3': 'surname:Gee;given-names:JC',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Epstein;given-names:CL',\n",
       "   'pub-id_pmid': '17659998',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '12',\n",
       "   'source': 'Medical Image Analysis',\n",
       "   'fpage': '26',\n",
       "   'year': '2008',\n",
       "   'lpage': '41',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Dementia induces correlated reductions in white matter integrity and cortical thickness: a multivariate neuroimaging study with sparse canonical correlation analysis',\n",
       "  'offset': 33234,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Ungar;given-names:L',\n",
       "   'name_3': 'surname:Gee;given-names:JC',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Cook;given-names:PA',\n",
       "   'name_4': 'surname:Grossman;given-names:M',\n",
       "   'pub-id_pmid': '20083207',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '50',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '1004',\n",
       "   'year': '2010a',\n",
       "   'lpage': '1016',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'An open source multivariate framework for n-tissue segmentation with evaluation on public data',\n",
       "  'offset': 33400,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Wu;given-names:J',\n",
       "   'name_3': 'surname:Cook;given-names:PA',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Tustison;given-names:NJ',\n",
       "   'name_4': 'surname:Gee;given-names:JC',\n",
       "   'pub-id_pmid': '21373993',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '9',\n",
       "   'source': 'Neuroinformatics',\n",
       "   'fpage': '381',\n",
       "   'year': '2011a',\n",
       "   'lpage': '400',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A reproducible evaluation of ANTs similarity metric performance in brain image registration',\n",
       "  'offset': 33495,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Song;given-names:G',\n",
       "   'name_3': 'surname:Cook;given-names:PA',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Tustison;given-names:NJ',\n",
       "   'name_4': 'surname:Kelin;given-names:A',\n",
       "   'name_5': 'surname:Gee;given-names:JC',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '54',\n",
       "   'pub-id_pmid': '20851191',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '2033',\n",
       "   'year': '2011b',\n",
       "   'lpage': '2044',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The optimal template effect in hippocampus studies of diseased populations',\n",
       "  'offset': 33587,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Pluta;given-names:J',\n",
       "   'name_3': 'surname:Minkoff;given-names:D',\n",
       "   'name_0': 'surname:Avants;given-names:BB',\n",
       "   'name_1': 'surname:Yushkevich;given-names:P',\n",
       "   'name_6': 'surname:Gee;given-names:JC',\n",
       "   'name_4': 'surname:Korczykowski;given-names:M',\n",
       "   'name_5': 'surname:detre;given-names:J',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '49',\n",
       "   'pub-id_pmid': '19818860',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '2457',\n",
       "   'year': '2010b',\n",
       "   'lpage': '2466',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Cortical sulci asymmetries in chimpanzees and macaques: A new look at an old idea',\n",
       "  'offset': 33662,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Schapiro;given-names:SJ',\n",
       "   'name_3': 'surname:Reamer;given-names:L',\n",
       "   'name_0': 'surname:Bogart;given-names:SL',\n",
       "   'name_1': 'surname:Mangin;given-names:J-F',\n",
       "   'name_6': 'surname:Hopkins;given-names:WD',\n",
       "   'name_4': 'surname:Bennett;given-names:AJ',\n",
       "   'name_5': 'surname:Pierre;given-names:PJ',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '61',\n",
       "   'pub-id_pmid': '22504765',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '533',\n",
       "   'year': '2012',\n",
       "   'lpage': '541',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Asymmetric Broca’s area in great apes',\n",
       "  'offset': 33744,\n",
       "  'relations': [],\n",
       "  'infons': {'name_0': 'surname:Cantalupo;given-names:C',\n",
       "   'name_1': 'surname:Hopkins;given-names:WD',\n",
       "   'pub-id_pmid': '11734839',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '414',\n",
       "   'source': 'Nature',\n",
       "   'fpage': '505',\n",
       "   'year': '2001',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Are planum temporale and sylvian fissure asymmetries directly related? A MRI study in great apes',\n",
       "  'offset': 33784,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Hopkins;given-names:WD',\n",
       "   'lpage': '1981',\n",
       "   'name_0': 'surname:Cantalupo;given-names:C',\n",
       "   'name_1': 'surname:Pilcher;given-names:D',\n",
       "   'pub-id_pmid': '14572530',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '41',\n",
       "   'source': 'Neuropsychologia',\n",
       "   'fpage': '1975',\n",
       "   'year': '2003',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The chimpanzee show human-like perisylvian asymmetries in white matter',\n",
       "  'offset': 33881,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Smith;given-names:J',\n",
       "   'name_3': 'surname:Nir;given-names:T',\n",
       "   'name_0': 'surname:Cantalupo;given-names:C',\n",
       "   'name_1': 'surname:Oliver;given-names:J',\n",
       "   'name_4': 'surname:Taglialatela;given-names:JP',\n",
       "   'name_5': 'surname:Hopkins;given-names:WD',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '30',\n",
       "   'pub-id_pmid': '19614754',\n",
       "   'source': 'Eur J Neurosci',\n",
       "   'fpage': '431',\n",
       "   'year': '2009',\n",
       "   'lpage': '438',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Principles underlying mammalian neocortical scaling',\n",
       "  'offset': 33952,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '215',\n",
       "   'name_0': 'surname:Changizi;given-names:MA',\n",
       "   'pub-id_pmid': '11252638',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '84',\n",
       "   'source': 'Biol Cybern',\n",
       "   'fpage': '207',\n",
       "   'year': '2001',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A comparison of voxel and surface based cortical thickness estimation methods',\n",
       "  'offset': 34004,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Ridgway;given-names:GR',\n",
       "   'name_3': 'surname:Modat;given-names:m',\n",
       "   'name_0': 'surname:Clarkson;given-names:MJ',\n",
       "   'name_1': 'surname:Cardoso;given-names:MJ',\n",
       "   'name_6': 'surname:Fox;given-names:NC',\n",
       "   'name_7': 'surname:Ourselin;given-names:S',\n",
       "   'name_4': 'surname:Leung;given-names:KK',\n",
       "   'name_5': 'surname:Rohrer;given-names:JD',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '57',\n",
       "   'pub-id_pmid': '21640841',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '856',\n",
       "   'year': '2011',\n",
       "   'lpage': '865',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '',\n",
       "  'offset': 34082,\n",
       "  'relations': [],\n",
       "  'infons': {'source': 'The lopsided brain: Evolution of the generative mind',\n",
       "   'name_0': 'surname:Corballis;given-names:MC',\n",
       "   'year': '1992',\n",
       "   'type': 'ref',\n",
       "   'section_type': 'REF'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A theory of the origin of cerebral asymmetry: Epigenetic variation superimposed on a fixed right-shift',\n",
       "  'offset': 34083,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '303',\n",
       "   'name_0': 'surname:Crow;given-names:TJ',\n",
       "   'pub-id_pmid': '19288302',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '15',\n",
       "   'source': 'Laterality',\n",
       "   'fpage': '289',\n",
       "   'year': '2009',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Registration based cortical thickness measurement',\n",
       "  'offset': 34186,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Grossman;given-names:M',\n",
       "   'name_3': 'surname:Gee;given-names:JC',\n",
       "   'name_0': 'surname:Das;given-names:SR',\n",
       "   'name_1': 'surname:Avants;given-names:BB',\n",
       "   'pub-id_pmid': '19150502',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '45',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '867',\n",
       "   'year': '2009',\n",
       "   'lpage': '879',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '',\n",
       "  'offset': 34236,\n",
       "  'relations': [],\n",
       "  'infons': {'source': 'The Symbolic Species: The coevolution of language and the brain',\n",
       "   'name_0': 'surname:Deacon;given-names:TW',\n",
       "   'year': '1997',\n",
       "   'type': 'ref',\n",
       "   'section_type': 'REF'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Brain development during childhood and adolescence: a longitudinal study',\n",
       "  'offset': 34237,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Jeffries;given-names:NO',\n",
       "   'name_3': 'surname:Castellanos;given-names:FX',\n",
       "   'name_0': 'surname:Giedd;given-names:JN',\n",
       "   'name_1': 'surname:Blumenthal;given-names:J',\n",
       "   'name_6': 'surname:Paus;given-names:T',\n",
       "   'name_7': 'surname:Evans;given-names:AC',\n",
       "   'name_4': 'surname:Liu;given-names:H',\n",
       "   'name_5': 'surname:Zijenbos;given-names:A',\n",
       "   'name_8': 'surname:Rapoport;given-names:JL',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '2',\n",
       "   'pub-id_pmid': '10491603',\n",
       "   'source': 'Nat Neurosci',\n",
       "   'fpage': '861',\n",
       "   'year': '1999',\n",
       "   'lpage': '863',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Correlations between hand preference and cortical thickness in the secondary somatosensory (SII) cortex of the common marmoset, Callithrix jacchus',\n",
       "  'offset': 34310,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Rogers;given-names:LJ',\n",
       "   'lpage': '1351',\n",
       "   'name_0': 'surname:Gorrie;given-names:C',\n",
       "   'name_1': 'surname:Waite;given-names:PM',\n",
       "   'pub-id_pmid': '19045953',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '122',\n",
       "   'source': 'Behav Neurosci',\n",
       "   'fpage': '1343',\n",
       "   'year': '2008',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The structure of the cerebral cortex across adult life: Age-related patterns of surface area, thickness and gyrification',\n",
       "  'offset': 34457,\n",
       "  'relations': [],\n",
       "  'infons': {'comment': '(in press)',\n",
       "   'name_2': 'surname:Walhovd;given-names:KB',\n",
       "   'name_0': 'surname:Hogstrom;given-names:LJ',\n",
       "   'name_1': 'surname:Westlye;given-names:LT',\n",
       "   'name_3': 'surname:Fjell;given-names:AM',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'Cerebral Cortex',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Planum temporale surface area and grey matter asymmetries in chimpanzees (Pan troglodytes): The effect of handedness and comparison within findings in humans',\n",
       "  'offset': 34578,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '443',\n",
       "   'name_0': 'surname:Hopkins;given-names:WD',\n",
       "   'name_1': 'surname:Nir;given-names:T',\n",
       "   'pub-id_pmid': '20035802',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '208',\n",
       "   'source': 'Behav Brain Res',\n",
       "   'fpage': '436',\n",
       "   'year': '2010',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A voxel-based morphometry analysis of white matter asymmetries in chimpanzees (Pan troglodytes)',\n",
       "  'offset': 34736,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Nir;given-names:T',\n",
       "   'name_3': 'surname:Schenker;given-names:NM',\n",
       "   'name_0': 'surname:Hopkins;given-names:WD',\n",
       "   'name_1': 'surname:Taglialatela;given-names:JP',\n",
       "   'name_4': 'surname:Sherwood;given-names:CC',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '76',\n",
       "   'source': 'Brain, Behavior and Evolution',\n",
       "   'fpage': '93',\n",
       "   'year': '2010',\n",
       "   'lpage': '100',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Gray matter asymmetries in chimpanzees as revealed by voxel-based morphometry',\n",
       "  'offset': 34832,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Meguerditchian;given-names:A',\n",
       "   'name_3': 'surname:Nir;given-names:T',\n",
       "   'name_0': 'surname:Hopkins;given-names:WD',\n",
       "   'name_1': 'surname:Taglialatela;given-names:JP',\n",
       "   'name_4': 'surname:Schenker;given-names:NM',\n",
       "   'name_5': 'surname:Sherwood;given-names:CC',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '42',\n",
       "   'pub-id_pmid': '18586523',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '491',\n",
       "   'year': '2008',\n",
       "   'lpage': '497',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Gender difference analysis of cortical thickness in healthy young adults with surface-based methods',\n",
       "  'offset': 34910,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Shin;given-names:YW',\n",
       "   'name_3': 'surname:Kim;given-names:IY',\n",
       "   'name_0': 'surname:Im;given-names:K',\n",
       "   'name_1': 'surname:Lee;given-names:JM',\n",
       "   'name_4': 'surname:Kwon;given-names:JS',\n",
       "   'name_5': 'surname:Kim;given-names:SI',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '31',\n",
       "   'pub-id_pmid': '16426865',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '31',\n",
       "   'year': '2006',\n",
       "   'lpage': '38',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A comparative magnetic resonance imaging study of the anatomy, variability and asymmetry of Broca’s area in the human and chimpanzee brain',\n",
       "  'offset': 35010,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Hopkins;given-names:WD',\n",
       "   'lpage': '14616',\n",
       "   'name_0': 'surname:Keller;given-names:SS',\n",
       "   'name_1': 'surname:Roberts;given-names:N',\n",
       "   'pub-id_pmid': '19923293',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '29',\n",
       "   'source': 'J Neurosci',\n",
       "   'fpage': '14607',\n",
       "   'year': '2009',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Evaluation of 14 nonlinear deformation algorithms applied to human brain MRI registration',\n",
       "  'offset': 35151,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Ardekani;given-names:BA',\n",
       "   'name_3': 'surname:Ashbumer;given-names:J',\n",
       "   'name_0': 'surname:Klein;given-names:A',\n",
       "   'name_1': 'surname:Andersson;given-names:J',\n",
       "   'name_6': 'surname:Christensen;given-names:GE',\n",
       "   'name_7': 'surname:Collins;given-names:L',\n",
       "   'name_4': 'surname:Avants;given-names:B',\n",
       "   'name_5': 'surname:Chiang;given-names:M',\n",
       "   'name_8': 'surname:Hellier;given-names:P',\n",
       "   'name_9': 'surname:Song;given-names:JH',\n",
       "   'year': '2009',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'Neuroimage',\n",
       "   'type': 'ref',\n",
       "   'name_10': 'surname:Jenkinson;given-names:M',\n",
       "   'name_11': 'surname:Lepage;given-names:C',\n",
       "   'name_12': 'surname:Rueckert;given-names:D',\n",
       "   'name_13': 'surname:Thompson;given-names:P',\n",
       "   'name_14': 'surname:Vercauteren;given-names:T',\n",
       "   'name_15': 'surname:Woods;given-names:RP',\n",
       "   'name_16': 'surname:Mann;given-names:JJ',\n",
       "   'name_17': 'surname:Parsey;given-names:RV'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Chimpanzee pre-central corticospinal system asymmetry and handedness: A diffusion magnetic reonance imaging study',\n",
       "  'offset': 35241,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Rilling;given-names:JK',\n",
       "   'name_3': 'surname:Hopkins;given-names:WD',\n",
       "   'name_0': 'surname:Li;given-names:L',\n",
       "   'name_1': 'surname:Preuss;given-names:TM',\n",
       "   'name_6': 'surname:Nana;given-names:R',\n",
       "   'name_7': 'surname:Zhang;given-names:X',\n",
       "   'name_4': 'surname:Glasser;given-names:MF',\n",
       "   'name_5': 'surname:Kumar;given-names:B',\n",
       "   'name_8': 'surname:Hu;given-names:X',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '5',\n",
       "   'source': 'PlosOne',\n",
       "   'fpage': 'e12886',\n",
       "   'year': '2009',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Hemispheric asymmetries in cortical thickness',\n",
       "  'offset': 35355,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Thompson;given-names:PM',\n",
       "   'name_3': 'surname:Rex;given-names:DE',\n",
       "   'name_0': 'surname:Luders;given-names:E',\n",
       "   'name_1': 'surname:Narr;given-names:KL',\n",
       "   'name_4': 'surname:Jancke;given-names:J',\n",
       "   'name_5': 'surname:Toga;given-names:AW',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '16',\n",
       "   'pub-id_pmid': '16267139',\n",
       "   'source': 'Cerebral Cortex',\n",
       "   'fpage': '1232',\n",
       "   'year': '2006a',\n",
       "   'lpage': '1238',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Gender effects on cortial thickness and the influence of scaling',\n",
       "  'offset': 35401,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Thompson;given-names:PM',\n",
       "   'name_3': 'surname:Rex;given-names:DE',\n",
       "   'name_0': 'surname:Luders;given-names:E',\n",
       "   'name_1': 'surname:Narr;given-names:KL',\n",
       "   'name_6': 'surname:Jancke;given-names:L',\n",
       "   'name_7': 'surname:Toga;given-names:AW',\n",
       "   'name_4': 'surname:Woods;given-names:RP',\n",
       "   'name_5': 'surname:DeLuca;given-names:H',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '27',\n",
       "   'pub-id_pmid': '16124013',\n",
       "   'source': 'Hum Brain Mapp',\n",
       "   'fpage': '314',\n",
       "   'year': '2006b',\n",
       "   'lpage': '324',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Planum temporale grey matter asymmetries in chimpanzees (Pan troglodytes), vervet (Chlorocebus aethiops sabaeus), rhesus (Macaca mulatta) and bonnet (Macaca radiata) monkeys',\n",
       "  'offset': 35466,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Bennett;given-names:AJ',\n",
       "   'name_3': 'surname:Fears;given-names:SC',\n",
       "   'name_0': 'surname:Lyn;given-names:HL',\n",
       "   'name_1': 'surname:Pierre;given-names:P',\n",
       "   'name_4': 'surname:Woods;given-names:RP',\n",
       "   'name_5': 'surname:Hopkins;given-names:WD',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '49',\n",
       "   'pub-id_pmid': '21447349',\n",
       "   'source': 'Neuropsychologia',\n",
       "   'fpage': '2004',\n",
       "   'year': '2011',\n",
       "   'lpage': '2012',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'On the other hand: Current issues in and meta-analysis of the behavioral laterality of hand function in non-human primates',\n",
       "  'offset': 35640,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '232',\n",
       "   'name_0': 'surname:McGrew;given-names:WC',\n",
       "   'name_1': 'surname:Marchant;given-names:LF',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '40',\n",
       "   'source': 'Yearb Phys Anthropol',\n",
       "   'fpage': '201',\n",
       "   'year': '1997',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Gender differences in the human cerebral cortex: More neurons in males, more processes in females',\n",
       "  'offset': 35763,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:McFDonald-Comber;given-names:J',\n",
       "   'name_3': 'surname:Courten-Myers;given-names:GM',\n",
       "   'name_0': 'surname:Rabinowicz;given-names:T',\n",
       "   'name_1': 'surname:Dean;given-names:DE',\n",
       "   'pub-id_pmid': '10073431',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '14',\n",
       "   'source': 'J Child Neurol',\n",
       "   'fpage': '98',\n",
       "   'year': '1999',\n",
       "   'lpage': '107',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Human and non-human primate brains: are they allometrically scaled versions of the same design?',\n",
       "  'offset': 35861,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '77',\n",
       "   'name_0': 'surname:Rilling;given-names:JK',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '15',\n",
       "   'source': 'Evol Anthropol',\n",
       "   'fpage': '65',\n",
       "   'year': '2006',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Differential expansion of neural projection systems in primate brain evolution',\n",
       "  'offset': 35957,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '1459',\n",
       "   'name_0': 'surname:Rilling;given-names:JK',\n",
       "   'name_1': 'surname:Insel;given-names:TR',\n",
       "   'pub-id_pmid': '10380962',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '10',\n",
       "   'source': 'Neuroreport',\n",
       "   'fpage': '1453',\n",
       "   'year': '1999a',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The primate neocortex in comparative perspective using magnetic resonance imaging',\n",
       "  'offset': 36036,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '223',\n",
       "   'name_0': 'surname:Rilling;given-names:JK',\n",
       "   'name_1': 'surname:Insel;given-names:TR',\n",
       "   'pub-id_pmid': '10444351',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '37',\n",
       "   'source': 'J Hum Evol',\n",
       "   'fpage': '191',\n",
       "   'year': '1999b',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'A quantitative morphometric comparative analysis of the primate temporal lobe',\n",
       "  'offset': 36118,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '533',\n",
       "   'name_0': 'surname:Rilling;given-names:JK',\n",
       "   'name_1': 'surname:Seligman;given-names:RA',\n",
       "   'pub-id_pmid': '11969295',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '42',\n",
       "   'source': 'J Hum Evol',\n",
       "   'fpage': '505',\n",
       "   'year': '2002',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Continuity, divergence and the evolution of brain language pathways',\n",
       "  'offset': 36196,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Jbabdi;given-names:S',\n",
       "   'name_3': 'surname:Andersson;given-names:J',\n",
       "   'name_0': 'surname:Rilling;given-names:JK',\n",
       "   'name_1': 'surname:Glasser;given-names:MF',\n",
       "   'name_4': 'surname:Preuss;given-names:TM',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '3',\n",
       "   'source': 'Frontiers in Evolutionary Neuroscience',\n",
       "   'fpage': '1',\n",
       "   'year': '2012',\n",
       "   'lpage': '6',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'On the genetic architecture of cortical folding and brain volume in primates',\n",
       "  'offset': 36264,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Zilles;given-names:K',\n",
       "   'name_3': 'surname:Shelledy;given-names:W',\n",
       "   'name_0': 'surname:Rogers;given-names:J',\n",
       "   'name_1': 'surname:Kochunov;given-names:PV',\n",
       "   'name_6': 'surname:Duggirala;given-names:R',\n",
       "   'name_7': 'surname:Blangero;given-names:J',\n",
       "   'name_4': 'surname:Lancaster;given-names:JL',\n",
       "   'name_5': 'surname:Thompson;given-names:P',\n",
       "   'name_8': 'surname:Fox;given-names:PT',\n",
       "   'name_9': 'surname:Glahn;given-names:DC',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '53',\n",
       "   'pub-id_pmid': '20176115',\n",
       "   'source': 'Neuroimage',\n",
       "   'fpage': '1103',\n",
       "   'year': '2010',\n",
       "   'lpage': '1108',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Evolution of brain and intelligence',\n",
       "  'offset': 36341,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '257',\n",
       "   'name_0': 'surname:Roth;given-names:G',\n",
       "   'name_1': 'surname:Dicke;given-names:U',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '9',\n",
       "   'source': 'Trends in Cognitive Science',\n",
       "   'fpage': '250',\n",
       "   'year': '2005',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Broca’s area homologue in chimpanzees (Pan troglodytes): probabilistic mapping, asymmetry and comparison to humans',\n",
       "  'offset': 36377,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Spocter;given-names:MA',\n",
       "   'name_3': 'surname:Garrison;given-names:A',\n",
       "   'name_0': 'surname:Schenker;given-names:NM',\n",
       "   'name_1': 'surname:Hopkins;given-names:WD',\n",
       "   'name_6': 'surname:Hof;given-names:PR',\n",
       "   'name_7': 'surname:Sherwood;given-names:CC',\n",
       "   'name_4': 'surname:Stimpson;given-names:CD',\n",
       "   'name_5': 'surname:Erwin;given-names:JM',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '20',\n",
       "   'pub-id_pmid': '19620620',\n",
       "   'source': 'Cerebral Cortex',\n",
       "   'fpage': '730',\n",
       "   'year': '2010',\n",
       "   'lpage': '742',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Evolution of size and functional areas of the human brain',\n",
       "  'offset': 36494,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '406',\n",
       "   'name_0': 'surname:Schoenemann;given-names:PT',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '35',\n",
       "   'source': 'Annu Rev Anthropol',\n",
       "   'fpage': '379',\n",
       "   'year': '2006',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Prefrontal white matter volume is disproportionately larger in humans than in other primates',\n",
       "  'offset': 36552,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Glotzer;given-names:LD',\n",
       "   'lpage': '252',\n",
       "   'name_0': 'surname:Schoenemann;given-names:PT',\n",
       "   'name_1': 'surname:Sheehan;given-names:MJ',\n",
       "   'pub-id_pmid': '15665874',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '8',\n",
       "   'source': 'Nat Neurosci',\n",
       "   'fpage': '242',\n",
       "   'year': '2005',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Does brain white matter growth expand the cortex like a balloon> Hypothesis and consequences',\n",
       "  'offset': 36645,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '95',\n",
       "   'name_0': 'surname:Seldon;given-names:SH',\n",
       "   'pub-id_pmid': '15841825',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '10',\n",
       "   'source': 'Laterality',\n",
       "   'fpage': '81',\n",
       "   'year': '2005',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The brain and its main anatomical subdivisions in living hominids using magnetic resonance imaging',\n",
       "  'offset': 36738,\n",
       "  'relations': [],\n",
       "  'infons': {'lpage': '332',\n",
       "   'name_0': 'surname:Semendeferi;given-names:K',\n",
       "   'name_1': 'surname:Damasio;given-names:H',\n",
       "   'pub-id_pmid': '10656781',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '38',\n",
       "   'source': 'J Hum Evol',\n",
       "   'fpage': '317',\n",
       "   'year': '2000',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The evolution of the frontal lobes: a volumetric analysis based on three dimensional reconstructions of the magnetic resonance scans of human and ape brains',\n",
       "  'offset': 36837,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Frank;given-names:R',\n",
       "   'name_3': 'surname:Van Hoesen;given-names:GW',\n",
       "   'name_0': 'surname:Semendeferi;given-names:K',\n",
       "   'name_1': 'surname:Damasio;given-names:H',\n",
       "   'pub-id_pmid': '9085187',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '32',\n",
       "   'source': 'J Hum Evol',\n",
       "   'fpage': '375',\n",
       "   'year': '1997',\n",
       "   'lpage': '388',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Limbic frontal cortex in Hominoids: A comparative study of area 13',\n",
       "  'offset': 36994,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Schleicher;given-names:A',\n",
       "   'name_3': 'surname:Zilles;given-names:K',\n",
       "   'name_0': 'surname:Semendeferi;given-names:K',\n",
       "   'name_1': 'surname:Armstrong;given-names:E',\n",
       "   'name_4': 'surname:Van Hoesen;given-names:GW',\n",
       "   'pub-id_pmid': '9637180',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '106',\n",
       "   'source': 'Am J Phys Anthropol',\n",
       "   'fpage': '129',\n",
       "   'year': '1998',\n",
       "   'lpage': '155',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Prefrontal cortex in humans and apes: A comparative study of area 10',\n",
       "  'offset': 37061,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Schleicher;given-names:A',\n",
       "   'name_3': 'surname:Zilles;given-names:K',\n",
       "   'name_0': 'surname:Semendeferi;given-names:K',\n",
       "   'name_1': 'surname:Armstrong;given-names:E',\n",
       "   'name_4': 'surname:Van Hoesen;given-names:GW',\n",
       "   'pub-id_pmid': '11241188',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '114',\n",
       "   'source': 'Am J Phys Anthropol',\n",
       "   'fpage': '224',\n",
       "   'year': '2001',\n",
       "   'lpage': '241',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Intellectual ability and cortical development in children and adolescents',\n",
       "  'offset': 37130,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Lerch;given-names:JP',\n",
       "   'name_3': 'surname:Clasen;given-names:L',\n",
       "   'name_0': 'surname:Shaw;given-names:P',\n",
       "   'name_1': 'surname:Greenstein;given-names:D',\n",
       "   'name_6': 'surname:Evans;given-names:A',\n",
       "   'name_7': 'surname:Rapoport;given-names:JL',\n",
       "   'name_4': 'surname:Lenroot;given-names:R',\n",
       "   'name_5': 'surname:Gogtay;given-names:N',\n",
       "   'name_8': 'surname:Giedd;given-names:JN',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '440',\n",
       "   'pub-id_pmid': '16572172',\n",
       "   'source': 'Nature',\n",
       "   'fpage': '676',\n",
       "   'year': '2006',\n",
       "   'lpage': '679',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'The evolution of neuron types and cortical histology in apes and humans',\n",
       "  'offset': 37204,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Preuss;given-names:TM',\n",
       "   'name_3': 'surname:Kaas;given-names:J',\n",
       "   'name_0': 'surname:Sherwood;given-names:CC',\n",
       "   'name_1': 'surname:Hof;given-names:PR',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'The evolution of primate nervous systems. Evolution of nervous systems',\n",
       "   'year': '2007',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Human brain evolution writ large and small',\n",
       "  'offset': 37276,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Bianchi;given-names:S',\n",
       "   'name_3': 'surname:Raghanti;given-names:MA',\n",
       "   'name_0': 'surname:Sherwood;given-names:CC',\n",
       "   'name_1': 'surname:Baurernfeind;given-names:AL',\n",
       "   'name_6': 'surname:Falk;given-names:D',\n",
       "   'name_4': 'surname:Hof;given-names:PR',\n",
       "   'name_5': 'surname:Hofman;given-names:MA',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'Progress in Brain Research',\n",
       "   'year': '2012',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Advances in functional and structural MR image analysis and implementation of FSL',\n",
       "  'offset': 37319,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Woolrich;given-names:MW',\n",
       "   'name_3': 'surname:Beckmann;given-names:CF',\n",
       "   'name_0': 'surname:Smith;given-names:SM',\n",
       "   'name_1': 'surname:Jenkinson;given-names:M',\n",
       "   'name_6': 'surname:Bannister;given-names:PR',\n",
       "   'name_7': 'surname:De Luca;given-names:M',\n",
       "   'name_4': 'surname:Behrens;given-names:TEJ',\n",
       "   'name_5': 'surname:Johansen-Berg;given-names:H',\n",
       "   'name_8': 'surname:Drobniak;given-names:I',\n",
       "   'name_9': 'surname:Flitney;given-names:DE',\n",
       "   'lpage': '219',\n",
       "   'fpage': '208',\n",
       "   'year': '2004',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'Neuroimage',\n",
       "   'issue': 'S1',\n",
       "   'volume': '23 ',\n",
       "   'type': 'ref',\n",
       "   'name_10': 'surname:Niazy;given-names:R',\n",
       "   'name_11': 'surname:Saunders;given-names:J',\n",
       "   'name_12': 'surname:Vickers;given-names:J',\n",
       "   'name_13': 'surname:Zhang;given-names:Y',\n",
       "   'name_14': 'surname:De Stafano;given-names:N',\n",
       "   'name_15': 'surname:Brady;given-names:JM',\n",
       "   'name_16': 'surname:Matthews;given-names:PM'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Longitudinal mapping of cortical thickness and brain growth in normal children',\n",
       "  'offset': 37401,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Leonard;given-names:CM',\n",
       "   'name_3': 'surname:Welcome;given-names:SE',\n",
       "   'name_0': 'surname:Sowell;given-names:ER',\n",
       "   'name_1': 'surname:Thompson;given-names:PM',\n",
       "   'name_4': 'surname:Kan;given-names:E',\n",
       "   'name_5': 'surname:Toga;given-names:AW',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '24',\n",
       "   'pub-id_pmid': '15385605',\n",
       "   'source': 'The Journal of Neuroscience',\n",
       "   'fpage': '8223',\n",
       "   'year': '2004',\n",
       "   'lpage': '8231',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Wernicke’s area homolog in chimpanzees (Pan troglodytes): Probabilstic mapping, asymmetry and comparison with humans',\n",
       "  'offset': 37480,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Garrison;given-names:AR',\n",
       "   'name_3': 'surname:Stimpson;given-names:CD',\n",
       "   'name_0': 'surname:Spocter;given-names:MA',\n",
       "   'name_1': 'surname:Hopkins;given-names:WD',\n",
       "   'name_6': 'surname:Sherwood;given-names:CS',\n",
       "   'name_4': 'surname:Erwin;given-names:JM',\n",
       "   'name_5': 'surname:Hof;given-names:PR',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '277',\n",
       "   'source': 'Proceedings of the Royal Society B, Biological Sciences',\n",
       "   'fpage': '2165',\n",
       "   'year': '2010',\n",
       "   'lpage': '2174',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Automatic brain segmentation in rhesus monkeys',\n",
       "  'offset': 37599,\n",
       "  'relations': [],\n",
       "  'infons': {'name_2': 'surname:Joshi;given-names:S',\n",
       "   'name_3': 'surname:Coe;given-names:C',\n",
       "   'name_0': 'surname:Styner;given-names:M',\n",
       "   'name_1': 'surname:Knickmeyer;given-names:R',\n",
       "   'name_4': 'surname:Short;given-names:SJ',\n",
       "   'name_5': 'surname:Gilmore;given-names:J',\n",
       "   'section_type': 'REF',\n",
       "   'volume': '65122',\n",
       "   'source': 'SpieEE',\n",
       "   'fpage': 'L65121',\n",
       "   'year': '2007',\n",
       "   'lpage': 'L65128',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Parcellations and hemispheric asymmetries of human cerebral cortex analyzed on surface-based atlases',\n",
       "  'offset': 37646,\n",
       "  'relations': [],\n",
       "  'infons': {'comment': '(in press)',\n",
       "   'name_2': 'surname:Dierker;given-names:DL',\n",
       "   'name_0': 'surname:Van Essen;given-names:DC',\n",
       "   'name_1': 'surname:Glasser;given-names:MF',\n",
       "   'name_4': 'surname:Coalson;given-names:T',\n",
       "   'name_3': 'surname:Harwell;given-names:J',\n",
       "   'section_type': 'REF',\n",
       "   'source': 'Cerebral Cortex',\n",
       "   'type': 'ref'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Lower panel: 3D-reconstruction of chimpanzee template brain with vertical white lines representing regions outlined above (a, b) a) Coronal view of the chimpanzee brain with the orbital (red), dorsal (green) and mesial (yellow) prefrontal cortex regions outlined on the scan, FO = fronto-orbital sulcus, RG = rectus b) Coronal view of the chimpanzee brain with the superior (red), middle (green) and inferiorl (yellow) temporal gyri regions outlined on the scan. SF = sylvian fissure, STS = superior temporal sulcus, ITS = inferior temporal sulcus.',\n",
       "  'offset': 37747,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'fig_caption',\n",
       "   'section_type': 'FIG',\n",
       "   'id': 'F1',\n",
       "   'file': 'nihms458577f1.jpg'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Lower panel: 3D-reconstruction of chimpanzee template brain with horizontal white lines representing regions outlined above (a, b, c) a) Transverse view of the chimpanzee brain with superior parietal lobe b) supramarginal gyrus and c) pre-central (red) and post-central (green) outlined on the scan, IPS = intraparietal sulcus, SPS = supurior parietal sulcus, CS = central sulcus, PCS = post central sulcus, PCI = precentral inferior sulcus.',\n",
       "  'offset': 38296,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'fig_caption',\n",
       "   'section_type': 'FIG',\n",
       "   'id': 'F2',\n",
       "   'file': 'nihms458577f2.jpg'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Lower panel: 3D-reconstruction of chimpanzee template brain with horizontal and vertical white lines representing regions outlined above (a, b). a) Transverse view of the chimpanzee template brain with the IFG outlined on the scan b) Coronal view of the chimpanzee template brain with the posterior superior temporal gyrus outlined on the scan. SF = sylvian fissure, STS = superior temporal sulcus.',\n",
       "  'offset': 38738,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'fig_caption',\n",
       "   'section_type': 'FIG',\n",
       "   'id': 'F3',\n",
       "   'file': 'nihms458577f3.jpg'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Mean cortical thickness of the left and right hemispheres. Color bar indicates variation in cortical thickness throughout cortex. Upper panel left and right lateral view. Lower panel, left and right medial views.',\n",
       "  'offset': 39137,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'fig_caption',\n",
       "   'section_type': 'FIG',\n",
       "   'id': 'F4',\n",
       "   'file': 'nihms458577f4.jpg'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Mean cortical thickness (+/− s.e.) for male and female chimpanzees for each of the 12 regions. Bolded labels indicate significant male-female differences at p<.05 after adjustment for overall brain volume.',\n",
       "  'offset': 39350,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'fig_caption',\n",
       "   'section_type': 'FIG',\n",
       "   'id': 'F5',\n",
       "   'file': 'nihms458577f5.jpg'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Analyses of interhemispheric differences in grey matter thickness were performed using pared samples t–tests. Values indicate grey matter thickness in mm.',\n",
       "  'offset': 39558,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                    ',\n",
       "  'offset': 39715,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                ',\n",
       "  'offset': 39736,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Mean Cortical Thickness (+/− s.e.) for Each Hemisphere and Region',\n",
       "  'offset': 39753,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_caption',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T1',\n",
       "   'file': 'T1.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '\\n        \\n          \\n            \\t\\n            Left\\t\\n            s.e.\\t\\n            Right\\t\\n            s.e.\\t\\n            t\\t\\n            p\\t\\n          \\n\\n        \\n        \\n          \\n            \\n              Frontal Lobe\\n            \\t\\n          \\n\\n          \\n            \\u2003Dorsal PFC\\t\\n            2.67\\t\\n            .03\\t\\n            2.71\\t\\n            .03\\t\\n            3.801\\t\\n            .000\\t\\n          \\n\\n          \\n            \\u2003Orbital PFC\\t\\n            2.56\\t\\n            .03\\t\\n            2.53\\t\\n            .04\\t\\n            3.269\\t\\n            .001\\t\\n          \\n\\n          \\n            \\u2003Mesial PFC\\t\\n            2.43\\t\\n            .04\\t\\n            2.46\\t\\n            .03\\t\\n            1.213\\t\\n            .229\\t\\n          \\n\\n          \\n            \\u2003IFG\\t\\n            2.78\\t\\n            .05\\t\\n            2.78\\t\\n            .05\\t\\n            3.151\\t\\n            .002\\t\\n          \\n\\n          \\n            \\u2003Pre-central\\t\\n            1.67\\t\\n            .04\\t\\n            1.75\\t\\n            .04\\t\\n            6.643\\t\\n            .000\\t\\n          \\n\\n          \\n            \\n              Parietal\\n            \\t\\n          \\n\\n          \\n            \\u2003Post-central\\t\\n            1.62\\t\\n            .04\\t\\n            1.71\\t\\n            .04\\t\\n            5.163\\t\\n            .000\\t\\n          \\n\\n          \\n            \\u2003Superior\\t\\n            2.76\\t\\n            .06\\t\\n            2.83\\t\\n            .06\\t\\n            3.499\\t\\n            .001\\t\\n          \\n\\n          \\n            \\u2003Supramarginal\\t\\n            2.45\\t\\n            .04\\t\\n            2.51\\t\\n            .05\\t\\n            3.807\\t\\n            .000\\t\\n          \\n\\n          \\n            \\n              Temporal\\n            \\t\\n            \\t\\n            \\t\\n            \\t\\n            \\t\\n            \\t\\n            \\t\\n          \\n\\n          \\n            \\u2003Superior\\t\\n            2.16\\t\\n            .05\\t\\n            2.23\\t\\n            .05\\t\\n            4.820\\t\\n            .000\\t\\n          \\n\\n          \\n            \\u2003Middle\\t\\n            2.70\\t\\n            .04\\t\\n            2.83\\t\\n            .04\\t\\n            8.131\\t\\n            .000\\t\\n          \\n\\n          \\n            \\u2003Inferior\\t\\n            2.65\\t\\n            .04\\t\\n            2.61\\t\\n            .05\\t\\n            −1.425\\t\\n            .158\\t\\n          \\n\\n          \\n            \\u2003pSTG\\t\\n            2.08\\t\\n            .06\\t\\n            2.18\\t\\n            .06\\t\\n            3.396\\t\\n            .001\\t\\n          \\n\\n        \\n      ',\n",
       "  'offset': 39821,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T1',\n",
       "   'File': 'T1.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Surface area, grey and white matter volumes are in CC. Numbers in the parentheses are standard errors.',\n",
       "  'offset': 42183,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                    ',\n",
       "  'offset': 42286,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                ',\n",
       "  'offset': 42307,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Mean Surface Area, Grey and White Matter Volumes for the Left and Right Hemispheres in Male and Females Chimpanzees',\n",
       "  'offset': 42324,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_caption',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T2',\n",
       "   'file': 'T2.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '\\n        \\n          \\n            \\t\\n            Males\\t\\n            Females\\t\\n            Total\\t\\n          \\n\\n        \\n        \\n          \\n            \\n              Grey Matter\\n            \\t\\n          \\n\\n          \\n            \\u2003LH\\t\\n            93.73 (1.94)\\t\\n            83.50 (1.35\\t\\n            88.62 (1.18)\\t\\n          \\n\\n          \\n            \\u2003RH\\t\\n            99.34 (.178)\\t\\n            89.42 (1.24)\\t\\n            94.38 (1.08)\\t\\n          \\n\\n          \\n            \\n              White Matter\\n            \\t\\n          \\n\\n          \\n            \\u2003LH\\t\\n            73.15 (1.56)\\t\\n            72.14 (1.07)\\t\\n            72.65 (1.37)\\t\\n          \\n\\n          \\n            \\u2003RH\\t\\n            72.67 (1.55)\\t\\n            70.64 (1.07)\\t\\n            71.65 (1.31)\\t\\n          \\n\\n          \\n            \\n              Surface Area\\n            \\t\\n          \\n\\n          \\n            \\u2003LH\\t\\n            187.96 (2.45)\\t\\n            178.05 (1.69)\\t\\n            183.00 (1.49)\\t\\n          \\n\\n          \\n            \\u2003RH\\t\\n            187.43 (2.45)\\t\\n            176.52 (1.74)\\t\\n            181.98 (1.53)\\t\\n          \\n\\n        \\n      ',\n",
       "  'offset': 42440,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T2',\n",
       "   'File': 'T2.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'p < .001,',\n",
       "  'offset': 43535,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                               ',\n",
       "  'offset': 43545,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'p < .01,',\n",
       "  'offset': 43577,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                               ',\n",
       "  'offset': 43586,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'p < .05:',\n",
       "  'offset': 43618,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                               ',\n",
       "  'offset': 43627,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'WM = − white matter volume, GM = grey matter volume, SA = surface area, Thickness = average cortical thickness',\n",
       "  'offset': 43659,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                    ',\n",
       "  'offset': 43772,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '                                           ',\n",
       "  'offset': 43793,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table', 'section_type': 'TABLE'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': 'Intercorrelations between Volume and Asymmetry Values for SA, GM, WM and Cortical Thickness',\n",
       "  'offset': 43837,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table_caption',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T3',\n",
       "   'file': 'T3.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []},\n",
       " {'text': '\\n        \\n          \\n            \\t\\n            WM\\t\\n            GM\\t\\n            SA\\t\\n            Thickness\\t\\n          \\n\\n        \\n        \\n          \\n            \\n              Volume\\n            \\t\\n          \\n\\n          \\n            WM\\t\\n            1.0\\t\\n            \\t\\n            \\t\\n            \\t\\n          \\n\\n          \\n            GM\\t\\n            .457***\\t\\n            1.0\\t\\n            \\t\\n            \\t\\n          \\n\\n          \\n            SA\\t\\n            .254\\t\\n            .585\\t\\n            1.0\\t\\n            \\t\\n          \\n\\n          \\n            Thickness\\t\\n            −.506***\\t\\n            −.263*\\t\\n            −.148\\t\\n            1.0\\t\\n          \\n\\n          \\n            \\n              Asymmetry\\n            \\t\\n          \\n\\n          \\n            WM\\t\\n            1.0\\t\\n            \\t\\n            \\t\\n            \\t\\n          \\n\\n          \\n            GM\\t\\n            .045\\t\\n            1.0\\t\\n            \\t\\n            \\t\\n          \\n\\n          \\n            SA\\t\\n            .241*\\t\\n            .056\\t\\n            1.0\\t\\n            \\t\\n          \\n\\n          \\n            Thickness\\t\\n            −.529***\\t\\n            .195\\t\\n            −.054\\t\\n            1.0\\t\\n          \\n\\n        \\n      ',\n",
       "  'offset': 43929,\n",
       "  'relations': [],\n",
       "  'infons': {'type': 'table',\n",
       "   'section_type': 'TABLE',\n",
       "   'id': 'T3',\n",
       "   'File': 'T3.xml'},\n",
       "  'sentences': [],\n",
       "  'annotations': []}]"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmcid_to_doc[no_hits[2]]['passages']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3104413'"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_hits[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+\n",
    "\n",
    "All our code is hosted on Github (http://github.com/intermine/) and contributions are welcome.\n",
    "\n",
    "The raw data and complete source code to reproduce the results of the analysis is freely available at https://github.com/akastrin/kastrin2014large.\n",
    "\n",
    "```\n",
    "{'text': '2Software available at https://github.com/vellamike/optimizer, online documentation at http://optimizer.readthedocs.org/',\n",
    "   'offset': 67424,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'footnote', 'section_type': 'CONCL'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}\n",
    "   \n",
    "{'text': '– To handle the RRF syntax we have developed the UMLS2RDF project.1 UMLS2RDF is the set of scripts that connect to the UMLS MySQL release and transforms its content into RDF triples.',\n",
    "   'offset': 17722,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}   \n",
    "{'text': '           https://github.com/ncbo/umls2rdf         ',\n",
    "   'offset': 20138,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'footnote', 'section_type': 'CONCL'},\n",
    "   'sentences': [],\n",
    "   'annotations': []},\n",
    "   \n",
    "{'passages': [{'text': 'Open-Source Syringe Pump Library',\n",
    "   'offset': 0,\n",
    "   'relations': [],\n",
    "   'infons': {'name_2': 'surname:Anzalone;given-names:Gerald C.',\n",
    "    'license': 'This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.',\n",
    "    'name_0': 'surname:Wijnen;given-names:Bas',\n",
    "    'name_1': 'surname:Hunt;given-names:Emily J.',\n",
    "    'alt-title': 'Open-Source Syringe Pump Library',\n",
    "    'article-id_publisher-id': 'PONE-D-14-11249',\n",
    "    'name_3': 'surname:Pearce;given-names:Joshua M.',\n",
    "    'type': 'front',\n",
    "    'elocation-id': 'e107216',\n",
    "    'article-id_doi': '10.1371/journal.pone.0107216',\n",
    "    'section_type': 'TITLE',\n",
    "    'article-id_pmid': '25229451',\n",
    "    'volume': '9',\n",
    "    'article-id_pmc': '4167991',\n",
    "    'year': '2014',\n",
    "    'title': 'Data Availability',\n",
    "    'name_4': 'surname:Gilestro;given-names:Giorgio F.',\n",
    "    'issue': '9',\n",
    "    'notes': 'The authors confirm that all data underlying the findings are fully available without restriction. All files needed to construct the device are held at https://github.com/mtu-most/linear-actuator All data for the paper presented within it.'},\n",
    "```\n",
    "\n",
    "To foster reproducibility and fast development of future work based upon these results, we have published functional graph analysis tools under an open source, GPLv3 license, available here: https://github.com/ssgrn/GraphInvariantsNeocortex.'\n",
    "\n",
    "Software and user manual are available at http://bioinformaticstools.mayo.edu/research/patterncnv/, and R package at https://github.com/topsoil/patternCNV/.\n",
    "\n",
    "\n",
    "Availability and implementation: The software for PatternCNV is implemented using Perl and R, and can be used in Mac or Linux environments. Software and user manual are available at http://bioinformaticstools.mayo.edu/research/patterncnv/, and R package at https://github.com/topsoil/patternCNV/.\n",
    "\n",
    "\n",
    "To quantify the number of c-Fos positive and β-gal positive cells, the red (c-Fos) and green (β-gal) color channels in each image were first filtered with a stringent threshold (mean +2 standard deviations of background pixel intensity level), and each color channel was then converted to a binary, black/white (BW) image using ImageBWconvertGUI- a custom-made program running in the 2013a Matlab Computing Environment with the Image Processing Toolbox (The MathWorks, Natick, MA; program available on Github: https://github.com/neuropil/ImageBWconvert/).\n",
    "\n",
    "In addition to quantification of the number of c-Fos and β-gal positive cell bodies, we also quantified, across different experimental conditions, the number and percentage of pixels within each nTS polygon with a pixel intensity value that exceeded 2 standard deviations for each stain, which includes both histological-positive cell bodies as well as associated cellular morphology (e.g., dendrites and axons) using ROIImageAnalysis - another custom-made MATLAB program (available on Github: https://github.com/neuropil/ROIImageAnalysis/).\n",
    "\n",
    "The complete R scripts to reproduce the analysis can be downloaded from http://sgibb.github.io/Culicoides/.\n",
    "\n",
    "Project home page:ftp://ftp.ebi.ac.uk/pub/databases/chembl/text-mining, and https://github.com/chembl/chembl_literature_classifier\n",
    "\n",
    "Proteome clustering results and matrix, scripts, and sequences alignments are available from GitHub (https://github.com/ocisse/Pneumocystis_comparative, last accessed July 27, 2014).\n",
    "\n",
    "All the figures for this article were produced using matplotlib, and figure scripts are available from https://github.com/rougier/ten-rules.\n",
    "\n",
    "New scripts written for this protocol, an example data set, and any future updates are available at https://github.com/listonlab/.\n",
    "\n",
    "A copy of the macro code is available as open source freeware; https://github.com/graemeball/ij_scripts/blob/master/Macros/Sum_Masked_Signal.ijm.\n",
    "\n",
    "The code is also available online at http://github.com/uygarsumbul/rgc.\n",
    "\n",
    "Source code repository URL: http://github.com/CovertLab/WholeCellSimDB\n",
    "\n",
    "The raw count data, script, script output, addition sample metadata, and saved Cytoscape session are available (https://github.com/brwnj/interaction_network).\n",
    "\n",
    "The R code used for all of the analysis has been made available on an open source basis (https://github.com/SEEG-Oxford/ebola_zoonotic).\n",
    "\n",
    "The data and source code used to generate this table are provided through http://jbloom.github.io/phyloExpCM/example_2014Analysis_lactamase.html (last accessed July 28, 2014).\n",
    "\n",
    "Code developed for this project, as well as issue trackers and full documentation for deploying the software are all available on github at https://github.com/apfejes/epigenetics-software.\n",
    "\n",
    "Source code can be freely obtained from https://github.com/bps10/color/tree/JOSA.\n",
    "\n",
    "The R package can be downloaded from our website (http://genemed.uchicago.edu/~pgeeleher/pRRophetic) or GitHub (https://github.com/paulgeeleher/pRRophetic).\n",
    "\n",
    "Phages genes were analyzed using Phamerator , an open-source program (GNU general public license) designed to compare phage genes and genomes. For this study, Phamerator was adapted and stored in a GitHub repository ( http://github.com/byuphamerator/phamerator-dev) separate from the original version.\n",
    "\n",
    "A reference implementation of SILP2 is provided at https://github.com/jim-bo/silp2.\n",
    "\n",
    "The code used for generating most of the reported results, including the implementation of the ALN50 metric, is available at https://github.com/jim-bo/scafathon.\n",
    "\n",
    "ARYANA with complete source code can be obtained from http://github.com/aryana-aligner\n",
    "\n",
    "```\n",
    "  {'text': 'https://github.com/alibashir/EMMCKmer',\n",
    "   'offset': 1688,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}\n",
    "```\n",
    "\n",
    "Latest source code: https://github.com/GeneMANIA/genemania.\n",
    "\n",
    "Computational times for convergence of our MCMC schemes (code is available at http://github.com/jhncl/LNA.git) can be compared using estimates for the minimum effective sample size per second (ESSmin/sec).\n",
    "\n",
    "Maire N (n.d.) SwissTPH/data-diff. GitHub. Available: https://github.com/SwissTPH/data-diff. Accessed 5 December 2013.\n",
    "\n",
    "-\n",
    "\n",
    "To process huge numbers of queries in parallel in the background, we use the Resque library (https://github.com/resque/resque).\n",
    "\n",
    "Networks were constructed using the online application Population Graphs v2 (http://dyerlab.github.io/popgraph/. Accessed 2014 June 16), and the analyses were performed with the software Genetic Studio.\n",
    "\n",
    "The phylogenetic analyses were performed using the software package phyloExpCM (phylogenetic analyses with experimental codon models, https://github.com/jbloom/phyloExpCM, last accessed July 28, 2014), which primarily serves as an interface to run HYPHY.\n",
    "\n",
    "The PyJade module (https://github.com/SyrusAkbary/pyjade) is used to render Jade templates into HTML to generate the interface.\n",
    "\n",
    "```\n",
    "{'text': 'MGFLI. count_genes_per_transcript.pl. https://github.com/MGFLI/RNAseq/tree/master',\n",
    "   'offset': 25770,\n",
    "   'relations': [],\n",
    "   'infons': {'type': 'ref', 'section_type': 'REF'},\n",
    "   'sentences': [],\n",
    "   'annotations': []}\n",
    "   \n",
    "4Retrieved from https://github.com/opencobra/cobratoolbox/tree/cd4368bda07ef4d63486a3683865f4d9f3bc53fe\n",
    "```\n",
    "\n",
    "The system of differential equations were solved using a fourth-order Runge-Kutta method implemented in the Python programming language (v2.7.6) [https://www.python.org/]. Parameter estimation was conducted using lsqfit (v4.6.1) [https://github.com/gplepage/lsqfit], a software package for non-linear least-squares fitting of noisy data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The system of differential equations were solved using a fourth-order Runge-Kutta method implemented in the Python programming language (v2.7.6) [https://www.python.org/]. Parameter estimation was conducted using lsqfit (v4.6.1) [https://github.com/gplepage/lsqfit], a software package for non-linear least-squares fitting of noisy data.']"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[t['text'] for t in pmcid_to_doc[ids_to_check[528]]['passages'] if 'github' in t['text']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4168813'"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids_to_check[525]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'passages': [{'text': 'Modeling toxicodynamic effects of trichloroethylene on liver in mouse model of autoimmune hepatitis',\n",
       "   'offset': 0,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Zurlinden;given-names:Todd',\n",
       "    'name_3': 'surname:Kreps;given-names:Meagan N.',\n",
       "    'article-id_manuscript': 'NIHMS625241',\n",
       "    'name_1': 'surname:Reisfeld;given-names:Brad',\n",
       "    'kwd': 'Trichloroethylene liver IL-6 immunotoxicity modeling',\n",
       "    'license': '\\n          This file is available for text mining. It may also be used consistent with the principles of fair use under the copyright law.\\n        ',\n",
       "    'name_5': 'surname:Blossom;given-names:Sarah J.',\n",
       "    'issue': '3',\n",
       "    'article-id_doi': '10.1016/j.taap.2014.07.003',\n",
       "    'section_type': 'TITLE',\n",
       "    'name_0': 'surname:Gilbert;given-names:Kathleen M.',\n",
       "    'article-id_pmid': '25026505',\n",
       "    'lpage': '293',\n",
       "    'article-id_pmc': '4171219',\n",
       "    'fpage': '284',\n",
       "    'year': '2015',\n",
       "    'name_4': 'surname:Erickson;given-names:Stephen W.',\n",
       "    'type': 'front',\n",
       "    'volume': '279'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chronic exposure to industrial solvent and water pollutant trichloroethylene (TCE) in female MRL+/+ mice generates disease similar to human autoimmune hepatitis. The current study was initiated to investigate why TCE-induced autoimmunity targeted the liver. Compared to other tissues the liver has an unusually robust capacity for repair and regeneration. This investigation examined both time-dependent and dose-dependent effects of TCE on hepatoprotective and pro-inflammatory events in liver and macrophages from female MRL+/+ mice. After a 12-week exposure to TCE in drinking water a dose-dependent decrease in macrophage production of IL-6 at both the transcriptional and protein level was observed. A longitudinal study similarly showed that TCE inhibited macrophage IL-6 production. In terms of the liver, TCE had little effect on expression of pro-inflammatory genes (Tnfa, Saa2 or Cscl1) until the end of the 40-week exposure. Instead, TCE suppressed hepatic expression of genes involved in IL-6 signaling (Il6r, gp130, and Egr1). Linear regression analysis confirmed liver histopathology in the TCE-treated mice correlated with decreased expression of Il6r. A toxicodynamic model was developed to estimate the effects of TCE on IL-6 signaling and liver pathology under different levels of exposure and rates of repair. This study underlined the importance of longitudinal studies in mechanistic evaluations of immuntoxicants. It showed that later-occurring liver pathology caused by TCE was associated with early suppression of hepatoprotection rather than an increase in conventional pro-inflammatory events. This information was used to create a novel toxicodynamic model of IL-6-mediated TCE-induced liver inflammation.',\n",
       "   'offset': 100,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Introduction',\n",
       "   'offset': 1833,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene (TCE) is a chlorinated hydrocarbon that has been used as a degreasing agent since the 1920s. Because of inappropriate disposal over the years TCE is now a common pollutant at Superfund toxic waste sites and at many industry and government facilities. It is found in soil and surface water as a result of direct discharges and in groundwater due to leaching from disposal operations. As noted by a recent National Research Council report evidence on human health hazards from TCE exposure, either occupational or environmental, has strengthened in recent years. One of the predominant non-cancer outcome associated with TCE exposure in humans is immunotoxicity, most notably the development of hypersensitivity responses including autoimmune scleroderma and autoimmune liver diseases. TCE is still widely used in Asia, where it has become a serious work-related health concern due to the induction of dermal and systemic hypersensitivity diseases often accompanied by non-viral, immune-mediated hepatitis.',\n",
       "   'offset': 1846,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Using a mouse model we and others have found that long-term exposure to TCE in drinking water at concentrations lower than sanctioned occupational exposure levels generated a T cell-mediated liver disease commensurate with human idiopathic autoimmune hepatitis (AIH). This TCE-induced liver inflammation was not usually accompanied by markers of acute liver injury such as increased blood levels of alanine transaminase or liver fibrosis, but was associated with the development of antibodies specific for liver microsomal proteins similar to those in patients with type 2 AIH.',\n",
       "   'offset': 2869,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The development of toxicant-induced immune pathology such as the autoimmune hepatitis caused by TCE exposure is almost certainly a complex multifactorial process. Developing conceptual models can be a way to delineate and quantify the contribution of different toxicant-induced alterations to the actual pathology. As a first step in this direction a model was developed here to describe a specific part of the process, namely IL-6-mediated liver events. IL-6 is one of the most important regulators of hepatic inflammation. The pathogenesis of AIH requires circumvention of the well-known propensity of the liver to induce T cell tolerance. Pre-existing inflammation in the liver may subvert its tolerogenicity and help sustain an immune response by entering T cells. The ability of toxicant exposure to generate such inflammation depends on opposing forces of tissue injury and tissue repair. Distress signals triggered during initiation of toxicant-induced liver injury (e.g. lipid peroxidation, reactive intermediate formation) can promote inflammation. However, they also stimulate protective (anti-apoptotic) and regenerative (cell division) mechanisms in the liver. One of the mechanisms that determine whether toxicant exposure ultimately leads to tissue repair or to injury-induced inflammation is regulated by IL-6.',\n",
       "   'offset': 3447,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Treatments to prevent or reverse immunological liver injury in mouse models have been associated with an increase in liver expression of Il6. Disruption of IL-6, or its receptors IL-6R or Gp130, has been shown to promote liver inflammation and/or mortality following partial hepatectomy, ethanol-induced liver disease, carbon tetrachloride-induced liver necrosis, obesity-associated insulin resistance, autoimmune cholangitis, and Con A-induced hepatitis. Thus, IL-6 appears to prevent immunological liver injury. In addition to its documented ability to promote liver regeneration and/or protection in the face of damage or trauma IL-6 also appears to be required for normal liver maintenance. Liver weight and total DNA and protein contents were decreased 26–28% in older (5–10month-old) female IL-6-deficient mice as compared to age-matched wild-type controls. This suggests that IL-6 is needed for normal hepatocyte turnover, and that over time a loss of this cytokine is detrimental to liver function.',\n",
       "   'offset': 4773,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In an attempt to define why TCE-induced autoimmunity targets the liver, mice exposed to a single dose of TCE for 4, 10, 16, 22, 28, 34 or 40 weeks were evaluated in the current study for time-dependent alterations in IL-6 as well as other pro-inflammatory mediators. This was complemented by a second study that examined the dose-dependent effects of TCE on these mediators at a single time point.',\n",
       "   'offset': 5784,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The development of autoimmune hepatitis in our mouse model of TCE exposure involves alterations in both the liver and the immune system. This multi-factorial process mimics the complicated etiologies of human autoimmune diseases. Developing conceptual models can be a way to delineate and quantify the contribution of different disease-induced alterations to actual pathology. As a first step in this direction the results obtained here were used to model the part of the TCE-induced disease process revealed in the current study, namely the effect of TCE on IL-6-mediated liver events. Taken together, the results suggest that later-occurring TCE-induced liver damage was due to an early decrease in IL-6-mediated hepatoprotection rather than an increase in pro-inflammatory events.',\n",
       "   'offset': 6182,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Materials and Methods',\n",
       "   'offset': 6966,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mouse treatment',\n",
       "   'offset': 6988,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Eight week-old female MRL+/+ mice (Jackson Laboratories; Bar Harbor, ME) were housed in polycarbonate ventilated cages and provided with drinking water (ultrapure from Milli-Q Integral Water Purification System, Millipore) ad libitum. TCE (purity 99± %; Aldrich Chemical Co. Inc.; Milwaukee, WI) was suspended in drinking water with 1% emulsifier Alkamuls EL-620 from Rhone-Poulenc (Cranbury, NJ). Freshly made TCE-containing drinking water was provided every 2–3 days. In one experiment the mice (12 mice/group) received either 0, 0.02, 0.1 mg/ml or 0.5 mg/ml TCE in their drinking water for 12 weeks. In a second experiment the mice (8–9 mice/group) received 0 or 0.5 mg/ml TCE in their drinking water for 4, 10, 16, 22, 28, 34 or 40 weeks. The mice were weighed weekly and water consumption was monitored. All studies were approved by the Animal Care and Use Committee at the University of Arkansas for Medical Sciences. When the mice were sacrificed at the different time points adherent macrophages isolated from pooled peritoneal exudates from 2–3 mice (n=3–4/treatment group) were incubated for 20 hours in the presence or absence of LPS (1 μg/ml). Approximately 80% of adherent peritoneal exudate cells (PEC), regardless of treatment group, expressed the transmembrane protein F4/80, a marker of mature macrophages (data not shown). Culture supernatants from the peritoneal macrophages were then collected for cytokine evaluation. RLT Lysis Buffer (Qiagen Sciences, Germantown, MD) was then added directly to the remaining adherent cells before freezing for subsequent qRT-PCR analysis. Liver tissue harvested at the time of sacrifice was stained with H&E and examined for liver pathology. Liver and sections were examined microscopically and scored in a blinded manner by a veterinary pathologist for the severity of inflammation and fibrosis based on a four point scale (0 – 3), ranging from no change to severe, respectively) as described. Some liver tissue was instead frozen for subsequent qRT-PCR analysis.',\n",
       "   'offset': 7004,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'qRT-PCR',\n",
       "   'offset': 9037,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fluorescence-based quantitative reverse transcriptase polymerase chain reaction (qRT-PCR) was conducted using RNA isolated (using RNeasy, Qiagen, Germantown, MD) from peritoneal macrophages or liver. Details of the technique as well as quality controls and rationale for choice of reference gene Eef2 (eukaryotic translation elongation factor 2) has been described.',\n",
       "   'offset': 9045,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cytokine Analysis',\n",
       "   'offset': 9411,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The macrophage culture supernatants were examined using the Mouse IL-1β, IL-6 and TNF alpha Ready-Set-Go! ELISA kits from eBioscience, Inc. (San Diego, CA). The amounts of IL-6 produced by macrophages in the 12-week study were generally lower than those generated in the 40-week study. This is likely due to difference in the number of macrophages/well. In the 12-week study pooled peritoneal cells from 3 mice were used to generate four 1ml wells (two unstimulated and two LPS-treated) for an n=4 per treatment group. With 8–9 mice/group in the 40-week study pooled peritoneal cells from 2–3 mice were used to generate two 1ml wells (one unstimulated and one LPS-treated) for an n=3 group.',\n",
       "   'offset': 9429,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Levels of IL-6R in the livers of individual mice were examined by Western blotting using goat IgG anti-mouse IL-6Rα (R&D Systems, Minneapolis, MN). The results were represented as mean ± standard deviation of the densitometric analysis of the IL-6R compared to reference protein GAPDH (glyceraldehyde 3-phosphate dehydrogenase) run in the same lane, and detected by rabbit IgG anti-GAPDH (Trevigen, Gaithersberg, MD).',\n",
       "   'offset': 10125,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Antibody Production',\n",
       "   'offset': 10548,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Using previously described methodology, microsomal liver protein (30 μg) obtained from an untreated MRL+/+ mouse was separated on 12% SDS-PAGE, electrotransferred onto nitrocellulose, and subsequently probed with pooled sera (1:500) obtained from control or TCE-treated MRL+/+ mice followed by HRP-conjugated polyclonal goat anti-mouse IgG (1:4,000). Densitometric analysis of mouse myeloma IgG run in adjoining lanes was used to normalize exposure times for the individual Western blots.',\n",
       "   'offset': 10568,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Statistics',\n",
       "   'offset': 11058,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The data are presented as means ± standard deviations. Assays were conducted using samples from 8–12 individual mice per treatment group or samples from equal numbers of pooled cells for n=3 or 4 per treatment group. The threshold for statistical significance was set at α = 0.05. Differences between experimental groups were tested first with analysis of variance (ANOVA), and where the F test was significant, subsequent pairwise contrasts were tested using a two-sample t-test. Homogeneity of variance between groups was tested using studentized Breusch-Pagan, and normality of residuals using Shapiro-Wilk. Where significant deviations from homoscedasticity or normality were observed, the non-parametric Kruskal-Wallis and Wilcoxon rank sum tests were applied instead of ANOVA/t-test. For experiments involving multiple timepoints, ordinal logistic regression models were fitted and the significance of TCE exposure was computed using a likelihood-ratio test; this is a two-factor generalization of Kruskal-Wallis. Linear regressions were fitted to evaluate the relationship between gene expression and histopathology, with statistical significance judged using an F test. Macrophage concentration and gene expression values were right-skewed, and therefore these data were log-transformed for statistical analyses.',\n",
       "   'offset': 11069,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Mechanistic Toxicodynamic Modeling',\n",
       "   'offset': 12395,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The effects of IL-6 signaling on liver events in TCE-treated mice were modeled to link changes in TCE mediated IL-6 signaling outcomes to the observed pathology following low-dose chronic exposure to TCE. The objective in developing a mathematical model for chronic, low dose exposure to TCE was to provide a means to quantitatively describe the role of IL-6 as a maintenance mechanism and predict downstream effects, such as changes in pathology, due to modifications of this IL-6 repair pathway. To this end, a time-dependent mathematical description of the health state of discrete volumes of liver (“liver units”) and the IL-6 and TCE-dependent transition between these health states was developed. Using in vivo results reported in this study, this toxicodynamic model will create a link between TCE exposure and the resulting histopathology. While not measurable in vivo, the state and number density of individual liver units serve as an intermediate measure to quantify the relationship between impaired cytokine signaling and the resulting autoimmune hepatitis.',\n",
       "   'offset': 12430,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Liver unit health state model',\n",
       "   'offset': 13505,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'For the purposes of mathematical modeling, the characteristics of the liver units (LUs) were as follows: the entire liver comprises LUs, each of which is of equal volume; an LU is relatively small in volume compared to that of the entire liver, but consists of a large enough number of cells to be represented as a continuum; each LU exists in one of three health states: healthy (H), compromised (C), and inflamed (I), and is homogeneous with respect to its properties and health state; and the health state of the entire liver may be estimated through a number-weighted average of the health states of the constituent LUs. Table 1 lists the characteristics and assumptions for each mathematical state.',\n",
       "   'offset': 13535,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The resulting transition between health states is described by',\n",
       "   'offset': 14239,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '  where the k’s represent transition rates, the subscripts DAM and REP refer to damage- and repair-associated phenomena, respectively, and the subscripts H-C and C-I refer to their respective transition pathways.',\n",
       "   'offset': 14302,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The corresponding system of differential equations governing the time-dependent fraction of LUs in each state may be written as',\n",
       "   'offset': 14517,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Here, t is time and [H], [C], and [I] are the fractions of LUs in the healthy, compromised, and inflamed states, respectively. It is assumed that initially (t=0), [H]=1 and [C]=[I]=0.',\n",
       "   'offset': 14645,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'To complete the mathematical description of this system, four major assumptions were made:',\n",
       "   'offset': 14829,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'LUs normally exist in a state of IL-6-mediated hepatocyte turnover and protection.',\n",
       "   'offset': 14920,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Events such as TCE exposure can initiate inflammatory processes and move the LUs into the “C” state. However, protective mechanisms mediated by IL-6 normally restore the LUs from the “C” to the “H” state. The rates of repair, kREP,H-C and kREP,C-I, are dependent on the fraction of IL-6 produced from the macrophage and the IL-6r expressed by the hepatocyte. If homeostasis levels of IL-6 and IL-6r are present, these pathways operate at the optimal repair rates.',\n",
       "   'offset': 15003,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE initiates inflammatory processes (e.g. redox disequilibrium) that move the LUs from the “H” to “C” state. It also decreases the protective effects of IL-6-signaling that would normally restore the LUs to the “H” state. These dual effects of TCE allow the inflammatory processes to progress and to move the LUs from the “C” to the “I” state.',\n",
       "   'offset': 15479,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Autoimmune hepatitis pathology does not occur without TCE, even if the IL-6 pathway is impaired in some other way.',\n",
       "   'offset': 15844,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Consistent with these assumptions, the rate terms above can be further specified as follows:   where k1, k2, k3, and k4 are constants to be estimated using experimental data, fTCE represents the administered TCE dose normalized by the upper dose used in this study (0.5 mg/ml or a time-weighted average of approximately 54 mg/kg/day), and fIL6 is the fraction of IL-6 expressed by the macrophage compared to control levels. To express the dependence of fIL6 on TCE dose, a sub-model based on a saturation mechanism was used:   where α and β are constants to be derived from experimental data.',\n",
       "   'offset': 15959,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Predicting liver pathology scores',\n",
       "   'offset': 16559,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'To compute overall liver pathology scores, the [H], [C], and [I] calculated from equations (2), (3), and (4) at the desired time point were used as weighting factors for the individual PS values corresponding to each of the model states. Mathematically, this can be expressed as',\n",
       "   'offset': 16593,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '  where PSs is the pathology score of a LU in state s (see Table 1).',\n",
       "   'offset': 16872,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Software and modeling tools',\n",
       "   'offset': 16941,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The system of differential equations were solved using a fourth-order Runge-Kutta method implemented in the Python programming language (v2.7.6) [https://www.python.org/]. Parameter estimation was conducted using lsqfit (v4.6.1) [https://github.com/gplepage/lsqfit], a software package for non-linear least-squares fitting of noisy data.',\n",
       "   'offset': 16969,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Results',\n",
       "   'offset': 17307,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Dose-dependent effects of TCE on peritoneal macrophage activity',\n",
       "   'offset': 17315,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Since autoimmune diseases and hypersensitivity disorders in humans involve an ill-defined genetic component, we use young “autoimmune-prone” female MRL+/+ mice to study the immunotoxicity of TCE. As observed previously, TCE exposure did not alter weight gain or water consumption (data not shown). Peritoneal macrophages from the mice exposed to different concentrations of TCE for 12 weeks were examined for the production of macrophage-derived cytokines IL-6 and IL-1β. Macrophage secretion of IL-1β was unchanged by exposure to TCE (Figure 1). The peritoneal macrophages collected from control mice secreted low but measurable levels of IL-6 even in the absence of LPS. Stimulation with LPS increased IL-6 production in all groups. However, both LPS-dependent and LPS-independent IL-6 production was suppressed in a dose-dependent manner in peritoneal macrophages from mice treated for 12 weeks with TCE. For example, LPS-induced IL-6 production in mice exposed to 0.5 mg/ml TCE was 70% lower than that of controls.',\n",
       "   'offset': 17379,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IL-6 was also inhibited at the transcriptional level in macrophages from TCE-treated mice (Figure 2). Although LPS stimulation increased Il6 expression, this effect was significantly suppressed in macrophages from mice treated with 0.1 or 0.5 mg/ml TCE as compared to control mice. Once again the suppressive effects of TCE were confined to IL-6, and did not encompass expression of genes for other macrophage-derived cytokines, including Lt-α, IL-12, or IL-10. Taken together, a 12-week exposure to TCE selectively suppressed IL-6 gene expression and protein production by peritoneal macrophages in a dose-dependent manner. The ability of TCE to alter expression of genes for other macrophage-derived cytokines was intermittent and not dose-dependent.',\n",
       "   'offset': 18404,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Time-dependent effects of TCE on peritoneal macrophage gene expression',\n",
       "   'offset': 19161,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In a second study designed to examine time-dependency of TCE-induced effects mice were given drinking water alone or with 0.5 mg/ml TCE for 4, 10, 16, 22, 28, 34 or 40 weeks. TCE exposure did not alter the number of PEC recovered at any of the time points (data not shown). Once again TCE suppressed production of IL-6 (Figure 3). Also evident, but as yet unexplained, was the general time-dependent decrease in IL-6 production in both treatment and control groups. Production of TNF-α was not affected by TCE exposure. A longitudinal evaluation of cytokine gene expression showed that the TCE-induced decrease in Il6 expression by peritoneal macrophages was evident by 16 weeks of exposure (Figure 4). The time-dependent expression of several other genes for macrophage-derived cytokines, IL1b, Il12, and Mmp12 was for the most part unaltered by exposure to TCE (Figure 4 and data not shown). Thus, the primary effects of exposure to TCE on peritoneal macrophages was a decrease in Il6 that was maintained for the duration of the study.',\n",
       "   'offset': 19232,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Time-dependent effects of TCE on liver events',\n",
       "   'offset': 20274,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Most of the protective and/or regenerative events in T cell-mediated liver injury are triggered by IL-6 signaling that is initiated when IL-6 binds to a complex comprised of the transmembrane protein gp130 and the IL-6R on hepatocytes. As shown in Figure 5 hepatic expression of Il6r was suppressed by TCE at several time points, and only approached control values at the last time point. Protein levels of IL-6R were also lower in the livers of the TCE-treated mice. The gene that encoded for the other subunit in the IL-6R family, Gp130, was suppressed by TCE at early time points. Expression of IL-6 itself in the liver was undetectable (data not shown). Another molecule important in hepatoprotection is the transcription factor EGR-1. EGR-1 binds to the promoter region of Il6, and reciprocally, is important in mediating signaling from the IL-6R/STAT3 pathway. Expression of egr1 in the liver was suppressed midway through the TCE exposure, but then rebounded at the final 40-week time point. Increased levels of pro-inflammatory cytokines/chemokines such as TNF-α, osteopontin, serum amyloid A (SAA) and CXCL1 have been implicated in the induction or progression of chronic liver inflammation. Hepatic expression of these Saa2, Cxcl1 and Spp1 (encodes for osteopontin) were for the most part unchanged or decreased during all but the last 40-week time point of TCE exposure. Thus, unlike IL-6R associated genes hepatic expression of several pro-inflammatory cytokines and chemokines was primarily unchanged or decreased by TCE exposure until the last time point when expression was dramatically reversed in select TCE-treated mice. These results showed that during most of the exposure TCE appeared to negatively impact liver repair rather than directly promote inflammation. Only at the last time point was this reversed; several pro-inflammatory cytokines/chemokines increased expression while the negative effect on hepatoprotective genes was overturned.',\n",
       "   'offset': 20320,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Histopathology in the form of lymphoplasmacytic portal infiltrate and lobular inflammation in the liver was not noted until week 28 of TCE exposure, and became more robust during the course of the 40-week experiment (Figure 6A). This pathology was characteristic of the early stages of autoimmune hepatitis; hepatocellular necrosis was only noted in a couple of instances. The mice were also examined for the generation of anti-liver antibodies as another readout of immune-mediated liver disease (Figure 6B). MRL+/+ mice are noted for their age-dependent increase in the production of autoantibodies such as anti-nuclear antibodies, even in the absence of toxicant exposure. In accord with this predisposition the baseline production of anti-liver antibodies became more abundant in control mice at the 40 week time point. However, exposure to TCE further increased the levels and diversity of the anti-liver antibodies. Thus, the MRL+/+ mice treated with TCE for 40 weeks demonstrated liver inflammation and anti-liver autoantibody production consistent with AIH.',\n",
       "   'offset': 22289,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'To help determine functional relevance hepatic gene expression in individual mice at the 40-week time point were plotted against liver pathology scores in the same mice. The linear regression showed that liver pathology was most closely correlated with a decrease in Il6r (p=0.003)(Figure 6C). Correlations between liver histopathology and expression of Egr1 and Spp1 were also moderately significant (p<0.07). There was no significant correlation between liver histopathology and expression of Tnfa or Cxcl1. This evaluation suggested that liver pathology was most closely correlated with a decrease in IL-6 signaling as reflected in a decrease in the expression of Il6r and Egr1 in the liver. There was also correlation with expression of Spp1, suggesting that the reversal of the TCE-induced decrease in OPN observed in the liver played a role in liver pathology.',\n",
       "   'offset': 23355,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Toxicodynamic model for liver response to TCE exposure',\n",
       "   'offset': 24222,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to develop a model to describe the effect of TCE on IL-6-mediated liver events certain required parameters were estimated based on the results described above.',\n",
       "   'offset': 24277,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Parameter estimation',\n",
       "   'offset': 24446,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In order to fit a curve that could be used to extrapolate IL-6 effects across a range of TCE doses values of α and β in the IL-6 submodel, Eq. (4), were estimated using a nonlinear least-squares approach with the non-LPS induced IL-6 results presented in Fig. 1. The resulting parameters values, mean (variance), were found to be α = 1.01 (0.01) and β = 0.071 (0.003). Figure 7A illustrates the resulting fit of the experimental data to the IL-6 submodel.',\n",
       "   'offset': 24467,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'It was similarly necessary to fit a curve to extrapolate liver pathology based on time of TCE exposure. The rate constants, ki, defined in Eq. (3), were estimated based on experimental time-course pathology scores (Figure 6A) to be k1 = 101.5 (98.0), k2 = 0.39 (0.18), k3 = 1.02 (0.08), and k4 = 0.21 (0.16). The resulting fit of the data to the mathematical model is depicted in Figure 7B. The uncertainty shown in model simulations results from both the uncertainty in the parameters associated with the IL-6 submodel and that from in vivo pathology scores.',\n",
       "   'offset': 24937,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Simulations of liver unit health states and the dose response',\n",
       "   'offset': 25497,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Following parameter estimation, simulations of time-course LU health were conducted. Figure 8 illustrates results from several such studies, where the fraction of LUs in a particular health state are shown as a function of time at the two highest doses used in the experimental study. For the 0.1 mg/ml dose (Figure 8A), almost all of the LUs are in a healthy state. However, as the external TCE dose is increased to 0.5 mg/ml (Figure 8B), the abundance of healthy LUs decreases while those in the compromised/inflamed state increase in a non-linear manner. At doses less than 0.1 mg/ml, simulations indicated that virtually all of the LUs were in the healthy (H) state over time.',\n",
       "   'offset': 25559,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'One of the benefits of the mathematical model is the prediction of system variables and endpoints not directly measured during the course of the studies. For instance, using 40-week pathology scores as an endpoint, the model was used to generate a dose-response curve (Figure 9). This curve can be used to relate this endpoint to any dose within the predicted range. As an example, for a benchmark response level (BML) corresponding to mild inflammation of 25% or less of the portal regions of the liver (PS=2), the benchmark dose (BMD) was estimated to be fTCE = 0.55, corresponding to a 0.27 mg/ml dose, or approximately 37 mg/kg/day, of continuous TCE exposure.',\n",
       "   'offset': 26240,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Examining the impact of varying relative rates of damage and repair',\n",
       "   'offset': 26905,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_3', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Another important benefit to the mathematical modeling is the ability to vary system parameters and observe the effects on system states of interest. Here, the effects of varying the relative rates of damage and repair in the H-C and C-I state transitions were investigated through a parametric study. Since each damage and repair pathway is first order with respect to the abundance of LUs, ratios of rate constants were defined:  ',\n",
       "   'offset': 26973,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'By conducting simulations with varying values for κ (Figure 10), the effects of relative rates of repair and damage in the system could be examined. If κ ≫ 1, the repair mechanism dominates and the LUs tend toward a relatively low value of PS, even at higher TCE dose; conversely, if κ ≪ 1, the damage mechanism dominates for the given pathway and LUs may acquire high values of PS values, even at relatively low doses. As can be seen, the shape and nonlinearity of the dose-response curve is highly dependent on the relative rates of repair and damage in both of the health state transitions. Overall, such predictions can help in the understanding of the interactions in this system and lend insight into the effects of non-TCE mediated events, such as additional stress from other hepatotoxicants or further impaired IL-6 repair mechanisms.',\n",
       "   'offset': 27406,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'DISCUSSION',\n",
       "   'offset': 28264,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_1', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'MRL+/+ mice can spontaneously develop autoimmune diseases such as lupus nephritis, pancreatitis, and Sjogren’s syndrome late in life (after 1-year of age). However, before they reach one year of age most female MRL+/+ mice do not exhibit autoimmune tissue pathology, and are often used to examine the autoimmune-promoting capacity of a toxicant such as TCE. Based on water consumption and TCE degradation in the water bottles, the mice given water containing TCE at 0.02, 0.1 or 0.5 mg/ml for 12 weeks were exposed to TCE at time-weighted levels of approximately 3, 14 or 64 mg/kg/day, respectively. Even the highest exposure is lower than the current 8-hour Permissible Exposure Limit [established by the Occupational Safety and Health Administration (OSHA)] for TCE of 100 ppm or approximately 76 mg/kg/day.',\n",
       "   'offset': 28275,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The effects of TCE on macrophage activity have been primarily studied in inhalation models. Inhalation exposure to TCE was shown to increase susceptibility to respiratory bacterial infection in mice, and to suppress phagocytosis in lung macrophages. Similarly, multiple inhalation exposures to TCE lowered resistance to respiratory streptococcus infection. Although the mechanism for this suppressive effect of inhaled TCE on macrophages was not defined, others have shown that an IL-6 deficiency increases susceptibility to viral and bacterial respiratory infections. The results of the current study showed that oral exposure to TCE suppressed IL-6 at the level of protein production and gene expression in macrophages.',\n",
       "   'offset': 29087,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IL-6 is a pleiotropic cytokine, which can make it difficult to predict the cumulative impact of its altered production. Elevated levels of IL-6 in the blood have been observed in a number of pathological conditions associated with chronic inflammation including rheumatoid arthritis, systemic lupus erythematosus, and active disease in Guillain-Barre syndrome. IL-6 did not reach detectable levels in the blood of control or TCE-treated mice in the current study. Circulating levels of IL-6 are increased in children with AIH type 1, but not with AIH type 2, the type of AIH that most closely resembles TCE-induced disease in MRL+/+ mice. Some studies of idiopathic autoimmune liver disease in humans have found increased levels of IL-6 in liver biopsies, while other studies of autoimmune hepatitis have demonstrated decreased expression of hepatic Il6 in the liver. On the other hand, treatments to prevent or reverse immunological liver injury in mouse models have been associated with an increase in liver expression of Il6. Thus, the majority of studies suggest that in the liver IL-6 is primarily protective. Increases in hepatic levels of IL-6 in some humans with AIH may represent a compensatory rather than pathological mechanism. Alternatively, changes in IL-6 may be specific for a certain stage of disease development, type of autoimmune hepatitis (e.g. type 1 vs type 2), or cell type (e.g. peritoneal exudate macrophages vs Kupffer cells). Deletion of IL-6 in mice deficient in TGF-β receptor II improved colitis but exacerbated autoimmune cholangitis in association with increased numbers of activated T cells.',\n",
       "   'offset': 29809,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cytokine production by macrophages from MRL+/+ mice is reportedly aberrant even in the absence of TCE exposure. LPS-induced production of IL-6, IL-1β, TNF-α, and IL-12 by macrophages from untreated MRL+/+ mice were all significantly decreased in comparison to macrophages from C57BL/6, BALB/c or A/J mice. Of these macrophage-derived cytokines only IL-6 was found in the present study to be further decreased by TCE exposure.',\n",
       "   'offset': 31438,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'In addition to a decrease in macrophage-derived IL-6, TCE suppressed liver expression of Il6r and gp130, the dual components of the IL-6R. This TCE-induced decrease would seem to further ensure the lack of IL-6 signaling in the liver. The IL-6-induced liver protection to T cell-mediated liver injury has been attributed to a downstream increase in acute phase protein serum amyloid A2, (SAA2). TCE suppressed hepatic expression of Saa2 at two time points late in the exposure period, thus seeming to prevent the upregulation of this molecules needed for liver regeneration. Egr1 is a transcription factor required for wound healing, and which has been identified as a negative regulator of carbon tetrachloride-induced hepatotoxicity. Egr1 has been described as both a trigger and a target for IL-6. Only at the final time point did TCE increase expression of Egr1 and Saa2. It is not known why the earlier TCE-induced suppression was reversed, but presumably the late recovery of these genes was not sufficient to protect against liver damage.',\n",
       "   'offset': 31871,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The contribution of TCE to AIH in the present model is multidimensional; the healthy-to-inflamed state model described here can be amended to include more immune parameters such as the contribution of CD4+ T cells as they are characterized. However, even in its present state, the model facilitated point-of-departure predictions based on dose-dependent changes in liver pathology. The model stemmed from the linear regression analyses showing that liver pathology in TCE-treated mice was best correlated with the decreased liver expression of macrophage Il-6r. We now have the tools to predict liver pathology based on relative rates of liver repair and damage. In addition to its predicted effect on IL-6 signaling the model also infers that TCE initiates inflammatory processes that transition LUs from “H” to “C”. These processes were not investigated in this study, but probably include, but are not restricted to, alterations in redox equilibrium. In a previous study, a metabolomics analysis following chronic 32 week exposure to 0.5 mg/ml in MRL+/+ mice revealed significant alterations in several metabolites (e.g., cystathionine) involved in the generation of glutathione, which functions as the major intracellular antioxidant against oxidative stress and plays an important role in the detoxification of reactive oxygen species and subsequent oxidative damage from pro-oxidant environmental exposures. Others have shown the functional significance of oxidative stress in TCE-induced liver pathology. IL-6 has been shown to inhibit oxidative stress and steatosis in the liver. Consequently, a TCE-induced loss of IL-6 signaling in the liver would be expected to exacerbate associated oxidative-stress and resulting inflammation. The first stage model development described here (i.e. generation of equations and description of parameters) was based on data from two different experiments, albeit with some differences in experimental design. Obtaining new data to validate and extend this model will be included in the design of future chronic TCE exposure studies.',\n",
       "   'offset': 32917,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'This is a PDF file of an unedited manuscript that has been accepted for publication. As a service to our customers we are providing this early version of the manuscript. The manuscript will undergo copyediting, typesetting, and review of the resulting proof before it is published in its final citable form. Please note that during the production process errors may be discovered which could affect the content, and all legal disclaimers that apply to the journal pertain.',\n",
       "   'offset': 35002,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'footnote', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '                ',\n",
       "   'offset': 35475,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'footnote', 'section_type': 'DISCUSS'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Abbreviations',\n",
       "   'offset': 35492,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE',\n",
       "   'offset': 35506,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'trichloroethylene',\n",
       "   'offset': 35510,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'ABBR'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Reference List',\n",
       "   'offset': 35528,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Aberrant macrophage cytokine production is a conserved feature among autoimmune-prone mouse strains: elevated interleukin (IL)-12 and an imbalance in tumor necrosis factor-alpha and IL-10 define a unique cytokine profile in macrophages from young nonobese diabetic mice',\n",
       "   'offset': 35543,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Grant;given-names:C',\n",
       "    'name_3': 'surname:Kaser;given-names:SB',\n",
       "    'name_0': 'surname:Alleva;given-names:DG',\n",
       "    'name_1': 'surname:Pavlovich;given-names:RP',\n",
       "    'name_4': 'surname:Beller;given-names:DI',\n",
       "    'pub-id_pmid': '10909966',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '49',\n",
       "    'source': 'Diabetes',\n",
       "    'fpage': '1106',\n",
       "    'year': '2000',\n",
       "    'lpage': '1115',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The effects of inhalation of organic chemical air contaminants on murine lung host defenses',\n",
       "   'offset': 35813,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Graham;given-names:JA',\n",
       "    'name_3': 'surname:Miller;given-names:FJ',\n",
       "    'name_0': 'surname:Aranyi;given-names:C',\n",
       "    'name_1': 'surname:O’Shea;given-names:WJ',\n",
       "    'pub-id_pmid': '3519345',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '6',\n",
       "    'source': 'Fundam Appl Toxicol',\n",
       "    'fpage': '713',\n",
       "    'year': '1986',\n",
       "    'lpage': '720',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interleukin-6 protects hepatocytes from CCl4-mediated necrosis and apoptosis in mice by reducing MMP-2 expression',\n",
       "   'offset': 35905,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Gupta;given-names:R',\n",
       "    'name_3': 'surname:Li;given-names:W',\n",
       "    'name_0': 'surname:Bansal;given-names:MB',\n",
       "    'name_1': 'surname:Kovalovich;given-names:K',\n",
       "    'name_6': 'surname:Alvarez;given-names:CE',\n",
       "    'name_7': 'surname:Safadi;given-names:R',\n",
       "    'name_4': 'surname:Agarwal;given-names:A',\n",
       "    'name_5': 'surname:Radbill;given-names:B',\n",
       "    'name_8': 'surname:Fiel;given-names:MI',\n",
       "    'name_9': 'surname:Friedman;given-names:SL',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '42',\n",
       "    'pub-id_pmid': '15763341',\n",
       "    'source': 'J Hepatol',\n",
       "    'fpage': '548',\n",
       "    'year': '2005',\n",
       "    'name_10': 'surname:Taub;given-names:RA',\n",
       "    'lpage': '556',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Association between clinical symptoms and lymphocyte abnormalities in a population with chronic domestic exposure to industrial solvent-contaminated domestic water supply and a high incidence of leukemia',\n",
       "   'offset': 36019,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Ozonoff;given-names:DM',\n",
       "    'name_3': 'surname:Baldwin;given-names:RW',\n",
       "    'name_0': 'surname:Byers;given-names:VS',\n",
       "    'name_1': 'surname:Levin;given-names:AS',\n",
       "    'pub-id_pmid': '3260823',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '27',\n",
       "    'source': 'Cancer Immunol Immunother',\n",
       "    'fpage': '77',\n",
       "    'year': '1988',\n",
       "    'lpage': '82',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Chronic exposure to trichloroethene causes early onset of SLE-like disease in female MRL +/+ mice',\n",
       "   'offset': 36223,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Boor;given-names:PJ',\n",
       "    'name_3': 'surname:Kondraganti;given-names:S',\n",
       "    'name_0': 'surname:Cai;given-names:P',\n",
       "    'name_1': 'surname:Konig;given-names:R',\n",
       "    'name_6': 'surname:Ansari;given-names:GA',\n",
       "    'name_4': 'surname:Kaphalia;given-names:BS',\n",
       "    'name_5': 'surname:Khan;given-names:MF',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '228',\n",
       "    'pub-id_pmid': '18234256',\n",
       "    'source': 'Toxicol Appl Pharmacol',\n",
       "    'fpage': '68',\n",
       "    'year': '2008',\n",
       "    'lpage': '75',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'CD4 T cells in hepatic immune tolerance',\n",
       "   'offset': 36321,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '28',\n",
       "    'name_0': 'surname:Carambia;given-names:A',\n",
       "    'name_1': 'surname:Herkel;given-names:J',\n",
       "    'pub-id_pmid': '19720498',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '34',\n",
       "    'source': 'J Autoimmun',\n",
       "    'fpage': '23',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cytokine IL-6 and IL-10 as biomarkers in systemic lupus erythematosus',\n",
       "   'offset': 36361,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kim;given-names:HA',\n",
       "    'name_3': 'surname:Yun;given-names:JM',\n",
       "    'name_0': 'surname:Chun;given-names:HY',\n",
       "    'name_1': 'surname:Chung;given-names:JW',\n",
       "    'name_6': 'surname:Kim;given-names:SH',\n",
       "    'name_7': 'surname:Park;given-names:HS',\n",
       "    'name_4': 'surname:Jeon;given-names:JY',\n",
       "    'name_5': 'surname:Ye;given-names:YM',\n",
       "    'name_8': 'surname:Suh;given-names:CH',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '27',\n",
       "    'pub-id_pmid': '17587156',\n",
       "    'source': 'J Clin Immunol',\n",
       "    'fpage': '461',\n",
       "    'year': '2007',\n",
       "    'lpage': '466',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '',\n",
       "   'offset': 36431,\n",
       "   'relations': [],\n",
       "   'infons': {'source': 'Assessing the Human Health Risks of Trichloroethylene: Key Scientific Issues',\n",
       "    'year': '2006',\n",
       "    'type': 'ref',\n",
       "    'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The liver as a lymphoid organ',\n",
       "   'offset': 36432,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '63',\n",
       "    'name_0': 'surname:Crispe;given-names:IN',\n",
       "    'pub-id_pmid': '19302037',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '27',\n",
       "    'source': 'Annu Rev Immunol',\n",
       "    'fpage': '147',\n",
       "    'year': '2009',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Localized scleroderma after exposure to organic solvents',\n",
       "   'offset': 36462,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Szegedi;given-names:G',\n",
       "    'lpage': '401',\n",
       "    'name_0': 'surname:Czirjak;given-names:L',\n",
       "    'name_1': 'surname:Pocs;given-names:E',\n",
       "    'pub-id_pmid': '7873829',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '189',\n",
       "    'source': 'Dermatology',\n",
       "    'fpage': '399',\n",
       "    'year': '1994',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cause-specific mortality among Rhode Island jewelry workers',\n",
       "   'offset': 36519,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '593',\n",
       "    'name_0': 'surname:Dubrow;given-names:R',\n",
       "    'name_1': 'surname:Gute;given-names:DM',\n",
       "    'pub-id_pmid': '2961258',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '12',\n",
       "    'source': 'Am J Ind Med',\n",
       "    'fpage': '579',\n",
       "    'year': '1987',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IL-6-deficient mice are susceptible to ethanol-induced hepatic steatosis: IL-6 protects against ethanol-induced oxidative stress and mitochondrial permeability transition in the liver',\n",
       "   'offset': 36579,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kim;given-names:WH',\n",
       "    'name_3': 'surname:Radaeva;given-names:S',\n",
       "    'name_0': 'surname:El-Assal;given-names:O',\n",
       "    'name_1': 'surname:Hong;given-names:F',\n",
       "    'name_4': 'surname:Gao;given-names:B',\n",
       "    'pub-id_pmid': '16219169',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '1',\n",
       "    'source': 'Cell Mol Immunol',\n",
       "    'fpage': '205',\n",
       "    'year': '2004',\n",
       "    'lpage': '211',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Scleroderma after occupational exposure to trichloroethylene and trichlorethane',\n",
       "   'offset': 36763,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '181',\n",
       "    'name_0': 'surname:Flindt-Hansen;given-names:H',\n",
       "    'name_1': 'surname:Isager;given-names:H',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '95',\n",
       "    'source': 'Toxicol Lett',\n",
       "    'fpage': '173',\n",
       "    'year': '1987',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hepatoprotective and anti-inflammatory cytokines in alcoholic liver disease',\n",
       "   'offset': 36843,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '93',\n",
       "    'name_0': 'surname:Gao;given-names:B',\n",
       "    'pub-id_pmid': '22320924',\n",
       "    'issue': 'Suppl 2',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '27',\n",
       "    'source': 'J Gastroenterol Hepatol',\n",
       "    'fpage': '89',\n",
       "    'year': '2012',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Use of transcriptomics and metabolomics to delineate liver events associated with trichloroethylene-induced autoimmune hepatitis',\n",
       "   'offset': 36919,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Pumford;given-names:NR',\n",
       "    'name_3': 'surname:Han;given-names:T',\n",
       "    'name_0': 'surname:Gilbert;given-names:KM',\n",
       "    'name_1': 'surname:Przybyla;given-names:B',\n",
       "    'name_6': 'surname:Holland;given-names:RD',\n",
       "    'name_7': 'surname:Doss;given-names:JC',\n",
       "    'name_4': 'surname:Fuscoe;given-names:J',\n",
       "    'name_5': 'surname:Schnackenberg;given-names:LK',\n",
       "    'name_8': 'surname:MacMillan-Crow;given-names:LA',\n",
       "    'name_9': 'surname:Blossom;given-names:SJ',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '22',\n",
       "    'source': 'Chem Res Tox',\n",
       "    'fpage': '626',\n",
       "    'year': '2008',\n",
       "    'lpage': '632',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene--a review of the literature from a health effects perspective',\n",
       "   'offset': 37048,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '307',\n",
       "    'name_0': 'surname:Gist;given-names:GL',\n",
       "    'name_1': 'surname:Burg;given-names:JR',\n",
       "    'pub-id_pmid': '7482570',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '11',\n",
       "    'source': 'Toxicol Ind Health',\n",
       "    'fpage': '253',\n",
       "    'year': '1995',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hepatic catabolism of serum amyloid A during an acute phase response and chronic inflammation',\n",
       "   'offset': 37128,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '250',\n",
       "    'name_0': 'surname:Gollaher;given-names:CJ',\n",
       "    'name_1': 'surname:Bausserman;given-names:LL',\n",
       "    'pub-id_pmid': '2113283',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '194',\n",
       "    'source': 'Proc Soc Exp Biol Med',\n",
       "    'fpage': '245',\n",
       "    'year': '1990',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Serum IL-6 and IL-21 are associated with markers of B cell activation and structural progression in early rheumatoid arthritis: results from the ESPOIR cohort',\n",
       "   'offset': 37222,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Lukas;given-names:C',\n",
       "    'name_3': 'surname:Ducot;given-names:B',\n",
       "    'name_0': 'surname:Gottenberg;given-names:JE',\n",
       "    'name_1': 'surname:Dayer;given-names:JM',\n",
       "    'name_6': 'surname:Saraux;given-names:A',\n",
       "    'name_7': 'surname:Roux-Lombard;given-names:P',\n",
       "    'name_4': 'surname:Chiocchia;given-names:G',\n",
       "    'name_5': 'surname:Cantagrel;given-names:A',\n",
       "    'name_8': 'surname:Mariette;given-names:X',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '71',\n",
       "    'pub-id_pmid': '22532637',\n",
       "    'source': 'Ann Rheum Dis',\n",
       "    'fpage': '1243',\n",
       "    'year': '2012',\n",
       "    'lpage': '1248',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'CD4+ T cell activation and induction of autoimmune hepatitis following trichloroethylene treatment in MRL+/+ mice',\n",
       "   'offset': 37381,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Lamps;given-names:LW',\n",
       "    'name_3': 'surname:Pumford;given-names:NR',\n",
       "    'name_0': 'surname:Griffin;given-names:JM',\n",
       "    'name_1': 'surname:Gilbert;given-names:KM',\n",
       "    'pub-id_pmid': '11006364',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '57',\n",
       "    'source': 'Toxicol Sci',\n",
       "    'fpage': '345',\n",
       "    'year': '2000',\n",
       "    'lpage': '352',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A scleroderma-resembling disease-exposure to trichloroethylene and trichloroethane, is there a causal connection?',\n",
       "   'offset': 37495,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '808',\n",
       "    'name_0': 'surname:Hansen;given-names:BL',\n",
       "    'name_1': 'surname:Isager;given-names:H',\n",
       "    'pub-id_pmid': '3363711',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '150',\n",
       "    'source': 'Ugeskr Laeger',\n",
       "    'fpage': '805',\n",
       "    'year': '1988',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Aberrant cytokine regulation in macrophages from young autoimmune-prone mice: evidence that the intrinsic defect in MRL macrophage IL-1 expression is transcriptionally controlled',\n",
       "   'offset': 37609,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Levine;given-names:JS',\n",
       "    'name_3': 'surname:Beller;given-names:DI',\n",
       "    'name_0': 'surname:Hartwell;given-names:DW',\n",
       "    'name_1': 'surname:Fenton;given-names:MJ',\n",
       "    'pub-id_pmid': '7544869',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '32',\n",
       "    'source': 'Mol Immunol',\n",
       "    'fpage': '743',\n",
       "    'year': '1995',\n",
       "    'lpage': '751',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Transcriptional regulation of EGR-1 by the interleukin-1-JNK-MKK7-c-Jun pathway',\n",
       "   'offset': 37788,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Wolter;given-names:S',\n",
       "    'name_3': 'surname:Doerrie;given-names:A',\n",
       "    'name_0': 'surname:Hoffmann;given-names:E',\n",
       "    'name_1': 'surname:Ashouri;given-names:J',\n",
       "    'name_6': 'surname:Wagner;given-names:EF',\n",
       "    'name_7': 'surname:Troppmair;given-names:J',\n",
       "    'name_4': 'surname:Dittrich-Breiholz;given-names:O',\n",
       "    'name_5': 'surname:Schneider;given-names:H',\n",
       "    'name_8': 'surname:Mackman;given-names:N',\n",
       "    'name_9': 'surname:Kracht;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '283',\n",
       "    'pub-id_pmid': '18281687',\n",
       "    'source': 'J Biol Chem',\n",
       "    'fpage': '12120',\n",
       "    'year': '2008',\n",
       "    'lpage': '12128',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TNF-alpha is essential in the induction of fatal autoimmune hepatitis in mice through upregulation of hepatic CCL20 expression',\n",
       "   'offset': 37868,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Aoki;given-names:N',\n",
       "    'name_3': 'surname:Nishiura;given-names:H',\n",
       "    'name_0': 'surname:Iwamoto;given-names:S',\n",
       "    'name_1': 'surname:Kido;given-names:M',\n",
       "    'name_6': 'surname:Okazaki;given-names:T',\n",
       "    'name_7': 'surname:Chiba;given-names:T',\n",
       "    'name_4': 'surname:Maruoka;given-names:R',\n",
       "    'name_5': 'surname:Ikeda;given-names:A',\n",
       "    'name_8': 'surname:Watanabe;given-names:N',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '146',\n",
       "    'pub-id_pmid': '23178752',\n",
       "    'source': 'Clin Immunol',\n",
       "    'fpage': '15',\n",
       "    'year': '2013',\n",
       "    'lpage': '25',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Roles of interleukin-6 in activation of STAT proteins and recruitment of neutrophils during Escherichia coli pneumonia',\n",
       "   'offset': 37995,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Simms;given-names:BT',\n",
       "    'name_3': 'surname:Lupa;given-names:MM',\n",
       "    'name_0': 'surname:Jones;given-names:MR',\n",
       "    'name_1': 'surname:Quinton;given-names:LJ',\n",
       "    'name_4': 'surname:Kogan;given-names:MS',\n",
       "    'name_5': 'surname:Mizgerd;given-names:JP',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '193',\n",
       "    'pub-id_pmid': '16388483',\n",
       "    'source': 'J Infect Dis',\n",
       "    'fpage': '360',\n",
       "    'year': '2006',\n",
       "    'lpage': '369',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene causes generalized hypersensitivity skin disorders complicated by hepatitis',\n",
       "   'offset': 38114,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Huang;given-names:H',\n",
       "    'name_3': 'surname:Li;given-names:L',\n",
       "    'name_0': 'surname:Kamijima;given-names:M',\n",
       "    'name_1': 'surname:Wang;given-names:H',\n",
       "    'name_6': 'surname:Sakai;given-names:K',\n",
       "    'name_7': 'surname:Liu;given-names:H',\n",
       "    'name_4': 'surname:Shibata;given-names:E',\n",
       "    'name_5': 'surname:Lin;given-names:B',\n",
       "    'name_8': 'surname:Tsuchiyama;given-names:F',\n",
       "    'name_9': 'surname:Chen;given-names:J',\n",
       "    'lpage': '338',\n",
       "    'fpage': '328',\n",
       "    'year': '2008',\n",
       "    'section_type': 'REF',\n",
       "    'source': 'J Occup Health',\n",
       "    'type': 'ref',\n",
       "    'volume': '50',\n",
       "    'pub-id_pmid': '18540116',\n",
       "    'issue': '4',\n",
       "    'name_10': 'surname:Okamura;given-names:A',\n",
       "    'name_11': 'surname:Huang;given-names:X',\n",
       "    'name_12': 'surname:Hisanaga;given-names:N',\n",
       "    'name_13': 'surname:Huang;given-names:Z',\n",
       "    'name_14': 'surname:Ito;given-names:Y',\n",
       "    'name_15': 'surname:Takeuchi;given-names:Y',\n",
       "    'name_16': 'surname:Nakajima;given-names:T'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Spontaneous development of pancreatitis in the MRL/Mp strain of mice in autoimmune mechanism',\n",
       "   'offset': 38208,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Itoh;given-names:J',\n",
       "    'name_3': 'surname:Taniguchi;given-names:Y',\n",
       "    'name_0': 'surname:Kanno;given-names:H',\n",
       "    'name_1': 'surname:Nose;given-names:M',\n",
       "    'name_4': 'surname:Kyogoku;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '89',\n",
       "    'source': 'Clin Ex Immunol',\n",
       "    'fpage': '68',\n",
       "    'year': '1992',\n",
       "    'lpage': '73',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'The IL-6-gp130-STAT3 pathway in hepatocytes triggers liver protection in T cell-mediated liver injury',\n",
       "   'offset': 38301,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Assmus;given-names:U',\n",
       "    'name_3': 'surname:Roskams;given-names:T',\n",
       "    'name_0': 'surname:Klein;given-names:C',\n",
       "    'name_1': 'surname:Wustefeld;given-names:T',\n",
       "    'name_6': 'surname:Manns;given-names:MP',\n",
       "    'name_7': 'surname:Ernst;given-names:M',\n",
       "    'name_4': 'surname:Rose-John;given-names:S',\n",
       "    'name_5': 'surname:Muller;given-names:M',\n",
       "    'name_8': 'surname:Trautwein;given-names:C',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '115',\n",
       "    'pub-id_pmid': '15761498',\n",
       "    'source': 'J Clin Invest',\n",
       "    'fpage': '860',\n",
       "    'year': '2005',\n",
       "    'lpage': '869',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Protective effect of paeoniflorin on immunological liver injury induced by bacillus Calmette-Guerin plus lipopolysaccharide: modulation of tumour necrosis factor-alpha and interleukin-6 MRNA',\n",
       "   'offset': 38403,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Song;given-names:LH',\n",
       "    'lpage': '339',\n",
       "    'name_0': 'surname:Liu;given-names:DF',\n",
       "    'name_1': 'surname:Wei;given-names:W',\n",
       "    'pub-id_pmid': '16620297',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '33',\n",
       "    'source': 'Clin Exp Pharmacol Physiol',\n",
       "    'fpage': '332',\n",
       "    'year': '2006',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Progressive systemic sclerosis associated with exposure to trichloroethylene',\n",
       "   'offset': 38594,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Cannon;given-names:GW',\n",
       "    'name_3': 'surname:Colby;given-names:TV',\n",
       "    'name_0': 'surname:Lockey;given-names:JE',\n",
       "    'name_1': 'surname:Kelly;given-names:CR',\n",
       "    'name_4': 'surname:Aldrich;given-names:V',\n",
       "    'name_5': 'surname:Livingston;given-names:GK',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '29',\n",
       "    'pub-id_pmid': '3612322',\n",
       "    'source': 'J Occup Med',\n",
       "    'fpage': '493',\n",
       "    'year': '1997',\n",
       "    'lpage': '496',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Deletion of gp130 in myeloid cells modulates IL-6-release and is associated with more severe liver injury of Con A hepatitis',\n",
       "   'offset': 38671,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Kroy;given-names:DC',\n",
       "    'name_3': 'surname:Gassler;given-names:N',\n",
       "    'name_0': 'surname:Lutz;given-names:HH',\n",
       "    'name_1': 'surname:Sackett;given-names:SD',\n",
       "    'name_4': 'surname:Trautwein;given-names:C',\n",
       "    'pub-id_pmid': '22018663',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '91',\n",
       "    'source': 'Eur J Cell Biol',\n",
       "    'fpage': '576',\n",
       "    'year': '2012',\n",
       "    'lpage': '581',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Porphyromonas gingivalis antigens and interleukin-6 stimulate the production of monocyte chemoattractant protein-1 via the upregulation of early growth response-1 transcription in human coronary artery endothelial cells',\n",
       "   'offset': 38796,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Honda;given-names:T',\n",
       "    'name_3': 'surname:Yonezawa;given-names:D',\n",
       "    'name_0': 'surname:Maekawa;given-names:T',\n",
       "    'name_1': 'surname:Takahashi;given-names:N',\n",
       "    'name_6': 'surname:Tabeta;given-names:K',\n",
       "    'name_7': 'surname:Yamazaki;given-names:K',\n",
       "    'name_4': 'surname:Miyashita;given-names:H',\n",
       "    'name_5': 'surname:Okui;given-names:T',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '47',\n",
       "    'pub-id_pmid': '20016208',\n",
       "    'source': 'J Vasc Res',\n",
       "    'fpage': '346',\n",
       "    'year': '2010',\n",
       "    'lpage': '354',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Circulating levels of interleukin-6, interleukin-8, and tumor necrosis factor-alpha in children with autoimmune hepatitis',\n",
       "   'offset': 39016,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Massa;given-names:M',\n",
       "    'name_3': 'surname:Pignatti;given-names:P',\n",
       "    'name_0': 'surname:Maggiore;given-names:G',\n",
       "    'name_1': 'surname:De;given-names:BF',\n",
       "    'name_4': 'surname:Martini;given-names:A',\n",
       "    'pub-id_pmid': '7884614',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '20',\n",
       "    'source': 'J Pediatr Gastroenterol Nutr',\n",
       "    'fpage': '23',\n",
       "    'year': '1995',\n",
       "    'lpage': '27',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Effect of IL-6 deficiency on susceptibility to HSV-1 respiratory infection and intrinsic macrophage antiviral resistance',\n",
       "   'offset': 39138,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Brown;given-names:AS',\n",
       "    'name_3': 'surname:Carmichael;given-names:MD',\n",
       "    'name_0': 'surname:Murphy;given-names:EA',\n",
       "    'name_1': 'surname:Davis;given-names:JM',\n",
       "    'name_4': 'surname:Ghaffar;given-names:A',\n",
       "    'name_5': 'surname:Mayer;given-names:EP',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '28',\n",
       "    'pub-id_pmid': '18778200',\n",
       "    'source': 'J Interferon Cytokine Res',\n",
       "    'fpage': '589',\n",
       "    'year': '2008',\n",
       "    'lpage': '595',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Osteopontin: Versatile modulator of liver diseases',\n",
       "   'offset': 39259,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '30',\n",
       "    'name_0': 'surname:Nagoshi;given-names:S',\n",
       "    'pub-id_pmid': '23701387',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '44',\n",
       "    'source': 'Hepatol Res',\n",
       "    'fpage': '22',\n",
       "    'year': '2014',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Early growth response (EGR)-1 is required for timely cell-cycle entry and progression in hepatocytes after acute carbon tetrachloride exposure in mice',\n",
       "   'offset': 39310,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Nagy;given-names:LE',\n",
       "    'lpage': 'G1131',\n",
       "    'name_0': 'surname:Pritchard;given-names:MT',\n",
       "    'name_1': 'surname:Malinak;given-names:RN',\n",
       "    'pub-id_pmid': '21415413',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '300',\n",
       "    'source': 'Am J Physiol Gastrointest Liver Physiol',\n",
       "    'fpage': 'G1124',\n",
       "    'year': '2011',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Hepatic fibrosis is enhanced and accompanied by robust oval cell activation after chronic carbon tetrachloride administration to Egr-1-deficient mice',\n",
       "   'offset': 39461,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '2752',\n",
       "    'name_0': 'surname:Pritchard;given-names:MT',\n",
       "    'name_1': 'surname:Nagy;given-names:LE',\n",
       "    'pub-id_pmid': '20395449',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '176',\n",
       "    'source': 'Am J Pathol',\n",
       "    'fpage': '2743',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A new syndrome with pigmentation, scleroderma, gynaecomastia, Raynaud’s phenomenon and peripheral neuropathy',\n",
       "   'offset': 39611,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Heaton;given-names:KW',\n",
       "    'lpage': '440',\n",
       "    'name_0': 'surname:Saihan;given-names:EM',\n",
       "    'name_1': 'surname:Burton;given-names:JL',\n",
       "    'pub-id_pmid': '213095',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '99',\n",
       "    'source': 'Br J Dermatol',\n",
       "    'fpage': '437',\n",
       "    'year': '1978',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Suppression of pulmonary host defenses and enhanced susceptibility to respiratory bacterial infection in mice following inhalation exposure to trichloroethylene and chloroform',\n",
       "   'offset': 39722,\n",
       "   'relations': [],\n",
       "   'infons': {'lpage': '356',\n",
       "    'name_0': 'surname:Selgrade;given-names:MK',\n",
       "    'name_1': 'surname:Gilmour;given-names:MI',\n",
       "    'pub-id_pmid': '20925451',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '7',\n",
       "    'source': 'J Immunotoxicol',\n",
       "    'fpage': '350',\n",
       "    'year': '2010',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Impact of gender on exocrine gland inflammation in mouse models of Sjogren’s syndrome',\n",
       "   'offset': 39898,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Rocha;given-names:EM',\n",
       "    'name_3': 'surname:Da Silveira;given-names:LA',\n",
       "    'name_0': 'surname:Toda;given-names:I',\n",
       "    'name_1': 'surname:Sullivan;given-names:BD',\n",
       "    'name_4': 'surname:Wickham;given-names:LA',\n",
       "    'name_5': 'surname:Sullivan;given-names:DA',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '69',\n",
       "    'pub-id_pmid': '10504269',\n",
       "    'source': 'Exp Eye Res',\n",
       "    'fpage': '355',\n",
       "    'year': '1999',\n",
       "    'lpage': '366',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Genes for interleukin-1, interleukin-6, and tumor necrosis factor are expressed at markedly reduced levels in the livers of patients with severe liver disease',\n",
       "   'offset': 39986,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Guymarho;given-names:J',\n",
       "    'name_3': 'surname:Blanchard;given-names:B',\n",
       "    'name_0': 'surname:Tovey;given-names:MG',\n",
       "    'name_1': 'surname:Gugenheim;given-names:J',\n",
       "    'name_6': 'surname:Bismuth;given-names:H',\n",
       "    'name_7': 'surname:Reynes;given-names:M',\n",
       "    'name_4': 'surname:Vanden Broecke;given-names:C',\n",
       "    'name_5': 'surname:Gresser;given-names:I',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '10',\n",
       "    'pub-id_pmid': '1772964',\n",
       "    'source': 'Autoimmunity',\n",
       "    'fpage': '297',\n",
       "    'year': '1991',\n",
       "    'lpage': '310',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Retarded liver growth in interleukin-6-deficient and tumor necrosis factor receptor-1-deficient mice',\n",
       "   'offset': 40145,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Hisaoka;given-names:M',\n",
       "    'name_3': 'surname:Sandstedt;given-names:J',\n",
       "    'name_0': 'surname:Wallenius;given-names:V',\n",
       "    'name_1': 'surname:Wallenius;given-names:K',\n",
       "    'name_6': 'surname:Jansson;given-names:JO',\n",
       "    'name_4': 'surname:Ohlsson;given-names:C',\n",
       "    'name_5': 'surname:Kopf;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '142',\n",
       "    'pub-id_pmid': '11416016',\n",
       "    'source': 'Endocrinology',\n",
       "    'fpage': '2953',\n",
       "    'year': '2001',\n",
       "    'lpage': '2960',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Oxidative and nitrosative stress in trichloroethene-mediated autoimmune response',\n",
       "   'offset': 40246,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Ansari;given-names:GA',\n",
       "    'name_3': 'surname:Khan;given-names:MF',\n",
       "    'name_0': 'surname:Wang;given-names:G',\n",
       "    'name_1': 'surname:Cai;given-names:P',\n",
       "    'pub-id_pmid': '17123686',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '229',\n",
       "    'source': 'Toxicology',\n",
       "    'fpage': '186',\n",
       "    'year': '2007',\n",
       "    'lpage': '193',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'N-Acetylcysteine protects against trichloroethene-mediated autoimmunity by attenuating oxidative stress',\n",
       "   'offset': 40327,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Ma;given-names:H',\n",
       "    'name_3': 'surname:Ansari;given-names:GA',\n",
       "    'name_0': 'surname:Wang;given-names:G',\n",
       "    'name_1': 'surname:Wang;given-names:J',\n",
       "    'name_4': 'surname:Khan;given-names:MF',\n",
       "    'pub-id_pmid': '23993974',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '273',\n",
       "    'source': 'Toxicol Appl Pharmacol',\n",
       "    'fpage': '189',\n",
       "    'year': '2013',\n",
       "    'lpage': '195',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Comparative analysis of cytokine patterns in immunological, infectious, and oncological neurological disorders',\n",
       "   'offset': 40431,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Sommer;given-names:N',\n",
       "    'name_3': 'surname:Melms;given-names:A',\n",
       "    'name_0': 'surname:Weller;given-names:M',\n",
       "    'name_1': 'surname:Stevens;given-names:A',\n",
       "    'name_4': 'surname:Dichgans;given-names:J',\n",
       "    'name_5': 'surname:Wietholter;given-names:H',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '104',\n",
       "    'pub-id_pmid': '1940975',\n",
       "    'source': 'J Neurol Sci',\n",
       "    'fpage': '215',\n",
       "    'year': '1991',\n",
       "    'lpage': '221',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interleukin-6/glycoprotein 130-dependent pathways are protective during liver regeneration',\n",
       "   'offset': 40542,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Streetz;given-names:KL',\n",
       "    'name_3': 'surname:Betz;given-names:U',\n",
       "    'name_0': 'surname:Wuestefeld;given-names:T',\n",
       "    'name_1': 'surname:Klein;given-names:C',\n",
       "    'name_6': 'surname:Manns;given-names:MP',\n",
       "    'name_7': 'surname:Muller;given-names:W',\n",
       "    'name_4': 'surname:Lauber;given-names:J',\n",
       "    'name_5': 'surname:Buer;given-names:J',\n",
       "    'name_8': 'surname:Trautwein;given-names:C',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '278',\n",
       "    'pub-id_pmid': '12509437',\n",
       "    'source': 'J Biol Chem',\n",
       "    'fpage': '11281',\n",
       "    'year': '2003',\n",
       "    'lpage': '11288',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interleukin-6 signaling in liver-parenchymal cells suppresses hepatic inflammation and improves systemic insulin action',\n",
       "   'offset': 40633,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Konner;given-names:AC',\n",
       "    'name_3': 'surname:Gruber;given-names:S',\n",
       "    'name_0': 'surname:Wunderlich;given-names:FT',\n",
       "    'name_1': 'surname:Strohle;given-names:P',\n",
       "    'name_6': 'surname:Juntti-Berggren;given-names:L',\n",
       "    'name_7': 'surname:Li;given-names:LS',\n",
       "    'name_4': 'surname:Tovar;given-names:S',\n",
       "    'name_5': 'surname:Bronneke;given-names:HS',\n",
       "    'name_8': 'surname:van;given-names:RN',\n",
       "    'name_9': 'surname:Libert;given-names:C',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '12',\n",
       "    'pub-id_pmid': '20816090',\n",
       "    'source': 'Cell Metab',\n",
       "    'fpage': '237',\n",
       "    'year': '2010',\n",
       "    'name_10': 'surname:Berggren;given-names:PO',\n",
       "    'name_11': 'surname:Bruning;given-names:JC',\n",
       "    'lpage': '249',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Silica and trichloroethylene-induced progressive systemic sclerosis',\n",
       "   'offset': 40753,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Unamuno;given-names:P',\n",
       "    'name_3': 'surname:Armijo;given-names:M',\n",
       "    'name_0': 'surname:Yanez Diaz;given-names:S',\n",
       "    'name_1': 'surname:Moran;given-names:M',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '184',\n",
       "    'source': 'Dermatol',\n",
       "    'fpage': '98',\n",
       "    'year': '1992',\n",
       "    'lpage': '102',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Studies on age-related functional changes in regulatory T cells and B cells involved in the autoantibody production of MRL/MpJ- +/+ mice',\n",
       "   'offset': 40821,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Nishi;given-names:Y',\n",
       "    'name_3': 'surname:Koyama;given-names:K',\n",
       "    'name_0': 'surname:Yoshida;given-names:A',\n",
       "    'name_1': 'surname:Hosokawa;given-names:T',\n",
       "    'name_6': 'surname:Rokutan;given-names:K',\n",
       "    'name_7': 'surname:Aoike;given-names:A',\n",
       "    'name_4': 'surname:Nakamura;given-names:K',\n",
       "    'name_5': 'surname:Marui;given-names:N',\n",
       "    'name_8': 'surname:Kawai;given-names:K',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '50',\n",
       "    'pub-id_pmid': '2601416',\n",
       "    'source': 'Mech Ageing Dev',\n",
       "    'fpage': '179',\n",
       "    'year': '1989',\n",
       "    'lpage': '192',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A newly identified microRNA, mmu-miR-7578, functions as a negative regulator on inflammatory cytokines tumor necrosis factor-alpha and interleukin-6 via targeting Egr1 in vivo',\n",
       "   'offset': 40958,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Ma;given-names:W',\n",
       "    'name_3': 'surname:Teng;given-names:Y',\n",
       "    'name_0': 'surname:Zhang;given-names:J',\n",
       "    'name_1': 'surname:Xie;given-names:S',\n",
       "    'name_6': 'surname:Zhang;given-names:Y',\n",
       "    'name_4': 'surname:Tian;given-names:Y',\n",
       "    'name_5': 'surname:Huang;given-names:X',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '288',\n",
       "    'pub-id_pmid': '23184950',\n",
       "    'source': 'J Biol Chem',\n",
       "    'fpage': '4310',\n",
       "    'year': '2013',\n",
       "    'lpage': '4320',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Deletion of interleukin-6 in mice with the dominant negative form of transforming growth factor beta receptor II improves colitis but exacerbates autoimmune cholangitis',\n",
       "   'offset': 41134,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Yang;given-names:GX',\n",
       "    'name_3': 'surname:Tsuneyama;given-names:K',\n",
       "    'name_0': 'surname:Zhang;given-names:W',\n",
       "    'name_1': 'surname:Tsuda;given-names:M',\n",
       "    'name_6': 'surname:Ansari;given-names:AA',\n",
       "    'name_7': 'surname:Flavell;given-names:RA',\n",
       "    'name_4': 'surname:Rong;given-names:G',\n",
       "    'name_5': 'surname:Ridgway;given-names:WM',\n",
       "    'name_8': 'surname:Coppel;given-names:RL',\n",
       "    'name_9': 'surname:Lian;given-names:ZX',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '52',\n",
       "    'pub-id_pmid': '20578264',\n",
       "    'source': 'Hepatology',\n",
       "    'fpage': '215',\n",
       "    'year': '2010',\n",
       "    'name_10': 'surname:Gershwin;given-names:ME',\n",
       "    'lpage': '222',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'IL-17 neutralization significantly ameliorates hepatic granulomatous inflammation and liver damage in Schistosoma japonicum infected mice',\n",
       "   'offset': 41303,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:Gao;given-names:W',\n",
       "    'name_3': 'surname:Hou;given-names:X',\n",
       "    'name_0': 'surname:Zhang;given-names:Y',\n",
       "    'name_1': 'surname:Chen;given-names:L',\n",
       "    'name_6': 'surname:Huang;given-names:D',\n",
       "    'name_7': 'surname:Liu;given-names:M',\n",
       "    'name_4': 'surname:Gu;given-names:Y',\n",
       "    'name_5': 'surname:Gui;given-names:L',\n",
       "    'name_8': 'surname:Ren;given-names:C',\n",
       "    'name_9': 'surname:Wang;given-names:S',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '42',\n",
       "    'pub-id_pmid': '22678906',\n",
       "    'source': 'Eur J Immunol',\n",
       "    'fpage': '1523',\n",
       "    'year': '2012',\n",
       "    'name_10': 'surname:Shen;given-names:J',\n",
       "    'lpage': '1535',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Interleukin-17 contributes to the pathogenesis of autoimmune hepatitis through inducing hepatic interleukin-6 expression',\n",
       "   'offset': 41441,\n",
       "   'relations': [],\n",
       "   'infons': {'name_2': 'surname:You;given-names:Z',\n",
       "    'name_3': 'surname:Wang;given-names:Q',\n",
       "    'name_0': 'surname:Zhao;given-names:L',\n",
       "    'name_1': 'surname:Tang;given-names:Y',\n",
       "    'name_6': 'surname:Qiu;given-names:D',\n",
       "    'name_7': 'surname:Wei;given-names:J',\n",
       "    'name_4': 'surname:Liang;given-names:S',\n",
       "    'name_5': 'surname:Han;given-names:X',\n",
       "    'name_8': 'surname:Liu;given-names:Y',\n",
       "    'name_9': 'surname:Shen;given-names:L',\n",
       "    'section_type': 'REF',\n",
       "    'volume': '6',\n",
       "    'pub-id_pmid': '21526159',\n",
       "    'source': 'PLoS One',\n",
       "    'fpage': 'e18909',\n",
       "    'year': '2011',\n",
       "    'name_10': 'surname:Chen;given-names:X',\n",
       "    'name_11': 'surname:Peng;given-names:Y',\n",
       "    'name_12': 'surname:Li;given-names:Z',\n",
       "    'name_13': 'surname:Ma;given-names:X',\n",
       "    'type': 'ref'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE inhibits macrophage IL-6 production in dose-dependent manner',\n",
       "   'offset': 41562,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F1',\n",
       "    'file': 'nihms625241f1.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Peritoneal macrophages were incubated with (open bars) or without LPS (shaded bars) following isolation from untreated control mice or from mice exposed to TCE at different concentrations for 12 weeks. Culture supernatants were examined for cytokines (mean ± SD). *Significantly different (α<0.05) compared to control values.',\n",
       "   'offset': 41627,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F1',\n",
       "    'file': 'nihms625241f1.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE inhibits macrophage Il6 expression in dose-dependent manner',\n",
       "   'offset': 41958,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F2',\n",
       "    'file': 'nihms625241f2.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cytokine gene expression was examined in peritoneal macrophages incubated with (open bars) or without LPS (shaded bars) after isolation from untreated control mice or from mice exposed to TCE for 12 weeks. The data represents the mean ± SD. *Significantly different (α<0.05) compared to control values.',\n",
       "   'offset': 42022,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F2',\n",
       "    'file': 'nihms625241f2.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE inhibition IL-6 production is maintained over time',\n",
       "   'offset': 42330,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F3',\n",
       "    'file': 'nihms625241f3.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Peritoneal macrophages were incubated with LPS following isolation from untreated control mice or from mice exposed to TCE (0.5 mg/ml) for up to 40 weeks. Culture supernatants were examined for cytokines (mean ± SD). *Significantly different (α<0.05) compared to control values.',\n",
       "   'offset': 42385,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F3',\n",
       "    'file': 'nihms625241f3.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE inhibition of Il6 expression is maintained over time',\n",
       "   'offset': 42669,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F4',\n",
       "    'file': 'nihms625241f4.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Cytokine gene expression was examined in peritoneal macrophages incubated with or without LPS after isolation from untreated control mice or from mice exposed to TCE (0.5 mg/ml) for up to 40 weeks. The data represents the mean ± SD. *Significantly different (α<0.05) compared to control values.',\n",
       "   'offset': 42726,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F4',\n",
       "    'file': 'nihms625241f4.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE alters expression of hepatic genes over time',\n",
       "   'offset': 43026,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F5',\n",
       "    'file': 'nihms625241f5.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A. Gene expression in individual liver tissue isolated from untreated control mice or from mice exposed to TCE (0.5 mg/ml) for up to 40 weeks. The data represents the mean ± SD from 6–8 individual mice/treatment/time point. *Significantly different (α<0.05) compared to control values. B. Relative protein levels (percentage reference protein GAPDH) of IL-6R in individual livers from untreated control mice or mice exposed to TCE (0.5 mg/ml) for 16 weeks (mean ± SD).',\n",
       "   'offset': 43075,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F5',\n",
       "    'file': 'nihms625241f5.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'TCE liver pathology correlates with loss of hepatic Il-6r expression',\n",
       "   'offset': 43550,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F6',\n",
       "    'file': 'nihms625241f6.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A. Liver pathology based on immune cell infiltration and inflammation was assessed in mice exposed to TCE (0.5mg/ml) for 28, 34 or40 weeks. B. Equal amounts of liver protein from an untreated mouse were separated in 4 lanes of SDS-PAGE, each of which were immunoblotted with pooled sera obtained from control MRL+/+ mice or mice treated with 0.5 mg/ml TCE for 4 or 40 weeks. C. Hepatic gene expression in from mice exposed to TCE (0.5 mg/ml) for 40 weeks was plotted against liver histopathology in the same mice. Gene expression values are shown in log scale because of right skewness. Regression p-values were computed using an F test of the null hypothesis of horizontal slope.',\n",
       "   'offset': 43619,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F6',\n",
       "    'file': 'nihms625241f6.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Submodel for parameter estimation',\n",
       "   'offset': 44300,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F7',\n",
       "    'file': 'nihms625241f7.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'A. An IL-6 submodel was developed for estimating dose-dependent reduction in the fraction of IL-6 expressed by the macrophage. Points and error bars represent data and uncertainty, while solid and dashed lines are the mean and 95% confidence intervals from model predictions. B. Time-course pathology scores were used to extrapolate liver pathology based on time of TCE exposure. Points and error bars represent data and uncertainty, while solid and dashed lines are the mean and 95% confidence intervals from model predictions.',\n",
       "   'offset': 44334,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F7',\n",
       "    'file': 'nihms625241f7.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Liver unit state predictions based on the model',\n",
       "   'offset': 44863,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F8',\n",
       "    'file': 'nihms625241f8.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Fraction of liver units in each state for the 0.1 (A) and 0.5 (B) mg/ml experimental doses. This reflects the overall health of the liver rather than specific changes in IL-6 production. Solid lines represent the H state, while vertical and dashed lines correspond to the C and I states, respectively.',\n",
       "   'offset': 44911,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F8',\n",
       "    'file': 'nihms625241f8.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Dose response curve for current study',\n",
       "   'offset': 45213,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F9',\n",
       "    'file': 'nihms625241f9.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Predicted dose response curves for pathology scores (PS) 40 weeks post TCE exposure. The mean values and 95% confidence intervals are shown as solid and dashed lines, respectively. The point represents the value of the benchmark dose (BMD) corresponding to the benchmark response level (BMR) described in the text.',\n",
       "   'offset': 45251,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F9',\n",
       "    'file': 'nihms625241f9.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Simulations illustrate the effects of relative rates of repair and damage on liver damage',\n",
       "   'offset': 45566,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption_title',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F10',\n",
       "    'file': 'nihms625241f10.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Model predictions for varying relative levels of repair in (A) the H-C pathway (κH-C=[1, 10, 100, 1000], κC-I=1) and (B) the C-I pathway (κH-C=100, κC-I=[0.1, 1, 10, 100]). κH-C=100 and κC-I=1 are the values that that relate to the pathology scores from the current study. The two dotted lines represent the kappas (k) used in this study (k1=100 and k2=1). Dashed lines are representative of hypothetical pathways with an order of magnitude increase in repair mechanisms. Conversely, solid lines represent scenarios where external factors result in an order of magnitude decrease in repair.',\n",
       "   'offset': 45656,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'fig_caption',\n",
       "    'section_type': 'FIG',\n",
       "    'id': 'F10',\n",
       "    'file': 'nihms625241f10.jpg'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Description of LU states',\n",
       "   'offset': 46271,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table_caption',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T1',\n",
       "    'file': 'T1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': '\\n        \\n          \\n            LU State\\t\\n            Distinguishing Characteristics\\t\\n            Relevant Pathology Score (PS)\\t\\n          \\n\\n        \\n        \\n          \\n            Healthy (H)\\t\\n            Normal liver function, homeostatic levels of IL-6 and IL-6r\\t\\n            1\\t\\n          \\n\\n          \\n            Compromised (C)\\t\\n            Intermediate state, events initiate inflammatory pathways which are normally countered by IL-6 signaling.\\t\\n            2\\t\\n          \\n\\n          \\n            Inflamed (I)\\t\\n            Contains markers for the early stages of auto-immune hepatitis, including inflammation and lymphoplasmacytic portal infiltration\\t\\n            4\\t\\n          \\n\\n        \\n      ',\n",
       "   'offset': 46296,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'table',\n",
       "    'section_type': 'TABLE',\n",
       "    'id': 'T1',\n",
       "    'File': 'T1.xml'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Highlights',\n",
       "   'offset': 47000,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'title_caption', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We developed a toxicodynamic model to study effects of trichloroethylene on liver.',\n",
       "   'offset': 47011,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'We examined protective as well as pro-inflammatory events in the liver.',\n",
       "   'offset': 47094,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene inhibits IL-6 production by macrophages.',\n",
       "   'offset': 47166,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene inhibits components of the IL-6R in the liver.',\n",
       "   'offset': 47225,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []},\n",
       "  {'text': 'Trichloroethylene inhibits events associated with IL-6-mediated hepatoprotection.',\n",
       "   'offset': 47290,\n",
       "   'relations': [],\n",
       "   'infons': {'type': 'paragraph', 'section_type': 'REF'},\n",
       "   'sentences': [],\n",
       "   'annotations': []}],\n",
       " 'infons': {'license': 'author_manuscript'},\n",
       " 'id': '4171219',\n",
       " 'relations': []}"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmcid_to_doc[ids_to_check[528]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
